<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>Airway: DE analysis</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="intro.html">1.1 Position</a>
    </li>
    <li>
      <a href="recapGeneralLinearModel.html">1.2 Recap linear model</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="docs/martens_proteomics_bioinformatics.pdf">1. Proteomics Bioinformatics</a>
    </li>
    <li>
      <a href="TargetDecoy.html">2.1 Identification - FDR - QC</a>
    </li>
    <li>
      <a href="pda_quantification_preprocessing.html">2.2.1. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">2.2.2. Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">2.3.1. Quantification - Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="pda_blocking_wrapup.html">2.3.2. Wrap-up DA - Blocking</a>
    </li>
    <li>
      <a href="technicalDetailsProteomics.html">2.3.3 Technical Details</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencing_intro.html">1.  Introduction to RNA-seq</a>
    </li>
    <li>
      <a href="sequencing_countData.html">2.  Working with count data and GLMs</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="multipleRegression_KPNA2.html">1 Intro KPNA2</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="https://www.compomics.com/bioinformatics-for-proteomics/">1. Bioinformatics for Proteomics</a>
    </li>
    <li>
      <a href="TargetDecoyTutorial.html">2.1. Target Decoy QC</a>
    </li>
    <li>
      <a href="pda_tutorialPreprocessing.html">2.2. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">2.3. Quantification - Design</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencingTutorial1_mapping.html">1. Read mapping</a>
    </li>
    <li>
      <a href="sequencingTutorial2_DE.html">2. Working with Count Data</a>
    </li>
    <li>
      <a href="scRNA-seq-Tutorial.html">3. Single Cell RNA-seq Data Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Airway: DE analysis</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date">statOmics, Ghent University (<a
href="https://statomics.github.io"
class="uri">https://statomics.github.io</a>)</h4>

</div>


<div id="background" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Background</h1>
<p>The data used in this workflow comes from an RNA-seq experiment where
airway smooth muscle cells were treated with dexamethasone, a synthetic
glucocorticoid steroid with anti-inflammatory effects (Himes et
al. 2014). Glucocorticoids are used, for example, by people with asthma
to reduce inflammation of the airways. In the experiment, four human
airway smooth muscle cell lines were treated with 1 micromolar
dexamethasone for 18 hours. For each of the four cell lines, we have a
treated and an untreated sample. For more description of the experiment
see the article, PubMed entry 24926665, and for raw data see the GEO
entry GSE52778.</p>
<p>Many parts of this tutorial are based on parts of a published RNA-seq
workflow available via Love et al. 2015 <a
href="http://f1000research.com/articles/4-1070">F1000Research</a> and as
a <a
href="https://www.bioconductor.org/packages/release/workflows/html/rnaseqGene.html">Bioconductor
package</a> and on Charlotte Soneson’s material from the <a
href="https://uclouvain-cbio.github.io/BSS2019/rnaseq_gene_summerschool_belgium_2019.html">bss2019</a>
workshop.</p>
</div>
<div id="data" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Data</h1>
<p>FastQ files with a small subset of the reads can be found on <a
href="https://github.com/statOmics/SGA2019/tree/data-rnaseq"
class="uri">https://github.com/statOmics/SGA2019/tree/data-rnaseq</a>
Here, we will not use our count table because it is based on a small
subset of the reads. We will use the count table that was provided after
mapping all the reads with the read mapper star.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span></code></pre></div>
<pre><code>## Loading required package: limma</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
<pre><code>## ── Attaching packages
## ───────────────────────────────────────
## tidyverse 1.3.2 ──</code></pre>
<pre><code>## ✔ ggplot2 3.3.6     ✔ purrr   0.3.4
## ✔ tibble  3.1.7     ✔ dplyr   1.0.9
## ✔ tidyr   1.2.0     ✔ stringr 1.4.1
## ✔ readr   2.1.2     ✔ forcats 0.5.1
## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span></code></pre></div>
<pre><code>## Loading required package: Biobase
## Loading required package: BiocGenerics
## 
## Attaching package: &#39;BiocGenerics&#39;
## 
## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     combine, intersect, setdiff, union
## 
## The following object is masked from &#39;package:limma&#39;:
## 
##     plotMA
## 
## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs
## 
## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min
## 
## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.
## 
## Setting options(&#39;download.file.method.GEOquery&#39;=&#39;auto&#39;)
## Setting options(&#39;GEOquery.inmemory.gpl&#39;=FALSE)</code></pre>
</div>
<div id="metadata" class="section level1" number="3">
<h1><span class="header-section-number">3</span> MetaData</h1>
<div id="geo-data" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> GEO data</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>gse<span class="ot">&lt;-</span><span class="fu">getGEO</span>(<span class="st">&quot;GSE52778&quot;</span>)</span></code></pre></div>
<pre><code>## Found 1 file(s)</code></pre>
<pre><code>## GSE52778_series_matrix.txt.gz</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>pdata<span class="ot">&lt;-</span><span class="fu">pData</span>(gse[[<span class="dv">1</span>]])</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>pdata<span class="sc">$</span>SampleName<span class="ot">&lt;-</span><span class="fu">rownames</span>(pdata) <span class="sc">%&gt;%</span> as.factor</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="fu">head</span>(pdata)[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>,]</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["title"],"name":[1],"type":["chr"],"align":["left"]},{"label":["geo_accession"],"name":[2],"type":["chr"],"align":["left"]},{"label":["status"],"name":[3],"type":["chr"],"align":["left"]},{"label":["submission_date"],"name":[4],"type":["chr"],"align":["left"]},{"label":["last_update_date"],"name":[5],"type":["chr"],"align":["left"]},{"label":["type"],"name":[6],"type":["chr"],"align":["left"]},{"label":["channel_count"],"name":[7],"type":["chr"],"align":["left"]},{"label":["source_name_ch1"],"name":[8],"type":["chr"],"align":["left"]},{"label":["organism_ch1"],"name":[9],"type":["chr"],"align":["left"]},{"label":["characteristics_ch1"],"name":[10],"type":["chr"],"align":["left"]},{"label":["characteristics_ch1.1"],"name":[11],"type":["chr"],"align":["left"]},{"label":["characteristics_ch1.2"],"name":[12],"type":["chr"],"align":["left"]},{"label":["characteristics_ch1.3"],"name":[13],"type":["chr"],"align":["left"]},{"label":["characteristics_ch1.4"],"name":[14],"type":["chr"],"align":["left"]},{"label":["treatment_protocol_ch1"],"name":[15],"type":["chr"],"align":["left"]},{"label":["growth_protocol_ch1"],"name":[16],"type":["chr"],"align":["left"]},{"label":["molecule_ch1"],"name":[17],"type":["chr"],"align":["left"]},{"label":["extract_protocol_ch1"],"name":[18],"type":["chr"],"align":["left"]},{"label":["extract_protocol_ch1.1"],"name":[19],"type":["chr"],"align":["left"]},{"label":["taxid_ch1"],"name":[20],"type":["chr"],"align":["left"]},{"label":["data_processing"],"name":[21],"type":["chr"],"align":["left"]},{"label":["data_processing.1"],"name":[22],"type":["chr"],"align":["left"]},{"label":["data_processing.2"],"name":[23],"type":["chr"],"align":["left"]},{"label":["data_processing.3"],"name":[24],"type":["chr"],"align":["left"]},{"label":["data_processing.4"],"name":[25],"type":["chr"],"align":["left"]},{"label":["data_processing.5"],"name":[26],"type":["chr"],"align":["left"]},{"label":["data_processing.6"],"name":[27],"type":["chr"],"align":["left"]},{"label":["platform_id"],"name":[28],"type":["chr"],"align":["left"]},{"label":["contact_name"],"name":[29],"type":["chr"],"align":["left"]},{"label":["contact_email"],"name":[30],"type":["chr"],"align":["left"]},{"label":["contact_laboratory"],"name":[31],"type":["chr"],"align":["left"]},{"label":["contact_department"],"name":[32],"type":["chr"],"align":["left"]},{"label":["contact_institute"],"name":[33],"type":["chr"],"align":["left"]},{"label":["contact_address"],"name":[34],"type":["chr"],"align":["left"]},{"label":["contact_city"],"name":[35],"type":["chr"],"align":["left"]},{"label":["contact_state"],"name":[36],"type":["chr"],"align":["left"]},{"label":["contact_zip/postal_code"],"name":[37],"type":["chr"],"align":["left"]},{"label":["contact_country"],"name":[38],"type":["chr"],"align":["left"]},{"label":["data_row_count"],"name":[39],"type":["chr"],"align":["left"]},{"label":["instrument_model"],"name":[40],"type":["chr"],"align":["left"]},{"label":["library_selection"],"name":[41],"type":["chr"],"align":["left"]},{"label":["library_source"],"name":[42],"type":["chr"],"align":["left"]},{"label":["library_strategy"],"name":[43],"type":["chr"],"align":["left"]},{"label":["relation"],"name":[44],"type":["chr"],"align":["left"]},{"label":["relation.1"],"name":[45],"type":["chr"],"align":["left"]},{"label":["supplementary_file_1"],"name":[46],"type":["chr"],"align":["left"]},{"label":["cell line:ch1"],"name":[47],"type":["chr"],"align":["left"]},{"label":["cell type:ch1"],"name":[48],"type":["chr"],"align":["left"]},{"label":["ercc_mix:ch1"],"name":[49],"type":["chr"],"align":["left"]},{"label":["tissue:ch1"],"name":[50],"type":["chr"],"align":["left"]},{"label":["treatment:ch1"],"name":[51],"type":["chr"],"align":["left"]},{"label":["SampleName"],"name":[52],"type":["fct"],"align":["left"]}],"data":[{"1":"N61311_untreated","2":"GSM1275862","3":"Public on Jan 01 2014","4":"Nov 26 2013","5":"May 15 2019","6":"SRA","7":"1","8":"airway smooth muscle cells","9":"Homo sapiens","10":"treatment: Untreated","11":"tissue: human airway smooth muscle cells","12":"ercc_mix: -","13":"cell line: N61311","14":"cell type: airway smooth muscle cells","15":"Cells from each donor were treated with 1 μM dexamethasone,  1 μM albuterol, both dexamethasone and albuterol, or control vehicle for 18 h.","16":"Primary ASM cells were isolated from four white aborted lung transplant donors with no chronic illness. Passages 4 to 7 ASM cells maintained in Ham's F12 medium supplemented with 10% FBS were used in all experiments.","17":"total RNA","18":"Total RNA was extracted using the miRNAeasy mini kit (Qiagen Sciences, Inc., Germantown, MD).","19":"TruSeq RNA Sample Prep Kit v2 (Illumina, Inc., San Diego, CA).","20":"9606","21":"Illumina Casava1.8 software used for basecalling.","22":"Sequenced reads were trimmed by 12bp then mapped to hg19 whole genome using tophat v.2.0.4 with parameters -G [hg19_ERCC_gtf] --no-novel-juncs --transcriptome-only","23":"For each sample, Cufflinks (v.2.0.2) was used to quantify External RNA Controls Consortium (ERCC) Spike-In and hg19 transcripts based on reads that mapped to the provided hg19 and ERCC reference files.","24":"Differential expression of genes and transcripts in samples treated with Dex vs. Untreated samples was obtained using Cuffdiff (v.2.0.2) with the quantified transcripts computed by Cufflinks (v.2.0.2), while applying bias correction. Options: -p 12 --min-outlier-p 0.05 -b [hg19_reference_index] -u [hg19_gtf] -L Dex,Control  [List of accepted_hits.bam for each sample sorted by condition (i.e. Dex,Control)]","25":"CummeRbund R package (v.0.1.3) was used to create FPKM matrix of all samples","26":"Genome_build: hg19","27":"Supplementary_files_format_and_content: tab-delimited text files on Series record include FPKM values for each sample (ALL_Sample_FPKM_Matrix.txt) and gene-based differential expression results for Dex vs. Untreated condition (Dex_vs_Untreated_gene_exp.diff)","28":"GPL11154","29":"Blanca,E,Himes","30":"bhimes@pennmedicine.upenn.edu","31":"402 Blockley Hall","32":"Department of Biostatistics, Epidemiology and Informatics","33":"University of Pennsylvania","34":"423 Guardian Dr","35":"Philadelphia","36":"PA","37":"19104","38":"USA","39":"0","40":"Illumina HiSeq 2000","41":"cDNA","42":"transcriptomic","43":"RNA-Seq","44":"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02422669","45":"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX384345","46":"NONE","47":"N61311","48":"airway smooth muscle cells","49":"-","50":"human airway smooth muscle cells","51":"Untreated","52":"GSM1275862","_rn_":"GSM1275862"},{"1":"N61311_Dex","2":"GSM1275863","3":"Public on Jan 01 2014","4":"Nov 26 2013","5":"May 15 2019","6":"SRA","7":"1","8":"airway smooth muscle cells","9":"Homo sapiens","10":"treatment: Dexamethasone","11":"tissue: human airway smooth muscle cells","12":"ercc_mix: -","13":"cell line: N61311","14":"cell type: airway smooth muscle cells","15":"Cells from each donor were treated with 1 μM dexamethasone,  1 μM albuterol, both dexamethasone and albuterol, or control vehicle for 18 h.","16":"Primary ASM cells were isolated from four white aborted lung transplant donors with no chronic illness. Passages 4 to 7 ASM cells maintained in Ham's F12 medium supplemented with 10% FBS were used in all experiments.","17":"total RNA","18":"Total RNA was extracted using the miRNAeasy mini kit (Qiagen Sciences, Inc., Germantown, MD).","19":"TruSeq RNA Sample Prep Kit v2 (Illumina, Inc., San Diego, CA).","20":"9606","21":"Illumina Casava1.8 software used for basecalling.","22":"Sequenced reads were trimmed by 12bp then mapped to hg19 whole genome using tophat v.2.0.4 with parameters -G [hg19_ERCC_gtf] --no-novel-juncs --transcriptome-only","23":"For each sample, Cufflinks (v.2.0.2) was used to quantify External RNA Controls Consortium (ERCC) Spike-In and hg19 transcripts based on reads that mapped to the provided hg19 and ERCC reference files.","24":"Differential expression of genes and transcripts in samples treated with Dex vs. Untreated samples was obtained using Cuffdiff (v.2.0.2) with the quantified transcripts computed by Cufflinks (v.2.0.2), while applying bias correction. Options: -p 12 --min-outlier-p 0.05 -b [hg19_reference_index] -u [hg19_gtf] -L Dex,Control  [List of accepted_hits.bam for each sample sorted by condition (i.e. Dex,Control)]","25":"CummeRbund R package (v.0.1.3) was used to create FPKM matrix of all samples","26":"Genome_build: hg19","27":"Supplementary_files_format_and_content: tab-delimited text files on Series record include FPKM values for each sample (ALL_Sample_FPKM_Matrix.txt) and gene-based differential expression results for Dex vs. Untreated condition (Dex_vs_Untreated_gene_exp.diff)","28":"GPL11154","29":"Blanca,E,Himes","30":"bhimes@pennmedicine.upenn.edu","31":"402 Blockley Hall","32":"Department of Biostatistics, Epidemiology and Informatics","33":"University of Pennsylvania","34":"423 Guardian Dr","35":"Philadelphia","36":"PA","37":"19104","38":"USA","39":"0","40":"Illumina HiSeq 2000","41":"cDNA","42":"transcriptomic","43":"RNA-Seq","44":"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02422675","45":"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX384346","46":"NONE","47":"N61311","48":"airway smooth muscle cells","49":"-","50":"human airway smooth muscle cells","51":"Dexamethasone","52":"GSM1275863","_rn_":"GSM1275863"},{"1":"N61311_Alb","2":"GSM1275864","3":"Public on Jan 01 2014","4":"Nov 26 2013","5":"May 15 2019","6":"SRA","7":"1","8":"airway smooth muscle cells","9":"Homo sapiens","10":"treatment: Albuterol","11":"tissue: human airway smooth muscle cells","12":"ercc_mix: -","13":"cell line: N61311","14":"cell type: airway smooth muscle cells","15":"Cells from each donor were treated with 1 μM dexamethasone,  1 μM albuterol, both dexamethasone and albuterol, or control vehicle for 18 h.","16":"Primary ASM cells were isolated from four white aborted lung transplant donors with no chronic illness. Passages 4 to 7 ASM cells maintained in Ham's F12 medium supplemented with 10% FBS were used in all experiments.","17":"total RNA","18":"Total RNA was extracted using the miRNAeasy mini kit (Qiagen Sciences, Inc., Germantown, MD).","19":"TruSeq RNA Sample Prep Kit v2 (Illumina, Inc., San Diego, CA).","20":"9606","21":"Illumina Casava1.8 software used for basecalling.","22":"Sequenced reads were trimmed by 12bp then mapped to hg19 whole genome using tophat v.2.0.4 with parameters -G [hg19_ERCC_gtf] --no-novel-juncs --transcriptome-only","23":"For each sample, Cufflinks (v.2.0.2) was used to quantify External RNA Controls Consortium (ERCC) Spike-In and hg19 transcripts based on reads that mapped to the provided hg19 and ERCC reference files.","24":"Differential expression of genes and transcripts in samples treated with Dex vs. Untreated samples was obtained using Cuffdiff (v.2.0.2) with the quantified transcripts computed by Cufflinks (v.2.0.2), while applying bias correction. Options: -p 12 --min-outlier-p 0.05 -b [hg19_reference_index] -u [hg19_gtf] -L Dex,Control  [List of accepted_hits.bam for each sample sorted by condition (i.e. Dex,Control)]","25":"CummeRbund R package (v.0.1.3) was used to create FPKM matrix of all samples","26":"Genome_build: hg19","27":"Supplementary_files_format_and_content: tab-delimited text files on Series record include FPKM values for each sample (ALL_Sample_FPKM_Matrix.txt) and gene-based differential expression results for Dex vs. Untreated condition (Dex_vs_Untreated_gene_exp.diff)","28":"GPL11154","29":"Blanca,E,Himes","30":"bhimes@pennmedicine.upenn.edu","31":"402 Blockley Hall","32":"Department of Biostatistics, Epidemiology and Informatics","33":"University of Pennsylvania","34":"423 Guardian Dr","35":"Philadelphia","36":"PA","37":"19104","38":"USA","39":"0","40":"Illumina HiSeq 2000","41":"cDNA","42":"transcriptomic","43":"RNA-Seq","44":"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02422668","45":"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX384347","46":"NONE","47":"N61311","48":"airway smooth muscle cells","49":"-","50":"human airway smooth muscle cells","51":"Albuterol","52":"GSM1275864","_rn_":"GSM1275864"},{"1":"N61311_Alb_Dex","2":"GSM1275865","3":"Public on Jan 01 2014","4":"Nov 26 2013","5":"May 15 2019","6":"SRA","7":"1","8":"airway smooth muscle cells","9":"Homo sapiens","10":"treatment: Albuterol_Dexamethasone","11":"tissue: human airway smooth muscle cells","12":"ercc_mix: -","13":"cell line: N61311","14":"cell type: airway smooth muscle cells","15":"Cells from each donor were treated with 1 μM dexamethasone,  1 μM albuterol, both dexamethasone and albuterol, or control vehicle for 18 h.","16":"Primary ASM cells were isolated from four white aborted lung transplant donors with no chronic illness. Passages 4 to 7 ASM cells maintained in Ham's F12 medium supplemented with 10% FBS were used in all experiments.","17":"total RNA","18":"Total RNA was extracted using the miRNAeasy mini kit (Qiagen Sciences, Inc., Germantown, MD).","19":"TruSeq RNA Sample Prep Kit v2 (Illumina, Inc., San Diego, CA).","20":"9606","21":"Illumina Casava1.8 software used for basecalling.","22":"Sequenced reads were trimmed by 12bp then mapped to hg19 whole genome using tophat v.2.0.4 with parameters -G [hg19_ERCC_gtf] --no-novel-juncs --transcriptome-only","23":"For each sample, Cufflinks (v.2.0.2) was used to quantify External RNA Controls Consortium (ERCC) Spike-In and hg19 transcripts based on reads that mapped to the provided hg19 and ERCC reference files.","24":"Differential expression of genes and transcripts in samples treated with Dex vs. Untreated samples was obtained using Cuffdiff (v.2.0.2) with the quantified transcripts computed by Cufflinks (v.2.0.2), while applying bias correction. Options: -p 12 --min-outlier-p 0.05 -b [hg19_reference_index] -u [hg19_gtf] -L Dex,Control  [List of accepted_hits.bam for each sample sorted by condition (i.e. Dex,Control)]","25":"CummeRbund R package (v.0.1.3) was used to create FPKM matrix of all samples","26":"Genome_build: hg19","27":"Supplementary_files_format_and_content: tab-delimited text files on Series record include FPKM values for each sample (ALL_Sample_FPKM_Matrix.txt) and gene-based differential expression results for Dex vs. Untreated condition (Dex_vs_Untreated_gene_exp.diff)","28":"GPL11154","29":"Blanca,E,Himes","30":"bhimes@pennmedicine.upenn.edu","31":"402 Blockley Hall","32":"Department of Biostatistics, Epidemiology and Informatics","33":"University of Pennsylvania","34":"423 Guardian Dr","35":"Philadelphia","36":"PA","37":"19104","38":"USA","39":"0","40":"Illumina HiSeq 2000","41":"cDNA","42":"transcriptomic","43":"RNA-Seq","44":"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02422667","45":"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX384348","46":"NONE","47":"N61311","48":"airway smooth muscle cells","49":"-","50":"human airway smooth muscle cells","51":"Albuterol_Dexamethasone","52":"GSM1275865","_rn_":"GSM1275865"},{"1":"N052611_untreated","2":"GSM1275866","3":"Public on Jan 01 2014","4":"Nov 26 2013","5":"May 15 2019","6":"SRA","7":"1","8":"airway smooth muscle cells","9":"Homo sapiens","10":"treatment: Untreated","11":"tissue: human airway smooth muscle cells","12":"ercc_mix: -","13":"cell line: N052611","14":"cell type: airway smooth muscle cells","15":"Cells from each donor were treated with 1 μM dexamethasone,  1 μM albuterol, both dexamethasone and albuterol, or control vehicle for 18 h.","16":"Primary ASM cells were isolated from four white aborted lung transplant donors with no chronic illness. Passages 4 to 7 ASM cells maintained in Ham's F12 medium supplemented with 10% FBS were used in all experiments.","17":"total RNA","18":"Total RNA was extracted using the miRNAeasy mini kit (Qiagen Sciences, Inc., Germantown, MD).","19":"TruSeq RNA Sample Prep Kit v2 (Illumina, Inc., San Diego, CA).","20":"9606","21":"Illumina Casava1.8 software used for basecalling.","22":"Sequenced reads were trimmed by 12bp then mapped to hg19 whole genome using tophat v.2.0.4 with parameters -G [hg19_ERCC_gtf] --no-novel-juncs --transcriptome-only","23":"For each sample, Cufflinks (v.2.0.2) was used to quantify External RNA Controls Consortium (ERCC) Spike-In and hg19 transcripts based on reads that mapped to the provided hg19 and ERCC reference files.","24":"Differential expression of genes and transcripts in samples treated with Dex vs. Untreated samples was obtained using Cuffdiff (v.2.0.2) with the quantified transcripts computed by Cufflinks (v.2.0.2), while applying bias correction. Options: -p 12 --min-outlier-p 0.05 -b [hg19_reference_index] -u [hg19_gtf] -L Dex,Control  [List of accepted_hits.bam for each sample sorted by condition (i.e. Dex,Control)]","25":"CummeRbund R package (v.0.1.3) was used to create FPKM matrix of all samples","26":"Genome_build: hg19","27":"Supplementary_files_format_and_content: tab-delimited text files on Series record include FPKM values for each sample (ALL_Sample_FPKM_Matrix.txt) and gene-based differential expression results for Dex vs. Untreated condition (Dex_vs_Untreated_gene_exp.diff)","28":"GPL11154","29":"Blanca,E,Himes","30":"bhimes@pennmedicine.upenn.edu","31":"402 Blockley Hall","32":"Department of Biostatistics, Epidemiology and Informatics","33":"University of Pennsylvania","34":"423 Guardian Dr","35":"Philadelphia","36":"PA","37":"19104","38":"USA","39":"0","40":"Illumina HiSeq 2000","41":"cDNA","42":"transcriptomic","43":"RNA-Seq","44":"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02422678","45":"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX384349","46":"NONE","47":"N052611","48":"airway smooth muscle cells","49":"-","50":"human airway smooth muscle cells","51":"Untreated","52":"GSM1275866","_rn_":"GSM1275866"},{"1":"N052611_Dex","2":"GSM1275867","3":"Public on Jan 01 2014","4":"Nov 26 2013","5":"May 15 2019","6":"SRA","7":"1","8":"airway smooth muscle cells","9":"Homo sapiens","10":"treatment: Dexamethasone","11":"tissue: human airway smooth muscle cells","12":"ercc_mix: 1","13":"cell line: N052611","14":"cell type: airway smooth muscle cells","15":"Cells from each donor were treated with 1 μM dexamethasone,  1 μM albuterol, both dexamethasone and albuterol, or control vehicle for 18 h.","16":"Primary ASM cells were isolated from four white aborted lung transplant donors with no chronic illness. Passages 4 to 7 ASM cells maintained in Ham's F12 medium supplemented with 10% FBS were used in all experiments.","17":"total RNA","18":"Total RNA was extracted using the miRNAeasy mini kit (Qiagen Sciences, Inc., Germantown, MD).","19":"TruSeq RNA Sample Prep Kit v2 (Illumina, Inc., San Diego, CA).","20":"9606","21":"Illumina Casava1.8 software used for basecalling.","22":"Sequenced reads were trimmed by 12bp then mapped to hg19 whole genome using tophat v.2.0.4 with parameters -G [hg19_ERCC_gtf] --no-novel-juncs --transcriptome-only","23":"For each sample, Cufflinks (v.2.0.2) was used to quantify External RNA Controls Consortium (ERCC) Spike-In and hg19 transcripts based on reads that mapped to the provided hg19 and ERCC reference files.","24":"Differential expression of genes and transcripts in samples treated with Dex vs. Untreated samples was obtained using Cuffdiff (v.2.0.2) with the quantified transcripts computed by Cufflinks (v.2.0.2), while applying bias correction. Options: -p 12 --min-outlier-p 0.05 -b [hg19_reference_index] -u [hg19_gtf] -L Dex,Control  [List of accepted_hits.bam for each sample sorted by condition (i.e. Dex,Control)]","25":"CummeRbund R package (v.0.1.3) was used to create FPKM matrix of all samples","26":"Genome_build: hg19","27":"Supplementary_files_format_and_content: tab-delimited text files on Series record include FPKM values for each sample (ALL_Sample_FPKM_Matrix.txt) and gene-based differential expression results for Dex vs. Untreated condition (Dex_vs_Untreated_gene_exp.diff)","28":"GPL11154","29":"Blanca,E,Himes","30":"bhimes@pennmedicine.upenn.edu","31":"402 Blockley Hall","32":"Department of Biostatistics, Epidemiology and Informatics","33":"University of Pennsylvania","34":"423 Guardian Dr","35":"Philadelphia","36":"PA","37":"19104","38":"USA","39":"0","40":"Illumina HiSeq 2000","41":"cDNA","42":"transcriptomic","43":"RNA-Seq","44":"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02422670","45":"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX384350","46":"NONE","47":"N052611","48":"airway smooth muscle cells","49":"1","50":"human airway smooth muscle cells","51":"Dexamethasone","52":"GSM1275867","_rn_":"GSM1275867"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="sra-info" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> SRA info</h2>
<p>Download SRA info. To link sample info to info sequencing: Go to
corresponding SRA page and save the information via the “Send to: File
button” This file can also be used to make a script to download
sequencing files from the web. Note that sra files can be converted to
fastq files via the fastq-dump function of the sra-tools.</p>
<p>File is also available on course website.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>sraInfo<span class="ot">&lt;-</span><span class="fu">read.csv</span>(<span class="st">&quot;https://raw.githubusercontent.com/statOmics/SGA/airwaySeqData/SraRunInfo.csv&quot;</span>)</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>sraInfo<span class="sc">$</span>SampleName <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(sraInfo<span class="sc">$</span>SampleName)</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="fu">levels</span>(pdata<span class="sc">$</span>SampleName)<span class="sc">==</span><span class="fu">levels</span>(sraInfo<span class="sc">$</span>SampleName)</span></code></pre></div>
<pre><code>##  [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
## [16] TRUE</code></pre>
<p>SampleNames are can be linked.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>pdata<span class="ot">&lt;-</span><span class="fu">merge</span>(pdata,sraInfo,<span class="at">by=</span><span class="st">&quot;SampleName&quot;</span>)</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>pdata<span class="sc">$</span>Run</span></code></pre></div>
<pre><code>##  [1] &quot;SRR1039508&quot; &quot;SRR1039509&quot; &quot;SRR1039510&quot; &quot;SRR1039511&quot; &quot;SRR1039512&quot;
##  [6] &quot;SRR1039513&quot; &quot;SRR1039514&quot; &quot;SRR1039515&quot; &quot;SRR1039516&quot; &quot;SRR1039517&quot;
## [11] &quot;SRR1039518&quot; &quot;SRR1039519&quot; &quot;SRR1039520&quot; &quot;SRR1039521&quot; &quot;SRR1039522&quot;
## [16] &quot;SRR1039523&quot;</code></pre>
</div>
</div>
<div id="read-featurecounts-object" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Read featurecounts
object</h1>
<p>We import the featurecounts object that we have stored.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>fc <span class="ot">&lt;-</span>  <span class="fu">readRDS</span>(</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>  <span class="fu">url</span>(<span class="st">&quot;https://github.com/statOmics/SGA2020/blob/data-rnaseq/airway/featureCounts/star_featurecounts.rds?raw=true&quot;</span>)</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>  )</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a><span class="fu">colnames</span>(fc<span class="sc">$</span>counts)</span></code></pre></div>
<pre><code>## [1] &quot;SRR1039508&quot; &quot;SRR1039509&quot; &quot;SRR1039512&quot; &quot;SRR1039513&quot; &quot;SRR1039516&quot;
## [6] &quot;SRR1039517&quot; &quot;SRR1039520&quot; &quot;SRR1039521&quot;</code></pre>
<div id="read-meta-data" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Read Meta Data</h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="co">#pdata &lt;- readRDS(&quot;airwayMetaData.rds&quot;)</span></span></code></pre></div>
</div>
</div>
<div id="data-analysis" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Data Analysis</h1>
<div id="setup-count-object-edger" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Setup count object
edgeR</h2>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(fc<span class="sc">$</span>counts)</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="fu">colnames</span>(dge)<span class="sc">==</span>pdata<span class="sc">$</span>Run</span></code></pre></div>
<pre><code>##  [1]  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
## [13] FALSE FALSE FALSE FALSE</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>pdata<span class="sc">$</span>Run<span class="sc">%in%</span><span class="fu">colnames</span>(dge)</span></code></pre></div>
<pre><code>##  [1]  TRUE  TRUE FALSE FALSE  TRUE  TRUE FALSE FALSE  TRUE  TRUE FALSE FALSE
## [13]  TRUE  TRUE FALSE FALSE</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="fu">rownames</span>(pdata) <span class="ot">&lt;-</span> pdata<span class="sc">$</span>Run</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>pdata <span class="ot">&lt;-</span> pdata[<span class="fu">colnames</span>(dge),]</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>pdata[,<span class="fu">grep</span>(<span class="st">&quot;:ch1&quot;</span>,<span class="fu">colnames</span>(pdata))]</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["cell line:ch1"],"name":[1],"type":["chr"],"align":["left"]},{"label":["cell type:ch1"],"name":[2],"type":["chr"],"align":["left"]},{"label":["ercc_mix:ch1"],"name":[3],"type":["chr"],"align":["left"]},{"label":["tissue:ch1"],"name":[4],"type":["chr"],"align":["left"]},{"label":["treatment:ch1"],"name":[5],"type":["chr"],"align":["left"]}],"data":[{"1":"N61311","2":"airway smooth muscle cells","3":"-","4":"human airway smooth muscle cells","5":"Untreated","_rn_":"SRR1039508"},{"1":"N61311","2":"airway smooth muscle cells","3":"-","4":"human airway smooth muscle cells","5":"Dexamethasone","_rn_":"SRR1039509"},{"1":"N052611","2":"airway smooth muscle cells","3":"-","4":"human airway smooth muscle cells","5":"Untreated","_rn_":"SRR1039512"},{"1":"N052611","2":"airway smooth muscle cells","3":"1","4":"human airway smooth muscle cells","5":"Dexamethasone","_rn_":"SRR1039513"},{"1":"N080611","2":"airway smooth muscle cells","3":"-","4":"human airway smooth muscle cells","5":"Untreated","_rn_":"SRR1039516"},{"1":"N080611","2":"airway smooth muscle cells","3":"-","4":"human airway smooth muscle cells","5":"Dexamethasone","_rn_":"SRR1039517"},{"1":"N061011","2":"airway smooth muscle cells","3":"1","4":"human airway smooth muscle cells","5":"Untreated","_rn_":"SRR1039520"},{"1":"N061011","2":"airway smooth muscle cells","3":"1","4":"human airway smooth muscle cells","5":"Dexamethasone","_rn_":"SRR1039521"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>pdata<span class="sc">$</span>cellLine <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(pdata[,<span class="fu">grep</span>(<span class="st">&quot;cell line:ch1&quot;</span>,<span class="fu">colnames</span>(pdata))])</span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>pdata<span class="sc">$</span>treatment <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(pdata[,<span class="fu">grep</span>(<span class="st">&quot;treatment:ch1&quot;</span>,<span class="fu">colnames</span>(pdata))])</span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>pdata<span class="sc">$</span>treatment <span class="ot">&lt;-</span> <span class="fu">relevel</span>(pdata<span class="sc">$</span>treatment,<span class="st">&quot;Untreated&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="fu">colnames</span>(dge) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>  <span class="fu">substr</span>(pdata<span class="sc">$</span>cellLine, <span class="dv">1</span>, <span class="dv">3</span>),</span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>  <span class="st">&quot;_&quot;</span>,</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a>  <span class="fu">substr</span>(pdata<span class="sc">$</span>treatment, <span class="dv">1</span>, <span class="dv">3</span>)</span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a>  )</span></code></pre></div>
</div>
<div id="filtering-and-normalisation" class="section level2"
number="5.2">
<h2><span class="header-section-number">5.2</span> Filtering and
normalisation</h2>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>( <span class="sc">~</span> treatment <span class="sc">+</span> cellLine, <span class="at">data =</span> pdata)</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(dge, design)</span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>dge <span class="ot">&lt;-</span> dge[keep, , keep.lib.sizes<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>dge<span class="ot">&lt;-</span><span class="fu">calcNormFactors</span>(dge)</span></code></pre></div>
</div>
<div id="data-exploration" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Data exploration</h2>
<p>One way to reduce dimensionality is the use of multidimensional
scaling (MDS). For MDS, we first have to calculate all pairwise
distances between our objects (samples in this case), and then create a
(typically) two-dimensional representation where these pre-calculated
distances are represented as accurately as possible. This means that
depending on how the pairwise sample distances are defined, the
two-dimensional plot can be very different, and it is important to
choose a distance that is suitable for the type of data at hand.</p>
<p>edgeR contains a function plotMDS, which operates on a DGEList object
and generates a two-dimensional MDS representation of the samples. The
default distance between two samples can be interpreted as the “typical”
log fold change between the two samples, for the genes that are most
different between them (by default, the top 500 genes, but this can be
modified). We generate an MDS plot from the DGEList object dge, coloring
by the treatment and using different plot symbols for different cell
lines.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="fu">plotMDS</span>(dge, <span class="at">top =</span> <span class="dv">500</span>,<span class="at">col=</span><span class="fu">as.double</span>(pdata<span class="sc">$</span>cellLine))</span></code></pre></div>
<p><img src="airway_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>
<div id="differential-analysis" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> Differential
analysis</h2>
<div id="model" class="section level3" number="5.4.1">
<h3><span class="header-section-number">5.4.1</span> Model</h3>
<p>We first estimate the overdispersion.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(dge, design)</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a><span class="fu">plotBCV</span>(dge)</span></code></pre></div>
<p><img src="airway_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p>Finally, we fit the generalized linear model and perform the test. In
the glmLRT function, we indicate which coefficient (which column in the
design matrix) that we would like to test for. It is possible to test
more general contrasts as well, and the user guide contains many
examples on how to do this. The topTags function extracts the top-ranked
genes. You can indicate the adjusted p-value cutoff, and/or the number
of genes to keep.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(dge, design)</span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a>lrt <span class="ot">&lt;-</span> <span class="fu">glmLRT</span>(fit, <span class="at">coef =</span> <span class="st">&quot;treatmentDexamethasone&quot;</span>)</span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a>ttAll <span class="ot">&lt;-</span> <span class="fu">topTags</span>(lrt, <span class="at">n =</span> <span class="fu">nrow</span>(dge)) <span class="co"># all genes</span></span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a><span class="fu">hist</span>(ttAll<span class="sc">$</span>table<span class="sc">$</span>PValue)</span></code></pre></div>
<p><img src="airway_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>tt <span class="ot">&lt;-</span> <span class="fu">topTags</span>(lrt, <span class="at">n =</span> <span class="fu">nrow</span>(dge), <span class="at">p.value =</span> <span class="fl">0.05</span>) <span class="co"># genes with adj.p&lt;0.05</span></span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a><span class="fu">nrow</span>(tt)</span></code></pre></div>
<pre><code>## [1] 5713</code></pre>
<p>There are 5713 genes statistiscally significant at the 5% FDR
level.</p>
</div>
<div id="plots" class="section level3" number="5.4.2">
<h3><span class="header-section-number">5.4.2</span> Plots</h3>
<p>We first make a volcanoplot and an MA plot.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(ttAll<span class="sc">$</span>table,<span class="fu">aes</span>(<span class="at">x=</span>logFC,<span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(PValue),<span class="at">color=</span>FDR<span class="sc">&lt;</span><span class="fl">0.05</span>)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;black&quot;</span>,<span class="st">&quot;red&quot;</span>))</span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>volcano</span></code></pre></div>
<p><img src="airway_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="fu">plotSmear</span>(lrt, <span class="at">de.tags =</span> <span class="fu">row.names</span>(tt<span class="sc">$</span>table))</span></code></pre></div>
<p><img src="airway_files/figure-html/unnamed-chunk-15-2.png" width="672" /></p>
<p>Another way of representing the results of a differential expression
analysis is to construct a heatmap of the top differentially expressed
genes. Here, we would expect the contrasted sample groups to cluster
separately. A heatmap is a “color coded expression matrix”, where the
rows and columns are clustered using hierarchical clustering. Typically,
it should not be applied to counts, but works better with transformed
values. Here we show how it can be applied to the variance-stabilized
values generated above. We choose the top 30 differentially expressed
genes. There are many functions in R that can generate heatmaps, here we
show the one from the pheatmap package.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="fu">heatmap</span>(<span class="fu">cpm</span>(dge,<span class="at">log=</span><span class="cn">TRUE</span>)[<span class="fu">rownames</span>(tt<span class="sc">$</span>table)[<span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>],])</span></code></pre></div>
<p><img src="airway_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="issues" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Issues</h1>
<p>We find a huge number of DE genes. If this is the case it is always
important to carefully check your results.</p>
<div id="p-values" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> p-values</h2>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>ttAll<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>PValue)) <span class="sc">+</span></span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a>  <span class="fu">geom_histogram</span>()</span></code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="airway_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>We indeed see a large enrichment of small p-values.</p>
<p>We know that the inference is only asymptotically valid, i.e. for
large experiments.</p>
<p>EdgeR is using a likelihood ratio test for statistical inference. The
null distribution of this test statistic follows an asymptotic
chi-square distribution.</p>
<p>If we test for a single contrast (e.g. one model parameter or one
linear combination of model parameters) then the chi-square distribution
has one degree of freedom and you will perform a two-sided test.</p>
</div>
<div id="empirical-null-distribution" class="section level2"
number="6.2">
<h2><span class="header-section-number">6.2</span> Empirical null
distribution</h2>
<p>Efron 2008 relaxes the use of a theoretical null distribution for
assessing statistical significance.</p>
<ul>
<li><p>He shows that we can exploit the massive parallel data structure
of high throughput experiments to estimate the empirical null
distribution.</p></li>
<li><p>He assumes that the test statistics follow a normal distribution
under the null, but proposes to estimate the location and the variance
empirically.</p></li>
</ul>
<p>Assuming that the test statistic has a standard normal null
distribution is not restrictive. For example, suppose that <span
class="math inline">\(t\)</span>-tests have been applied and that the
null distribution is <span class="math inline">\(t_d\)</span>, with
<span class="math inline">\(d\)</span> representing the degrees of
freedom. Let <span class="math inline">\(F_{td}\)</span> denote the
distribution function of <span class="math inline">\(t_d\)</span> and
let <span class="math inline">\(\Phi\)</span> denote the distribution
function of the standard normal distribution. If <span
class="math inline">\(T\)</span> denotes the <span
class="math inline">\(t\)</span>-test statistic, then, under the null
hypothesis, <span class="math display">\[
  T \sim t_d
\]</span> and hence <span class="math display">\[
  F_{td}(T) \sim U[0,1]
\]</span> and <span class="math display">\[
  Z = \Phi^{-1}(F_{td}(T)) \sim N(0,1).
\]</span></p>
<p>If all <span class="math inline">\(m\)</span> null hypotheses are
true, then each of the <span class="math inline">\(Z_g\)</span> is <span
class="math inline">\(N(0,1)\)</span> and the set of <span
class="math inline">\(m\)</span> calculated <span
class="math inline">\(z_g\)</span> test statistics may be considered as
a sample from <span class="math inline">\(N(0,1)\)</span>. Hence, under
these conditions we expect the histogram of the <span
class="math inline">\(m\)</span> <span
class="math inline">\(z_g\)</span>’s to look like the density of the
null distribution.</p>
<p>If a LRT test is conducted for a single contrast, we can obtain the
corresponding <span class="math inline">\(Z_g\)</span> by converting the
p-values into quantiles of the normal, i.e.</p>
<p><span class="math inline">\(z_g = \Phi^{-1} (1-p/2) \times
\text{sign}(FC)\)</span></p>
<p>The normal distribution is very useful for diagnostics because we are
used to work with it.</p>
<p>So in general we</p>
<ol style="list-style-type: decimal">
<li>Start of the two-sided p-values.</li>
<li>We divide the p-values by two to have the probability in one of the
tails</li>
<li>We calculate which quantile of the normal corresponds to it</li>
<li>We give the z-value the sign of our contrast.</li>
</ol>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>ttAll<span class="sc">$</span>table <span class="ot">&lt;-</span> ttAll<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">z =</span> <span class="fu">sign</span>(logFC) <span class="sc">*</span> <span class="fu">abs</span>(<span class="fu">qnorm</span>(PValue<span class="sc">/</span><span class="dv">2</span>)))</span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a></span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a>ttAll<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>z)) <span class="sc">+</span></span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">y =</span> ..density..), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb37-7"><a href="#cb37-7" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun =</span> dnorm,</span>
<span id="cb37-8"><a href="#cb37-8" tabindex="-1"></a>      <span class="at">args =</span> <span class="fu">list</span>(</span>
<span id="cb37-9"><a href="#cb37-9" tabindex="-1"></a>    <span class="at">mean =</span> <span class="dv">0</span>,</span>
<span id="cb37-10"><a href="#cb37-10" tabindex="-1"></a>    <span class="at">sd=</span><span class="dv">1</span>)</span>
<span id="cb37-11"><a href="#cb37-11" tabindex="-1"></a>  )</span></code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="airway_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<p>The histogram clearly shows that the theoretical null distribution
N(0,1) does not seen to match the null distribution in the airway
dataset.</p>
<p>We can empirically estimate the null distribution using the locfdr
package. The locfdr package allows you to calculate local fdrs. It uses
the two groups model:</p>
<p><span class="math display">\[f(z) = \pi_0f_0(z) + (1-\pi_0)
f_1(z)\]</span></p>
<p>The local fdr or posterior probability that the gene with a certain
test statistic z is a false positive then becomes</p>
<p><span class="math display">\[lfdr = \frac{\pi_0 f_0(z)}{f(z)}
\]</span></p>
<p>Efron suggests to use a cutoff for the lfdr of 20%. So we return all
genes for which the probability that they are false positives is below
20%. He claims that this corresponds to a conventional FDR of about 5%
in the set that you return.</p>
<p>It can be shown that the FDR:</p>
<p><span class="math display">\[
FDR = E[lfdr] \approx \frac{\sum\limits_{k \in DE-set}
lfdr_k}{n_\text{DE}}
\]</span></p>
<p>We can also estimate the global FDR by simply calculating the average
lfdr in the set that we return.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="fu">library</span>(locfdr)</span>
<span id="cb39-2"><a href="#cb39-2" tabindex="-1"></a>ttAll<span class="sc">$</span>table <span class="ot">&lt;-</span> ttAll<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb39-3"><a href="#cb39-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">lfdr =</span> <span class="fu">locfdr</span>(z)<span class="sc">$</span>fdr)</span></code></pre></div>
<pre><code>## Warning in locfdr(z): f(z) misfit = 11.1. Rerun with increased df</code></pre>
<p><img src="airway_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<p>In the plot we see the histogram of the z-values.</p>
<ul>
<li>The green line is the estimate of the marginal distribution <span
class="math inline">\(f(z)\)</span>. A non-parameteric density estimator
is used for this purpose.</li>
<li>The blue line is the estimate of the null distribution.</li>
<li>We see that null distribution appears to be much broader than the
standard normal and it shows a slight shift.</li>
<li>We see that the marginal distribution (green distribution) is not
estimated well.</li>
<li>We also get a warning message that indicates that we can increase
the degrees of freedom.</li>
</ul>
<p>We re-estimate <span class="math inline">\(f(z)\)</span> by using
more degrees of freedom, e.g. <code>df = 20</code>.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a>ttAll<span class="sc">$</span>table <span class="ot">&lt;-</span> ttAll<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">lfdr =</span> <span class="fu">locfdr</span>(z, <span class="at">df =</span> <span class="dv">20</span>)<span class="sc">$</span>fdr)</span></code></pre></div>
<p><img src="airway_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<p>Estimated FDR corresponding to the set of genes with lfdr &lt; 0.2
and number of significant genes with lfdr &lt; 0.2?</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a>ttAll<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a>  <span class="fu">filter</span>(lfdr <span class="sc">&lt;</span> <span class="fl">0.2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a>  <span class="fu">summarize</span>(</span>
<span id="cb42-4"><a href="#cb42-4" tabindex="-1"></a>    <span class="at">FDR =</span> <span class="fu">mean</span>(lfdr, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb42-5"><a href="#cb42-5" tabindex="-1"></a>    <span class="at">nSig =</span> lfdr <span class="sc">%&gt;%</span></span>
<span id="cb42-6"><a href="#cb42-6" tabindex="-1"></a>      is.na <span class="sc">%&gt;%</span></span>
<span id="cb42-7"><a href="#cb42-7" tabindex="-1"></a>      <span class="st">`</span><span class="at">!</span><span class="st">`</span> <span class="sc">%&gt;%</span></span>
<span id="cb42-8"><a href="#cb42-8" tabindex="-1"></a>      sum)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["FDR"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["nSig"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"0.04545589","2":"1436"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Note, that</p>
<ul>
<li>the FDR of the set that is returned is indeed close to 0.5.</li>
<li>the number of significant genes has been reduced considerably.</li>
<li>It is very important to evaluate the fit of the null and marginal
distribution in the plot.</li>
</ul>
<p>If you would like to opt to use the FDR method of Benjamini and
Hochberg, you also could convert the z-values back in p-values by using
a normal <span class="math inline">\(N(\hat \mu, \hat
\sigma^2)\)</span>, where we use the estimated parameters for the null
distribution by locfdr.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>h <span class="ot">&lt;-</span> ttAll<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a>  <span class="fu">pull</span>(z) <span class="sc">%&gt;%</span></span>
<span id="cb43-3"><a href="#cb43-3" tabindex="-1"></a>  <span class="fu">locfdr</span>(<span class="at">df =</span> <span class="dv">20</span>, <span class="at">plot=</span><span class="cn">FALSE</span>)</span>
<span id="cb43-4"><a href="#cb43-4" tabindex="-1"></a></span>
<span id="cb43-5"><a href="#cb43-5" tabindex="-1"></a>ttAll<span class="sc">$</span>table <span class="ot">&lt;-</span> ttAll<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb43-6"><a href="#cb43-6" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pEmp =</span> <span class="fu">pnorm</span>(</span>
<span id="cb43-7"><a href="#cb43-7" tabindex="-1"></a>    <span class="fu">abs</span>(z),</span>
<span id="cb43-8"><a href="#cb43-8" tabindex="-1"></a>    <span class="at">mean =</span> h<span class="sc">$</span>fp0[<span class="st">&quot;mlest&quot;</span>,<span class="st">&quot;delta&quot;</span>],</span>
<span id="cb43-9"><a href="#cb43-9" tabindex="-1"></a>    <span class="at">sd =</span> h<span class="sc">$</span>fp0[<span class="st">&quot;mlest&quot;</span>,<span class="st">&quot;sigma&quot;</span>],</span>
<span id="cb43-10"><a href="#cb43-10" tabindex="-1"></a>    <span class="at">lower =</span> <span class="cn">FALSE</span>) <span class="sc">*</span> <span class="dv">2</span></span>
<span id="cb43-11"><a href="#cb43-11" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb43-12"><a href="#cb43-12" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pAdjEmp =</span> <span class="fu">p.adjust</span>(</span>
<span id="cb43-13"><a href="#cb43-13" tabindex="-1"></a>    pEmp,</span>
<span id="cb43-14"><a href="#cb43-14" tabindex="-1"></a>    <span class="at">method=</span><span class="st">&quot;BH&quot;</span>)</span>
<span id="cb43-15"><a href="#cb43-15" tabindex="-1"></a>    )</span>
<span id="cb43-16"><a href="#cb43-16" tabindex="-1"></a></span>
<span id="cb43-17"><a href="#cb43-17" tabindex="-1"></a>ttAll<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb43-18"><a href="#cb43-18" tabindex="-1"></a>  <span class="fu">filter</span>(pAdjEmp <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-19"><a href="#cb43-19" tabindex="-1"></a>  nrow</span></code></pre></div>
<pre><code>## [1] 1500</code></pre>
<p>We observe that the number of significant genes reduces to about the
same amount as with the lfdr method of Efron. We, however, did not have
to use a density estimator for the marginal cumulative distribution
function of the test statistics. Instead, the BH method uses the
empirical distribution of the statistics.</p>
</div>
<div id="assess-why-deviations-from-the-null-occur"
class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Assess why deviations
from the null occur?</h2>
<div id="failed-mathematical-assumptions" class="section level3"
number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Failed mathematical
assumptions?</h3>
<p>If we permute the gene expression data of all observations
differently for every gene, we simulate expressions under the null and
breakup the correlation structure between the genes.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1004</span>)</span>
<span id="cb45-2"><a href="#cb45-2" tabindex="-1"></a>dgePermAll <span class="ot">&lt;-</span> dge</span>
<span id="cb45-3"><a href="#cb45-3" tabindex="-1"></a>dgePermAll<span class="sc">$</span>counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(dge<span class="sc">$</span>counts, <span class="dv">1</span>, sample)  <span class="sc">%&gt;%</span> t</span>
<span id="cb45-4"><a href="#cb45-4" tabindex="-1"></a>dgePermAll<span class="sc">$</span>samples<span class="sc">$</span>lib.size <span class="ot">&lt;-</span>  dgePermAll<span class="sc">$</span>counts <span class="sc">%&gt;%</span> colSums</span>
<span id="cb45-5"><a href="#cb45-5" tabindex="-1"></a>dgePermAll <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dgePermAll)</span>
<span id="cb45-6"><a href="#cb45-6" tabindex="-1"></a></span>
<span id="cb45-7"><a href="#cb45-7" tabindex="-1"></a>dgePermAll <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(dgePermAll, design)</span>
<span id="cb45-8"><a href="#cb45-8" tabindex="-1"></a><span class="fu">plotBCV</span>(dgePermAll)</span></code></pre></div>
<p><img src="airway_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a>fitPermAll <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(dgePermAll, design)</span>
<span id="cb46-2"><a href="#cb46-2" tabindex="-1"></a>lrtPermAll <span class="ot">&lt;-</span> <span class="fu">glmLRT</span>(fitPermAll, <span class="at">coef =</span> <span class="st">&quot;treatmentDexamethasone&quot;</span>)</span>
<span id="cb46-3"><a href="#cb46-3" tabindex="-1"></a>ttAllPermAll <span class="ot">&lt;-</span> <span class="fu">topTags</span>(lrtPermAll, <span class="at">n =</span> <span class="fu">nrow</span>(dgePermAll))</span>
<span id="cb46-4"><a href="#cb46-4" tabindex="-1"></a></span>
<span id="cb46-5"><a href="#cb46-5" tabindex="-1"></a>ttAllPermAll<span class="sc">$</span>table <span class="ot">&lt;-</span> ttAllPermAll<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb46-6"><a href="#cb46-6" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">z =</span> <span class="fu">sign</span>(logFC) <span class="sc">*</span> <span class="fu">abs</span>(<span class="fu">qnorm</span>(PValue<span class="sc">/</span><span class="dv">2</span>)))</span>
<span id="cb46-7"><a href="#cb46-7" tabindex="-1"></a></span>
<span id="cb46-8"><a href="#cb46-8" tabindex="-1"></a>ttAllPermAll<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb46-9"><a href="#cb46-9" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>z)) <span class="sc">+</span></span>
<span id="cb46-10"><a href="#cb46-10" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">y =</span> ..density..), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb46-11"><a href="#cb46-11" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun =</span> dnorm,</span>
<span id="cb46-12"><a href="#cb46-12" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(</span>
<span id="cb46-13"><a href="#cb46-13" tabindex="-1"></a>    <span class="at">mean =</span> <span class="dv">0</span>,</span>
<span id="cb46-14"><a href="#cb46-14" tabindex="-1"></a>    <span class="at">sd=</span><span class="dv">1</span>)</span>
<span id="cb46-15"><a href="#cb46-15" tabindex="-1"></a>  )</span></code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="airway_files/figure-html/unnamed-chunk-23-2.png" width="672" /></p>
<p>We observe that the null distribution is approximately following a
standard normal distribution. There is no shift and the standard
deviation in no longer larger than 1.</p>
<p>Hence, there do not seem to be issues related to failed mathematical
assumptions.</p>
</div>
<div id="correlation-of-the-genes" class="section level3"
number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Correlation of the
genes?</h3>
<p>We will now only permute the samples across conditions, but, we keep
all genes together, e.g. by permuting the design matrix.</p>
<p>This simulates data from the null and breaks the association between
the predictors and potential confounders, however, correlation between
the genes remains.</p>
<p>Here, we have a block design so we will permute within block. We will
perform a balanced permutation. We will select two cellines at random
and we will permute the labels of the treatment.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a>designPerm <span class="ot">&lt;-</span> design</span>
<span id="cb48-2"><a href="#cb48-2" tabindex="-1"></a></span>
<span id="cb48-3"><a href="#cb48-3" tabindex="-1"></a>cellLine <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">2</span>)</span>
<span id="cb48-4"><a href="#cb48-4" tabindex="-1"></a>cellLine</span></code></pre></div>
<pre><code>## [1] 1 2</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> cellLine)</span>
<span id="cb50-2"><a href="#cb50-2" tabindex="-1"></a>  designPerm[i<span class="sc">*</span><span class="dv">2</span> <span class="sc">+</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb50-3"><a href="#cb50-3" tabindex="-1"></a>designPerm</span></code></pre></div>
<pre><code>##            (Intercept) treatmentDexamethasone cellLineN061011 cellLineN080611
## SRR1039508           1                      0               0               0
## SRR1039509           1                      1               0               0
## SRR1039512           1                      1               0               0
## SRR1039513           1                      0               0               0
## SRR1039516           1                      1               0               1
## SRR1039517           1                      0               0               1
## SRR1039520           1                      0               1               0
## SRR1039521           1                      1               1               0
##            cellLineN61311
## SRR1039508              1
## SRR1039509              1
## SRR1039512              0
## SRR1039513              0
## SRR1039516              0
## SRR1039517              0
## SRR1039520              0
## SRR1039521              0
## attr(,&quot;assign&quot;)
## [1] 0 1 2 2 2
## attr(,&quot;contrasts&quot;)
## attr(,&quot;contrasts&quot;)$treatment
## [1] &quot;contr.treatment&quot;
## 
## attr(,&quot;contrasts&quot;)$cellLine
## [1] &quot;contr.treatment&quot;</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" tabindex="-1"></a>fitPermSamp <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(dge, designPerm)</span>
<span id="cb52-2"><a href="#cb52-2" tabindex="-1"></a>lrtPermSamp <span class="ot">&lt;-</span> <span class="fu">glmLRT</span>(fitPermSamp, <span class="at">coef =</span> <span class="st">&quot;treatmentDexamethasone&quot;</span>)</span>
<span id="cb52-3"><a href="#cb52-3" tabindex="-1"></a>ttAllPermSamp <span class="ot">&lt;-</span> <span class="fu">topTags</span>(lrtPermSamp, <span class="at">n =</span> <span class="fu">nrow</span>(dge))</span>
<span id="cb52-4"><a href="#cb52-4" tabindex="-1"></a></span>
<span id="cb52-5"><a href="#cb52-5" tabindex="-1"></a>ttAllPermSamp<span class="sc">$</span>table <span class="ot">&lt;-</span> ttAllPermSamp<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb52-6"><a href="#cb52-6" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">z =</span> <span class="fu">sign</span>(logFC) <span class="sc">*</span> <span class="fu">abs</span>(<span class="fu">qnorm</span>(PValue<span class="sc">/</span><span class="dv">2</span>)))</span>
<span id="cb52-7"><a href="#cb52-7" tabindex="-1"></a></span>
<span id="cb52-8"><a href="#cb52-8" tabindex="-1"></a>ttAllPermSamp<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb52-9"><a href="#cb52-9" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>z)) <span class="sc">+</span></span>
<span id="cb52-10"><a href="#cb52-10" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">y =</span> ..density..), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb52-11"><a href="#cb52-11" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun =</span> dnorm,</span>
<span id="cb52-12"><a href="#cb52-12" tabindex="-1"></a>  <span class="at">args =</span> <span class="fu">list</span>(</span>
<span id="cb52-13"><a href="#cb52-13" tabindex="-1"></a>    <span class="at">mean =</span> <span class="dv">0</span>,</span>
<span id="cb52-14"><a href="#cb52-14" tabindex="-1"></a>    <span class="at">sd=</span><span class="dv">1</span>)</span>
<span id="cb52-15"><a href="#cb52-15" tabindex="-1"></a>  )</span></code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="airway_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<p>We observe that the null distribution is approximately matching with
a standard normal. There is no shift and the standard deviation in no
longer larger than 1. Here, the distribution seems to be a bit more
narrow than the the theoretical distribution.</p>
<div id="repeat-for-other-permutation" class="section level4"
number="6.3.2.1">
<h4><span class="header-section-number">6.3.2.1</span> Repeat for other
permutation:</h4>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" tabindex="-1"></a>designPerm <span class="ot">&lt;-</span> design</span>
<span id="cb54-2"><a href="#cb54-2" tabindex="-1"></a></span>
<span id="cb54-3"><a href="#cb54-3" tabindex="-1"></a>cellLine <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">2</span>)</span>
<span id="cb54-4"><a href="#cb54-4" tabindex="-1"></a>cellLine</span></code></pre></div>
<pre><code>## [1] 0 2</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> cellLine)</span>
<span id="cb56-2"><a href="#cb56-2" tabindex="-1"></a>  designPerm[i<span class="sc">*</span><span class="dv">2</span> <span class="sc">+</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb56-3"><a href="#cb56-3" tabindex="-1"></a>designPerm</span></code></pre></div>
<pre><code>##            (Intercept) treatmentDexamethasone cellLineN061011 cellLineN080611
## SRR1039508           1                      1               0               0
## SRR1039509           1                      0               0               0
## SRR1039512           1                      0               0               0
## SRR1039513           1                      1               0               0
## SRR1039516           1                      1               0               1
## SRR1039517           1                      0               0               1
## SRR1039520           1                      0               1               0
## SRR1039521           1                      1               1               0
##            cellLineN61311
## SRR1039508              1
## SRR1039509              1
## SRR1039512              0
## SRR1039513              0
## SRR1039516              0
## SRR1039517              0
## SRR1039520              0
## SRR1039521              0
## attr(,&quot;assign&quot;)
## [1] 0 1 2 2 2
## attr(,&quot;contrasts&quot;)
## attr(,&quot;contrasts&quot;)$treatment
## [1] &quot;contr.treatment&quot;
## 
## attr(,&quot;contrasts&quot;)$cellLine
## [1] &quot;contr.treatment&quot;</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" tabindex="-1"></a>fitPermSamp <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(dge, designPerm)</span>
<span id="cb58-2"><a href="#cb58-2" tabindex="-1"></a>lrtPermSamp <span class="ot">&lt;-</span> <span class="fu">glmLRT</span>(fitPermSamp, <span class="at">coef =</span> <span class="st">&quot;treatmentDexamethasone&quot;</span>)</span>
<span id="cb58-3"><a href="#cb58-3" tabindex="-1"></a>ttAllPermSamp <span class="ot">&lt;-</span> <span class="fu">topTags</span>(lrtPermSamp, <span class="at">n =</span> <span class="fu">nrow</span>(dge))</span>
<span id="cb58-4"><a href="#cb58-4" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" tabindex="-1"></a>ttAllPermSamp<span class="sc">$</span>table <span class="ot">&lt;-</span> ttAllPermSamp<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb58-6"><a href="#cb58-6" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">z =</span> <span class="fu">sign</span>(logFC) <span class="sc">*</span> <span class="fu">abs</span>(<span class="fu">qnorm</span>(PValue<span class="sc">/</span><span class="dv">2</span>)))</span>
<span id="cb58-7"><a href="#cb58-7" tabindex="-1"></a></span>
<span id="cb58-8"><a href="#cb58-8" tabindex="-1"></a>pPermSamp <span class="ot">&lt;-</span> ttAllPermSamp<span class="sc">$</span>table <span class="sc">%&gt;%</span></span>
<span id="cb58-9"><a href="#cb58-9" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>z)) <span class="sc">+</span></span>
<span id="cb58-10"><a href="#cb58-10" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">y =</span> ..density..), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb58-11"><a href="#cb58-11" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun =</span> dnorm,</span>
<span id="cb58-12"><a href="#cb58-12" tabindex="-1"></a>  <span class="at">args =</span> <span class="fu">list</span>(</span>
<span id="cb58-13"><a href="#cb58-13" tabindex="-1"></a>    <span class="at">mean =</span> <span class="dv">0</span>,</span>
<span id="cb58-14"><a href="#cb58-14" tabindex="-1"></a>    <span class="at">sd=</span><span class="dv">1</span>)</span>
<span id="cb58-15"><a href="#cb58-15" tabindex="-1"></a>  )</span>
<span id="cb58-16"><a href="#cb58-16" tabindex="-1"></a>pPermSamp</span></code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="airway_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" tabindex="-1"></a>pPermSamp <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">4</span>,<span class="dv">4</span>)</span></code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<pre><code>## Warning: Removed 3 rows containing non-finite values (stat_bin).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_bar).</code></pre>
<p><img src="airway_files/figure-html/unnamed-chunk-25-2.png" width="672" /></p>
<p>Both permutations show that there are no deviations from the
theoretical null when we simulate under the null and break potential
confounding but leave correlation between genes intact.</p>
<p>So there seems to be another reason for deviations of the null, than
failed mathematical assumptions and correlation between genes!</p>
<p>Perhaps confounders that we do not account for?</p>
</div>
</div>
</div>
<div id="concluding-remark" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> Concluding
remark</h2>
<p>This example also clearly shows that although permutation methods
offer a simple and attractive way to deal with mathematically
complicated hypothesis testing situations, they often cannot remedy
failures of the theoretical null.</p>
<p>So the use of permutations to establish the null distribution should
also be used with care in high throughput contexts. Indeed, depending on
the permutation scheme that is used you might break correlations between
genes, samples and/or potential confounding.</p>
<p>So even if you use a permutation approach in a high throughput
context, e.g. a wilcoxon rank sum test or other permutation based tests,
you still might need to check for deviations of the null!</p>
</div>
</div>

<div id="rmd-source-code">---
title: "Airway: DE analysis"
author: "Lieven Clement"
date: "statOmics, Ghent University (https://statomics.github.io)"
output:
    html_document:
      code_download: true
      theme: cosmo
      toc: true
      toc_float: true
      highlight: tango
      number_sections: true
---

# Background
The data used in this workflow comes from an RNA-seq experiment where airway smooth muscle cells were treated with dexamethasone, a synthetic glucocorticoid steroid with anti-inflammatory effects (Himes et al. 2014). Glucocorticoids are used, for example, by people with asthma to reduce inflammation of the airways. In the experiment, four human airway smooth muscle cell lines were treated with 1 micromolar dexamethasone for 18 hours. For each of the four cell lines, we have a treated and an untreated sample.
For more description of the experiment see the article, PubMed entry 24926665, and for raw data see the GEO entry GSE52778.

Many parts of this tutorial are based on parts of a published RNA-seq workflow
available via Love et al. 2015 [F1000Research](http://f1000research.com/articles/4-1070) and as a [Bioconductor
package](https://www.bioconductor.org/packages/release/workflows/html/rnaseqGene.html) and on Charlotte Soneson's material from the [bss2019](https://uclouvain-cbio.github.io/BSS2019/rnaseq_gene_summerschool_belgium_2019.html) workshop.

# Data
FastQ files with a small subset of the reads can be found on https://github.com/statOmics/SGA2019/tree/data-rnaseq
Here, we will not use our count table because it is based on a small subset of the reads.
We will use the count table that was provided after mapping all the reads with the read mapper star.


```{r}
library(edgeR)
library(tidyverse)
library(GEOquery)
```

# MetaData 

## GEO data

```{r}
gse<-getGEO("GSE52778")
pdata<-pData(gse[[1]])
pdata$SampleName<-rownames(pdata) %>% as.factor
head(pdata)[1:6,]
```
## SRA info 
Download SRA info. To link sample info to info sequencing: Go to corresponding SRA page and save the information via the “Send to: File button” This file can also be used to make a script to download sequencing files from the web. Note that sra files can be converted to fastq files via the fastq-dump function of the sra-tools.

File is also available on course website. 

```{r}
sraInfo<-read.csv("https://raw.githubusercontent.com/statOmics/SGA/airwaySeqData/SraRunInfo.csv")
sraInfo$SampleName <- as.factor(sraInfo$SampleName)
levels(pdata$SampleName)==levels(sraInfo$SampleName)
```

SampleNames are can be linked.

```{r}
pdata<-merge(pdata,sraInfo,by="SampleName")
pdata$Run
```

# Read featurecounts object

We import the featurecounts object that we have stored.

```{r}
fc <-  readRDS(
  url("https://github.com/statOmics/SGA2020/blob/data-rnaseq/airway/featureCounts/star_featurecounts.rds?raw=true")
  )
colnames(fc$counts)
```

## Read Meta Data

```{r}
#pdata <- readRDS("airwayMetaData.rds")
```

# Data Analysis
## Setup count object edgeR

```{r}
dge <- DGEList(fc$counts)
colnames(dge)==pdata$Run
pdata$Run%in%colnames(dge)
```

```{r}
rownames(pdata) <- pdata$Run
pdata <- pdata[colnames(dge),]
pdata[,grep(":ch1",colnames(pdata))]
```

```{r}
pdata$cellLine <- as.factor(pdata[,grep("cell line:ch1",colnames(pdata))])
pdata$treatment <- as.factor(pdata[,grep("treatment:ch1",colnames(pdata))])
pdata$treatment <- relevel(pdata$treatment,"Untreated")
```


```{r}
colnames(dge) <- paste0(
  substr(pdata$cellLine, 1, 3),
  "_",
  substr(pdata$treatment, 1, 3)
  )
```

## Filtering and normalisation

```{r}
design <- model.matrix( ~ treatment + cellLine, data = pdata)
keep <- filterByExpr(dge, design)
dge <- dge[keep, , keep.lib.sizes=FALSE]
dge<-calcNormFactors(dge)
```




## Data exploration

One way to reduce dimensionality is the use of multidimensional scaling (MDS). For MDS, we first have to calculate all pairwise distances between our objects (samples in this case), and then create a (typically) two-dimensional representation where these pre-calculated distances are represented as accurately as possible. This means that depending on how the pairwise sample distances are defined, the two-dimensional plot can be very different, and it is important to choose a distance that is suitable for the type of data at hand.

edgeR contains a function plotMDS, which operates on a DGEList object and generates a two-dimensional MDS representation of the samples. The default distance between two samples can be interpreted as the "typical" log fold change between the two samples, for the genes that are most different between them (by default, the top 500 genes, but this can be modified). We generate an MDS plot from the DGEList object dge, coloring by the treatment and using different plot symbols for different cell lines.

```{r}
plotMDS(dge, top = 500,col=as.double(pdata$cellLine))
```




## Differential analysis

### Model

We first estimate the overdispersion.

```{r}
dge <- estimateDisp(dge, design)
plotBCV(dge)
```


Finally, we fit the generalized linear model and perform the test. In the glmLRT function, we indicate which coefficient (which column in the design matrix) that we would like to test for. It is possible to test more general contrasts as well, and the user guide contains many examples on how to do this. The topTags function extracts the top-ranked genes. You can indicate the adjusted p-value cutoff, and/or the number of genes to keep.

```{r}
fit <- glmFit(dge, design)
lrt <- glmLRT(fit, coef = "treatmentDexamethasone")
ttAll <- topTags(lrt, n = nrow(dge)) # all genes
hist(ttAll$table$PValue)
tt <- topTags(lrt, n = nrow(dge), p.value = 0.05) # genes with adj.p<0.05
nrow(tt)
```

There are `r nrow(tt)` genes statistiscally significant at the 5% FDR level.

### Plots

We first make a volcanoplot and an MA plot.

```{r}
library(ggplot2)
volcano <- ggplot(ttAll$table,aes(x=logFC,y=-log10(PValue),color=FDR<0.05)) + geom_point() + scale_color_manual(values=c("black","red"))
volcano
plotSmear(lrt, de.tags = row.names(tt$table))
```

Another way of representing the results of a differential expression analysis is to construct a heatmap of the top differentially expressed genes. Here, we would expect the contrasted sample groups to cluster separately. A heatmap is a "color coded expression matrix", where the rows and columns are clustered using hierarchical clustering. Typically, it should not be applied to counts, but works better with transformed values. Here we show how it can be applied to the variance-stabilized values generated above. We choose the top 30 differentially expressed genes. There are many functions in R that can generate heatmaps, here we show the one from the pheatmap package.

```{r}
heatmap(cpm(dge,log=TRUE)[rownames(tt$table)[1:30],])
```

# Issues

We find a huge number of DE genes. If this is the case it is always important to carefully check your results.

## p-values

```{r}
ttAll$table %>%
  ggplot(aes(x=PValue)) +
  geom_histogram()
```

We indeed see a large enrichment of small p-values.

We know that the inference is only asymptotically valid, i.e. for large experiments.

EdgeR is using a likelihood ratio test for statistical inference. The null distribution of this test statistic follows an asymptotic chi-square distribution.

If we test for a single contrast (e.g. one model parameter or one linear combination of model parameters) then the chi-square distribution has one degree of freedom and you will perform a two-sided test.

## Empirical null distribution

Efron 2008 relaxes the use of a theoretical null distribution for assessing statistical significance.

- He shows that we can exploit the massive parallel data structure of high throughput experiments to estimate the empirical null distribution.

- He assumes that the test statistics follow a normal distribution under the null, but proposes to estimate the location and the variance empirically.

Assuming that the test statistic has a standard normal null distribution is not restrictive. For example, suppose that $t$-tests have been applied and that the null distribution is $t_d$, with $d$ representing the degrees of freedom. Let $F_{td}$ denote the distribution function of $t_d$ and let $\Phi$ denote the distribution function of the standard normal distribution. If $T$ denotes the $t$-test statistic, then, under the null hypothesis,
\[
  T \sim t_d
\]
and hence
\[
  F_{td}(T) \sim U[0,1]
\]
and
\[
  Z = \Phi^{-1}(F_{td}(T)) \sim N(0,1).
\]

If all $m$ null hypotheses are true, then each of the $Z_g$ is $N(0,1)$ and the set of $m$ calculated $z_g$ test statistics may be considered as a sample from $N(0,1)$. Hence, under these conditions we expect the histogram of the $m$ $z_g$'s to look like the density of the null distribution.

If a LRT test is conducted for a single contrast, we can obtain the corresponding $Z_g$ by converting the p-values into quantiles of the normal, i.e.

$z_g = \Phi^{-1} (1-p/2) \times \text{sign}(FC)$

The normal distribution is very useful for diagnostics because we are used to work with it.

So in general we

1. Start of the two-sided p-values.
2. We divide the p-values by two to have the probability in one of the tails
3. We calculate which quantile of the normal corresponds to it
4. We give the z-value the sign of our contrast.

```{r}
ttAll$table <- ttAll$table %>%
  mutate(z = sign(logFC) * abs(qnorm(PValue/2)))

ttAll$table %>%
  ggplot(aes(x=z)) +
  geom_histogram(aes(y = ..density..), color = "black") +
  stat_function(fun = dnorm,
      args = list(
    mean = 0,
    sd=1)
  )
```

The histogram clearly shows that the theoretical null distribution N(0,1) does not seen to match the null distribution in the airway dataset.

We can empirically estimate the null distribution using the locfdr package.
The locfdr package allows you to calculate local fdrs.
It uses the two groups model:

\[f(z) = \pi_0f_0(z) + (1-\pi_0) f_1(z)\]

The local fdr or posterior probability that the gene with a certain test statistic z is a false positive then becomes

\[lfdr = \frac{\pi_0 f_0(z)}{f(z)} \]

Efron suggests to use a cutoff for the lfdr of 20%. So we return all genes for which the probability that they are false positives is below 20%. He claims that this corresponds to a conventional FDR of about 5% in the set that you return.

It can be shown that the FDR:

\[
FDR = E[lfdr] \approx \frac{\sum\limits_{k \in DE-set} lfdr_k}{n_\text{DE}}
\]

We can also estimate the global FDR by simply calculating the average lfdr in the set that we return.

```{r}
library(locfdr)
ttAll$table <- ttAll$table %>%
  mutate(lfdr = locfdr(z)$fdr)
```

In the plot we see the histogram of the z-values.

- The green line is the estimate of the marginal distribution $f(z)$. A non-parameteric density estimator is used for this purpose.
- The blue line is the estimate of the null distribution.
- We see that null distribution appears to be much broader than the standard normal and it shows a slight shift.
- We see that the marginal distribution (green distribution) is not estimated well.
- We also get a warning message that indicates that we can increase the degrees of freedom.

We re-estimate $f(z)$ by using more degrees of freedom, e.g. `df = 20`.

```{r}
ttAll$table <- ttAll$table %>%
  mutate(lfdr = locfdr(z, df = 20)$fdr)
```



Estimated FDR corresponding to the set of genes with lfdr < 0.2 and number of significant genes with lfdr < 0.2?

```{r}
ttAll$table %>%
  filter(lfdr < 0.2) %>%
  summarize(
    FDR = mean(lfdr, na.rm = TRUE),
    nSig = lfdr %>%
      is.na %>%
      `!` %>%
      sum)
```

Note, that

- the FDR of the set that is returned is indeed close to 0.5.
- the number of significant genes has been reduced considerably.
- It is very important to evaluate the fit of the null and marginal distribution in the plot.

If you would like to opt to use the FDR method of Benjamini and Hochberg, you also could convert the z-values back in p-values by using a normal $N(\hat \mu, \hat \sigma^2)$, where we use the estimated parameters for the null distribution by locfdr.

```{r}
h <- ttAll$table %>%
  pull(z) %>%
  locfdr(df = 20, plot=FALSE)

ttAll$table <- ttAll$table %>%
  mutate(pEmp = pnorm(
    abs(z),
    mean = h$fp0["mlest","delta"],
    sd = h$fp0["mlest","sigma"],
    lower = FALSE) * 2
    ) %>%
  mutate(pAdjEmp = p.adjust(
    pEmp,
    method="BH")
    )

ttAll$table %>%
  filter(pAdjEmp < 0.05) %>%
  nrow
```

We observe that the number of significant genes reduces to about the same amount as with the lfdr method of Efron.
We, however, did not have to use a density estimator for the marginal cumulative distribution function of the test statistics. Instead, the BH method uses the empirical distribution of the statistics.

##  Assess why deviations from the null occur?

### Failed mathematical assumptions?

If we permute the gene expression data of all observations differently for every gene, we simulate expressions under the null and breakup the correlation structure between the genes.

```{r}
set.seed(1004)
dgePermAll <- dge
dgePermAll$counts <- apply(dge$counts, 1, sample)  %>% t
dgePermAll$samples$lib.size <-  dgePermAll$counts %>% colSums
dgePermAll <- calcNormFactors(dgePermAll)

dgePermAll <- estimateDisp(dgePermAll, design)
plotBCV(dgePermAll)

fitPermAll <- glmFit(dgePermAll, design)
lrtPermAll <- glmLRT(fitPermAll, coef = "treatmentDexamethasone")
ttAllPermAll <- topTags(lrtPermAll, n = nrow(dgePermAll))

ttAllPermAll$table <- ttAllPermAll$table %>%
  mutate(z = sign(logFC) * abs(qnorm(PValue/2)))

ttAllPermAll$table %>%
  ggplot(aes(x=z)) +
  geom_histogram(aes(y = ..density..), color = "black") +
  stat_function(fun = dnorm,
    args = list(
    mean = 0,
    sd=1)
  )
```

We observe that the null distribution is approximately following a standard normal distribution.
There is no shift and the standard deviation in no longer larger than 1.

Hence, there do not seem to be issues related to failed mathematical assumptions.

### Correlation of the genes?

We will now only permute the samples across conditions, but, we keep all genes together, e.g. by permuting the design matrix.

This simulates data from the null and breaks the association between the predictors and potential confounders, however, correlation between the genes remains.

Here, we have a block design so we will permute within block.
We will perform a balanced permutation.
We will select two cellines at random and we will permute the labels of the treatment.

```{r}
designPerm <- design

cellLine <- sample(0:3,2)
cellLine

for (i in cellLine)
  designPerm[i*2 + 1:2, 2] <- c(1,0)
designPerm


fitPermSamp <- glmFit(dge, designPerm)
lrtPermSamp <- glmLRT(fitPermSamp, coef = "treatmentDexamethasone")
ttAllPermSamp <- topTags(lrtPermSamp, n = nrow(dge))

ttAllPermSamp$table <- ttAllPermSamp$table %>%
  mutate(z = sign(logFC) * abs(qnorm(PValue/2)))

ttAllPermSamp$table %>%
  ggplot(aes(x=z)) +
  geom_histogram(aes(y = ..density..), color = "black") +
  stat_function(fun = dnorm,
  args = list(
    mean = 0,
    sd=1)
  )
```

We observe that the null distribution is approximately matching with a standard normal.
There is no shift and the standard deviation in no longer larger than 1.
Here, the distribution seems to be a bit more narrow than the the theoretical distribution.  


#### Repeat for other permutation:

```{r}
designPerm <- design

cellLine <- c(0,2)
cellLine

for (i in cellLine)
  designPerm[i*2 + 1:2, 2] <- c(1,0)
designPerm

fitPermSamp <- glmFit(dge, designPerm)
lrtPermSamp <- glmLRT(fitPermSamp, coef = "treatmentDexamethasone")
ttAllPermSamp <- topTags(lrtPermSamp, n = nrow(dge))

ttAllPermSamp$table <- ttAllPermSamp$table %>%
  mutate(z = sign(logFC) * abs(qnorm(PValue/2)))

pPermSamp <- ttAllPermSamp$table %>%
  ggplot(aes(x=z)) +
  geom_histogram(aes(y = ..density..), color = "black") +
  stat_function(fun = dnorm,
  args = list(
    mean = 0,
    sd=1)
  )
pPermSamp
pPermSamp + xlim(-4,4)
```

Both permutations show that there are no deviations from the theoretical null when we simulate under the null and break potential confounding but leave correlation between genes intact.

So there seems to be another reason for deviations of the null, than failed mathematical assumptions and correlation between genes!

Perhaps confounders that we do not account for?

## Concluding remark

This example also clearly shows that although permutation methods offer a simple and attractive way to deal with mathematically complicated hypothesis testing situations,  they often cannot remedy failures of the theoretical null.

So the use of permutations to establish the null distribution should also be used with care in high throughput contexts. Indeed, depending on the permutation scheme that is used you might break correlations between genes, samples and/or potential confounding.  

So even if you use a permutation approach in a high throughput context, e.g. a wilcoxon rank sum test or other permutation based tests, you still might need to check for deviations of the null!
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("airway.rmd");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
