<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>Elegans: DE analysis</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="intro.html">1.1 Position</a>
    </li>
    <li>
      <a href="recapGeneralLinearModel.html">1.2 Recap linear model</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="docs/martens_proteomics_bioinformatics.pdf">1. Proteomics Bioinformatics</a>
    </li>
    <li>
      <a href="TargetDecoy.html">2.1 Identification - FDR - QC</a>
    </li>
    <li>
      <a href="pda_quantification_preprocessing.html">2.2.1. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">2.2.2. Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">2.3.1. Quantification - Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="pda_blocking_wrapup.html">2.3.2. Wrap-up DA - Blocking</a>
    </li>
    <li>
      <a href="technicalDetailsProteomics.html">2.3.3 Technical Details</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencing_intro.html">1.  Introduction to RNA-seq</a>
    </li>
    <li>
      <a href="sequencing_countData.html">2.  Working with count data and GLMs</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="multipleRegression_KPNA2.html">1 Intro KPNA2</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="https://www.compomics.com/bioinformatics-for-proteomics/">1. Bioinformatics for Proteomics</a>
    </li>
    <li>
      <a href="TargetDecoyTutorial.html">2.1. Target Decoy QC</a>
    </li>
    <li>
      <a href="pda_tutorialPreprocessing.html">2.2. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">2.3. Quantification - Design</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencingTutorial1_mapping.html">1. Read mapping</a>
    </li>
    <li>
      <a href="sequencingTutorial2_DE.html">2. Working with Count Data</a>
    </li>
    <li>
      <a href="scRNA-seq-Tutorial.html">3. Single Cell RNA-seq Data Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Elegans: DE analysis</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date">statOmics, Ghent University (<a
href="https://statomics.github.io"
class="uri">https://statomics.github.io</a>)</h4>

</div>


<div id="background" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Background</h1>
<p>After fertilization but prior to the onset of zygotic transcription,
the C. elegans zygote cleaves asymmetrically to create the anterior AB
and posterior P1 blastomeres, each of which goes on to generate distinct
cell lineages. To understand how patterns of RNA inheritance and
abundance arise after this first asymmetric cell division, we pooled
hand-dissected AB and P1 blastomeres and performed RNA-seq (Study <a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE59943">GSE59943</a>).</p>
<div id="read-alignment-and-count-table" class="section level2"
number="1.1">
<h2><span class="header-section-number">1.1</span> Read alignment and
Count table</h2>
<p>We downloaded the SRA files, converted them to fastQ files and
aligned them using the Hisat2 mapper through the QuasR and RHisat2
package in the script <a
href="./elegansAlignmentCountTable.html">elegansAlignmentCountTable</a>
on the SGA course website.</p>
<p>Next, we import the count table derived from the full fastQ files for
the elegans experiment. The count table can be found on the elegansFastq
branch of the course website.</p>
<p>Note, that you can also work with the count table that you generated
using the small fastQ files that contained 2% of the reads of the full
fastQ files.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>geneCounts <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;https://raw.githubusercontent.com/statOmics/SGA/elegansFastq/elegansCountTableFull.csv&quot;</span>,<span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">head</span>(geneCounts)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["width"],"name":[1],"type":["int"],"align":["right"]},{"label":["SRR1532959"],"name":[2],"type":["int"],"align":["right"]},{"label":["SRR1532960"],"name":[3],"type":["int"],"align":["right"]},{"label":["SRR1532961"],"name":[4],"type":["int"],"align":["right"]},{"label":["SRR1532962"],"name":[5],"type":["int"],"align":["right"]},{"label":["SRR1532963"],"name":[6],"type":["int"],"align":["right"]},{"label":["SRR1532964"],"name":[7],"type":["int"],"align":["right"]}],"data":[{"1":"1806","2":"5983","3":"2771","4":"7089","5":"3641","6":"10083","7":"7498","_rn_":"WBGene00000001"},{"1":"1739","2":"23","3":"1","4":"3","5":"10","6":"14","7":"14","_rn_":"WBGene00000002"},{"1":"1738","2":"86","3":"95","4":"230","5":"152","6":"255","7":"266","_rn_":"WBGene00000003"},{"1":"2027","2":"244","3":"139","4":"50","5":"27","6":"134","7":"110","_rn_":"WBGene00000004"},{"1":"1821","2":"0","3":"0","4":"0","5":"0","6":"0","7":"0","_rn_":"WBGene00000005"},{"1":"1937","2":"1","3":"0","4":"21","5":"2","6":"50","7":"44","_rn_":"WBGene00000006"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Note, that</p>
<ul>
<li><p>The first column is the size of the gene.</p></li>
<li><p>The remaining columns are the count for each fastq
files.</p></li>
<li><p>The column names of the count columns are the names of the SRA
files, which can be used to link them to the experiment.</p></li>
</ul>
</div>
<div id="meta-data" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Meta Data</h2>
<p>The information which connects the sample information from GEO with
the SRA run id is downloaded from <a
href="https://www.ncbi.nlm.nih.gov/sra?term=SRP045110">SRA</a> using the
Send to: File button.</p>
<p><img
src="https://raw.githubusercontent.com/statOmics/SGA/master/images_sequencing/elegans_GSE59943.png" /></p>
<p>Download SRA info to link sample info to info sequencing:</p>
<ol style="list-style-type: decimal">
<li><p>Go to corresponding <a
href="https://www.ncbi.nlm.nih.gov/sra?term=SRP045110">SRA</a> page and
save the information via the “Send to: File button”.</p></li>
<li><p>Select RunInfo!</p></li>
</ol>
<p><img
src="https://raw.githubusercontent.com/statOmics/SGA/master/images_sequencing/elegans_sraInfo.png" />
Note, that this file is already included on the elegansFastq branch of
the course github site.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>sraInfo <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;https://raw.githubusercontent.com/statOmics/SGA/elegansFastq/SraRunInfo.csv&quot;</span>)</span></code></pre></div>
<div id="read-background-experiment" class="section level3"
number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Read background
experiment</h3>
<p>Via the GEOquery package we can access the meta data from experiments
that are deposited to GEO.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>    <span class="fu">library</span>( <span class="st">&quot;GEOquery&quot;</span> )</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>    })</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>gse <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="st">&quot;GSE59943&quot;</span>)</span></code></pre></div>
<pre><code>## Found 2 file(s)</code></pre>
<pre><code>## GSE59943-GPL13657_series_matrix.txt.gz</code></pre>
<pre><code>## GSE59943-GPL9269_series_matrix.txt.gz</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">length</span>(gse)</span></code></pre></div>
<pre><code>## [1] 2</code></pre>
<p>There are two objects because there were runs with two different
machines. Combine the data from both files and add sample name column in
order to be able to link the info to that from SRA names in the count
table.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>pdata <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">pData</span>(gse[[<span class="dv">1</span>]]),<span class="fu">pData</span>(gse[[<span class="dv">2</span>]]))</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>pdata<span class="sc">$</span>SampleName <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pdata)</span></code></pre></div>
</div>
<div
id="combine-experiment-metadata-with-metadata-on-the-sra-sequencing-files"
class="section level3" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Combine experiment
metadata with metadata on the SRA sequencing files</h3>
<p>The SampleName column that we made in the pdata object can be used to
combine the metadata from SRA files to that of GEO experiment data.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>sraInfo<span class="sc">$</span>SampleName</span></code></pre></div>
<pre><code>## [1] &quot;GSM1462555&quot; &quot;GSM1462556&quot; &quot;GSM1462557&quot; &quot;GSM1462558&quot; &quot;GSM1462559&quot;
## [6] &quot;GSM1462560&quot;</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>pdata<span class="sc">$</span>SampleName</span></code></pre></div>
<pre><code>## [1] &quot;GSM1462557&quot; &quot;GSM1462558&quot; &quot;GSM1462559&quot; &quot;GSM1462560&quot; &quot;GSM1462555&quot;
## [6] &quot;GSM1462556&quot;</code></pre>
<p>The order is different, but we can use merge to combine them
correctly.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>pdata <span class="ot">&lt;-</span> <span class="fu">merge</span>(pdata,sraInfo,<span class="at">by=</span><span class="st">&quot;SampleName&quot;</span>)</span></code></pre></div>
<p>The run is also the name of the SRA file so we will be able to link
alignment file name to the experiment via the SRA file info.</p>
</div>
</div>
</div>
<div id="differential-analysis" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Differential
analysis</h1>
<div id="preprocessing" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Preprocessing</h2>
<div id="setup-count-object-edger" class="section level3"
number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Setup count object
edgeR</h3>
<ul>
<li>First column of geneCounts is the size of the gene, so we will not
use it to setup the count object for edgeR.</li>
</ul>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>    <span class="fu">library</span>(edgeR)</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>})</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(geneCounts[,<span class="sc">-</span><span class="dv">1</span>])</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="fu">cbind</span>(pdata<span class="sc">$</span>Run,<span class="fu">colnames</span>(dge))</span></code></pre></div>
<pre><code>##      [,1]         [,2]        
## [1,] &quot;SRR1532959&quot; &quot;SRR1532959&quot;
## [2,] &quot;SRR1532960&quot; &quot;SRR1532960&quot;
## [3,] &quot;SRR1532961&quot; &quot;SRR1532961&quot;
## [4,] &quot;SRR1532962&quot; &quot;SRR1532962&quot;
## [5,] &quot;SRR1532963&quot; &quot;SRR1532963&quot;
## [6,] &quot;SRR1532964&quot; &quot;SRR1532964&quot;</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>pdata<span class="sc">$</span>Run<span class="sc">==</span><span class="fu">colnames</span>(dge)</span></code></pre></div>
<pre><code>## [1] TRUE TRUE TRUE TRUE TRUE TRUE</code></pre>
<ul>
<li><p>The order of the samples in the metadata and count table is the
same</p></li>
<li><p>We replace the name of the SRA file with the title of the
experiment, which is more informative</p></li>
</ul>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">colnames</span>(dge) <span class="ot">&lt;-</span> pdata<span class="sc">$</span>title</span></code></pre></div>
<ul>
<li>The pdata contains many variables with long names. We extract rename
the relevant data and convert it into a factor.</li>
</ul>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>pdata <span class="ot">&lt;-</span> pdata <span class="sc">%&gt;%</span> </span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">cellType =</span> <span class="st">`</span><span class="at">cell type:ch1</span><span class="st">`</span>,</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a>                <span class="at">rep =</span> <span class="st">`</span><span class="at">replicate:ch1</span><span class="st">`</span>, </span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a>                <span class="at">model =</span> Model) <span class="sc">%&gt;%</span> </span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a>        <span class="at">cellType =</span> <span class="fu">as.factor</span>(cellType),</span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a>        <span class="at">rep =</span> <span class="fu">as.factor</span>(rep), </span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a>        <span class="at">model =</span> <span class="fu">as.factor</span>(model)</span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a>        )</span></code></pre></div>
</div>
<div id="filtering" class="section level3" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Filtering</h3>
<p>We typically filter out lowly abundant genes. Note, that the
filtering is independent from the downstream analysis.</p>
<p>The main rationale is to keeps genes that have at least min.count
reads in a worthwhile number samples. The latter is derived from the
design matrix.</p>
<p>Indeed, genes with many zeros do not contain a lot of information and
the DE analysis is typically underpowered for these genes.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>cellType, pdata)</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(dge, design)</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>dge <span class="ot">&lt;-</span> dge[keep, ,keep.lib.sizes<span class="ot">=</span><span class="cn">FALSE</span>]</span></code></pre></div>
<p>The option keep.lib.sizes=FALSE causes the library sizes to be
recomputed after the filtering. This is generally recommended, although
the effect on the downstream analysis is usually small.</p>
</div>
<div
id="normalisation-to-correct-for-differences-in-effective-library-size"
class="section level3" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> Normalisation to
correct for differences in effective library size</h3>
<p>The normalisation factors/offsets have to be calculated upon
filtering.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dge)</span></code></pre></div>
</div>
</div>
<div id="data-exploration" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Data exploration</h2>
<p>One way to reduce dimensionality is the use of multidimensional
scaling (MDS). For MDS, we first have to calculate all pairwise
distances between our objects (samples in this case), and then create a
(typically) two-dimensional representation where these pre-calculated
distances are represented as accurately as possible. This means that
depending on how the pairwise sample distances are defined, the
two-dimensional plot can be very different, and it is important to
choose a distance that is suitable for the type of data at hand.</p>
<p>edgeR contains a function plotMDS, which operates on a DGEList object
and generates a two-dimensional MDS representation of the samples. The
default distance between two samples can be interpreted as the “typical”
log fold change between the two samples, for the genes that are most
different between them (by default, the top 500 genes, but this can be
modified). We generate an MDS plot from the DGEList object dge, coloring
by the treatment and using different plot symbols for different cell
lines.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="fu">plotMDS</span>(dge, <span class="at">top =</span> <span class="dv">500</span>, <span class="at">col=</span><span class="fu">as.double</span>(pdata<span class="sc">$</span>cellType))</span></code></pre></div>
<p><img src="elegansDE_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<p>The cell types of the same repeat seem to line up.</p>
</div>
<div id="modeling" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Modeling</h2>
<div id="estimation-of-the-dispersion" class="section level3"
number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Estimation of the
dispersion</h3>
<p>We first estimate the overdispersion.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(dge, design)</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a><span class="fu">plotBCV</span>(dge)</span></code></pre></div>
<p><img src="elegansDE_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
<div id="fitting-and-inference" class="section level3" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Fitting and
inference</h3>
<p>Finally, we fit the generalized linear model and perform the test. In
the glmLRT function, we indicate which coefficient (which column in the
design matrix) that we would like to test for. It is possible to test
more general contrasts as well, and the user guide contains many
examples on how to do this. The topTags function extracts the top-ranked
genes. You can indicate the adjusted p-value cutoff, and/or the number
of genes to keep.</p>
<p>Here we keep all genes in the toptable:
<code>n = nrow(dge)</code></p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(dge, design)</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>lrt <span class="ot">&lt;-</span> <span class="fu">glmLRT</span>(fit, <span class="at">coef =</span> <span class="st">&quot;cellTypeP1&quot;</span>)</span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>ttAll <span class="ot">&lt;-</span><span class="fu">topTags</span>(lrt, <span class="at">n =</span> <span class="fu">nrow</span>(dge)) <span class="co"># all genes</span></span></code></pre></div>
<p>Note, that the workflow with glmFit and glmLRT is no longer the
default workflow of edgeR. We will discuss this when we focus on the
technical aspects of differential analysis of RNASeq data.</p>
</div>
</div>
<div id="plots" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Plots</h2>
<div id="p-values" class="section level3" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> P-values</h3>
<p>We first assess the p-values.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="fu">qplot</span>(ttAll<span class="sc">$</span>table<span class="sc">$</span>PValue,<span class="at">geom =</span> <span class="st">&quot;histogram&quot;</span>, <span class="at">binwidth=</span>.<span class="dv">05</span>,<span class="at">center=</span><span class="fl">0.025</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;pvalue&quot;</span>)</span></code></pre></div>
<p><img src="elegansDE_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
</div>
<div id="volcano-and-ma-plot" class="section level3" number="2.4.2">
<h3><span class="header-section-number">2.4.2</span> Volcano and MA
plot</h3>
<p>Next we make a volcano plot and an MA plot.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>volcano<span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a>    ttAll<span class="sc">$</span>table,</span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x=</span>logFC,<span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(PValue),</span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a>    <span class="at">color=</span>FDR<span class="sc">&lt;</span><span class="fl">0.05</span>)) <span class="sc">+</span> </span>
<span id="cb28-6"><a href="#cb28-6" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;black&quot;</span>,<span class="st">&quot;red&quot;</span>))</span>
<span id="cb28-7"><a href="#cb28-7" tabindex="-1"></a>volcano</span></code></pre></div>
<p><img src="elegansDE_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="fu">plotSmear</span>(lrt, <span class="at">de.tags =</span> <span class="fu">row.names</span>(ttAll<span class="sc">$</span>table <span class="sc">|&gt;</span> <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>)))</span></code></pre></div>
<p><img src="elegansDE_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
<div id="heatmap" class="section level3" number="2.4.3">
<h3><span class="header-section-number">2.4.3</span> Heatmap</h3>
<p>Another way of representing the results of a differential expression
analysis is to construct a heatmap of the top differentially expressed
genes. Here, we would expect the contrasted sample groups to cluster
separately. A heatmap is a “color coded expression matrix”, where the
rows and columns are clustered using hierarchical clustering. Typically,
it should not be applied to counts, but works better with transformed
values. Here we show how it can be applied to log CPM values.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>sigNames <span class="ot">&lt;-</span> <span class="fu">row.names</span>(ttAll<span class="sc">$</span>table <span class="sc">|&gt;</span> <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a><span class="fu">heatmap</span>(<span class="fu">cpm</span>(dge)[sigNames, ])</span></code></pre></div>
<p><img src="elegansDE_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
</div>
</div>
<div id="list-of-significant-genes" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> List of significant
genes</h2>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a>ttAll<span class="sc">$</span>table <span class="sc">|&gt;</span> <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["logFC"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["logCPM"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["LR"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["PValue"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["FDR"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"-1.8853337","2":"7.4922772","3":"76.85618","4":"1.838716e-18","5":"1.642525e-14","_rn_":"WBGene00017985"},{"1":"2.2292087","2":"7.0981125","3":"72.03442","4":"2.114760e-17","5":"9.445575e-14","_rn_":"WBGene00011409"},{"1":"-1.9873271","2":"6.8091004","3":"65.84952","4":"4.866996e-16","5":"1.449229e-12","_rn_":"WBGene00016824"},{"1":"-1.4756944","2":"7.5425057","3":"49.97596","4":"1.556410e-12","5":"3.475854e-09","_rn_":"WBGene00000388"},{"1":"1.7655610","2":"6.6071887","3":"48.18348","4":"3.881445e-12","5":"6.934589e-09","_rn_":"WBGene00019811"},{"1":"2.6014948","2":"6.1722312","3":"46.95434","4":"7.266009e-12","5":"1.081788e-08","_rn_":"WBGene00000496"},{"1":"2.7614417","2":"5.7436043","3":"46.33210","4":"9.981537e-12","5":"1.273787e-08","_rn_":"WBGene00021200"},{"1":"-1.5143579","2":"6.2564460","3":"45.22149","4":"1.759637e-11","5":"1.964854e-08","_rn_":"WBGene00008080"},{"1":"-1.4305444","2":"9.8593894","3":"44.23609","4":"2.910653e-11","5":"2.686532e-08","_rn_":"WBGene00020948"},{"1":"1.4752729","2":"5.7337673","3":"44.17207","4":"3.007425e-11","5":"2.686532e-08","_rn_":"WBGene00019001"},{"1":"1.9544416","2":"4.6064156","3":"38.92522","4":"4.403548e-10","5":"3.576082e-07","_rn_":"WBGene00001402"},{"1":"3.0122317","2":"3.3073046","3":"37.94819","4":"7.264847e-10","5":"5.408073e-07","_rn_":"WBGene00010354"},{"1":"-1.8380784","2":"7.0076492","3":"37.29512","4":"1.015386e-09","5":"6.977262e-07","_rn_":"WBGene00009606"},{"1":"1.7051669","2":"4.9517029","3":"36.53337","4":"1.500747e-09","5":"9.575841e-07","_rn_":"WBGene00017397"},{"1":"1.3040646","2":"7.7419921","3":"34.72438","4":"3.798446e-09","5":"2.262101e-06","_rn_":"WBGene00009264"},{"1":"-1.4134302","2":"6.8549569","3":"32.80072","4":"1.021071e-08","5":"5.700766e-06","_rn_":"WBGene00008823"},{"1":"1.5911204","2":"4.7696158","3":"32.04028","4":"1.510087e-08","5":"7.935062e-06","_rn_":"WBGene00015515"},{"1":"-2.1783192","2":"5.9476085","3":"31.45727","4":"2.038783e-08","5":"1.011803e-05","_rn_":"WBGene00021009"},{"1":"-1.2943705","2":"6.9623012","3":"30.74838","4":"2.937489e-08","5":"1.381084e-05","_rn_":"WBGene00000863"},{"1":"3.0490767","2":"1.6346424","3":"30.43605","4":"3.450563e-08","5":"1.541194e-05","_rn_":"WBGene00010781"},{"1":"1.2152727","2":"6.7824844","3":"30.25010","4":"3.797722e-08","5":"1.599752e-05","_rn_":"WBGene00020721"},{"1":"1.1919907","2":"7.6667953","3":"30.17885","4":"3.939835e-08","5":"1.599752e-05","_rn_":"WBGene00003994"},{"1":"1.1572793","2":"7.5292340","3":"29.30523","4":"6.182863e-08","5":"2.401370e-05","_rn_":"WBGene00015463"},{"1":"1.1718527","2":"8.3127487","3":"28.88039","4":"7.698851e-08","5":"2.862546e-05","_rn_":"WBGene00006562"},{"1":"-2.3443518","2":"6.7481522","3":"28.80337","4":"8.011155e-08","5":"2.862546e-05","_rn_":"WBGene00001333"},{"1":"1.9268358","2":"3.4878715","3":"28.60665","4":"8.867720e-08","5":"3.046744e-05","_rn_":"WBGene00018226"},{"1":"1.3125678","2":"8.2021029","3":"27.97514","4":"1.228841e-07","5":"4.065644e-05","_rn_":"WBGene00004374"},{"1":"-1.3278863","2":"6.0908644","3":"27.38847","4":"1.664203e-07","5":"5.309403e-05","_rn_":"WBGene00008464"},{"1":"-1.2551457","2":"5.5539793","3":"27.13670","4":"1.895649e-07","5":"5.839253e-05","_rn_":"WBGene00016812"},{"1":"1.7116881","2":"5.3308108","3":"26.60919","4":"2.490566e-07","5":"7.416076e-05","_rn_":"WBGene00010036"},{"1":"-1.6348069","2":"6.8153959","3":"26.06410","4":"3.302675e-07","5":"9.517031e-05","_rn_":"WBGene00016823"},{"1":"-1.6935250","2":"7.7264978","3":"26.00059","4":"3.413138e-07","5":"9.527987e-05","_rn_":"WBGene00002276"},{"1":"2.0530718","2":"4.0410905","3":"25.70142","4":"3.985339e-07","5":"1.065737e-04","_rn_":"WBGene00011225"},{"1":"-1.3783384","2":"4.9173541","3":"25.66735","4":"4.056313e-07","5":"1.065737e-04","_rn_":"WBGene00011883"},{"1":"-1.3711871","2":"7.5017504","3":"25.00433","4":"5.720170e-07","5":"1.459951e-04","_rn_":"WBGene00009921"},{"1":"1.5672996","2":"5.2838982","3":"24.49173","4":"7.462929e-07","5":"1.851843e-04","_rn_":"WBGene00005026"},{"1":"1.1291281","2":"6.6355774","3":"24.41419","4":"7.769451e-07","5":"1.875797e-04","_rn_":"WBGene00010048"},{"1":"-1.3306954","2":"7.5041501","3":"23.99400","4":"9.663642e-07","5":"2.271719e-04","_rn_":"WBGene00016955"},{"1":"-1.2900209","2":"6.5018352","3":"23.93363","4":"9.971467e-07","5":"2.283977e-04","_rn_":"WBGene00009793"},{"1":"2.3654574","2":"2.8724714","3":"23.80329","4":"1.066999e-06","5":"2.295164e-04","_rn_":"WBGene00005022"},{"1":"2.0361775","2":"2.2136529","3":"23.79704","4":"1.070469e-06","5":"2.295164e-04","_rn_":"WBGene00009940"},{"1":"-1.3498348","2":"5.1925632","3":"23.78157","4":"1.079110e-06","5":"2.295164e-04","_rn_":"WBGene00019792"},{"1":"-1.1941469","2":"9.9686588","3":"23.44589","4":"1.284768e-06","5":"2.669031e-04","_rn_":"WBGene00020375"},{"1":"1.8208718","2":"6.9659976","3":"22.58462","4":"2.010877e-06","5":"4.082536e-04","_rn_":"WBGene00000550"},{"1":"1.3024385","2":"4.9814307","3":"22.48027","4":"2.123134e-06","5":"4.214657e-04","_rn_":"WBGene00021051"},{"1":"-1.1805968","2":"5.4314800","3":"22.02920","4":"2.685344e-06","5":"5.214821e-04","_rn_":"WBGene00017037"},{"1":"1.7821228","2":"3.6066346","3":"21.78002","4":"3.057681e-06","5":"5.811545e-04","_rn_":"WBGene00009035"},{"1":"1.0782208","2":"6.8456108","3":"21.67220","4":"3.234451e-06","5":"6.019447e-04","_rn_":"WBGene00004258"},{"1":"1.4021624","2":"7.0221847","3":"21.51095","4":"3.518133e-06","5":"6.413772e-04","_rn_":"WBGene00010678"},{"1":"1.1446989","2":"5.9570823","3":"21.30070","4":"3.925879e-06","5":"6.987966e-04","_rn_":"WBGene00011543"},{"1":"-1.1097233","2":"6.2447362","3":"21.26986","4":"3.989547e-06","5":"6.987966e-04","_rn_":"WBGene00000872"},{"1":"1.2142320","2":"6.7972708","3":"20.95086","4":"4.712151e-06","5":"8.094932e-04","_rn_":"WBGene00019890"},{"1":"-1.3123657","2":"4.6077608","3":"20.85051","4":"4.965597e-06","5":"8.369372e-04","_rn_":"WBGene00020771"},{"1":"-1.1115249","2":"7.3893687","3":"20.53020","4":"5.869780e-06","5":"9.710138e-04","_rn_":"WBGene00018009"},{"1":"-1.0512383","2":"9.3194062","3":"20.44382","4":"6.140761e-06","5":"9.973713e-04","_rn_":"WBGene00008825"},{"1":"1.3710920","2":"3.3982199","3":"20.06350","4":"7.491298e-06","5":"1.194996e-03","_rn_":"WBGene00004758"},{"1":"1.1079082","2":"5.0304081","3":"19.97639","4":"7.840426e-06","5":"1.208103e-03","_rn_":"WBGene00022172"},{"1":"0.9545922","2":"6.9395005","3":"19.94418","4":"7.973620e-06","5":"1.208103e-03","_rn_":"WBGene00007145"},{"1":"-1.0865604","2":"8.4355718","3":"19.94285","4":"7.979189e-06","5":"1.208103e-03","_rn_":"WBGene00008789"},{"1":"1.6572219","2":"5.0135823","3":"19.88086","4":"8.242128e-06","5":"1.227116e-03","_rn_":"WBGene00013765"},{"1":"2.1859759","2":"4.2606032","3":"19.55651","4":"9.766755e-06","5":"1.424859e-03","_rn_":"WBGene00011350"},{"1":"-1.0133818","2":"7.9299902","3":"19.53268","4":"9.889318e-06","5":"1.424859e-03","_rn_":"WBGene00020269"},{"1":"1.0345902","2":"5.7669892","3":"19.30600","4":"1.113558e-05","5":"1.578955e-03","_rn_":"WBGene00019892"},{"1":"-1.8204015","2":"3.2090880","3":"19.15716","4":"1.203848e-05","5":"1.680309e-03","_rn_":"WBGene00010011"},{"1":"-0.9640113","2":"10.5763952","3":"18.99211","4":"1.312602e-05","5":"1.803918e-03","_rn_":"WBGene00003229"},{"1":"-1.1028236","2":"5.5928069","3":"18.93628","4":"1.351579e-05","5":"1.829341e-03","_rn_":"WBGene00006805"},{"1":"-1.2249389","2":"7.3689375","3":"18.43063","4":"1.762030e-05","5":"2.349286e-03","_rn_":"WBGene00000090"},{"1":"0.8986511","2":"6.6888512","3":"17.92457","4":"2.298342e-05","5":"2.976147e-03","_rn_":"WBGene00021440"},{"1":"1.5665800","2":"3.6719713","3":"17.92417","4":"2.298826e-05","5":"2.976147e-03","_rn_":"WBGene00007720"},{"1":"-0.9489161","2":"7.9003232","3":"17.88513","4":"2.346474e-05","5":"2.994436e-03","_rn_":"WBGene00021467"},{"1":"1.4851524","2":"3.1331767","3":"17.46977","4":"2.919119e-05","5":"3.651822e-03","_rn_":"WBGene00012466"},{"1":"1.0171973","2":"7.0879031","3":"17.45405","4":"2.943369e-05","5":"3.651822e-03","_rn_":"WBGene00005019"},{"1":"-0.9005820","2":"6.3470880","3":"17.20264","4":"3.359696e-05","5":"4.111256e-03","_rn_":"WBGene00013529"},{"1":"-1.0641584","2":"9.3056486","3":"17.00192","4":"3.734207e-05","5":"4.454453e-03","_rn_":"WBGene00000146"},{"1":"1.2301606","2":"3.7286626","3":"16.99903","4":"3.739886e-05","5":"4.454453e-03","_rn_":"WBGene00013381"},{"1":"1.0654648","2":"5.1458634","3":"16.78101","4":"4.195114e-05","5":"4.930915e-03","_rn_":"WBGene00008118"},{"1":"1.1539282","2":"5.1474232","3":"16.51512","4":"4.826366e-05","5":"5.599211e-03","_rn_":"WBGene00011739"},{"1":"1.4632904","2":"3.6046399","3":"16.33305","4":"5.312913e-05","5":"6.062664e-03","_rn_":"WBGene00004094"},{"1":"-0.8248154","2":"8.5210781","3":"16.31577","4":"5.361586e-05","5":"6.062664e-03","_rn_":"WBGene00001979"},{"1":"1.6307016","2":"1.8801528","3":"16.20032","4":"5.698439e-05","5":"6.363020e-03","_rn_":"WBGene00016029"},{"1":"-0.9609502","2":"8.5981875","3":"16.15626","4":"5.832544e-05","5":"6.432360e-03","_rn_":"WBGene00017263"},{"1":"1.0048294","2":"7.3146308","3":"16.09054","4":"6.038495e-05","5":"6.578277e-03","_rn_":"WBGene00011775"},{"1":"1.3258751","2":"6.5365285","3":"15.97785","4":"6.408779e-05","5":"6.897545e-03","_rn_":"WBGene00004268"},{"1":"1.0610509","2":"6.2772396","3":"15.92808","4":"6.579524e-05","5":"6.997010e-03","_rn_":"WBGene00000537"},{"1":"1.2284771","2":"6.2743102","3":"15.68850","4":"7.467687e-05","5":"7.842131e-03","_rn_":"WBGene00011986"},{"1":"-0.8757970","2":"6.5596228","3":"15.66782","4":"7.549795e-05","5":"7.842131e-03","_rn_":"WBGene00013018"},{"1":"1.0398237","2":"7.8645347","3":"15.58777","4":"7.876235e-05","5":"8.087173e-03","_rn_":"WBGene00012716"},{"1":"1.1668300","2":"5.1305778","3":"15.55716","4":"8.004783e-05","5":"8.125765e-03","_rn_":"WBGene00001566"},{"1":"-1.1465407","2":"8.0051409","3":"15.51852","4":"8.170081e-05","5":"8.200375e-03","_rn_":"WBGene00008775"},{"1":"-0.9673504","2":"6.2685864","3":"15.41289","4":"8.639681e-05","5":"8.575364e-03","_rn_":"WBGene00004855"},{"1":"1.3278444","2":"5.9949255","3":"15.25407","4":"9.397411e-05","5":"9.224953e-03","_rn_":"WBGene00012031"},{"1":"-1.4585992","2":"6.2050503","3":"15.14299","4":"9.966757e-05","5":"9.677505e-03","_rn_":"WBGene00011153"},{"1":"-0.8969616","2":"6.4054566","3":"14.93494","4":"1.112830e-04","5":"1.068915e-02","_rn_":"WBGene00011272"},{"1":"1.0963649","2":"6.7945075","3":"14.67157","4":"1.279619e-04","5":"1.212116e-02","_rn_":"WBGene00000585"},{"1":"-1.0229393","2":"7.5652884","3":"14.65773","4":"1.289052e-04","5":"1.212116e-02","_rn_":"WBGene00008219"},{"1":"0.9545945","2":"5.9295732","3":"14.36672","4":"1.504376e-04","5":"1.387735e-02","_rn_":"WBGene00006934"},{"1":"0.9140934","2":"6.3585500","3":"14.36358","4":"1.506888e-04","5":"1.387735e-02","_rn_":"WBGene00020447"},{"1":"-0.9177007","2":"5.8454985","3":"14.17672","4":"1.664166e-04","5":"1.516938e-02","_rn_":"WBGene00016486"},{"1":"1.1770960","2":"3.5214621","3":"14.02000","4":"1.808769e-04","5":"1.632094e-02","_rn_":"WBGene00000109"},{"1":"1.5389219","2":"3.2075200","3":"13.99361","4":"1.834334e-04","5":"1.638610e-02","_rn_":"WBGene00012934"},{"1":"0.9868191","2":"6.0864838","3":"13.80067","4":"2.032643e-04","5":"1.797782e-02","_rn_":"WBGene00010194"},{"1":"0.7570550","2":"7.0558261","3":"13.61886","4":"2.239250e-04","5":"1.944753e-02","_rn_":"WBGene00008410"},{"1":"1.1882130","2":"4.7082276","3":"13.61626","4":"2.242354e-04","5":"1.944753e-02","_rn_":"WBGene00003026"},{"1":"0.7851022","2":"7.6895327","3":"13.57970","4":"2.286453e-04","5":"1.963932e-02","_rn_":"WBGene00009559"},{"1":"1.2707751","2":"4.4922962","3":"13.51016","4":"2.372755e-04","5":"2.018650e-02","_rn_":"WBGene00019710"},{"1":"2.9450946","2":"-1.2638701","3":"13.24787","4":"2.728899e-04","5":"2.299741e-02","_rn_":"WBGene00008213"},{"1":"-1.0244819","2":"6.9642407","3":"13.20305","4":"2.794940e-04","5":"2.333383e-02","_rn_":"WBGene00001949"},{"1":"0.8412346","2":"6.7587375","3":"13.18254","4":"2.825692e-04","5":"2.337214e-02","_rn_":"WBGene00003898"},{"1":"-1.0220495","2":"6.1593009","3":"13.13524","4":"2.897925e-04","5":"2.374969e-02","_rn_":"WBGene00000161"},{"1":"-1.0728550","2":"3.7681124","3":"12.90992","4":"3.268453e-04","5":"2.631263e-02","_rn_":"WBGene00012608"},{"1":"0.8273536","2":"6.2945923","3":"12.90928","4":"3.269565e-04","5":"2.631263e-02","_rn_":"WBGene00008877"},{"1":"0.9535130","2":"5.0537616","3":"12.85700","4":"3.362183e-04","5":"2.681641e-02","_rn_":"WBGene00021789"},{"1":"0.8672632","2":"5.4571993","3":"12.76289","4":"3.535648e-04","5":"2.795039e-02","_rn_":"WBGene00004044"},{"1":"-1.1250092","2":"5.5869671","3":"12.20310","4":"4.771017e-04","5":"3.706635e-02","_rn_":"WBGene00022534"},{"1":"-0.7682383","2":"6.7950152","3":"12.20176","4":"4.774437e-04","5":"3.706635e-02","_rn_":"WBGene00021784"},{"1":"1.3861158","2":"3.2631954","3":"12.18665","4":"4.813273e-04","5":"3.706635e-02","_rn_":"WBGene00000259"},{"1":"1.1684310","2":"6.3353337","3":"12.16222","4":"4.876738e-04","5":"3.723410e-02","_rn_":"WBGene00010061"},{"1":"3.8059975","2":"0.5936852","3":"12.13887","4":"4.938177e-04","5":"3.738367e-02","_rn_":"WBGene00020854"},{"1":"-0.7088707","2":"6.9000039","3":"12.02796","4":"5.240842e-04","5":"3.934155e-02","_rn_":"WBGene00016632"},{"1":"-0.7082242","2":"7.1684688","3":"11.99981","4":"5.320600e-04","5":"3.960743e-02","_rn_":"WBGene00021352"},{"1":"-1.2440097","2":"4.0556002","3":"11.81975","4":"5.860585e-04","5":"4.326662e-02","_rn_":"WBGene00020970"},{"1":"-0.9571997","2":"6.5720045","3":"11.59494","4":"6.613137e-04","5":"4.842226e-02","_rn_":"WBGene00001493"},{"1":"0.7716360","2":"5.9172730","3":"11.54440","4":"6.795380e-04","5":"4.935214e-02","_rn_":"WBGene00019836"},{"1":"-0.7316735","2":"6.3922338","3":"11.52798","4":"6.855665e-04","5":"4.938843e-02","_rn_":"WBGene00016583"},{"1":"-1.2419152","2":"2.4264359","3":"11.49803","4":"6.966998e-04","5":"4.978895e-02","_rn_":"WBGene00185001"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="what-did-we-forgot-to-account-for-in-the-data-analysis"
class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> What did we forgot to
account for in the data analysis?</h2>
<p>Assess the MDS plot and the first figure in the paper that published
the data (Osborne et all, 2013, <a
href="https://doi.org/10.1371/journal.pgen.1005117">DOI:
10.1371/journal.pgen.1005117</a>).</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;https://europepmc.org/articles/PMC4395330/bin/pgen.1005117.g001.jpg&quot;</span>)</span></code></pre></div>
<div class="float">
<img
src="https://europepmc.org/articles/PMC4395330/bin/pgen.1005117.g001.jpg"
alt="Differential transcript abundance in AB and P1 blastomeres following the first embryonic division. (Source: Osborne et all, 2013, DOI: 10.1371/journal.pgen.1005117)" />
<div class="figcaption">Differential transcript abundance in AB and P1
blastomeres following the first embryonic division. (Source: Osborne et
all, 2013, DOI: 10.1371/journal.pgen.1005117)</div>
</div>
<p>Which source of variability is not included in the analysis and how
could we account for this? Try to adjust the script accordingly.</p>
</div>
</div>
<div id="session-info" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Session Info</h1>
<p>With respect to reproducibility, it is highly recommended to include
a session info in your script so that readers of your output can see
your particular setup of R.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.2.3 (2023-03-15)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 22.04.4 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so
## 
## locale:
##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   
## 
## attached base packages:
## [1] stats     graphics  grDevices datasets  utils     methods   base     
## 
## other attached packages:
##  [1] edgeR_3.38.4        limma_3.52.1        GEOquery_2.64.2    
##  [4] Biobase_2.56.0      BiocGenerics_0.42.0 R.utils_2.11.0     
##  [7] R.oo_1.24.0         R.methodsS3_1.8.1   forcats_0.5.1      
## [10] stringr_1.4.1       dplyr_1.0.9         purrr_0.3.4        
## [13] readr_2.1.2         tidyr_1.2.0         tibble_3.1.7       
## [16] ggplot2_3.3.6       tidyverse_1.3.2    
## 
## loaded via a namespace (and not attached):
##  [1] httr_1.4.3          sass_0.4.1          splines_4.2.3      
##  [4] jsonlite_1.8.0      modelr_0.1.8        bslib_0.3.1        
##  [7] assertthat_0.2.1    highr_0.9           BiocManager_1.30.18
## [10] renv_0.15.4         googlesheets4_1.0.0 cellranger_1.1.0   
## [13] yaml_2.3.5          pillar_1.7.0        backports_1.4.1    
## [16] lattice_0.20-45     glue_1.6.2          digest_0.6.29      
## [19] rvest_1.0.2         colorspace_2.0-3    htmltools_0.5.2    
## [22] pkgconfig_2.0.3     broom_0.8.0         haven_2.5.0        
## [25] scales_1.2.0        tzdb_0.3.0          googledrive_2.0.0  
## [28] farver_2.1.0        generics_0.1.2      ellipsis_0.3.2     
## [31] withr_2.5.0         cli_3.3.0           magrittr_2.0.3     
## [34] crayon_1.5.1        readxl_1.4.0        evaluate_0.16      
## [37] fs_1.5.2            fansi_1.0.3         xml2_1.3.3         
## [40] tools_4.2.3         data.table_1.14.2   hms_1.1.1          
## [43] gargle_1.2.0        lifecycle_1.0.1     munsell_0.5.0      
## [46] reprex_2.0.1        locfit_1.5-9.5      compiler_4.2.3     
## [49] jquerylib_0.1.4     rlang_1.0.2         grid_4.2.3         
## [52] labeling_0.4.2      rmarkdown_2.14      gtable_0.3.0       
## [55] DBI_1.1.2           curl_4.3.2          R6_2.5.1           
## [58] lubridate_1.8.0     knitr_1.40.1        fastmap_1.1.0      
## [61] utf8_1.2.2          stringi_1.7.8       Rcpp_1.0.8.3       
## [64] vctrs_0.4.1         dbplyr_2.1.1        tidyselect_1.1.2   
## [67] xfun_0.33</code></pre>
</div>

<div id="rmd-source-code">---
title: "Elegans: DE analysis"
author: "Lieven Clement"
date: "statOmics, Ghent University (https://statomics.github.io)"
output:
    html_document:
      code_download: true
      theme: flatly
      toc: true
      toc_float: true
      highlight: tango
      number_sections: true
linkcolor: blue
urlcolor: blue
citecolor: blue
---

```{r, echo=FALSE}
suppressPackageStartupMessages({
  library(tidyverse)
  library(R.utils)
})
```
# Background 

After fertilization but prior to the onset of zygotic transcription, the C. elegans zygote cleaves asymmetrically to create the anterior AB and posterior P1 blastomeres, each of which goes on to generate distinct cell lineages. To understand how patterns of RNA inheritance and abundance arise after this first asymmetric cell division, we pooled hand-dissected AB and P1 blastomeres and performed RNA-seq (Study [GSE59943](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE59943)).



## Read alignment and Count table

We downloaded the SRA files, converted them to fastQ files and aligned them using the Hisat2 mapper through the QuasR and RHisat2 package in the script [elegansAlignmentCountTable](./elegansAlignmentCountTable.html) on the SGA course website.

Next, we import the count table derived from the full fastQ files for the elegans experiment. The count table can be found on the elegansFastq branch of the course website. 

Note, that you can also work with the count table that you generated using the small fastQ files that contained 2% of the reads of the full fastQ files. 

```{r}
geneCounts <- read.csv("https://raw.githubusercontent.com/statOmics/SGA/elegansFastq/elegansCountTableFull.csv",row.names = 1)
head(geneCounts)
```

Note, that 

- The first column is the size of the gene. 

- The remaining columns are the count for each fastq files.

- The column names of the count columns are the names of the SRA files, which can be used to link them to the experiment.  

## Meta Data

The information which connects the sample information from GEO with the SRA run id is downloaded from [SRA](https://www.ncbi.nlm.nih.gov/sra?term=SRP045110) using the Send to: File button.

![](https://raw.githubusercontent.com/statOmics/SGA/master/images_sequencing/elegans_GSE59943.png)

Download SRA info to link sample info to info sequencing: 

1. Go to corresponding [SRA](https://www.ncbi.nlm.nih.gov/sra?term=SRP045110) page and save the information via the "Send to: File button". 

2. Select RunInfo! 

![](https://raw.githubusercontent.com/statOmics/SGA/master/images_sequencing/elegans_sraInfo.png)
Note, that this file is already included on the elegansFastq branch of the course github site. 

```{r}
sraInfo <- read.csv("https://raw.githubusercontent.com/statOmics/SGA/elegansFastq/SraRunInfo.csv")
```


### Read background experiment

Via the GEOquery package we can access the meta data from experiments that are deposited to GEO. 

```{r}
suppressPackageStartupMessages({
    library( "GEOquery" )
    })
gse <- getGEO("GSE59943")
length(gse)
```

There are two objects because there were runs with two different machines. 
Combine the data from both files and add sample name column in order to be able to link the info to that from SRA names in the count table.

```{r}
pdata <- rbind(pData(gse[[1]]),pData(gse[[2]]))
pdata$SampleName <- rownames(pdata)
```

### Combine experiment metadata with metadata on the SRA sequencing files

The SampleName column that we made in the pdata object can be used to combine the metadata from SRA files to that of GEO experiment data. 

```{r}
sraInfo$SampleName
pdata$SampleName
```

The order is different, but we can use merge to combine them correctly. 

```{r}
pdata <- merge(pdata,sraInfo,by="SampleName")
```

The run is also the name of the SRA file so we will be able to link alignment file name to the experiment via the SRA file info.

# Differential analysis

## Preprocessing 

### Setup count object edgeR

- First column of geneCounts is the size of the gene, so we will not use it to setup the count object for edgeR. 

```{r}
suppressPackageStartupMessages({
    library(edgeR)
})
dge <- DGEList(geneCounts[,-1])
```

```{r}
cbind(pdata$Run,colnames(dge))
```

```{r}
pdata$Run==colnames(dge)
```

- The order of the samples in the metadata and count table is the same

- We replace the name of the SRA file with the title of the experiment, which is more informative

```{r}
colnames(dge) <- pdata$title
```

- The pdata contains many variables with long names. We extract rename the relevant data and convert it into a factor.

```{r}
pdata <- pdata %>% 
  dplyr::rename(cellType = `cell type:ch1`,
                rep = `replicate:ch1`, 
                model = Model) %>% 
  mutate(
        cellType = as.factor(cellType),
        rep = as.factor(rep), 
        model = as.factor(model)
        )
```


### Filtering

We typically filter out lowly abundant genes. 
Note, that the filtering is independent from the downstream analysis. 

The main rationale is to keeps genes that have at least min.count reads in a worthwhile number samples. The latter is derived from the design matrix. 

Indeed, genes with many zeros do not contain a lot of information and the DE analysis is typically underpowered for these genes. 

```{r}
design <- model.matrix(~cellType, pdata)
keep <- filterByExpr(dge, design)
dge <- dge[keep, ,keep.lib.sizes=FALSE]
```

The option keep.lib.sizes=FALSE causes the library sizes to be recomputed after the filtering. This is generally recommended, although the effect on the downstream analysis is usually small.

### Normalisation to correct for differences in effective library size

The normalisation factors/offsets have to be calculated upon filtering.

```{r}
dge <- calcNormFactors(dge)
```

## Data exploration 

One way to reduce dimensionality is the use of multidimensional scaling (MDS). For MDS, we first have to calculate all pairwise distances between our objects (samples in this case), and then create a (typically) two-dimensional representation where these pre-calculated distances are represented as accurately as possible. This means that depending on how the pairwise sample distances are defined, the two-dimensional plot can be very different, and it is important to choose a distance that is suitable for the type of data at hand.

edgeR contains a function plotMDS, which operates on a DGEList object and generates a two-dimensional MDS representation of the samples. The default distance between two samples can be interpreted as the "typical" log fold change between the two samples, for the genes that are most different between them (by default, the top 500 genes, but this can be modified). We generate an MDS plot from the DGEList object dge, coloring by the treatment and using different plot symbols for different cell lines.

```{r}
plotMDS(dge, top = 500, col=as.double(pdata$cellType))
```

The cell types of the same repeat seem to line up. 

## Modeling

### Estimation of the dispersion

We first estimate the overdispersion. 

```{r}
dge <- estimateDisp(dge, design)
plotBCV(dge)
```

### Fitting and inference 

Finally, we fit the generalized linear model and perform the test. In the glmLRT function, we indicate which coefficient (which column in the design matrix) that we would like to test for. It is possible to test more general contrasts as well, and the user guide contains many examples on how to do this. The topTags function extracts the top-ranked genes. You can indicate the adjusted p-value cutoff, and/or the number of genes to keep.

Here we keep all genes in the toptable: `n = nrow(dge)`

```{r}
fit <- glmFit(dge, design)
lrt <- glmLRT(fit, coef = "cellTypeP1")
ttAll <-topTags(lrt, n = nrow(dge)) # all genes
```

Note, that the workflow with glmFit and glmLRT is no longer the default workflow of edgeR. We will discuss this when we focus on the technical aspects of differential analysis of RNASeq data.

## Plots

### P-values

We first assess the p-values. 

```{r}
qplot(ttAll$table$PValue,geom = "histogram", binwidth=.05,center=0.025) + xlab("pvalue")
```

### Volcano and MA plot

Next we make a volcano plot and an MA plot.

```{r}
library(ggplot2)
volcano<- ggplot(
    ttAll$table,
    aes(x=logFC,y=-log10(PValue),
    color=FDR<0.05)) + 
    geom_point() + scale_color_manual(values=c("black","red"))
volcano
```

```{r}
plotSmear(lrt, de.tags = row.names(ttAll$table |> filter(FDR < 0.05)))
```

### Heatmap 

Another way of representing the results of a differential expression analysis is to construct a heatmap of the top differentially expressed genes. Here, we would expect the contrasted sample groups to cluster separately. A heatmap is a "color coded expression matrix", where the rows and columns are clustered using hierarchical clustering. Typically, it should not be applied to counts, but works better with transformed values. Here we show how it can be applied to log CPM values. 

```{r}
sigNames <- row.names(ttAll$table |> filter(FDR < 0.05))
heatmap(cpm(dge)[sigNames, ])
```


## List of significant genes

```{r}
ttAll$table |> filter(FDR < 0.05)
```

## What did we forgot to account for in the data analysis?

Assess the MDS plot and the first figure in the paper that published the data (Osborne et all, 2013, [DOI: 10.1371/journal.pgen.1005117](https://doi.org/10.1371/journal.pgen.1005117)).

```{r fig.cap='Differential transcript abundance in AB and P1 blastomeres following the first embryonic division. (Source: Osborne et all, 2013, DOI: 10.1371/journal.pgen.1005117)'}
knitr::include_graphics("https://europepmc.org/articles/PMC4395330/bin/pgen.1005117.g001.jpg")
```


Which source of variability is not included in the analysis and how could we account for this? Try to adjust the script accordingly.

# Session Info

With respect to reproducibility, it is highly recommended to include a session info in your script so that readers of your output can see your particular setup of R. 

```{r}
sessionInfo()
```

</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("elegansDE.Rmd");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
