<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Statistical Genomics Analysis 2023</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>










<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="intro.html">1.1 Position</a>
    </li>
    <li>
      <a href="recapGeneralLinearModel.html">1.2 Recap linear model</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="docs/martens_proteomics_bioinformatics.pdf">1. Proteomics Bioinformatics</a>
    </li>
    <li>
      <a href="TargetDecoy.html">2.1 Identification - FDR - QC</a>
    </li>
    <li>
      <a href="pda_quantification_preprocessing.html">2.2.1. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">2.2.2. Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">2.3.1. Quantification - Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="pda_blocking_wrapup.html">2.3.2. Wrap-up DA - Blocking</a>
    </li>
    <li>
      <a href="technicalDetailsProteomics.html">2.3.3 Technical Details</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencing_intro.html">1.  Introduction to RNA-seq</a>
    </li>
    <li>
      <a href="sequencing_countData.html">2.  Working with count data and GLMs</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="multipleRegression_KPNA2.html">1 Intro KPNA2</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="https://www.compomics.com/bioinformatics-for-proteomics/">1. Bioinformatics for Proteomics</a>
    </li>
    <li>
      <a href="TargetDecoyTutorial.html">2.1. Target Decoy QC</a>
    </li>
    <li>
      <a href="pda_tutorialPreprocessing.html">2.2. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">2.3. Quantification - Design</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencingTutorial1_mapping.html">1. Read mapping</a>
    </li>
    <li>
      <a href="sequencingTutorial2_DE.html">2. Working with Count Data</a>
    </li>
    <li>
      <a href="scRNA-seq-Tutorial.html">3. Single Cell RNA-seq Data Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Statistical Genomics Analysis 2023</h1>

</div>


<p><img src="figures/wpGraph.jpeg" width="986" /></p>
<div id="course-description" class="section level1">
<h1>Course Description</h1>
<p>High-throughput ’omics studies generate ever larger datasets and, as
a consequence, complex data interpretation challenges. This course
focusses on statistical concepts involved in preprocessing,
quantification and differential analysis of high-throughput ’omics data.
The core focus will be on shotgun proteomics and (bulk and single-cell)
RNA-sequencing. Experimental design is essential to allow for correct
interpretation in all ’omics studies, and we will cover how to design a
statistically efficient experiment, as well as discuss the impact
experimental design has on how we model ’omics data, introducing
concepts such as blocking. The course will rely exclusively on free and
user-friendly open-source tools in R/Bioconductor. We hope that this
will provide a solid basis for beginners, but will also bring new
perspectives to those already familiar with standard data analysis
workflows for proteomics and next-generation sequencing
applications.</p>
</div>
<div id="target-audience" class="section level1">
<h1>Target Audience</h1>
<p>This course is oriented towards biologists and bioinformaticians with
a particular interest in differential analysis for quantitative ’omics
data.</p>
</div>
<div id="prerequisites" class="section level1">
<h1>Prerequisites</h1>
<p>The prerequisites for the Statistical Genomics Analysis course are
the successful completion of a basic course of statistics that covers
topics on data exploration and descriptive statistics, statistical
modeling, and inference: linear models, confidence intervals, t-tests,
F-tests, anova, chi-squared test. The basis concepts may be revisited in
the online course at <a href="https://statomics.github.io/PSLS/"
class="uri">https://statomics.github.io/PSLS/</a> (English) and in <a
href="https://statomics.github.io/sbc/"
class="uri">https://statomics.github.io/sbc/</a> (Dutch).</p>
<p>In addition, knowledge of programming in <code>R</code> is preferred.
A primer to <code>R</code> and Data visualization in <code>R</code> can
be found at:</p>
<ul>
<li><code>R</code> Basics: <a
href="https://dodona.ugent.be/nl/courses/335/"
class="uri">https://dodona.ugent.be/nl/courses/335/</a></li>
<li><code>R</code> Data Exploration: <a
href="https://dodona.ugent.be/nl/courses/345/"
class="uri">https://dodona.ugent.be/nl/courses/345/</a></li>
</ul>
</div>
<div id="software" class="section level1">
<h1>Software</h1>
<ul>
<li><p>Participants are required to bring their own laptop with <a
href="https://www.r-project.org/">R</a> version 4.2 or greater.</p></li>
<li><p>We also recommend to also install the latest version of <a
href="https://www.rstudio.com/products/rstudio/download/">RStudio</a>.</p></li>
<li><p>Installation script: to install all required packages, please
copy and paste this line of code in your R console.</p></li>
</ul>
<pre><code>source(&quot;https://raw.githubusercontent.com/statOmics/SGA/master/install.R&quot;)</code></pre>
</div>
<div id="detailed-program" class="section level1">
<h1>Detailed Program</h1>
<div id="introduction" class="section level2">
<h2>1. Introduction</h2>
<p>1.1. Position of the course: <a href="./intro.html">HTML</a></p>
<p>1.2. Recap Linear Models (Week 1)</p>
<ul>
<li>Lecture: <a href="./recapGeneralLinearModel.html">HTML</a>, <a
href="./recapGeneralLinearModel.pdf">PDF</a></li>
<li>Tutorial KPNA2: <a href="./multipleRegression_KPNA2.html">HTML</a>,
<a href="./multipleRegression_KPNA2.pdf">PDF</a></li>
<li>link to a compressed figures directory: <a
href="https://raw.githubusercontent.com/statOmics/SGA/master/figures.tar.gz">download
figures</a></li>
</ul>
</div>
<div id="module-i-proteomics-data-analysis-week-1-5"
class="section level2">
<h2>Module I: Proteomics Data Analysis (Week 1-5)</h2>
<div id="bioinformatics-for-proteomics" class="section level3">
<h3>1. Bioinformatics for proteomics</h3>
<ul>
<li>Lecture: <a
href="./docs/martens_proteomics_bioinformatics.pdf">PDF</a>, <a
href="https://www.youtube.com/watch?v=ZgwNWRul98o">youtube</a></li>
<li>Tutorials: <a
href="https://www.compomics.com/bioinformatics-for-proteomics/">identification</a></li>
</ul>
</div>
<div id="statistics-for-proteomics-data-analysis"
class="section level3">
<h3>2. Statistics for Proteomics Data Analysis</h3>
<div id="identification" class="section level4">
<h4>2.1. Identification</h4>
<ul>
<li><a href="TargetDecoy.html">Target Decoy QC</a></li>
<li><a href="TargetDecoyTutorial.html">Tutorial</a>, <a
href="mzIDMsgfSwissprotExample.html">example</a></li>
<li>Solution: <a href="mzIDuniprotTutorial.html">Uniprot search</a>, <a
href="mzIDswissprotTutorial.html">Swissprot search</a></li>
</ul>
</div>
<div
id="preprocessing-analysis-of-label-free-quantitative-proteomics-experiments-with-simple-designs"
class="section level4">
<h4>2.2. Preprocessing &amp; Analysis of Label Free Quantitative
Proteomics Experiments with Simple Designs</h4>
<ul>
<li><p>Lecture: <a
href="./pda_quantification_preprocessing.html">html</a>, <a
href="./pda_quantification_preprocessing.pdf">pdf</a></p></li>
<li><p>Tutorial: <a
href="./pda_tutorialPreprocessing.html">preprocessing</a></p></li>
<li><p>Wrap-up Peptide-based Models: <a
href="./pda_robustSummarisation_peptideModels.html">html</a>, <a
href="./pda_robustSummarisation_peptideModels.pdf">pdf</a></p></li>
<li><p>Solutions: <a href="cancer3x3.html">cancer3x3</a>, <a
href="cancer6x6.html">cancer6x6</a>, <a
href="cancer9x9.html">cancer9x9</a></p></li>
</ul>
</div>
<div
id="statistical-inference-analysis-of-experiments-with-factorial-designs"
class="section level4">
<h4>2.3. Statistical Inference &amp; Analysis of Experiments with
Factorial Designs</h4>
<ul>
<li><p>Lecture Differential Abundance Analysis: <a
href="./pda_quantification_inference.html">html</a>, <a
href="./pda_quantification_inference.pdf">pdf</a></p></li>
<li><p>Tutorial: <a href="pda_tutorialDesign.html">Design</a></p></li>
<li><p>Wrap-up Blocking: <a href="./pda_blocking_wrapup.html">html</a>,
<a href="./pda_blocking_wrapup.pdf">pdf</a></p></li>
<li><p>Solution: <a href="heartMainInteraction.html">Heart
Example</a></p></li>
<li><p>Technical Details: <a
href="technicalDetailsProteomics.html">Technical Details</a></p></li>
</ul>
</div>
</div>
</div>
<div id="module-ii-next-generation-sequencing-week-6-12"
class="section level2">
<h2>Module II: Next-generation Sequencing (Week 6-12)</h2>
<div
id="introduction-to-sequencing-technology-qc-read-mapping-and-count-table"
class="section level3">
<h3>1. Introduction to sequencing technology, QC, read mapping and count
table</h3>
<ul>
<li>Lecture Introduction to Sequencing: <a
href="./sequencing_intro.html">html</a>,<a
href="./sequencing_intro.pdf">pdf</a></li>
<li>Tutorial: <a href="./sequencingTutorial1_mapping.html">handout</a>,
<a href="./docs/elegansAlignmentCountTable.html">C. elegans
Example</a></li>
</ul>
</div>
<div id="introduction-to-count-data-and-glms" class="section level3">
<h3>2. Introduction to count data and GLMs</h3>
<ul>
<li>Lecture Working with count data and GLMs: <a
href="./sequencing_countData.html">html</a>, <a
href="./sequencing_countData.pdf">pdf</a></li>
<li>Tutorial: <a href="sequencingTutorial2_DE.html">handout</a>, <a
href="./elegansDE.html">C. elegans example</a></li>
</ul>
</div>
<div id="technical-details-on-rna-seq-de-analysis"
class="section level3">
<h3>3. Technical details on RNA-seq DE analysis</h3>
<ul>
<li>Lecture: <a href="./sequencing_technicalDE.html">html</a>, <a
href="./sequencing_technicalDE.pdf">pdf</a></li>
</ul>
</div>
<div id="de-analysis-starting-from-transcript-level-counts"
class="section level3">
<h3>4. DE analysis starting from transcript level counts</h3>
<ul>
<li><a href="https://f1000research.com/articles/4-1521/v2">Soneson et
al. 2016</a></li>
<li>Tutorial: <a href="./sequencingTutorial3_DE.html">handout</a></li>
</ul>
</div>
<div id="solutions-bulk-rna-seq-data-analysis" class="section level3">
<h3>5. Solutions Bulk RNA-seq Data Analysis</h3>
<ul>
<li>airway example: <a href="./airway.html">html</a></li>
<li>Parathyroid Example (quasi-edgeR): <a
href="./parathyroid.html">html</a>, <a
href="./parathyroid.pdf">pdf</a></li>
<li>airway transcript quantification: <a
href="./airway_salmon_edgeR.html">html</a></li>
</ul>
</div>
<div id="introduction-to-single-cell-transcriptomics-scrna-seq"
class="section level3">
<h3>6. Introduction to Single Cell Transcriptomics (scRNA-seq)</h3>
<ul>
<li>Lecture Intro to scRNA-seq: <a
href="./sequencing_singleCell.html">html</a>, <a
href="./sequencing_singleCell.pdf">pdf</a></li>
<li>Advanced topics: Inference for multi-cell-multi-patient experiments
&amp; transcript usage <a
href="./docs/statgen_presentation_Jeroen.pdf">pdf</a></li>
<li>Tutorial on scRNA-seq: <a
href="./scRNA-seq-Tutorial.html">handout</a></li>
<li>Tutorial on differential transcript usage: <a
href="./dtuTutorial.html">handout</a></li>
</ul>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
