<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>Parathyroid: DE analysis</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="intro.html">1.1 Position</a>
    </li>
    <li>
      <a href="recapGeneralLinearModel.html">1.2 Recap linear model</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="docs/martens_proteomics_bioinformatics.pdf">1. Proteomics Bioinformatics</a>
    </li>
    <li>
      <a href="TargetDecoy.html">2.1 Identification - FDR - QC</a>
    </li>
    <li>
      <a href="pda_quantification_preprocessing.html">2.2.1. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">2.2.2. Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">2.3.1. Quantification - Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="pda_blocking_wrapup.html">2.3.2. Wrap-up DA - Blocking</a>
    </li>
    <li>
      <a href="technicalDetailsProteomics.html">2.3.3 Technical Details</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencing_intro.html">1.  Introduction to RNA-seq</a>
    </li>
    <li>
      <a href="sequencing_countData.html">2.  Working with count data and GLMs</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="multipleRegression_KPNA2.html">1 Intro KPNA2</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="https://www.compomics.com/bioinformatics-for-proteomics/">1. Bioinformatics for Proteomics</a>
    </li>
    <li>
      <a href="TargetDecoyTutorial.html">2.1. Target Decoy QC</a>
    </li>
    <li>
      <a href="pda_tutorialPreprocessing.html">2.2. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">2.3. Quantification - Design</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencingTutorial1_mapping.html">1. Read mapping</a>
    </li>
    <li>
      <a href="sequencingTutorial2_DE.html">2. Working with Count Data</a>
    </li>
    <li>
      <a href="scRNA-seq-Tutorial.html">3. Single Cell RNA-seq Data Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Parathyroid: DE analysis</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date">statOmics, Ghent University (<a
href="https://statomics.github.io"
class="uri">https://statomics.github.io</a>)</h4>

</div>


<pre><code>## &#39;getOption(&quot;repos&quot;)&#39; replaces Bioconductor standard repositories, see
## &#39;?repositories&#39; for details
## 
## replacement repositories:
##     CRAN: https://cran.rstudio.com</code></pre>
<pre><code>## Bioconductor version 3.15 (BiocManager 1.30.18), R 4.2.3 (2023-03-15)</code></pre>
<pre><code>## Installing package(s) &#39;parathyroidSE&#39;</code></pre>
<pre><code>## Installation paths not writeable, unable to update packages
##   path: /opt/R/4.2.3/lib/R/library
##   packages:
##     class, KernSmooth, lattice, rpart, spatial</code></pre>
<pre><code>## Old packages: &#39;abind&#39;, &#39;askpass&#39;, &#39;backports&#39;, &#39;BH&#39;, &#39;BiocManager&#39;,
##   &#39;BiocParallel&#39;, &#39;Biostrings&#39;, &#39;bit&#39;, &#39;bitops&#39;, &#39;blob&#39;, &#39;bookdown&#39;, &#39;boot&#39;,
##   &#39;brio&#39;, &#39;broom&#39;, &#39;bslib&#39;, &#39;cachem&#39;, &#39;callr&#39;, &#39;car&#39;, &#39;cli&#39;, &#39;clue&#39;, &#39;cluster&#39;,
##   &#39;codetools&#39;, &#39;colorspace&#39;, &#39;commonmark&#39;, &#39;corrplot&#39;, &#39;cowplot&#39;, &#39;cpp11&#39;,
##   &#39;crayon&#39;, &#39;crosstalk&#39;, &#39;curl&#39;, &#39;data.table&#39;, &#39;DBI&#39;, &#39;dbplyr&#39;,
##   &#39;DelayedMatrixStats&#39;, &#39;deldir&#39;, &#39;desc&#39;, &#39;digest&#39;, &#39;dplyr&#39;, &#39;dqrng&#39;, &#39;DT&#39;,
##   &#39;dtplyr&#39;, &#39;ensembldb&#39;, &#39;evaluate&#39;, &#39;fansi&#39;, &#39;farver&#39;, &#39;fastmap&#39;, &#39;filelock&#39;,
##   &#39;fontawesome&#39;, &#39;forcats&#39;, &#39;foreign&#39;, &#39;formatR&#39;, &#39;fs&#39;, &#39;gargle&#39;, &#39;generics&#39;,
##   &#39;GenomeInfoDb&#39;, &#39;GenomicAlignments&#39;, &#39;GenomicFeatures&#39;, &#39;ggbeeswarm&#39;,
##   &#39;ggplot2&#39;, &#39;ggpubr&#39;, &#39;ggrepel&#39;, &#39;ggsci&#39;, &#39;ggsignif&#39;, &#39;glue&#39;, &#39;googledrive&#39;,
##   &#39;googlesheets4&#39;, &#39;gtable&#39;, &#39;haven&#39;, &#39;highr&#39;, &#39;hms&#39;, &#39;htmltools&#39;,
##   &#39;htmlwidgets&#39;, &#39;httpuv&#39;, &#39;httr&#39;, &#39;igraph&#39;, &#39;interp&#39;, &#39;IRanges&#39;, &#39;isoband&#39;,
##   &#39;jpeg&#39;, &#39;jsonlite&#39;, &#39;KEGGREST&#39;, &#39;knitr&#39;, &#39;labeling&#39;, &#39;later&#39;, &#39;lifecycle&#39;,
##   &#39;limma&#39;, &#39;lme4&#39;, &#39;locfit&#39;, &#39;lubridate&#39;, &#39;MatrixGenerics&#39;, &#39;MatrixModels&#39;,
##   &#39;matrixStats&#39;, &#39;mgcv&#39;, &#39;minqa&#39;, &#39;modelr&#39;, &#39;multcomp&#39;, &#39;munsell&#39;, &#39;mvtnorm&#39;,
##   &#39;ncdf4&#39;, &#39;nlme&#39;, &#39;nloptr&#39;, &#39;nnet&#39;, &#39;openssl&#39;, &#39;pbkrtest&#39;, &#39;pillar&#39;,
##   &#39;pkgload&#39;, &#39;plotly&#39;, &#39;plyr&#39;, &#39;png&#39;, &#39;prettyunits&#39;, &#39;printr&#39;, &#39;processx&#39;,
##   &#39;progress&#39;, &#39;promises&#39;, &#39;ps&#39;, &#39;purrr&#39;, &#39;quantreg&#39;, &#39;R.methodsS3&#39;, &#39;R.oo&#39;,
##   &#39;R.utils&#39;, &#39;Rcpp&#39;, &#39;RcppArmadillo&#39;, &#39;RcppEigen&#39;, &#39;RcppHNSW&#39;, &#39;RCurl&#39;,
##   &#39;readr&#39;, &#39;readxl&#39;, &#39;rematch&#39;, &#39;renv&#39;, &#39;reprex&#39;, &#39;restfulr&#39;, &#39;rintrojs&#39;,
##   &#39;rlang&#39;, &#39;rmarkdown&#39;, &#39;rprojroot&#39;, &#39;RSQLite&#39;, &#39;rstatix&#39;, &#39;rstudioapi&#39;,
##   &#39;rtracklayer&#39;, &#39;Rtsne&#39;, &#39;RUnit&#39;, &#39;rvest&#39;, &#39;sandwich&#39;, &#39;sass&#39;, &#39;ScaledMatrix&#39;,
##   &#39;scales&#39;, &#39;scuttle&#39;, &#39;shiny&#39;, &#39;SingleCellExperiment&#39;, &#39;sourcetools&#39;, &#39;sp&#39;,
##   &#39;SparseM&#39;, &#39;statmod&#39;, &#39;stringi&#39;, &#39;stringr&#39;, &#39;survival&#39;, &#39;sys&#39;, &#39;testthat&#39;,
##   &#39;TH.data&#39;, &#39;tibble&#39;, &#39;tidyr&#39;, &#39;tidyselect&#39;, &#39;tidyverse&#39;, &#39;tinytex&#39;, &#39;tzdb&#39;,
##   &#39;utf8&#39;, &#39;uuid&#39;, &#39;vctrs&#39;, &#39;vipor&#39;, &#39;viridis&#39;, &#39;viridisLite&#39;, &#39;vroom&#39;, &#39;waldo&#39;,
##   &#39;withr&#39;, &#39;xfun&#39;, &#39;xgboost&#39;, &#39;XML&#39;, &#39;xml2&#39;, &#39;yaml&#39;, &#39;zoo&#39;</code></pre>
<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>Paired-end sequencing was performed on primary cultures from
parathyroid tumors of 4 patients at 2 time points over 3 conditions
(control, treatment with diarylpropionitrile (DPN) and treatment with
4-hydroxytamoxifen (OHT)). DPN is a selective estrogen receptor agonist
and OHT is a selective estrogen receptor modulator. One sample (patient
4, 24 hours, control) was omitted by the paper authors due to low
quality. Data, the count table and information on the experiment is
available at <a
href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE37211"
class="uri">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE37211</a>.</p>
<p>#Count data and meta data</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;parathyroidGenesSE&quot;</span>, <span class="at">package=</span><span class="st">&quot;parathyroidSE&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>se1 <span class="ot">&lt;-</span> parathyroidGenesSE</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="fu">rm</span>(parathyroidGenesSE)</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>dupExps <span class="ot">&lt;-</span> <span class="fu">colData</span>(se1) <span class="sc">%&gt;%</span> </span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">duplicated</span>(experiment))  <span class="sc">%&gt;%</span> </span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>  <span class="fu">pull</span>(experiment)</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">assays</span>(se1)<span class="sc">$</span>counts</span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>newCounts <span class="ot">&lt;-</span> counts</span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>cd <span class="ot">&lt;-</span> <span class="fu">colData</span>(se1)</span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="cf">for</span>(ss <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(dupExps)){</span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a>  <span class="co"># check which samples are duplicates</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a>  relevantId <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">colData</span>(se1)<span class="sc">$</span>experiment <span class="sc">==</span> dupExps[ss])</span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a>  <span class="co"># sum counts</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a>  newCounts[,relevantId[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts[,relevantId])</span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a>  <span class="co"># keep which columns / rows to remove.</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a>  <span class="cf">if</span>(ss <span class="sc">==</span> <span class="dv">1</span>){</span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a>    toRemove <span class="ot">&lt;-</span> relevantId[<span class="dv">2</span>]</span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a>    toRemove <span class="ot">&lt;-</span> <span class="fu">c</span>(toRemove, relevantId[<span class="dv">2</span>])</span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a>  }</span>
<span id="cb6-23"><a href="#cb6-23" tabindex="-1"></a>}</span>
<span id="cb6-24"><a href="#cb6-24" tabindex="-1"></a></span>
<span id="cb6-25"><a href="#cb6-25" tabindex="-1"></a><span class="co"># remove after summing counts (otherwise IDs get mixed up)</span></span>
<span id="cb6-26"><a href="#cb6-26" tabindex="-1"></a>newCounts <span class="ot">&lt;-</span> newCounts[,<span class="sc">-</span>toRemove]</span>
<span id="cb6-27"><a href="#cb6-27" tabindex="-1"></a>newCD <span class="ot">&lt;-</span> cd[<span class="sc">-</span>toRemove,]</span>
<span id="cb6-28"><a href="#cb6-28" tabindex="-1"></a></span>
<span id="cb6-29"><a href="#cb6-29" tabindex="-1"></a><span class="co"># Create new SummarizedExperiment</span></span>
<span id="cb6-30"><a href="#cb6-30" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">SummarizedExperiment</span>(<span class="at">assays =</span> <span class="fu">list</span>(<span class="st">&quot;counts&quot;</span> <span class="ot">=</span> newCounts),</span>
<span id="cb6-31"><a href="#cb6-31" tabindex="-1"></a>                            <span class="at">colData =</span> newCD,</span>
<span id="cb6-32"><a href="#cb6-32" tabindex="-1"></a>                            <span class="at">metadata =</span> <span class="fu">metadata</span>(se1))</span>
<span id="cb6-33"><a href="#cb6-33" tabindex="-1"></a><span class="fu">rm</span>(se1)</span></code></pre></div>
</div>
<div id="data-analysis" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Data analysis</h1>
<div id="count-object" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Count object</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts=</span><span class="fu">assay</span>(se))</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>dge<span class="sc">$</span>sample</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["lib.size"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["norm.factors"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"1","2":"9102683","3":"1","_rn_":"Sample1"},{"1":"1","2":"10827109","3":"1","_rn_":"Sample2"},{"1":"1","2":"5217761","3":"1","_rn_":"Sample3"},{"1":"1","2":"9706035","3":"1","_rn_":"Sample4"},{"1":"1","2":"5700022","3":"1","_rn_":"Sample5"},{"1":"1","2":"7854568","3":"1","_rn_":"Sample6"},{"1":"1","2":"8610014","3":"1","_rn_":"Sample7"},{"1":"1","2":"6844144","3":"1","_rn_":"Sample8"},{"1":"1","2":"24584280","3":"1","_rn_":"Sample9"},{"1":"1","2":"8267977","3":"1","_rn_":"Sample10"},{"1":"1","2":"23590411","3":"1","_rn_":"Sample11"},{"1":"1","2":"8247122","3":"1","_rn_":"Sample12"},{"1":"1","2":"7341000","3":"1","_rn_":"Sample13"},{"1":"1","2":"8064268","3":"1","_rn_":"Sample14"},{"1":"1","2":"12481958","3":"1","_rn_":"Sample15"},{"1":"1","2":"16310090","3":"1","_rn_":"Sample16"},{"1":"1","2":"23697329","3":"1","_rn_":"Sample17"},{"1":"1","2":"7642648","3":"1","_rn_":"Sample18"},{"1":"1","2":"7701432","3":"1","_rn_":"Sample19"},{"1":"1","2":"7135899","3":"1","_rn_":"Sample20"},{"1":"1","2":"13818393","3":"1","_rn_":"Sample21"},{"1":"1","2":"6099942","3":"1","_rn_":"Sample22"},{"1":"1","2":"15825211","3":"1","_rn_":"Sample23"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="design" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Design</h2>
<p>There can be an effect of agent, time interaction and agent x time
interaction. We also expect blocking for patient. We can assess all
effects of interest within patient.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>time<span class="sc">*</span>treatment<span class="sc">+</span>patient,<span class="fu">colData</span>(se))</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="fu">rownames</span>(design) <span class="ot">=</span> <span class="fu">colnames</span>(dge)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>design</span></code></pre></div>
<pre><code>##          (Intercept) time48h treatmentDPN treatmentOHT patient2 patient3
## Sample1            1       0            0            0        0        0
## Sample2            1       1            0            0        0        0
## Sample3            1       0            1            0        0        0
## Sample4            1       1            1            0        0        0
## Sample5            1       0            0            1        0        0
## Sample6            1       1            0            1        0        0
## Sample7            1       0            0            0        1        0
## Sample8            1       1            0            0        1        0
## Sample9            1       0            1            0        1        0
## Sample10           1       1            1            0        1        0
## Sample11           1       0            0            1        1        0
## Sample12           1       1            0            1        1        0
## Sample13           1       0            0            0        0        1
## Sample14           1       1            0            0        0        1
## Sample15           1       0            1            0        0        1
## Sample16           1       1            1            0        0        1
## Sample17           1       0            0            1        0        1
## Sample18           1       1            0            1        0        1
## Sample19           1       1            0            0        0        0
## Sample20           1       0            1            0        0        0
## Sample21           1       1            1            0        0        0
## Sample22           1       0            0            1        0        0
## Sample23           1       1            0            1        0        0
##          patient4 time48h:treatmentDPN time48h:treatmentOHT
## Sample1         0                    0                    0
## Sample2         0                    0                    0
## Sample3         0                    0                    0
## Sample4         0                    1                    0
## Sample5         0                    0                    0
## Sample6         0                    0                    1
## Sample7         0                    0                    0
## Sample8         0                    0                    0
## Sample9         0                    0                    0
## Sample10        0                    1                    0
## Sample11        0                    0                    0
## Sample12        0                    0                    1
## Sample13        0                    0                    0
## Sample14        0                    0                    0
## Sample15        0                    0                    0
## Sample16        0                    1                    0
## Sample17        0                    0                    0
## Sample18        0                    0                    1
## Sample19        1                    0                    0
## Sample20        1                    0                    0
## Sample21        1                    1                    0
## Sample22        1                    0                    0
## Sample23        1                    0                    1
## attr(,&quot;assign&quot;)
## [1] 0 1 2 2 3 3 3 4 4
## attr(,&quot;contrasts&quot;)
## attr(,&quot;contrasts&quot;)$time
## [1] &quot;contr.treatment&quot;
## 
## attr(,&quot;contrasts&quot;)$treatment
## [1] &quot;contr.treatment&quot;
## 
## attr(,&quot;contrasts&quot;)$patient
## [1] &quot;contr.treatment&quot;</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>ExploreModelMatrix<span class="sc">::</span><span class="fu">VisualizeDesign</span>(<span class="fu">colData</span>(se),<span class="sc">~</span> time<span class="sc">*</span>treatment <span class="sc">+</span> patient)<span class="sc">$</span>plotlist</span></code></pre></div>
<pre><code>## $`time = 24h`</code></pre>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre><code>## 
## $`time = 48h`</code></pre>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-4-2.png" width="672" /></p>
</div>
<div id="filtering" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Filtering</h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(dge,design)</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a><span class="fu">table</span>(keep)</span></code></pre></div>
<pre><code>## keep
## FALSE  TRUE 
## 46629 16564</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>dge <span class="ot">&lt;-</span> dge[keep, , keep.lib.sizes<span class="ot">=</span><span class="cn">FALSE</span>]</span></code></pre></div>
</div>
<div id="normalisation" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Normalisation</h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dge)</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>dge<span class="sc">$</span>samples</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["lib.size"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["norm.factors"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"1","2":"9089191","3":"0.9811632","_rn_":"Sample1"},{"1":"1","2":"10811574","3":"0.9707108","_rn_":"Sample2"},{"1":"1","2":"5210162","3":"0.9768753","_rn_":"Sample3"},{"1":"1","2":"9691321","3":"0.9927766","_rn_":"Sample4"},{"1":"1","2":"5691477","3":"0.9740861","_rn_":"Sample5"},{"1":"1","2":"7843482","3":"0.9831290","_rn_":"Sample6"},{"1":"1","2":"8598294","3":"0.9316231","_rn_":"Sample7"},{"1":"1","2":"6834621","3":"0.9436520","_rn_":"Sample8"},{"1":"1","2":"24548167","3":"0.9421338","_rn_":"Sample9"},{"1":"1","2":"8256710","3":"0.9324177","_rn_":"Sample10"},{"1":"1","2":"23556262","3":"0.9349302","_rn_":"Sample11"},{"1":"1","2":"8235902","3":"0.9306290","_rn_":"Sample12"},{"1":"1","2":"7326632","3":"1.0573853","_rn_":"Sample13"},{"1":"1","2":"8048185","3":"1.0477621","_rn_":"Sample14"},{"1":"1","2":"12458057","3":"1.0616393","_rn_":"Sample15"},{"1":"1","2":"16279311","3":"1.0416439","_rn_":"Sample16"},{"1":"1","2":"23652574","3":"1.0441526","_rn_":"Sample17"},{"1":"1","2":"7628292","3":"1.0367185","_rn_":"Sample18"},{"1":"1","2":"7687278","3":"1.0449427","_rn_":"Sample19"},{"1":"1","2":"7122747","3":"1.0561330","_rn_":"Sample20"},{"1":"1","2":"13792983","3":"1.0382934","_rn_":"Sample21"},{"1":"1","2":"6088410","3":"1.0629288","_rn_":"Sample22"},{"1":"1","2":"15795813","3":"1.0415665","_rn_":"Sample23"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="data-exploration" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Data exploration</h2>
<p>An MDS plot shows the leading fold changes (differential expression)
between the 23 samples.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="fu">plotMDS</span>(dge,<span class="at">labels=</span><span class="fu">paste</span>(<span class="fu">colData</span>(se)<span class="sc">$</span>treatment,<span class="fu">colData</span>(se)<span class="sc">$</span>time,<span class="fu">colData</span>(se)<span class="sc">$</span>patient,<span class="at">sep=</span><span class="st">&quot;-&quot;</span>),<span class="at">col=</span><span class="fu">as.double</span>(<span class="fu">colData</span>(se)<span class="sc">$</span>treatment))</span></code></pre></div>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>There is a very strong patient effect! To further assess the
treatment effects we can make MDS plots per patient</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="fu">plotMDS</span>(dge[,<span class="fu">colData</span>(se)<span class="sc">$</span>patient<span class="sc">==</span>i], <span class="at">col=</span><span class="fu">as.double</span>(<span class="fu">colData</span>(se)<span class="sc">$</span>treatment)[<span class="fu">colData</span>(se)<span class="sc">$</span>patient<span class="sc">==</span>i],</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a><span class="at">labels=</span><span class="fu">paste</span>(<span class="fu">colData</span>(se)<span class="sc">$</span>treatment[<span class="fu">colData</span>(se)<span class="sc">$</span>patient<span class="sc">==</span>i],</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>             <span class="fu">colData</span>(se)<span class="sc">$</span>time[<span class="fu">colData</span>(se)<span class="sc">$</span>patient<span class="sc">==</span>i],</span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a>             <span class="fu">colData</span>(se)<span class="sc">$</span>patient,<span class="at">sep=</span><span class="st">&quot;-&quot;</span>)[<span class="fu">colData</span>(se)<span class="sc">$</span>patient<span class="sc">==</span>i])</span></code></pre></div>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-8-1.png" width="672" /><img src="parathyroid_files/figure-html/unnamed-chunk-8-2.png" width="672" /><img src="parathyroid_files/figure-html/unnamed-chunk-8-3.png" width="672" /><img src="parathyroid_files/figure-html/unnamed-chunk-8-4.png" width="672" /></p>
</div>
<div id="parameter-estimation" class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> Parameter
estimation</h2>
<p>We will use the default Quasi likelihood approach of edgeR.</p>
<p>For quasi-likelihood we do not specify the full distribution, only
the first two moments: the mean and the variance, which is sufficient to
do inference on the mean.</p>
<p><span class="math display">\[
\left\{
\begin{array}{lcl}
E[y_{ig}\vert \mathbf{x}_{ig}]&amp;=&amp;\mu_{ig}\\
log(\mu_{ig})&amp;=&amp;\eta_{ig}\\
\eta_{ig}&amp;=&amp;\beta_0 + \beta_{t2} x_{t2,i} + \beta_{DPN}
x_{DPN,i} + \beta_\text{DPN:t2} x_{DPN,i}x_{t2,i} \\
&amp;&amp; \quad + \beta_{OHT}x_{OHT,i} + \beta_\text{OHT:t2}
x_{OHT,i}x_{t2,i} \\
&amp;&amp; \quad + \beta_{p2}x_{p2,i} + \beta_{p3}x_{p3,i} +
\beta_{p4}x_{p4,i}\ + \log N_i\\
\text{Var}[y_{ig}\vert
\mathbf{x}_{ig}]&amp;=&amp;\sigma^2_g\left(\mu_{ig}+\phi\mu_{ig}^2\right)
\end{array}\right.
\]</span></p>
<p>with <span class="math inline">\(\sigma^2_g\)</span> an additional
dispersion parameter the scales the negative binomial variance function,
<span class="math inline">\(x_{DPN,i}\)</span>, <span
class="math inline">\(x_{DPN,i}\)</span>, <span
class="math inline">\(x_{t2,i}\)</span>, <span
class="math inline">\(x_{p.,i}\)</span> dummy variables that is 1 if
cell line was treated with DPN, OHT, incubated for 48 h, from patient
<span class="math inline">\(p.\)</span>, respectively and is 0
otherwise, and, <span class="math inline">\(\log{N}_i\)</span> a
normalisation offset to correct for sequencing depth. Note, that <span
class="math inline">\(\beta_{DPN}\)</span> is the main effect for the
DPN treatment, and corresponds to the average log fold change between
treated and control mice after 24h. The interaction <span
class="math inline">\(\beta_\text{DPN:t2}\)</span> can be interpreted as
the average change in log2 FC between DPN treated and control cell lines
at the late and early timepoint. The researchers are also interested in
a assessing third contrast: the effect of the DPN treatment at the late
time point.</p>
<p><span class="math display">\[ \log_2\text{FC}^\text{48h}_\text{DPN -
C}= \beta_{DPN}+\beta_{DPN,t2}\]</span></p>
<p>For the OHT treatment we will assess similar contrasts.</p>
<p><span class="math display">\[ \log_2\text{FC}^\text{24h}_\text{OHT -
C}= \beta_{OHT}\]</span> <span class="math display">\[
\log_2\text{FC}^\text{48h}_\text{OHT - C}=
\beta_{OHT}+\beta_{OHT,t2}\]</span> <span class="math display">\[
\log_2\text{FC}^\text{48h}_\text{OHT - C}
-\log_2\text{FC}^\text{24h}_\text{OHT - C}= \beta_{OHT,t2}
\]</span></p>
<p>Finally, we also have to assess if there is a difference between DPN
and OHT treatment</p>
<p><span class="math display">\[ \log_2\text{FC}^\text{24h}_\text{OHT -
DPN}= \beta_{OHT} - \beta_{DPN}\]</span> <span class="math display">\[
\log_2\text{FC}^\text{48h}_\text{OHT - C}= \beta_{OHT}+\beta_{OHT,t2} -
\beta_{DPN} - \beta_{DPN,t2}
\]</span></p>
<p><span class="math display">\[ \log_2\text{FC}^\text{48h}_\text{OHT -
DPN} -\log_2\text{FC}^\text{24h}_\text{OHT - DPN}= \beta_{OHT,t2} -
\beta_{DPN,t2}
\]</span></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(dge, design)  </span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="fu">plotBCV</span>(dge)</span></code></pre></div>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>The quasi-negative binomial model can be fitted using the function
<code>glmQLFit</code></p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glmQLFit</span>(dge,design)</span></code></pre></div>
</div>
<div id="contrasts" class="section level2" number="2.7">
<h2><span class="header-section-number">2.7</span> Contrasts</h2>
<p>We now implement all 9 contrasts of interest</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>L <span class="ot">&lt;-</span> msqrob2<span class="sc">::</span><span class="fu">makeContrast</span>(</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">&quot;treatmentDPN = 0&quot;</span>,</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a>      <span class="st">&quot;treatmentOHT = 0&quot;</span>,</span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a>      <span class="st">&quot;treatmentOHT - treatmentDPN = 0&quot;</span>, </span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a>      <span class="st">&quot;treatmentDPN + time48h:treatmentDPN = 0&quot;</span>,</span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a>      <span class="st">&quot;treatmentOHT + time48h:treatmentOHT = 0&quot;</span>,</span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a>      <span class="st">&quot;treatmentOHT + time48h:treatmentOHT - treatmentDPN - time48h:treatmentDPN = 0&quot;</span>, </span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a>      <span class="st">&quot;time48h:treatmentDPN = 0&quot;</span>, </span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a>      <span class="st">&quot;time48h:treatmentOHT = 0&quot;</span>, </span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a>      <span class="st">&quot;time48h:treatmentOHT - time48h:treatmentDPN = 0&quot;</span>),</span>
<span id="cb21-11"><a href="#cb21-11" tabindex="-1"></a>    <span class="at">parameterNames =</span> <span class="fu">colnames</span>(design))</span>
<span id="cb21-12"><a href="#cb21-12" tabindex="-1"></a>L</span></code></pre></div>
<pre><code>##                      treatmentDPN treatmentOHT treatmentOHT - treatmentDPN
## (Intercept)                     0            0                           0
## time48h                         0            0                           0
## treatmentDPN                    1            0                          -1
## treatmentOHT                    0            1                           1
## patient2                        0            0                           0
## patient3                        0            0                           0
## patient4                        0            0                           0
## time48h:treatmentDPN            0            0                           0
## time48h:treatmentOHT            0            0                           0
##                      treatmentDPN + time48h:treatmentDPN
## (Intercept)                                            0
## time48h                                                0
## treatmentDPN                                           1
## treatmentOHT                                           0
## patient2                                               0
## patient3                                               0
## patient4                                               0
## time48h:treatmentDPN                                   1
## time48h:treatmentOHT                                   0
##                      treatmentOHT + time48h:treatmentOHT
## (Intercept)                                            0
## time48h                                                0
## treatmentDPN                                           0
## treatmentOHT                                           1
## patient2                                               0
## patient3                                               0
## patient4                                               0
## time48h:treatmentDPN                                   0
## time48h:treatmentOHT                                   1
##                      treatmentOHT + time48h:treatmentOHT - treatmentDPN - time48h:treatmentDPN
## (Intercept)                                                                                  0
## time48h                                                                                      0
## treatmentDPN                                                                                -1
## treatmentOHT                                                                                 1
## patient2                                                                                     0
## patient3                                                                                     0
## patient4                                                                                     0
## time48h:treatmentDPN                                                                        -1
## time48h:treatmentOHT                                                                         1
##                      time48h:treatmentDPN time48h:treatmentOHT
## (Intercept)                             0                    0
## time48h                                 0                    0
## treatmentDPN                            0                    0
## treatmentOHT                            0                    0
## patient2                                0                    0
## patient3                                0                    0
## patient4                                0                    0
## time48h:treatmentDPN                    1                    0
## time48h:treatmentOHT                    0                    1
##                      time48h:treatmentOHT - time48h:treatmentDPN
## (Intercept)                                                    0
## time48h                                                        0
## treatmentDPN                                                   0
## treatmentOHT                                                   0
## patient2                                                       0
## patient3                                                       0
## patient4                                                       0
## time48h:treatmentDPN                                          -1
## time48h:treatmentOHT                                           1</code></pre>
</div>
<div id="tests" class="section level2" number="2.8">
<h2><span class="header-section-number">2.8</span> Tests</h2>
<p>We have to perform a quasi- F-test for each contrast. The quasi
F-test involves fitting a different model for each contrast so that we
can compare the full model with a reduced model that implies that one
specific contrast is zero.</p>
<p>Because we estimated the additional dispersion parameter <span
class="math inline">\(\sigma^2_g\)</span> using a sum of squared
deviance residuals:</p>
<p>i.e.  <span class="math display">\[e_{i,d} = 2 (l_i(y_i,y_i) -
l_i(\mu_i,y_i)\]</span> and</p>
<p><span class="math display">\[
\hat \sigma_g^2 = \frac{\sum_{i=1}^n e_{i,d}^2}{n-p} =
\frac{2\left[l(\mathbf{y},\mathbf{y}) -
l(\boldsymbol{\mu},\mathbf{y})\right]}{n-p}
\]</span></p>
<p>With edgeR we will then further adopt empirical Bayes to borrow
strength across genes to stabilise the parameter estimator, which will
also increase the degrees if this gene wise dispersion parameter
estimator which we refer to as <span
class="math inline">\(df_\text{res}^{EB}\)</span>.</p>
<p>We can use a quasi F -test that can also correct for the degrees of
freedom that have been used to estimate mean model parameters and the
residual degrees of freedom that were available for estimating the
additional dispersion parameter. The quasi F test will thus perform
better in a small sample setting. It is defined as:</p>
<p><span class="math display">\[
F = \frac{\frac{LRT_\text{g, full - reduced}}{df_{LRT}}}{\sigma^2_g}
\]</span></p>
<p>which follows an F - distribution with <span
class="math inline">\(df_{LRT}\)</span> degrees of freedom in the
nominator and <span class="math inline">\(df_{res}^{EB}\)</span> degrees
of freedom in the denominator under the null hypothesis that the full
and reduced model are equivalent and that the assessed contrasts are
thus equal to zero. Indeed, the dispersion estimator in the denominator
follows a scaled <span class="math inline">\(\chi^2\)</span>
distribution with <span class="math inline">\(df_{res}^{EB}\)</span>
degrees of freedom.</p>
<p>We perform all tests and loop over the columns of L for this
purpose.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>testsF <span class="ot">&lt;-</span> <span class="fu">apply</span>(L, <span class="dv">2</span>, <span class="cf">function</span>(fit,contrast) </span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>    <span class="fu">glmQLFTest</span>(fit,<span class="at">contrast=</span>contrast), </span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>    <span class="at">fit =</span> fit)</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>topTablesF<span class="ot">&lt;-</span> <span class="fu">lapply</span>(testsF, topTags, <span class="at">n=</span><span class="fu">nrow</span>(dge))</span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a><span class="fu">sapply</span>(topTablesF, <span class="cf">function</span>(x) <span class="fu">sum</span>(x<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span><span class="fl">0.05</span>))</span></code></pre></div>
<pre><code>##                                                              treatmentDPN 
##                                                                         0 
##                                                              treatmentOHT 
##                                                                         0 
##                                               treatmentOHT - treatmentDPN 
##                                                                         0 
##                                       treatmentDPN + time48h:treatmentDPN 
##                                                                         0 
##                                       treatmentOHT + time48h:treatmentOHT 
##                                                                         4 
## treatmentOHT + time48h:treatmentOHT - treatmentDPN - time48h:treatmentDPN 
##                                                                         0 
##                                                      time48h:treatmentDPN 
##                                                                         0 
##                                                      time48h:treatmentOHT 
##                                                                         0 
##                               time48h:treatmentOHT - time48h:treatmentDPN 
##                                                                         0</code></pre>
<p>We only find significant fold changes for very few genes between OHT
and the control treatment at the late time point. We also did not find
significant interactions.</p>
</div>
</div>
<div id="plots" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Plots</h1>
<div id="volcano-plots" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Volcano plots</h2>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(L))</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>{</span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>   volcano<span class="ot">&lt;-</span> <span class="fu">ggplot</span>(topTablesF[[i]]<span class="sc">$</span>table,<span class="fu">aes</span>(<span class="at">x=</span>logFC,<span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(PValue),<span class="at">color=</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;black&quot;</span>,<span class="st">&quot;red&quot;</span>)) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="fu">paste</span>(<span class="st">&quot;contrast&quot;</span>,<span class="fu">names</span>(topTablesF)[i]))</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a><span class="fu">print</span>(volcano)</span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a>}</span></code></pre></div>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-13-1.png" width="672" /><img src="parathyroid_files/figure-html/unnamed-chunk-13-2.png" width="672" /><img src="parathyroid_files/figure-html/unnamed-chunk-13-3.png" width="672" /><img src="parathyroid_files/figure-html/unnamed-chunk-13-4.png" width="672" /><img src="parathyroid_files/figure-html/unnamed-chunk-13-5.png" width="672" /><img src="parathyroid_files/figure-html/unnamed-chunk-13-6.png" width="672" /><img src="parathyroid_files/figure-html/unnamed-chunk-13-7.png" width="672" /><img src="parathyroid_files/figure-html/unnamed-chunk-13-8.png" width="672" /><img src="parathyroid_files/figure-html/unnamed-chunk-13-9.png" width="672" /></p>
</div>
<div id="histograms-of-p-values" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Histograms of
p-values</h2>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>histsP <span class="ot">&lt;-</span> <span class="fu">lapply</span>(topTablesF, <span class="cf">function</span>(x) </span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>    x<span class="sc">$</span>table <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>        <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>PValue)) <span class="sc">+</span> </span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>        <span class="fu">geom_histogram</span>(<span class="at">breaks =</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,.<span class="dv">1</span>) ,<span class="at">col=</span><span class="dv">1</span>)</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>    )</span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>    </span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(L)) </span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a>    histsP[[i]] <span class="ot">&lt;-</span> histsP[[i]] <span class="sc">+</span></span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste</span>(<span class="st">&quot;contrast&quot;</span>,<span class="fu">names</span>(topTablesF)[i]))</span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a>histsP</span></code></pre></div>
<pre><code>## $treatmentDPN</code></pre>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<pre><code>## 
## $treatmentOHT</code></pre>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-14-2.png" width="672" /></p>
<pre><code>## 
## $`treatmentOHT - treatmentDPN`</code></pre>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-14-3.png" width="672" /></p>
<pre><code>## 
## $`treatmentDPN + time48h:treatmentDPN`</code></pre>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-14-4.png" width="672" /></p>
<pre><code>## 
## $`treatmentOHT + time48h:treatmentOHT`</code></pre>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-14-5.png" width="672" /></p>
<pre><code>## 
## $`treatmentOHT + time48h:treatmentOHT - treatmentDPN - time48h:treatmentDPN`</code></pre>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-14-6.png" width="672" /></p>
<pre><code>## 
## $`time48h:treatmentDPN`</code></pre>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-14-7.png" width="672" /></p>
<pre><code>## 
## $`time48h:treatmentOHT`</code></pre>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-14-8.png" width="672" /></p>
<pre><code>## 
## $`time48h:treatmentOHT - time48h:treatmentDPN`</code></pre>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-14-9.png" width="672" /></p>
</div>
<div id="heatmaps" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> heatmaps</h2>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(L))</span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a>{</span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a>    sigID <span class="ot">&lt;-</span> topTablesF[[i]]<span class="sc">$</span>table <span class="sc">%&gt;%</span> </span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a>        <span class="fu">filter</span>(FDR<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-5"><a href="#cb36-5" tabindex="-1"></a>        rownames</span>
<span id="cb36-6"><a href="#cb36-6" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(sigID)<span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb36-7"><a href="#cb36-7" tabindex="-1"></a>        <span class="fu">heatmap</span>(dge<span class="sc">$</span>counts[sigID,], <span class="at">main =</span> <span class="fu">colnames</span>(L)[i], <span class="at">cex.main=</span>.<span class="dv">2</span>)</span>
<span id="cb36-8"><a href="#cb36-8" tabindex="-1"></a>}</span></code></pre></div>
<p><img src="parathyroid_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
</div>
</div>
<div id="edger-traditional" class="section level1" number="4">
<h1><span class="header-section-number">4</span> EdgeR traditional</h1>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>fitGlm <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(dge,design)</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a>testLRT2 <span class="ot">&lt;-</span> <span class="fu">apply</span>(L, <span class="dv">2</span>, <span class="cf">function</span>(fit,contrast) </span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a>    <span class="fu">glmLRT</span>(fit,<span class="at">contrast=</span>contrast), </span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a>    <span class="at">fit =</span> fitGlm)</span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a>topTablesLRT2 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(testLRT2, topTags, <span class="at">n=</span><span class="fu">nrow</span>(dge))</span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a><span class="fu">sapply</span>(topTablesLRT2, <span class="cf">function</span>(x) <span class="fu">sum</span>(x<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span><span class="fl">0.05</span>))</span></code></pre></div>
<pre><code>##                                                              treatmentDPN 
##                                                                         2 
##                                                              treatmentOHT 
##                                                                         0 
##                                               treatmentOHT - treatmentDPN 
##                                                                         4 
##                                       treatmentDPN + time48h:treatmentDPN 
##                                                                        64 
##                                       treatmentOHT + time48h:treatmentOHT 
##                                                                        23 
## treatmentOHT + time48h:treatmentOHT - treatmentDPN - time48h:treatmentDPN 
##                                                                        11 
##                                                      time48h:treatmentDPN 
##                                                                         0 
##                                                      time48h:treatmentOHT 
##                                                                         0 
##                               time48h:treatmentOHT - time48h:treatmentDPN 
##                                                                         0</code></pre>
<p>We find more genes for the traditional edgeR workflow, however, it is
known that this workflow is often too liberal.</p>
</div>

<div id="rmd-source-code">---
title: "Parathyroid: DE analysis"
author: "Lieven Clement"
date: "statOmics, Ghent University (https://statomics.github.io)"
output:
    html_document:
      code_download: true
      theme: flatly
      toc: true
      toc_float: true
      highlight: tango
      number_sections: true
    pdf_document:
      toc: true
      number_sections: true
linkcolor: blue
urlcolor: blue
citecolor: blue
---



```{r,echo=FALSE}
suppressPackageStartupMessages({
library(edgeR)
library(SummarizedExperiment)
library(tidyverse)
  if(!"parathyroidSE" %in% installed.packages()[,1]) BiocManager::install("parathyroidSE")
})

```

# Introduction

Paired-end sequencing was performed on primary cultures from parathyroid tumors of 4 patients at 2 time points over 3 conditions (control, treatment with diarylpropionitrile (DPN) and treatment with 4-hydroxytamoxifen (OHT)). DPN is a selective estrogen receptor agonist and OHT is a selective estrogen receptor modulator. One sample (patient 4, 24 hours, control) was omitted by the paper authors due to low quality. Data, the count table and information on the experiment is available at http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE37211.

#Count data and meta data

```{r}
data("parathyroidGenesSE", package="parathyroidSE")
se1 <- parathyroidGenesSE
rm(parathyroidGenesSE)
dupExps <- colData(se1) %>% 
  as.data.frame() %>% 
  filter(duplicated(experiment))  %>% 
  pull(experiment)

counts <- assays(se1)$counts
newCounts <- counts
cd <- colData(se1)
for(ss in 1:length(dupExps)){
  # check which samples are duplicates
  relevantId <- which(colData(se1)$experiment == dupExps[ss])
  # sum counts
  newCounts[,relevantId[1]] <- rowSums(counts[,relevantId])
  # keep which columns / rows to remove.
  if(ss == 1){
    toRemove <- relevantId[2]
  } else {
    toRemove <- c(toRemove, relevantId[2])
  }
}

# remove after summing counts (otherwise IDs get mixed up)
newCounts <- newCounts[,-toRemove]
newCD <- cd[-toRemove,]

# Create new SummarizedExperiment
se <- SummarizedExperiment(assays = list("counts" = newCounts),
                            colData = newCD,
                            metadata = metadata(se1))
rm(se1)
```

# Data analysis

## Count object

```{r}
dge <- DGEList(counts=assay(se))
dge$sample
```

## Design

There can be an effect of agent, time interaction and agent x time interaction. We also expect blocking for patient. We can assess all effects of interest within patient.

```{r}
design <- model.matrix(~time*treatment+patient,colData(se))
rownames(design) = colnames(dge)
design

ExploreModelMatrix::VisualizeDesign(colData(se),~ time*treatment + patient)$plotlist
```

## Filtering

```{r}
keep <- filterByExpr(dge,design)
table(keep)
dge <- dge[keep, , keep.lib.sizes=FALSE]
```

## Normalisation
```{r}
dge <- calcNormFactors(dge)
dge$samples
```

## Data exploration

An MDS plot shows the leading fold changes (differential expression) between the 23 samples.

```{r}
plotMDS(dge,labels=paste(colData(se)$treatment,colData(se)$time,colData(se)$patient,sep="-"),col=as.double(colData(se)$treatment))
```

There is a very strong patient effect! To further assess the treatment effects we can make MDS plots per patient

```{r}
for (i in 1:4)
plotMDS(dge[,colData(se)$patient==i], col=as.double(colData(se)$treatment)[colData(se)$patient==i],
labels=paste(colData(se)$treatment[colData(se)$patient==i],
             colData(se)$time[colData(se)$patient==i],
             colData(se)$patient,sep="-")[colData(se)$patient==i])
```


## Parameter estimation

We will use the default Quasi likelihood approach of edgeR.

For quasi-likelihood we do not specify the full distribution, only the first two moments: the mean and the variance, which is sufficient to do inference on the mean.

$$
\left\{
\begin{array}{lcl}
E[y_{ig}\vert \mathbf{x}_{ig}]&=&\mu_{ig}\\
log(\mu_{ig})&=&\eta_{ig}\\
\eta_{ig}&=&\beta_0 + \beta_{t2} x_{t2,i} + \beta_{DPN} x_{DPN,i} + \beta_\text{DPN:t2} x_{DPN,i}x_{t2,i} \\ 
&& \quad + \beta_{OHT}x_{OHT,i} + \beta_\text{OHT:t2} x_{OHT,i}x_{t2,i} \\ 
&& \quad + \beta_{p2}x_{p2,i} + \beta_{p3}x_{p3,i} + \beta_{p4}x_{p4,i}\ + \log N_i\\
\text{Var}[y_{ig}\vert \mathbf{x}_{ig}]&=&\sigma^2_g\left(\mu_{ig}+\phi\mu_{ig}^2\right)
\end{array}\right.
$$

with $\sigma^2_g$ an additional dispersion parameter the scales the negative binomial variance function, $x_{DPN,i}$, $x_{DPN,i}$, $x_{t2,i}$, $x_{p.,i}$  dummy variables that is 1 if cell line was treated with DPN, OHT, incubated for 48 h, from patient $p.$, respectively and is 0 otherwise, and, $\log{N}_i$ a normalisation offset to correct for sequencing depth. Note, that $\beta_{DPN}$ is the main effect for the DPN treatment, and corresponds to the average log fold change between treated and control mice after 24h. The interaction  $\beta_\text{DPN:t2}$ can be interpreted as the average change in log2 FC between DPN treated and control cell lines at the late and early timepoint. The researchers are also interested in a assessing third contrast: the effect of the DPN treatment at the late time point.

$$ \log_2\text{FC}^\text{48h}_\text{DPN - C}= \beta_{DPN}+\beta_{DPN,t2}$$

For the OHT treatment we will assess similar contrasts. 

$$ \log_2\text{FC}^\text{24h}_\text{OHT - C}= \beta_{OHT}$$
$$ \log_2\text{FC}^\text{48h}_\text{OHT - C}= \beta_{OHT}+\beta_{OHT,t2}$$
$$ \log_2\text{FC}^\text{48h}_\text{OHT - C} -\log_2\text{FC}^\text{24h}_\text{OHT - C}= \beta_{OHT,t2}
$$

Finally, we also have to assess if there is a difference between DPN and OHT treatment

$$ \log_2\text{FC}^\text{24h}_\text{OHT - DPN}= \beta_{OHT} - \beta_{DPN}$$
$$ \log_2\text{FC}^\text{48h}_\text{OHT - C}= \beta_{OHT}+\beta_{OHT,t2} - \beta_{DPN} - \beta_{DPN,t2}
$$


$$ \log_2\text{FC}^\text{48h}_\text{OHT - DPN} -\log_2\text{FC}^\text{24h}_\text{OHT - DPN}= \beta_{OHT,t2} - \beta_{DPN,t2}
$$






```{r}
dge <- estimateDisp(dge, design)  
plotBCV(dge)
```

The quasi-negative binomial model can be fitted using the function `glmQLFit`
```{r}
fit <- glmQLFit(dge,design)
```


## Contrasts

We now implement all 9 contrasts of interest 

```{r}
L <- msqrob2::makeContrast(
    c("treatmentDPN = 0",
      "treatmentOHT = 0",
      "treatmentOHT - treatmentDPN = 0", 
      "treatmentDPN + time48h:treatmentDPN = 0",
      "treatmentOHT + time48h:treatmentOHT = 0",
      "treatmentOHT + time48h:treatmentOHT - treatmentDPN - time48h:treatmentDPN = 0", 
      "time48h:treatmentDPN = 0", 
      "time48h:treatmentOHT = 0", 
      "time48h:treatmentOHT - time48h:treatmentDPN = 0"),
    parameterNames = colnames(design))
L
```

## Tests

We have to perform a quasi- F-test for each contrast. 
The quasi F-test involves fitting a different model for each contrast so that we can compare the full model with a reduced model that implies that one specific contrast is zero. 

Because we estimated the additional dispersion parameter $\sigma^2_g$ using a sum of squared deviance residuals: 

i.e. 
$$e_{i,d} = 2 (l_i(y_i,y_i) - l_i(\mu_i,y_i)$$ 
and 

$$ 
\hat \sigma_g^2 = \frac{\sum_{i=1}^n e_{i,d}^2}{n-p} = \frac{2\left[l(\mathbf{y},\mathbf{y}) - l(\boldsymbol{\mu},\mathbf{y})\right]}{n-p}
$$

With edgeR we will then further adopt empirical Bayes to borrow strength across genes to stabilise the parameter estimator, which will also increase the degrees if this gene wise dispersion parameter estimator which we refer to as $df_\text{res}^{EB}$.

We can use a quasi F -test that can also correct for the degrees of freedom that have been used to estimate mean model parameters and the residual degrees of freedom that were available for estimating the additional dispersion parameter. The quasi F test will thus perform better in a small sample setting. It is defined as: 

$$
F = \frac{\frac{LRT_\text{g, full - reduced}}{df_{LRT}}}{\sigma^2_g} 
$$

which follows an F - distribution with $df_{LRT}$ degrees of freedom in the nominator and $df_{res}^{EB}$ degrees of freedom in the denominator under the null hypothesis that the full and reduced model are equivalent and that the assessed contrasts are thus equal to zero. Indeed, the dispersion estimator in the denominator follows a scaled $\chi^2$ distribution with $df_{res}^{EB}$ degrees of freedom. 

We perform all tests and loop over the columns of L for this purpose. 

```{r}
testsF <- apply(L, 2, function(fit,contrast) 
    glmQLFTest(fit,contrast=contrast), 
    fit = fit)
topTablesF<- lapply(testsF, topTags, n=nrow(dge))

sapply(topTablesF, function(x) sum(x$table$FDR <0.05))
```

We only find significant fold changes for very few genes between OHT and the control treatment at the late time point. 
We also did not find significant interactions. 

# Plots 
## Volcano plots 

```{r}
for (i in 1:ncol(L))
{
   volcano<- ggplot(topTablesF[[i]]$table,aes(x=logFC,y=-log10(PValue),color=FDR < 0.05)) + geom_point() + scale_color_manual(values=c("black","red")) + ggtitle(paste("contrast",names(topTablesF)[i]))
print(volcano)
}
```

## Histograms of p-values

```{r}
histsP <- lapply(topTablesF, function(x) 
    x$table %>% 
        ggplot(aes(x=PValue)) + 
        geom_histogram(breaks =seq(0,1,.1) ,col=1)
    )
    
for (i in 1:ncol(L)) 
    histsP[[i]] <- histsP[[i]] +
    ggtitle(paste("contrast",names(topTablesF)[i]))
histsP
```

## heatmaps

```{r}
for (i in 1:ncol(L))
{
    sigID <- topTablesF[[i]]$table %>% 
        filter(FDR<0.05) %>% 
        rownames
    if (length(sigID)>0)
        heatmap(dge$counts[sigID,], main = colnames(L)[i], cex.main=.2)
}
```

# EdgeR traditional 

```{r}
fitGlm <- glmFit(dge,design)
testLRT2 <- apply(L, 2, function(fit,contrast) 
    glmLRT(fit,contrast=contrast), 
    fit = fitGlm)
topTablesLRT2 <- lapply(testLRT2, topTags, n=nrow(dge))
sapply(topTablesLRT2, function(x) sum(x$table$FDR <0.05))
```

We find more genes for the traditional edgeR workflow, however, it is known that this workflow is often too liberal. 
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("parathyroid.Rmd");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
