<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Koen Van den Berge &amp; Lieven Clement" />


<title>Introduction to RNA-seq sequencing: Preprocessing</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="intro.html">1.1 Position</a>
    </li>
    <li>
      <a href="recapGeneralLinearModel.html">1.2 Recap linear model</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="docs/martens_proteomics_bioinformatics.pdf">1. Proteomics Bioinformatics</a>
    </li>
    <li>
      <a href="TargetDecoy.html">2.1 Identification - FDR - QC</a>
    </li>
    <li>
      <a href="pda_quantification_preprocessing.html">2.2.1. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">2.2.2. Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">2.3.1. Quantification - Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="pda_blocking_wrapup.html">2.3.2. Wrap-up DA - Blocking</a>
    </li>
    <li>
      <a href="technicalDetailsProteomics.html">2.3.3 Technical Details</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencing_intro.html">1.  Introduction to RNA-seq</a>
    </li>
    <li>
      <a href="sequencing_countData.html">2.  Working with count data and GLMs</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="multipleRegression_KPNA2.html">1 Intro KPNA2</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="https://www.compomics.com/bioinformatics-for-proteomics/">1. Bioinformatics for Proteomics</a>
    </li>
    <li>
      <a href="TargetDecoyTutorial.html">2.1. Target Decoy QC</a>
    </li>
    <li>
      <a href="pda_tutorialPreprocessing.html">2.2. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">2.3. Quantification - Design</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencingTutorial1_mapping.html">1. Read mapping</a>
    </li>
    <li>
      <a href="sequencingTutorial2_DE.html">2. Working with Count Data</a>
    </li>
    <li>
      <a href="scRNA-seq-Tutorial.html">3. Single Cell RNA-seq Data Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Introduction to RNA-seq sequencing:
Preprocessing</h1>
<h4 class="author">Koen Van den Berge &amp; Lieven Clement</h4>
<h4 class="date">Last edited on 16 September, 2024</h4>

</div>


<div id="the-study-of-gene-expression" class="section level1"
number="1">
<h1><span class="header-section-number">1</span> The study of gene
expression</h1>
<ul>
<li><p>The first part of this course focussed on proteomics, studying
the concentration of proteins in biological samples. We have seen that
identification of proteins and measuring their respective concentrations
are extremely challenging, leading to many technological and statistical
challenges in order to interpret these data.</p></li>
<li><p>In the second part of the course, we will focus on measuring gene
expression, i.e., measuring the concentration of mRNA molecules, that
may eventually be translated into proteins, but may also have functions
on their own.</p></li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">include_graphics</span>(<span class="st">&quot;./images_sequencing/centralParadigm.png&quot;</span>)</span></code></pre></div>
<div class="figure">
<img src="images_sequencing/centralParadigm.png" alt="Central Paradigm of Biology: a gene, a specific region in the DNA, is first transcribed into RNA and then into proteins. Note, that for RNA-genes the RNA molecule is the end product itself, which is referred to as non-coding RNA (ncRNA) (Source: Wikipedia)" width="709" />
<p class="caption">
Central Paradigm of Biology: a gene, a specific region in the DNA, is
first transcribed into RNA and then into proteins. Note, that for
RNA-genes the RNA molecule is the end product itself, which is referred
to as non-coding RNA (ncRNA) (Source: Wikipedia)
</p>
</div>
<div id="alternative-splicing" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Alternative
splicing</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">include_graphics</span>(<span class="st">&quot;./images_sequencing/alternative_splicing.png&quot;</span>)</span></code></pre></div>
<div class="figure">
<img src="images_sequencing/alternative_splicing.png" alt="Alternative splicing allows a single gene in Eucaryotes to code for multiple proteins. During alternative splicing particular exons of a gene may be included or excluded from the premature messenger RNA when producing the messenger RNA (mRNA) from that gene. Hence exons can be joined in different combinations, leading to different (alternative) mRNA strands. The proteins translated from alternatively spliced mRNAs thus differ in their amino acid sequence and, often, in their biological functions (Source: Wikipedia)" width="500" />
<p class="caption">
Alternative splicing allows a single gene in Eucaryotes to code for
multiple proteins. During alternative splicing particular exons of a
gene may be included or excluded from the premature messenger RNA when
producing the messenger RNA (mRNA) from that gene. Hence exons can be
joined in different combinations, leading to different (alternative)
mRNA strands. The proteins translated from alternatively spliced mRNAs
thus differ in their amino acid sequence and, often, in their biological
functions (Source: Wikipedia)
</p>
</div>
</div>
</div>
<div id="sequencing-technology" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Sequencing
technology</h1>
<ul>
<li>Measuring mRNA molecules typically happens through sequencing.</li>
<li>The technology continues to evolve at an incredible speed. The data
output of so-called `next generation’ sequencing machines has more than
doubled each year! Simultaneously, the cost of sequencing (in terms of $
per Gigabase) is dropping. Each year, we’re able to sequence more for
less money, providing more information, as well as also computational
and statistical challenges.</li>
<li>This tremendous technological revolution has revolutionized biology,
and genomic sequencing is now a core component of the modern-day
biologist’s toolkit.</li>
<li>The large majority of sequencing data is generated using
sequencing-by-synthesis using machines produced by the company Illumina.
While new players such as Pacific Biosciences and Oxford Nanopore have
entered the scene, these are typically most useful for (but not limited
to) DNA sequencing rather than gene expression studies, owing to their
capability of sequencing long molecules.</li>
</ul>
<div class="figure">
<img src="images_sequencing/seqTechnology_throughput.png" alt="Figure: The data output revolution of sequencing machines. Image from Illumina documentation." width="698" />
<p class="caption">
Figure: The data output revolution of sequencing machines. Image from
Illumina documentation.
</p>
</div>
<div id="from-sample-to-data-analysis" class="section level2"
number="2.1">
<h2><span class="header-section-number">2.1</span> From sample to data
analysis</h2>
<div class="figure">
<img src="images_sequencing/seqWorkflow3.png" alt="Figure: The sequencing workflow. Image adapted from Van den Berge et al. (2019)." width="1291" />
<p class="caption">
Figure: The sequencing workflow. Image adapted from Van den Berge et
al. (2019).
</p>
</div>
</div>
<div id="illumina-sequencing" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Illumina
Sequencing</h2>
<div class="figure">
<img src="images_sequencing/illuminaSeq-step1-6.jpg" alt="Figure: Illumina sequencing steps 1-6 (Source: Illumina)" width="318" />
<p class="caption">
Figure: Illumina sequencing steps 1-6 (Source: Illumina)
</p>
</div>
<div class="figure">
<img src="images_sequencing/illuminaSeq-step7-12.jpg" alt="Figure: Illumina sequencing steps 1-6 (Source: Illumina)" width="318" />
<p class="caption">
Figure: Illumina sequencing steps 1-6 (Source: Illumina)
</p>
</div>
</div>
<div id="single-vs-paired-end" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Single vs paired
end</h2>
<div class="figure">
<img src="images_sequencing/singleEndVsPairedEnd.png" alt="Figure: Single vs paired end sequencing. In contrast to single-end sequencing, paired-end sequencing allows users to sequence both ends of a fragment (Image adopted from Zhernakova et al., 2013, doi.org/10.1371/journal.pgen.1003594)." width="1223" />
<p class="caption">
Figure: Single vs paired end sequencing. In contrast to single-end
sequencing, paired-end sequencing allows users to sequence both ends of
a fragment (Image adopted from Zhernakova et al., 2013,
doi.org/10.1371/journal.pgen.1003594).
</p>
</div>
</div>
<div id="the-sequencing-workflow-with-details" class="section level2"
number="2.4">
<h2><span class="header-section-number">2.4</span> The sequencing
workflow with Details</h2>
<p>Library preparation steps</p>
<ol style="list-style-type: decimal">
<li>First, the biological <strong>samples of interest are
collected</strong>. Owing to the maturity of different protocols for
sequencing, several types of biological input samples are amenable to
sequencing, such as frozen tissues or FFPE-preserved samples.</li>
<li>The <strong>mRNA molecules from our sample are captured</strong>.
This typically involves cell lysis in order to release the mRNA
molecules from within the cells. The mRNA molecules are most often
captured using (i) polyA-capture to select for polyadenylated RNA, or
(ii) ribosomal depletion, where ribosomal and transfer RNAs are
depleted, and so also non-polyA-mRNA molecules may be captured, such as
long non-coding RNAs. In the case of `targeted sequencing’, where
relevant molecules are of main interest (e.g., a gene panel), these
targets can be specifically targeted in this step.</li>
<li><strong>Fragmentation</strong> of captured molecules. The captured
molecules are fragmented, either chemically or mechanically. The
appropriate size of fragments depends on the sequencing machines, but is
often in the range of 300 - 500bp.</li>
<li><strong>Reverse transcription</strong>. Current dominant sequencing
machines only sequence double-stranded DNA molecules. Therefore, in
order to measure single-stranded mRNA, we must first reverse transcribe
these molecules to a double-stranded complementary (cDNA) molecule.</li>
<li><strong>Adapter ligation</strong>. Adapters are oligonucleotides
(short sequences of nucleotides) that are platform-specific sequences
for fragment recognition by the sequencing machines. These are added
either to the 3’ or 5’ end of the cDNA molecules or used as primers in
the reverse transcription reaction. The final cDNA library consists of
cDNA inserts flanked by an adapter sequence on each end.</li>
<li><strong>PCR amplification</strong>. To increase concentration,
several PCR reactions are performed.</li>
<li>Loading the amplified cDNA library on the
<strong>sequencing</strong> machine. Find out how
sequencing-by-synthesis works through <a
href="https://www.youtube.com/watch?v=fCd6B5HRaZ8">this video</a>. Note
that the video shows paired-end sequencing, where a number of basepairs
are sequenced at each end of the fragment. All previous steps together
are described as `sample prep’ in that video.</li>
</ol>
<div class="figure">
<img src="images_sequencing/seqWorkflow1_clean.png" alt="Figure: The sequencing workflow. Image adapted from Van den Berge et al. (2019)." width="822" />
<p class="caption">
Figure: The sequencing workflow. Image adapted from Van den Berge et
al. (2019).
</p>
</div>
<p>Note that several variants of library preparation protocols are
available. The most important ones are:</p>
<ul>
<li>Single-end vs paired-end sequencing: In single-end sequencing, a
single end (3’ or 5’) of the cDNA fragment is sequenced. In paired-end
sequencing, both ends are sequenced, and depending on the size of the
fragment, the reads may or may not overlap.</li>
<li>Strand-specific protocols: Some library preparation protocols allow
measuring strand specificity, where the strand information (i.e.,
sense/antisense) of each read can be preserved.</li>
</ul>
</div>
<div id="the-sequencing-output-files" class="section level2"
number="2.5">
<h2><span class="header-section-number">2.5</span> The sequencing output
files</h2>
<ul>
<li>The typical output of a sequencing machine we will be working with
are FASTA or FASTQ files for each sample. Each of these files are
several gigbases large and contain millions of sequences, which we will
call <strong>reads</strong>. For paired-end sequencing, there are two
files for each sample, one for each end of the sequenced fragments.</li>
<li>The difference between a FASTA file and a FASTQ file, is that while
FASTA files only store the results of base calls (sequences), FASTQ
files also store the quality score of each base call (i.e., each called
nucleotide), which can be useful in downstream analyses such as mapping
or variant calling.</li>
<li>A FASTQ file contains four lines for each sequenced read:
<ol style="list-style-type: decimal">
<li>Sequence identifier line, starting with @.</li>
<li>The sequence.</li>
<li>Another sequence identifier line, now starting with +.</li>
<li>Quality scores.</li>
</ol></li>
</ul>
<div class="figure">
<img src="images_sequencing/fastqLine.png" alt="Figure: One read in a FASTQ file. Slide courtesy by Charlotte Soneson." width="5333" />
<p class="caption">
Figure: One read in a FASTQ file. Slide courtesy by Charlotte Soneson.
</p>
</div>
<p>As you’ll have noticed, the base call quality scores are encoded as
ASCII characters for efficient storage. These ASCII characters can be
converted into integers called Phred scores, which are logarithmically
related to the probability of an erroneous base call.</p>
<div id="quality-score" class="section level3" number="2.5.1">
<h3><span class="header-section-number">2.5.1</span> Quality score</h3>
<p>The Phred Score is a quality score <span class="math display">\[Q =
-10 \log_{10} p\]</span>, with p the probability on an incorrect base
call</p>
<table>
<thead>
<tr class="header">
<th align="right">Phred Score</th>
<th align="left">Probability on incorrect base call</th>
<th align="left">Base call accuracy</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">10</td>
<td align="left">1/10</td>
<td align="left">90%</td>
</tr>
<tr class="even">
<td align="right">20</td>
<td align="left">1/100</td>
<td align="left">99%</td>
</tr>
<tr class="odd">
<td align="right">30</td>
<td align="left">1/1000</td>
<td align="left">99.9%</td>
</tr>
<tr class="even">
<td align="right">40</td>
<td align="left">1/10000</td>
<td align="left">99.99%</td>
</tr>
<tr class="odd">
<td align="right">50</td>
<td align="left">1/10000</td>
<td align="left">99.999%</td>
</tr>
</tbody>
</table>
<p>Phred score encoded as an ASCII letter. E.g. Phred+33:</p>
<table style="width:100%;">
<colgroup>
<col width="8%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
</colgroup>
<tbody>
<tr class="odd">
<td align="left">Phred</td>
<td align="left">0</td>
<td align="left">1</td>
<td align="left">2</td>
<td align="left">3</td>
<td align="left">4</td>
<td align="left">5</td>
<td align="left">6</td>
<td align="left">7</td>
<td align="left">8</td>
<td align="left">9</td>
<td align="left">10</td>
<td align="left">11</td>
<td align="left">12</td>
<td align="left">13</td>
<td align="left">14</td>
<td align="left">15</td>
<td align="left">16</td>
<td align="left">17</td>
<td align="left">18</td>
<td align="left">19</td>
<td align="left">20</td>
<td align="left">21</td>
<td align="left">22</td>
<td align="left">23</td>
<td align="left">24</td>
<td align="left">25</td>
<td align="left">26</td>
<td align="left">27</td>
<td align="left">28</td>
<td align="left">29</td>
<td align="left">30</td>
<td align="left">31</td>
<td align="left">32</td>
<td align="left">33</td>
<td align="left">34</td>
<td align="left">35</td>
<td align="left">36</td>
<td align="left">37</td>
<td align="left">38</td>
<td align="left">39</td>
<td align="left">40</td>
</tr>
<tr class="even">
<td align="left">Phred + 33</td>
<td align="left">33</td>
<td align="left">34</td>
<td align="left">35</td>
<td align="left">36</td>
<td align="left">37</td>
<td align="left">38</td>
<td align="left">39</td>
<td align="left">40</td>
<td align="left">41</td>
<td align="left">42</td>
<td align="left">43</td>
<td align="left">44</td>
<td align="left">45</td>
<td align="left">46</td>
<td align="left">47</td>
<td align="left">48</td>
<td align="left">49</td>
<td align="left">50</td>
<td align="left">51</td>
<td align="left">52</td>
<td align="left">53</td>
<td align="left">54</td>
<td align="left">55</td>
<td align="left">56</td>
<td align="left">57</td>
<td align="left">58</td>
<td align="left">59</td>
<td align="left">60</td>
<td align="left">61</td>
<td align="left">62</td>
<td align="left">63</td>
<td align="left">64</td>
<td align="left">65</td>
<td align="left">66</td>
<td align="left">67</td>
<td align="left">68</td>
<td align="left">69</td>
<td align="left">70</td>
<td align="left">71</td>
<td align="left">72</td>
<td align="left">73</td>
</tr>
<tr class="odd">
<td align="left">ASCII</td>
<td align="left">!</td>
<td align="left">”</td>
<td align="left">#</td>
<td align="left">$</td>
<td align="left">%</td>
<td align="left">&amp;</td>
<td align="left">’</td>
<td align="left">(</td>
<td align="left">)</td>
<td align="left">*</td>
<td align="left">+</td>
<td align="left">,</td>
<td align="left">-</td>
<td align="left">.</td>
<td align="left">/</td>
<td align="left">0</td>
<td align="left">1</td>
<td align="left">2</td>
<td align="left">3</td>
<td align="left">4</td>
<td align="left">5</td>
<td align="left">6</td>
<td align="left">7</td>
<td align="left">8</td>
<td align="left">9</td>
<td align="left">:</td>
<td align="left">;</td>
<td align="left">&lt;</td>
<td align="left">=</td>
<td align="left">&gt;</td>
<td align="left">?</td>
<td align="left">@</td>
<td align="left">A</td>
<td align="left">B</td>
<td align="left">C</td>
<td align="left">D</td>
<td align="left">E</td>
<td align="left">F</td>
<td align="left">G</td>
<td align="left">H</td>
<td align="left">I</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="preprocessing-of-raw-sequencing-data" class="section level1"
number="3">
<h1><span class="header-section-number">3</span> Preprocessing of raw
sequencing data</h1>
<p>After sequencing, we typically do a quality control (QC) check to
verify the quality of the samples. During QC check, aberrant samples due
to e.g. degraded mRNA can be detected.</p>
<p>The sequencing reads on their own contain a lot of information, but
are most useful if we would be able to assign sequencing reads to
genomic features (genes, exons, transcripts, etc.), i.e., for each
sequencing read we will try to derive the (set of) feature(s) that could
have plausibly produced the fragment through the process of gene
expression. This process is called <strong>mapping</strong>. Most often
we map reads to genes.</p>
<div class="figure">
<img src="images_sequencing/seqWorkflow2.png" alt="Figure: An updated sequencing workflow, including sequencing and mapping. Image adapted from Van den Berge et al. (2019)." width="1288" />
<p class="caption">
Figure: An updated sequencing workflow, including sequencing and
mapping. Image adapted from Van den Berge et al. (2019).
</p>
</div>
<div id="quality-control" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Quality control</h2>
<p>During quality control, diagnostic plots are created for each sample
in order to determine its quality. The most popular QC tool for bulk
RNA-seq data is <a
href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>.
If many samples are sequenced, then <a
href="https://multiqc.info/">MultiQC</a> can be used to aggregate the QC
checks across samples in a conveniently organized overview.</p>
<p>The FastQC website provides interesting example reports for us to
look at and compare against. Here are example reports of <a
href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html">high-quality
Illumina data</a> and <a
href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html">low-quality
Illumina data</a>.</p>
</div>
<div id="read-trimming" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Read Trimming</h2>
<ul>
<li><p>Read Trimming</p>
<ul>
<li>Adaptor sequence</li>
<li>Bar code</li>
<li>(deteriorating bases at the end of reads)</li>
<li>often already done by the sequencing provider.</li>
<li>remaining polyA tails</li>
</ul></li>
<li><p>Read filtering</p>
<ul>
<li>low quality reads</li>
<li>PhiX reads (should be removed already by sequence provider) in<br />
</li>
<li>RNA-seq never remove duplicates because they can occur for highly
expressed transcripts</li>
</ul></li>
<li><p>Perform fastQC again</p></li>
</ul>
</div>
<div id="mapping" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Mapping</h2>
<ul>
<li>Mapping is a critical step in the interpretation of RNA-seq data,
where we are attributing reads to genomic features.</li>
<li>Allows us to measure how strong a feature such as a gene is
expressed: the number of reads mapping to a gene serve as a proxy for
how high that gene has been expressed in the sample.</li>
<li>While this opens the door to many opportunities, mapping is
hard.</li>
<li>We are <strong>typically unable to assign each individual read
uniquely</strong> to one specific gene; some reads cannot be
unambiguously mapped and are compatible with multiple genes. These reads
are said to be ‘multi-mapping’.</li>
</ul>
<p>Finally, a note on terminology. In this text we will use the words
‘read’ or ‘fragment’ (referring to the fragmented mRNA molecule being
sequenced) to designate a datum, note that this could be either a single
read (in single-end sequencing) or a read pair (in paired-end
sequencing). The literature may also use these words interchangeably,
although ‘fragment’ seems better at avoiding ambiguity between
single-end reads and paired-end read pairs.</p>
<div id="reference-files" class="section level3" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Reference
files</h3>
<ul>
<li>The alignment most often relies on a <strong>reference
genome</strong> of the species, which can be considered a
‘representative example’ of the genome sequence of that species.
Reference genomes are contiuously updated and released
periodically.</li>
<li>Reference genomes can be freely downloaded from several providers,
for example <a
href="http://www.ensembl.org/info/data/ftp/index.html">Ensembl</a> or <a
href="https://www.gencodegenes.org/human/">Gencode</a>.</li>
<li>Along with a reference genome, an annotation GFF or GTF file defines
the coordinates of specific genomic features.</li>
<li>While here we will focus on reference-based alignment, i.e.,
alignment where a reference genome or transcriptome is available, note
that a <em>de novo</em> construction of a reference transcriptome is
also possible, where the reference may be constructed from the observed
sequencing reads.</li>
</ul>
<div class="figure">
<embed src="./images_sequencing/humanReferenceGenome.pdf" title="slide courtesy Charlotte Soneson" width="100%" type="application/pdf" />
<p class="caption">
slide courtesy Charlotte Soneson
</p>
</div>
<div class="figure">
<img src="images_sequencing/referenceGenome.png" alt="Figure: A reference sequence of human chr1." width="787" />
<p class="caption">
Figure: A reference sequence of human chr1.
</p>
</div>
<div class="figure">
<embed src="./images_sequencing/humanReferenceTranscriptome.pdf" title="slide courtesy Charlotte Soneson" width="100%" type="application/pdf" />
<p class="caption">
slide courtesy Charlotte Soneson
</p>
</div>
<div class="figure">
<img src="images_sequencing/gtfFile.png" alt="Figure: An example GTF file." width="1084" />
<p class="caption">
Figure: An example GTF file.
</p>
</div>
<ul>
<li>More recently, mapping of RNA-seq data occurs more often against a
<strong>reference transcriptome</strong>, which is a reference file
containing the sequences all known isoforms of a particular species,
e.g., using <a
href="https://www.nature.com/articles/nbt.3519">kallisto</a> or <a
href="https://www.nature.com/articles/nmeth.4197">Salmon</a>.</li>
<li>The set of spliced transcripts is much smaller than the entire
genome, and therefore mapping against a reference transcriptome is
typically <strong>fast and memory efficient</strong>.</li>
<li>However, <a
href="https://www.biorxiv.org/content/10.1101/2021.05.05.442755v1">it
has been noted</a> that mapping against a reference transcriptome may
also introduce spurious expression for genes that are not expressed.
These observations can be explained by <strong>intronic reads</strong>
that share some sequence similarity with transcripts, and could map to
spliced transcript sequences. Recent methods, such as <a
href="https://www.biorxiv.org/content/10.1101/2021.06.29.450377v1">alevin-fry</a>,
avoid this by expanding the reference transcriptome to also include
intronic sequences.</li>
</ul>
</div>
<div id="alignment-based-workflows" class="section level3"
number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> Alignment-based
workflows</h3>
<ul>
<li>Traditionally, alignment-based workflows have been used to map
reads, where one tries to find the exact coordinates a read maps to on
the reference genome or the reference transcriptome.</li>
<li>Note that due to alternative splicing, reads do not necessarily map
contiguously on a reference genome, as a read can overlap with a
splicing junction, where an intron has been excised. When mapping
against a transcriptome, however, reads should be mapping
contiguously.</li>
<li>A main challenge in spliced alignment against a reference genome is
the proper alignment of reads that span a splice junction, especially
when these junctions are not annotated a priori. Indeed, in spliced
alignment reads can be split at any nucleotide, and the corresponding
subsequences can map several thousands of basepairs apart. Meanwhile,
the main challenge in unspliced alignment to a transcriptome is the
redundant sequence among related transcripts in the transcriptome, which
often leads to a high multi-mapping rate (i.e., reads that cannot be
unambiguously assigned to a single transcript).</li>
<li>Spliced alignment against a genome is therefore computationally a
much harder task. Since the transcript sequences are already spliced
when aligning to a reference transcriptome, reads should align
contiguously, and many of the computationally expensive steps and
heuristics can be avoided, there.</li>
</ul>
<div class="figure">
<img src="images_sequencing/splicedAlignment.png" alt="Figure: Unspliced and spliced alignment. Figure from Van den Berge et al. (2019)." width="802" />
<p class="caption">
Figure: Unspliced and spliced alignment. Figure from Van den Berge et
al. (2019).
</p>
</div>
</div>
<div id="alignment-free-workflows" class="section level3"
number="3.3.3">
<h3><span class="header-section-number">3.3.3</span> Alignment-free
workflows</h3>
<ul>
<li>Modern approaches avoid mapping each fragment individually (i.e., do
not attempt to find the exact coordinates of a read’s origin), and
instead posit a probabilistic model where transcript abundances are
typically defined using its constituent <span
class="math inline">\(k\)</span>-mers. These methods are sometimes
referred to as <em>lightweight</em>.</li>
<li>A <span class="math inline">\(k\)</span>-mer is a short sequence of
nucleotides of length <span class="math inline">\(k\)</span>. The space
of possible <span class="math inline">\(k\)</span>-mers and the
corresponding transcripts can be precomputed in advance using the
reference transcriptome, providing a computational advantage as it only
needs to be computed once.</li>
<li>For each fragment, the transcripts its <span
class="math inline">\(k\)</span>-mers are compatible with is searched
for using an indexed (efficiently searchable) transcriptome. The set of
compatible transcripts is called the ‘<span
class="math inline">\(k\)</span>-compatibility class’, ‘equivalence
class’ or ‘transcript compatibility class’ of the fragment.</li>
</ul>
<div class="figure">
<img src="images_sequencing/kallisto_equivalenceClasses.png" alt="Figure: Overview of kallisto, image from Bray et al. (2016)." width="478" />
<p class="caption">
Figure: Overview of kallisto, image from Bray et al. (2016).
</p>
</div>
<div class="figure">
<img src="images_sequencing/kallisto_equivalenceClasses_caption.png" alt="Figure: Overview of kallisto, image from Bray et al. (2016)." width="538" />
<p class="caption">
Figure: Overview of kallisto, image from Bray et al. (2016).
</p>
</div>
</div>
</div>
<div id="abundance-quantification" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Abundance
quantification</h2>
<p>Given a set of mappings, using either alignment-based or
alignment-free workflows, the estimation of expression of a
gene/transcript/exon may occur in several ways.</p>
<p><strong>Counting</strong>:</p>
<ul>
<li>In alignment-based workflows, one could do a direct counting of
fragments at the gene-level, counting the number fragments mapping to
each gene. This has been the dominant approach for the first decade of
RNA-seq data, often obtained using reference genome alignments.</li>
<li>Many heuristic choices need to be made: Do we count a fragment as
soon as it intersects with the gene’s coordinates, or do we require the
full fragment to map to the gene? Do we count intronic reads? Do we
count multi-mapping reads?</li>
</ul>
<div class="figure">
<img src="images_sequencing/readCounting.png" alt="Figure: Gene- and exon-level read counting. Image adapted from Charlotte Soneson." width="2844" />
<p class="caption">
Figure: Gene- and exon-level read counting. Image adapted from Charlotte
Soneson.
</p>
</div>
<p><strong>Estimation</strong>:</p>
<ul>
<li>Abundance quantification is more recently starting to shift from
counting towards using statistical models to estimate the expression
counts for a feature, which in this case is typically a transcript.</li>
<li>This approach is amenable to alignment-free workflows, since the
number of fragments in each equivalence class are sufficient statistics
for the abundance quantification, meaning that they contain all
information needed to estimate the parameters of the statistical model,
and hence the feature-level abundances. Since the expression counts in
this case are estimated, they are not necessarily integer counts, and
will be referred to as ‘estimated counts’.</li>
<li>In order to derive these, the EM-algorithm is often used, although
other approaches have been used by tools like Salmon. A big advantage of
the estimation approach is that it probabilistically assigns fragments
to transcripts, thereby automatically dealing with multi-mapping reads.
The total number of fragments mapping to each transcript is then the sum
of all fragment-level probabilities to be assigned to that respective
transcript.</li>
</ul>
<div class="figure">
<img src="images_sequencing/abundanceEMAlgorithm.png" alt="Figure: Abundance quantification using the EM algorithm. Figure from Van den Berge et al. (2019)." width="1300" />
<p class="caption">
Figure: Abundance quantification using the EM algorithm. Figure from Van
den Berge et al. (2019).
</p>
</div>
<div id="abundance-metrics" class="section level3" number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> Abundance
metrics</h3>
<ul>
<li>For simplicity, we have only been talking about feature-level counts
as in sums of fragments. However, this is merely one metric that can be
used as a proxy for expression, and several others exist.</li>
<li>Most of these were introduced to attempt to make the abundances more
comparable across samples or features, as compared to the simple counts.
These mainly serve to correct for technical biases such as transcript
length and sequencing depth, both of which have significant impact on
the observed counts.</li>
<li>Indeed, for a gene with the same mRNA concentration in two samples,
sequencing one sample deeper, will on average result in a higher
count.</li>
<li>Likewise, for two transcripts with the same mRNA concentration but
different transcript lengths, one will tend to observe more fragments
from the longer transcript due to the fragmentation step in the RNA-seq
protocol, where longer transcripts can be split into more fragments of
appropriate length.</li>
</ul>
<p>Below we introduce several relevant abundance metrics, but note that
most data analysis methods we will discuss in this course will work with
(estimated) counts. In what follows, let <span
class="math inline">\(Y_{fi}\)</span> denote the random variable
representing the expression counts of feature <span
class="math inline">\(f\)</span> in sample <span
class="math inline">\(i\)</span> (obtained either as a simple sum of
fragments or estimated using lightweight approaches), and let <span
class="math inline">\(N_i = \sum_f Y_{fi}\)</span> denote the sequencing
depth of sample <span class="math inline">\(i\)</span>.</p>
<ul>
<li><strong>Counts per million (CPM)</strong> are the counts one could
expect to observe if the sample was sequenced to a depth of one
million.</li>
</ul>
<p><span class="math display">\[ CPM_{fi} = \frac{Y_{fi}}{N_i} 10^6
\]</span></p>
<ul>
<li><strong>Transcripts per million (TPM)</strong> refers to the
concentration or proportion of your feature in the sample. TPMs take
into account the length of the feature, which is often reformulated into
an <em>effective length</em> <span
class="math inline">\(l_{fi}^{(eff)}\)</span>, relating to the number of
possible start sites that a feature may have in order to generate
fragments of a typical length observed in your dataset. This typical
length is often calculated using the observed fragment length
distribution from the data and defined as <span class="math display">\[
l_{fi}^{(eff)} = l_{f} - \hat{\bar{F}}_i + 1,\]</span> where <span
class="math inline">\(l_{f}\)</span> is the total length of a feature in
terms of number of nucleotides, and <span
class="math inline">\(\hat{\bar{F}}_i\)</span> is the estimated average
fragment length in sample <span class="math inline">\(i\)</span>. We can
use this to define <span class="math display">\[ TPM_{fi} =
\frac{Y_{fi}}{l_{fi}^{(eff)}} \left( \frac{1}{\sum_f
\frac{Y_{fi}}{l_{fi}^{(eff)}}} \right) 10^6.\]</span> Note that the
first part of the right-hand-side (RHS), <span
class="math inline">\(\frac{Y_{fi}}{l_{fi}^{(eff)}}\)</span> is the
expression counts normalized for the length of the feature. This
measure, however, is still affected by the sequencing depth, which is
then alleviated by dividing by the sum of the length-normalized counts
across all features, i.e., <span class="math inline">\(\sum_f
\frac{Y_{fi}}{l_{fi}^{(eff)}}\)</span>. TPMs hence normalize for the
feature length as well as sequencing depth.</li>
</ul>
</div>
<div id="the-final-countdown" class="section level3" number="3.4.2">
<h3><span class="header-section-number">3.4.2</span> The final
countdown</h3>
<p>Once abundances have been quantified, the (estimated) counts are
typically stored in a count matrix, with genes spanning the rows and
samples spanning the columns. This count matrix forms the basis of most
downstream analyses to interpret RNA-seq data, and it will be the main
object we will be working with in the following lectures.</p>
<div class="figure">
<img src="images_sequencing/seqWorkflow3.png" alt="Figure: An updated sequencing workflow. Image adapted from Van den Berge et al. (2019)." width="1291" />
<p class="caption">
Figure: An updated sequencing workflow. Image adapted from Van den Berge
et al. (2019).
</p>
</div>
</div>
</div>
</div>
<div id="references" class="section level1" number="4">
<h1><span class="header-section-number">4</span> References</h1>
</div>

<div id="rmd-source-code">---
title: "Introduction to RNA-seq sequencing: Preprocessing"
author: "Koen Van den Berge & Lieven Clement"
date: "Last edited on `r format(Sys.time(), '%d %B, %Y')`"
output: 
  html_document:
    toc: true
    toc_float: true
  pdf_document:
    toc: true
    number_sections: true
    latex_engine: xelatex
---

```{r functions, include=FALSE}
# A function for captioning and referencing images
fig <- local({
    i <- 0
    ref <- list()
    list(
        cap=function(refName, text) {
            i <<- i + 1
            ref[[refName]] <<- i
            paste("Figure ", i, ": ", text, sep="")
        },
        ref=function(refName) {
            ref[[refName]]
        })
})
``` 

```{r, echo=FALSE, eval=TRUE}
suppressPackageStartupMessages({
  library(knitr)
  library(rmarkdown)
  library(ggplot2)
  library(tidyverse)
})
```

# The study of gene expression

 - The first part of this course focussed on proteomics, studying the concentration of proteins in biological samples. We have seen that identification of proteins and measuring their respective concentrations are extremely challenging, leading to many technological and statistical challenges in order to interpret these data.
 
 - In the second part of the course, we will focus on measuring gene expression, i.e., measuring the concentration of mRNA molecules, that may eventually be translated into proteins, but may also have functions on their own.

```{r fig.cap='Central Paradigm of Biology: a gene, a specific region in the DNA, is first transcribed into RNA and then into proteins. Note, that for RNA-genes the RNA molecule is the end product itself, which is referred to as non-coding RNA (ncRNA) (Source: Wikipedia)'}
include_graphics("./images_sequencing/centralParadigm.png")
```

## Alternative splicing 

```{r fig.cap='Alternative splicing allows a single gene in Eucaryotes to code for multiple proteins. During alternative splicing particular exons of a gene may be included or excluded from the premature messenger RNA when producing the messenger RNA (mRNA) from that gene. Hence exons can be joined in different combinations, leading to different (alternative) mRNA strands. The proteins translated from alternatively spliced mRNAs thus differ in their amino acid sequence and, often, in their biological functions (Source: Wikipedia)'}
include_graphics("./images_sequencing/alternative_splicing.png")
```

# Sequencing technology

 - Measuring mRNA molecules typically happens through sequencing.
 - The technology continues to evolve at an incredible speed. The data output of so-called `next generation' sequencing machines has more than doubled each year! Simultaneously, the cost of sequencing (in terms of $ per Gigabase) is dropping. Each year, we're able to sequence more for less money, providing more information, as well as also computational and statistical challenges.
 - This tremendous technological revolution has revolutionized biology, and genomic sequencing is now a core component of the modern-day biologist's toolkit.
 - The large majority of sequencing data is generated using sequencing-by-synthesis using machines produced by the company Illumina. While new players such as Pacific Biosciences and Oxford Nanopore have entered the scene, these are typically most useful for (but not limited to) DNA sequencing rather than gene expression studies, owing to their capability of sequencing long molecules.

```{r, echo=FALSE, fig.cap=paste("Figure: The data output revolution of sequencing machines. Image from Illumina documentation.")}
# All defaults
include_graphics("./images_sequencing/seqTechnology_throughput.png")
```

## From sample to data analysis

```{r, echo=FALSE, fig.cap=paste("Figure: The sequencing workflow. Image adapted from Van den Berge et al. (2019).")}
include_graphics("./images_sequencing/seqWorkflow3.png")
```

## Illumina Sequencing 

```{r, echo=FALSE, fig.cap=paste("Figure: Illumina sequencing steps 1-6 (Source: Illumina)")}
include_graphics("./images_sequencing/illuminaSeq-step1-6.jpg")
```
```{r, echo=FALSE, fig.cap=paste("Figure: Illumina sequencing steps 1-6 (Source: Illumina)")}
include_graphics("./images_sequencing/illuminaSeq-step7-12.jpg")
```

## Single vs paired end

```{r, echo=FALSE, fig.cap=paste("Figure: Single vs paired end sequencing. In contrast to single-end sequencing, paired-end sequencing allows users to sequence both ends of a fragment (Image adopted from Zhernakova et al., 2013, doi.org/10.1371/journal.pgen.1003594).")}
include_graphics("./images_sequencing/singleEndVsPairedEnd.png")
```

## The sequencing workflow with Details

Library preparation steps

1. First, the biological **samples of interest are collected**. Owing to the maturity of different protocols for sequencing, several types of biological input samples are amenable to sequencing, such as frozen tissues or FFPE-preserved samples.
2. The **mRNA molecules from our sample are captured**. This typically involves cell lysis in order to release the mRNA molecules from within the cells. The mRNA molecules are most often captured using (i) polyA-capture to select for polyadenylated RNA, or (ii) ribosomal depletion, where ribosomal and transfer RNAs are depleted, and so also non-polyA-mRNA molecules may be captured, such as long non-coding RNAs. In the case of `targeted sequencing', where relevant molecules are of main interest (e.g., a gene panel), these targets can be specifically targeted in this step.
3. **Fragmentation** of captured molecules. The captured molecules are fragmented, either chemically or mechanically. The appropriate size of fragments depends on the sequencing machines, but is often in the range of 300 - 500bp.
4. **Reverse transcription**. Current dominant sequencing machines only sequence double-stranded DNA molecules. Therefore, in order to measure single-stranded mRNA, we must first reverse transcribe these molecules to a double-stranded complementary (cDNA) molecule.
5. **Adapter ligation**. Adapters are oligonucleotides (short sequences of nucleotides) that are platform-specific sequences for fragment recognition by the sequencing machines. These are added either to the 3' or 5' end of the cDNA molecules or used as primers in the reverse transcription reaction. The final cDNA library consists of cDNA inserts flanked by an adapter sequence on each end. 
6. **PCR amplification**. To increase concentration, several PCR reactions are performed.
7. Loading the amplified cDNA library on the **sequencing** machine. Find out how sequencing-by-synthesis works through [this video](https://www.youtube.com/watch?v=fCd6B5HRaZ8). Note that the video shows paired-end sequencing, where a number of basepairs are sequenced at each end of the fragment. All previous steps together are described as `sample prep' in that video.

```{r, echo=FALSE, fig.cap=paste("Figure: The sequencing workflow. Image adapted from Van den Berge et al. (2019).")}
include_graphics("./images_sequencing/seqWorkflow1_clean.png")
```

Note that several variants of library preparation protocols are available. The most important ones are:

 - Single-end vs paired-end sequencing: In single-end sequencing, a single end (3' or 5') of the cDNA fragment is sequenced. In paired-end sequencing, both ends are sequenced, and depending on the size of the fragment, the reads may or may not overlap.
 - Strand-specific protocols: Some library preparation protocols allow measuring strand specificity, where the strand information (i.e., sense/antisense) of each read can be preserved.

## The sequencing output files

- The typical output of a sequencing machine we will be working with are FASTA or FASTQ files for each sample. Each of these files are several gigbases large and contain millions of sequences, which we will call **reads**. For paired-end sequencing, there are two files for each sample, one for each end of the sequenced fragments.
 - The difference between a FASTA file and a FASTQ file, is that while FASTA files only store the results of base calls (sequences), FASTQ files also store the quality score of each base call (i.e., each called nucleotide), which can be useful in downstream analyses such as mapping or variant calling.
 - A FASTQ file contains four lines for each sequenced read:
    1. Sequence identifier line, starting with @.
    2. The sequence.
    3. Another sequence identifier line, now starting with +.
    4. Quality scores.

```{r, echo=FALSE, fig.cap=paste("Figure: One read in a FASTQ file. Slide courtesy by Charlotte Soneson.")}
include_graphics("./images_sequencing/fastqLine.png")
```

As you'll have noticed, the base call quality scores are encoded as ASCII characters for efficient storage. These ASCII characters can be converted into integers called Phred scores, which are logarithmically related to the probability of an erroneous base call.

### Quality score

The Phred Score is a quality score 
$$Q = -10 \log_{10} p$$, with p the probability on an incorrect base call


```{r echo=FALSE}
data.frame("Phred Score"=seq(10,50,10), 
           "Probability on incorrect base call"=c(paste0("1/",c("10","100","1000","10000","10000"))),
           "Base call accuracy" = paste0((1-1/10^(1:5))*100,"%")
          ) %>%  knitr::kable(.,col.names = c("Phred Score","Probability on incorrect base call", "Base call accuracy"))
```
Phred score encoded as an ASCII letter. E.g. Phred+33: 

```{r echo=FALSE}
rbind("Phred"=0:40,"Phred + 33"=(0:40)+33,"ASCII"=sapply(as.raw((0:40)+33),rawToChar)) %>%knitr::kable(.)
```

# Preprocessing of raw sequencing data

After sequencing, we typically do a quality control (QC) check to verify the quality of the samples. During QC check, aberrant samples due to e.g. degraded mRNA can be detected.

The sequencing reads on their own contain a lot of information, but are most useful if we would be able to assign sequencing reads to genomic features (genes, exons, transcripts, etc.), i.e., for each sequencing read we will try to derive the (set of) feature(s) that could have plausibly produced the fragment through the process of gene expression. This process is called **mapping**. Most often we map reads to genes. 

```{r, echo=FALSE, fig.cap=paste("Figure: An updated sequencing workflow, including sequencing and mapping. Image adapted from Van den Berge et al. (2019).")}
include_graphics("./images_sequencing/seqWorkflow2.png")
```

## Quality control

During quality control, diagnostic plots are created for each sample in order to determine its quality. The most popular QC tool for bulk RNA-seq data is [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/). If many samples are sequenced, then [MultiQC](https://multiqc.info/) can be used to aggregate the QC checks across samples in a conveniently organized overview.

The FastQC website provides interesting example reports for us to look at and compare against. Here are example reports of [high-quality Illumina data](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html) and [low-quality Illumina data](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html).

## Read Trimming 

- Read Trimming 

    - Adaptor sequence
    - Bar code
    - (deteriorating bases at the end of reads)
    - often already done by the sequencing provider. 
    - remaining polyA tails

- Read filtering
    
    - low quality reads
    - PhiX reads (should be removed already by sequence provider) in      
    - RNA-seq never remove duplicates because they can occur for highly expressed transcripts

- Perform fastQC again

## Mapping

 - Mapping is a critical step in the interpretation of RNA-seq data, where we are attributing reads to genomic features. 
 - Allows us to measure how strong a feature such as a gene is expressed: the number of reads mapping to a gene serve as a proxy for how high that gene has been expressed in the sample. 
 - While this opens the door to many opportunities, mapping is hard.
 - We are **typically unable to assign each individual read uniquely** to one specific gene; some reads cannot be unambiguously mapped and are compatible with multiple genes. These reads are said to be 'multi-mapping'.
 
Finally, a note on terminology. In this text we will use the words 'read' or 'fragment' (referring to the fragmented mRNA molecule being sequenced) to designate a datum, note that this could be either a single read (in single-end sequencing) or a read pair (in paired-end sequencing). The literature may also use these words interchangeably, although 'fragment' seems better at avoiding ambiguity between single-end reads and paired-end read pairs.

### Reference files

 - The alignment most often relies on a **reference genome** of the species, which can be considered a 'representative example' of the genome sequence of that species. Reference genomes are contiuously updated and released periodically.
 - Reference genomes can be freely downloaded from several providers, for example [Ensembl](http://www.ensembl.org/info/data/ftp/index.html) or [Gencode](https://www.gencodegenes.org/human/).
 - Along with a reference genome, an annotation GFF or GTF file defines the coordinates of specific genomic features.
 - While here we will focus on reference-based alignment, i.e., alignment where a reference genome or transcriptome is available, note that a *de novo* construction of a reference transcriptome is also possible, where the reference may be constructed from the observed sequencing reads.

```{r refGenome, echo=FALSE, fig.cap="slide courtesy Charlotte Soneson", out.width='100%'}
knitr::include_graphics("./images_sequencing/humanReferenceGenome.pdf")
```

 

```{r, echo=FALSE, fig.cap=paste("Figure: A reference sequence of human chr1.")}
include_graphics("./images_sequencing/referenceGenome.png")
```


```{r refTranscriptome, echo=FALSE, fig.cap="slide courtesy Charlotte Soneson", out.width='100%'}
knitr::include_graphics("./images_sequencing/humanReferenceTranscriptome.pdf")
```

```{r, echo=FALSE, fig.cap=paste("Figure: An example GTF file.")}
include_graphics("./images_sequencing/gtfFile.png")
```

 - More recently, mapping of RNA-seq data occurs more often against a **reference transcriptome**, which is a reference file containing the sequences all known isoforms of a particular species, e.g., using [kallisto](https://www.nature.com/articles/nbt.3519) or [Salmon](https://www.nature.com/articles/nmeth.4197).
 - The set of spliced transcripts is much smaller than the entire genome, and therefore mapping against a reference transcriptome is typically **fast and memory efficient**.
 - However, [it has been noted](https://www.biorxiv.org/content/10.1101/2021.05.05.442755v1) that mapping against a reference transcriptome may also introduce spurious expression for genes that are not expressed. These observations can be explained by **intronic reads** that share some sequence similarity with transcripts, and could map to spliced transcript sequences. Recent methods, such as [alevin-fry](https://www.biorxiv.org/content/10.1101/2021.06.29.450377v1), avoid this by expanding the reference transcriptome to also include intronic sequences.

### Alignment-based workflows

 - Traditionally, alignment-based workflows have been used to map reads, where one tries to find the exact coordinates a read maps to on the reference genome or the reference transcriptome.
 - Note that due to alternative splicing, reads do not necessarily map contiguously on a reference genome, as a read can overlap with a splicing junction, where an intron has been excised. When mapping against a transcriptome, however, reads should be mapping contiguously.
  - A main challenge in spliced alignment against a reference genome is the proper alignment of reads that span a splice junction, especially when these junctions are not annotated a priori. Indeed, in spliced alignment reads can be split at any nucleotide, and the corresponding subsequences can map several thousands of basepairs apart. Meanwhile, the main challenge in unspliced alignment to a transcriptome is the redundant sequence among related transcripts in the transcriptome, which often leads to a high multi-mapping rate (i.e., reads that cannot be unambiguously assigned to a single transcript).
 - Spliced alignment against a genome is therefore computationally a much harder task. Since the transcript sequences are already spliced when aligning to a reference transcriptome, reads should align contiguously, and many of the computationally expensive steps and heuristics can be avoided, there.

```{r, echo=FALSE, fig.cap=paste("Figure: Unspliced and spliced alignment. Figure from Van den Berge et al. (2019).")}
include_graphics("./images_sequencing/splicedAlignment.png")
```

### Alignment-free workflows

 - Modern approaches avoid mapping each fragment individually (i.e., do not attempt to find the exact coordinates of a read's origin), and instead posit a probabilistic model where transcript abundances are typically defined using its constituent $k$-mers. These methods are sometimes referred to as *lightweight*.
 - A $k$-mer is a short sequence of nucleotides of length $k$. The space of possible $k$-mers and the corresponding transcripts can be precomputed in advance using the reference transcriptome, providing a computational advantage as it only needs to be computed once.
 - For each fragment, the transcripts its $k$-mers are compatible with is searched for using an indexed (efficiently searchable) transcriptome. The set of compatible transcripts is called the '$k$-compatibility class', 'equivalence class' or 'transcript compatibility class' of the fragment.
 
```{r, echo=FALSE, fig.cap=paste("Figure: Overview of kallisto, image from Bray et al. (2016).")}
include_graphics("./images_sequencing/kallisto_equivalenceClasses.png")
include_graphics("./images_sequencing/kallisto_equivalenceClasses_caption.png")
```
 
## Abundance quantification

Given a set of mappings, using either alignment-based or alignment-free workflows, the estimation of expression of a gene/transcript/exon may occur in several ways.

**Counting**:

 - In alignment-based workflows, one could do a direct counting of fragments at the gene-level, counting the number fragments mapping to each gene. This has been the dominant approach for the first decade of RNA-seq data, often obtained using reference genome alignments. 
 - Many heuristic choices need to be made: Do we count a fragment as soon as it intersects with the gene's coordinates, or do we require the full fragment to map to the gene? Do we count intronic reads? Do we count multi-mapping reads?
 
```{r, echo=FALSE, fig.cap=paste("Figure: Gene- and exon-level read counting. Image adapted from Charlotte Soneson.")}
include_graphics("./images_sequencing/readCounting.png")
```
 
**Estimation**: 
 
 - Abundance quantification is more recently starting to shift from counting towards using statistical models to estimate the expression counts for a feature, which in this case is typically a transcript. 
 - This approach is amenable to alignment-free workflows, since the number of fragments in each equivalence class are sufficient statistics for the abundance quantification, meaning that they contain all information needed to estimate the parameters of the statistical model, and hence the feature-level abundances. Since the expression counts in this case are estimated, they are not necessarily integer counts, and will be referred to as 'estimated counts'. 
 - In order to derive these, the EM-algorithm is often used, although other approaches have been used by tools like Salmon. A big advantage of the estimation approach is that it probabilistically assigns fragments to transcripts, thereby automatically dealing with multi-mapping reads. The total number of fragments mapping to each transcript is then the sum of all fragment-level probabilities to be assigned to that respective transcript.
 
```{r, echo=FALSE, fig.cap=paste("Figure: Abundance quantification using the EM algorithm. Figure from Van den Berge et al. (2019).")}
include_graphics("./images_sequencing/abundanceEMAlgorithm.png")
```

### Abundance metrics

 - For simplicity, we have only been talking about feature-level counts as in sums of fragments. However, this is merely one metric that can be used as a proxy for expression, and several others exist. 
 - Most of these were introduced to attempt to make the abundances more comparable across samples or features, as compared to the simple counts. These mainly serve to correct for technical biases such as transcript length and sequencing depth, both of which have significant impact on the observed counts. 
  - Indeed, for a gene with the same mRNA concentration in two samples, sequencing one sample deeper, will on average result in a higher count. 
  - Likewise, for two transcripts with the same mRNA concentration but different transcript lengths, one will tend to observe more fragments from the longer transcript due to the fragmentation step in the RNA-seq protocol, where longer transcripts can be split into more fragments of appropriate length.


Below we introduce several relevant abundance metrics, but note that most data analysis methods we will discuss in this course will work with (estimated) counts. In what follows, let $Y_{fi}$ denote the random variable representing the expression counts of feature $f$ in sample $i$ (obtained either as a simple sum of fragments or estimated using lightweight approaches), and let $N_i = \sum_f Y_{fi}$ denote the sequencing depth of sample $i$.

 * **Counts per million (CPM)** are the counts one could expect to observe if the sample was sequenced to a depth of one million.
 
\[ CPM_{fi} = \frac{Y_{fi}}{N_i} 10^6 \]

 * **Transcripts per million (TPM)** refers to the concentration or proportion of your feature in the sample. TPMs take into account the length of the feature, which is often reformulated into an *effective length* $l_{fi}^{(eff)}$, relating to the number of possible start sites that a feature may have in order to generate fragments of a typical length observed in your dataset. This typical length is often calculated using the observed fragment length distribution from the data and defined as $$ l_{fi}^{(eff)} = l_{f} - \hat{\bar{F}}_i + 1,$$ where $l_{f}$ is the total length of a feature in terms of number of nucleotides, and $\hat{\bar{F}}_i$ is the estimated average fragment length in sample $i$. We can use this to define $$ TPM_{fi} = \frac{Y_{fi}}{l_{fi}^{(eff)}} \left( \frac{1}{\sum_f \frac{Y_{fi}}{l_{fi}^{(eff)}}} \right) 10^6.$$ Note that the first part of the right-hand-side (RHS), $\frac{Y_{fi}}{l_{fi}^{(eff)}}$ is the expression counts normalized for the length of the feature. This measure, however, is still affected by the sequencing depth, which is then alleviated by dividing by the sum of the length-normalized counts across all features, i.e., $\sum_f \frac{Y_{fi}}{l_{fi}^{(eff)}}$. TPMs hence normalize for the feature length as well as sequencing depth.

### The final countdown

Once abundances have been quantified, the (estimated) counts are typically stored in a count matrix, with genes spanning the rows and samples spanning the columns. This count matrix forms the basis of most downstream analyses to interpret RNA-seq data, and it will be the main object we will be working with in the following lectures.


```{r, echo=FALSE, fig.cap=paste("Figure: An updated sequencing workflow. Image adapted from Van den Berge et al. (2019).")}
include_graphics("./images_sequencing/seqWorkflow3.png")
```

# References


</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("sequencing_intro.Rmd");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
