<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Koen Van den Berge &amp; Lieven Clement" />


<title>Introduction to Sequencing: Working with RNA-seq count data</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="intro.html">1.1 Position</a>
    </li>
    <li>
      <a href="recapGeneralLinearModel.html">1.2 Recap linear model</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="docs/martens_proteomics_bioinformatics.pdf">1. Proteomics Bioinformatics</a>
    </li>
    <li>
      <a href="TargetDecoy.html">2.1 Identification - FDR - QC</a>
    </li>
    <li>
      <a href="pda_quantification_preprocessing.html">2.2.1. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">2.2.2. Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">2.3.1. Quantification - Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="pda_blocking_wrapup.html">2.3.2. Wrap-up DA - Blocking</a>
    </li>
    <li>
      <a href="technicalDetailsProteomics.html">2.3.3 Technical Details</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencing_intro.html">1.  Introduction to RNA-seq</a>
    </li>
    <li>
      <a href="sequencing_countData.html">2.  Working with count data and GLMs</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="multipleRegression_KPNA2.html">1 Intro KPNA2</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="https://www.compomics.com/bioinformatics-for-proteomics/">1. Bioinformatics for Proteomics</a>
    </li>
    <li>
      <a href="TargetDecoyTutorial.html">2.1. Target Decoy QC</a>
    </li>
    <li>
      <a href="pda_tutorialPreprocessing.html">2.2. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">2.3. Quantification - Design</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencingTutorial1_mapping.html">1. Read mapping</a>
    </li>
    <li>
      <a href="sequencingTutorial2_DE.html">2. Working with Count Data</a>
    </li>
    <li>
      <a href="scRNA-seq-Tutorial.html">3. Single Cell RNA-seq Data Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Introduction to Sequencing: Working with
RNA-seq count data</h1>
<h4 class="author">Koen Van den Berge &amp; Lieven Clement</h4>
<h4 class="date">Last edited on 16 September, 2024</h4>

</div>


<p>In this lecture we will introduce the main principles of working with
count data, and how to model these using generalized linear models
(GLMs). We focus on introducing the concept of generalized linear
models, and how they can be applied to genomics data analysis.</p>
<div id="the-poisson-distribution" class="section level1" number="1">
<h1><span class="header-section-number">1</span> The Poisson
distribution</h1>
<ul>
<li>The Poisson distribution is a typical count distribution that is
generally popular and fairly easy to work with. It is defined by a
single parameter: its mean <span class="math inline">\(\mu\)</span>. For
a Poisson distributed random variable <span
class="math inline">\(Y_i\)</span> with observations <span
class="math inline">\(i \in \{1, \ldots, n\}\)</span>, its variance is
equal to its mean. That is, if</li>
</ul>
<p><span class="math display">\[Y_i \sim Poi(\mu)\]</span> , then</p>
<p><span class="math display">\[E(Y_i) = Var(Y_i) = \mu.\]</span></p>
<ul>
<li><p>This immediately shows an important feature of count data: the
<strong>mean-variance relationship</strong>. Indeed, in count data, the
variance will always be a function of the mean.</p></li>
<li><p>This is quite intuitive. Consider the following example. You have
two bird cages, where in one bird cage there are <span
class="math inline">\(10\)</span> birds, while in the other there are
<span class="math inline">\(100\)</span> birds. You let a sample of
people look to the number of birds in either one of the cages. It seems
unlikely that a person in front of the 10-bird cage would come up with
an estimate of <span class="math inline">\(3\)</span>, while it seems
likely that someone in front of the 100-bird cage would come up with an
estimate of <span class="math inline">\(80\)</span>. Even though the
difference from the true value is the same, the exact value has an
impact on the plausible deviation around it.</p></li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">11</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>y1 <span class="ot">&lt;-</span> <span class="fu">rpois</span>(<span class="at">n=</span><span class="dv">500</span>, <span class="at">lambda=</span><span class="dv">10</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>y2 <span class="ot">&lt;-</span> <span class="fu">rpois</span>(<span class="at">n=</span><span class="dv">500</span>, <span class="at">lambda=</span><span class="dv">100</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">qplot</span>(y1, <span class="at">main=</span><span class="st">&quot;Poisson(10)&quot;</span>, <span class="at">geom =</span> <span class="st">&quot;histogram&quot;</span>, <span class="at">binwidth=</span>.<span class="dv">5</span>,<span class="at">center=</span><span class="dv">0</span>)</span></code></pre></div>
<p><img src="sequencing_countData_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">qplot</span>(y2, <span class="at">main=</span><span class="st">&quot;Poisson(100)&quot;</span>, <span class="at">geom =</span> <span class="st">&quot;histogram&quot;</span>, <span class="at">binwidth=</span>.<span class="dv">5</span>,<span class="at">center=</span><span class="dv">0</span>)</span></code></pre></div>
<p><img src="sequencing_countData_files/figure-html/unnamed-chunk-1-2.png" width="672" /></p>
<div id="the-poisson-distribution-in-rna-seq" class="section level2"
number="1.1">
<h2><span class="header-section-number">1.1</span> The Poisson
distribution in RNA-seq</h2>
<ul>
<li>In RNA-seq, technical replicates represent different aliquots of the
same sample being sequenced repeatedly. The underlying true expression
of a gene can hence safely be assumed to be equal across these technical
replicates.</li>
<li><a href="https://genome.cshlp.org/content/18/9/1509">Marioni <em>et
al.</em> (2008)</a> have shown that, for most genes, the distribution of
observed gene expression counts across technical replicates follow a
Poisson distribution. A small proportion of genes (<span
class="math inline">\(\sim 0.5\%\)</span>) do not follow this Poisson
model, however, and actually show evidence for <em>‘extra-Poisson
variation’</em>.</li>
</ul>
<div class="figure">
<img src="images_sequencing/marioniFigs_cropped.png" alt="Figure: Technical replication in RNA-seq. Figures from Marioni et al. (2008)." width="1996" />
<p class="caption">
Figure: Technical replication in RNA-seq. Figures from Marioni et
al. (2008).
</p>
</div>
</div>
<div id="relative-uncertainty-for-poisson-distributed-random-variables"
class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Relative uncertainty
for Poisson distributed random variables</h2>
<p>Take a minute to consider the following question:</p>
<ul>
<li>Suppose that we have a solid tumor sample from a cancer patient, as
well as a sample of surrounding healthy tissue. For each sample, we have
three technical replicates at our disposal. Let <span
class="math inline">\(Y_{grt}\)</span> denote the observed gene
expression values of gene <span class="math inline">\(g\)</span> in
replicate <span class="math inline">\(r \in \{1,2,3\}\)</span> from
tissue <span class="math inline">\(t \in \{0,1\}\)</span>, where <span
class="math inline">\(t=0\)</span> denotes healthy tissue and <span
class="math inline">\(t=1\)</span> denotes tumoral tissue.</li>
<li>We then know that the random variables <span
class="math inline">\(Y_{gr0}\)</span> and <span
class="math inline">\(Y_{gr1}\)</span> follow a Poisson distribution,
and we would estimate its mean as <span
class="math inline">\(\bar{Y}_{g0} = \frac{1}{3} \sum_{r=1}^3
Y_{gr0}\)</span> and <span class="math inline">\(\bar{Y}_{g1} =
\frac{1}{3} \sum_{r=1}^3 Y_{gr1}\)</span>, respectively.</li>
<li>Similar, for another gene <span class="math inline">\(k\)</span>, we
observe <span class="math inline">\(Y_{krt}\)</span>, and estimate <span
class="math inline">\(\bar{Y}_{k0}\)</span> and <span
class="math inline">\(\bar{Y}_{k1}\)</span> correspondingly.</li>
<li>Now suppose that <span class="math inline">\(\beta_{k} =
\bar{Y}_{k1} / \bar{Y}_{k0} = 5\)</span>, but also <span
class="math inline">\(\beta_g = \bar{Y}_{g1} / \bar{Y}_{g0} =
5\)</span>, i.e., the two genes have the same average expression ratio
(also often called a fold-change) across samples. However, they are
differently expressed as <span class="math inline">\(\bar{Y}_{k1} =
100\)</span>, and <span class="math inline">\(\bar{Y}_{g1} = 10\)</span>
(making <span class="math inline">\(\bar{Y}_{k0} = 20\)</span>, and
<span class="math inline">\(\bar{Y}_{g0} = 2\)</span>).</li>
<li>For which of the two genes is the uncertainty on the expression
ratio the highest? In other words, do we trust <span
class="math inline">\(\beta_k\)</span> more or do we trust <span
class="math inline">\(\beta_g\)</span> more?</li>
</ul>
<hr />
<p>Let’s approximate the uncertainty in <span
class="math inline">\(beta_g\)</span> and <span
class="math inline">\(\beta_k\)</span> using simulation:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>N <span class="ot">&lt;-</span> <span class="fl">1e3</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>beta_g <span class="ot">&lt;-</span> beta_k <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="at">length=</span>N)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="cf">for</span>(ii <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>N){</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  ygr1 <span class="ot">&lt;-</span> <span class="fu">rpois</span>(<span class="at">n=</span><span class="dv">3</span>, <span class="at">lambda=</span><span class="dv">10</span>)</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  ygr0 <span class="ot">&lt;-</span> <span class="fu">rpois</span>(<span class="at">n=</span><span class="dv">3</span>, <span class="at">lambda=</span><span class="dv">2</span>)</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  ykr1 <span class="ot">&lt;-</span> <span class="fu">rpois</span>(<span class="at">n=</span><span class="dv">3</span>, <span class="at">lambda=</span><span class="dv">100</span>)</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  ykr0 <span class="ot">&lt;-</span> <span class="fu">rpois</span>(<span class="at">n=</span><span class="dv">3</span>, <span class="at">lambda=</span><span class="dv">20</span>)</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>  beta_g[ii] <span class="ot">&lt;-</span> <span class="fu">mean</span>(ygr1) <span class="sc">/</span> <span class="fu">mean</span>(ygr0)</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>  beta_k[ii] <span class="ot">&lt;-</span> <span class="fu">mean</span>(ykr1) <span class="sc">/</span> <span class="fu">mean</span>(ykr0)</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>}</span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="at">mar=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>))</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="fu">hist</span>(beta_g, <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="at">by=</span><span class="dv">1</span>), <span class="at">xlim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">50</span>))</span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="fu">hist</span>(beta_k, <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="at">by=</span><span class="dv">1</span>), <span class="at">xlim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">50</span>))</span></code></pre></div>
<p><img src="sequencing_countData_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<hr />
<p>We clearly see that the uncertainty on <span
class="math inline">\(\beta_k\)</span> is much lower than on <span
class="math inline">\(\beta_g\)</span>. Even though the variance on the
counts of gene <span class="math inline">\(k\)</span> is higher, since
its mean is higher and it is distributed as a Poisson variable. How do
we explain this?</p>
<ul>
<li>We may explain this by considering the relative uncertainty on the
mean. Relative uncertainty may be defined as the coefficient of
variation <span class="math inline">\(CV = \frac{\sigma}{\mu}\)</span>
(this is, the standard deviation divided by the mean). Indeed, the CV
describes the relative deviation of the distribution relative to its
mean, where a low CV indicates low dispersion with respect to the
mean.</li>
<li>Calculating the CV shows that <strong>the relative uncertainty for
gene <span class="math inline">\(k\)</span> than for gene <span
class="math inline">\(g\)</span>, even though the variance on the raw
counts is higher for gene <span class="math inline">\(k\)</span> than
for gene <span class="math inline">\(g\)</span></strong>.</li>
<li>This lower relative uncertainty on the mean then propagates further
to a lower uncertainty on the fold-change. This basic result will be
essential for understanding the results of a differential expression
analysis!</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">sqrt</span>(<span class="dv">100</span>)<span class="sc">/</span><span class="dv">100</span> <span class="co">#CV for gene k</span></span></code></pre></div>
<pre><code>## [1] 0.1</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">sqrt</span>(<span class="dv">10</span>)<span class="sc">/</span><span class="dv">10</span> <span class="co">#CV for gene g</span></span></code></pre></div>
<pre><code>## [1] 0.3162278</code></pre>
</div>
</div>
<div id="modeling-count-data-generalized-linear-models"
class="section level1" number="2">
<h1><span class="header-section-number">2</span> Modeling count data:
Generalized linear models</h1>
<p>Just like we have modeled protein abundances in the proteomics module
of this course in order to assess differential protein abundance, we can
model gene expression counts to identify genes with differences in
average expression between groups of samples.</p>
<div id="why-we-cant-use-linear-models-to-model-count-data"
class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Why we can(’t) use
linear models to model count data</h2>
<ul>
<li>If we are using a linear model to model a response <span
class="math inline">\(Y_i\)</span>, with <span class="math inline">\(i
\in \{1, \ldots, n\}\)</span> in function of a single covariate <span
class="math inline">\(X_i\)</span>, the linear model can be defined as
follows:</li>
</ul>
<p><span class="math display">\[
\left\{
\begin{array}{ccc}
Y_i &amp; = &amp; \beta_0 + \beta_1 x_i + \epsilon_i \\
Y_i | x_i &amp; \sim &amp; N(\beta_0 + \beta_1 x_i, \sigma^2 ).
\end{array}
\right.
\]</span></p>
<ul>
<li>Or, equivalently, we can write it in matrix form as <span
class="math display">\[
  \left\{
  \begin{array}{ccc}
  Y &amp; = &amp; \mathbf{X}^T \boldsymbol{\beta} + \epsilon \\
  Y | \mathbf{X} &amp; \sim &amp; N(\mathbf{X} \boldsymbol{\beta},
\sigma^2 \mathbf{I}),
  \end{array}
  \right.
  \]</span> where <span class="math inline">\(\mathbf{X}\)</span> now
represents our <span class="math inline">\(n \times p\)</span> design
matrix, with row <span class="math inline">\(i\)</span> corresponding to
observation <span class="math inline">\(i\)</span>.</li>
</ul>
<hr />
<ul>
<li><p>The variance-covariance matrix of <span
class="math inline">\(\mathbf{Y}\)</span> is assumed a diagonal matrix
with <span class="math inline">\(\sigma^2\)</span> on the diagonal
elements and zero everywhere else. This means that the data points are
uncorrelated, and that every observation has the same variance <span
class="math inline">\(\sigma^2\)</span>, also referred to as
homoscedasticity.</p></li>
<li><p>The latter does not hold for count data, due to the mean-variance
relationship. This makes linear models, in its basic form, unsuitable to
model count data.</p></li>
<li><p>In addition, count data are non-negative, while there are no such
constraints in the standard linear model to make sure that our estimates
will be non-negative. Indeed, <span class="math inline">\(\hat{Y}_i =
\hat\mu_i = \mathbf{X}^T_i \hat{\beta} \in ] -\infty,
\infty[\)</span>.</p></li>
</ul>
</div>
<div id="generalized-linear-models" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Generalized linear
models</h2>
<ul>
<li>As the name suggests, generalized linear models (GLMs) extend linear
models. In GLMs, we extend two things with respect to the linear model:
<ul>
<li>The <strong>conditional distribution of the response variable <span
class="math inline">\(Y_i | X_i\)</span></strong> can be assumed to
follow any distribution that belongs to the <strong>exponential
family</strong> of distributions, which includes the Gaussian but also
other commonly known distributions, such as the Binomial, Gamma and
Poisson distribution.</li>
<li>The linear model assumed a linear relationship between <span
class="math inline">\(Y_i\)</span> and <span
class="math inline">\(X_i\)</span>, since we assumed that <span
class="math inline">\(E(Y_i | X_i) = \mathbf{X}^T_i \beta\)</span>. In
GLMs, we will allow a <strong>link function</strong> <span
class="math inline">\(g()\)</span> that links the conditional mean to
the covariates. Hence, in GLMs we have that <span
class="math inline">\(g(E(Y_i | X_i)) = \mathbf{X}^T_i \beta\)</span>.
Note that each family has got a canonical link function, which is the
identity link function <span class="math inline">\(g(\mu) = \mu\)</span>
for Gaussian, the log link function <span class="math inline">\(g(\mu) =
\log \mu\)</span> for Poisson, or the logit link function <span
class="math inline">\(g(\mu) = \log(\frac{\mu}{1-\mu})\)</span> for
Binomial.</li>
</ul></li>
</ul>
<div id="a-poisson-glm" class="section level3" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> A Poisson GLM</h3>
<ul>
<li>We can define a Poisson GLM as follows <span class="math display">\[
  \left\{
  \begin{array}{ccc}
  Y_i &amp; \sim &amp; Poi(\mu_i) \\
  \log \mu_i &amp; = &amp; \eta_i \\
  \eta_i &amp; = &amp; \mathbf{X}^T_i \beta \\
  \end{array}
  \right.
  \]</span> where <span class="math inline">\(Y_i\)</span> is the
response variable, with mean <span class="math inline">\(\mu_i\)</span>,
<span class="math inline">\(\eta_i\)</span> is the linear predictor,
<span class="math inline">\(\mathbf{X}\)</span> is the <span
class="math inline">\(n \times p\)</span> model matrix and <span
class="math inline">\(\beta\)</span> is the <span
class="math inline">\(p \times 1\)</span> matrix of regression
coefficients.</li>
<li>It is insightful to compare this model to a linear model where <span
class="math inline">\(Y_i\)</span> is log-transformed. Indeed, in the
linear model case, we would be modeling <span
class="math inline">\(E(\log Y_i )\)</span>, while in the GLM we are
modeling <span class="math inline">\(\log E(Y_i)\)</span>.</li>
<li>This shows that in the GLM setting we are modeling a transformed
version of the expected value, and after retransforming we can interpret
the fit in terms of the mean of our response variable. In the
transformed linear model, however, we are working with the expected
value of a transformed version of our response variable, and we will not
be able to interpret the fit in terms of the mean (because <span
class="math inline">\(E( \log Y_i) \ne \log E(Y_i)\)</span>. In this
specific case, we would have to resort to interpreting changes in terms
of a geometric mean.</li>
<li>Also note that <span class="math inline">\(\mathbf{X}^T_i \beta \in
]-\infty, \infty[\)</span>, while <span
class="math inline">\(Y_i\)</span> must be non-negative <span
class="math inline">\([0, \infty[\)</span>. The link function helps with
this, since the exponential function transforms any real number to a
non-negative number, i.e., <span
class="math inline">\(\exp(\mathbf{X}^T_i \beta) \in [0,
\infty[\)</span>.</li>
</ul>
</div>
</div>
</div>
<div id="sequencing-data" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Sequencing Data</h1>
<p>In this lecture we will use a subset of the real bulk RNA-seq dataset
from <a
href="https://academic.oup.com/jcem/article/97/12/4631/2536573">Haglund
<em>et al.</em> (2012)</a>.</p>
<p>Lets try to work out the experimental design using the following
paragraph from the Methods section of the paper.</p>
<div class="figure">
<img src="images_sequencing/expDesign_para.png" alt="Figure: A paragraph from the Methods section." width="689" />
<p class="caption">
Figure: A paragraph from the Methods section.
</p>
</div>
<div id="technical-repeats" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Technical
repeats</h2>
<p>There are technical repeats in the data!</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;parathyroidGenesSE&quot;</span>, <span class="at">package=</span><span class="st">&quot;parathyroidSE&quot;</span>)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>se <span class="ot">&lt;-</span> parathyroidGenesSE</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="fu">rm</span>(parathyroidGenesSE)</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> knitr<span class="sc">::</span><span class="fu">kable</span>(.)</span></code></pre></div>
<table>
<colgroup>
<col width="13%" />
<col width="14%" />
<col width="10%" />
<col width="13%" />
<col width="6%" />
<col width="14%" />
<col width="13%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">run</th>
<th align="left">experiment</th>
<th align="left">patient</th>
<th align="left">treatment</th>
<th align="left">time</th>
<th align="left">submission</th>
<th align="left">study</th>
<th align="left">sample</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">SRR479052</td>
<td align="left">SRX140503</td>
<td align="left">1</td>
<td align="left">Control</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308865</td>
</tr>
<tr class="even">
<td align="left">SRR479053</td>
<td align="left">SRX140504</td>
<td align="left">1</td>
<td align="left">Control</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308866</td>
</tr>
<tr class="odd">
<td align="left">SRR479054</td>
<td align="left">SRX140505</td>
<td align="left">1</td>
<td align="left">DPN</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308867</td>
</tr>
<tr class="even">
<td align="left">SRR479055</td>
<td align="left">SRX140506</td>
<td align="left">1</td>
<td align="left">DPN</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308868</td>
</tr>
<tr class="odd">
<td align="left">SRR479056</td>
<td align="left">SRX140507</td>
<td align="left">1</td>
<td align="left">OHT</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308869</td>
</tr>
<tr class="even">
<td align="left">SRR479057</td>
<td align="left">SRX140508</td>
<td align="left">1</td>
<td align="left">OHT</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308870</td>
</tr>
<tr class="odd">
<td align="left">SRR479058</td>
<td align="left">SRX140509</td>
<td align="left">2</td>
<td align="left">Control</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308871</td>
</tr>
<tr class="even">
<td align="left">SRR479059</td>
<td align="left">SRX140510</td>
<td align="left">2</td>
<td align="left">Control</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308872</td>
</tr>
<tr class="odd">
<td align="left">SRR479060</td>
<td align="left">SRX140511</td>
<td align="left">2</td>
<td align="left">DPN</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308873</td>
</tr>
<tr class="even">
<td align="left">SRR479061</td>
<td align="left">SRX140511</td>
<td align="left">2</td>
<td align="left">DPN</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308873</td>
</tr>
<tr class="odd">
<td align="left">SRR479062</td>
<td align="left">SRX140512</td>
<td align="left">2</td>
<td align="left">DPN</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308874</td>
</tr>
<tr class="even">
<td align="left">SRR479063</td>
<td align="left">SRX140513</td>
<td align="left">2</td>
<td align="left">OHT</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308875</td>
</tr>
<tr class="odd">
<td align="left">SRR479064</td>
<td align="left">SRX140513</td>
<td align="left">2</td>
<td align="left">OHT</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308875</td>
</tr>
<tr class="even">
<td align="left">SRR479065</td>
<td align="left">SRX140514</td>
<td align="left">2</td>
<td align="left">OHT</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308876</td>
</tr>
<tr class="odd">
<td align="left">SRR479066</td>
<td align="left">SRX140515</td>
<td align="left">3</td>
<td align="left">Control</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308877</td>
</tr>
<tr class="even">
<td align="left">SRR479067</td>
<td align="left">SRX140516</td>
<td align="left">3</td>
<td align="left">Control</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308878</td>
</tr>
<tr class="odd">
<td align="left">SRR479068</td>
<td align="left">SRX140517</td>
<td align="left">3</td>
<td align="left">DPN</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308879</td>
</tr>
<tr class="even">
<td align="left">SRR479069</td>
<td align="left">SRX140518</td>
<td align="left">3</td>
<td align="left">DPN</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308880</td>
</tr>
<tr class="odd">
<td align="left">SRR479070</td>
<td align="left">SRX140519</td>
<td align="left">3</td>
<td align="left">OHT</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308881</td>
</tr>
<tr class="even">
<td align="left">SRR479071</td>
<td align="left">SRX140520</td>
<td align="left">3</td>
<td align="left">OHT</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308882</td>
</tr>
<tr class="odd">
<td align="left">SRR479072</td>
<td align="left">SRX140521</td>
<td align="left">4</td>
<td align="left">Control</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308883</td>
</tr>
<tr class="even">
<td align="left">SRR479073</td>
<td align="left">SRX140522</td>
<td align="left">4</td>
<td align="left">DPN</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308884</td>
</tr>
<tr class="odd">
<td align="left">SRR479074</td>
<td align="left">SRX140523</td>
<td align="left">4</td>
<td align="left">DPN</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308885</td>
</tr>
<tr class="even">
<td align="left">SRR479075</td>
<td align="left">SRX140523</td>
<td align="left">4</td>
<td align="left">DPN</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308885</td>
</tr>
<tr class="odd">
<td align="left">SRR479076</td>
<td align="left">SRX140524</td>
<td align="left">4</td>
<td align="left">OHT</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308886</td>
</tr>
<tr class="even">
<td align="left">SRR479077</td>
<td align="left">SRX140525</td>
<td align="left">4</td>
<td align="left">OHT</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308887</td>
</tr>
<tr class="odd">
<td align="left">SRR479078</td>
<td align="left">SRX140525</td>
<td align="left">4</td>
<td align="left">OHT</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308887</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>se <span class="sc">%&gt;%</span> </span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  colData <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  as.data.frame <span class="sc">%&gt;%</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  <span class="fu">pull</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>  nlevels</span></code></pre></div>
<pre><code>## [1] 23</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>se <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>  colData <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>  nrow</span></code></pre></div>
<pre><code>## [1] 27</code></pre>
<p>Extract sample names of duplicates!</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>dupl <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">table</span>(<span class="fu">colData</span>(se)<span class="sc">$</span>sample) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> names</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>techreps <span class="ot">&lt;-</span> <span class="fu">assays</span>(se)<span class="sc">$</span>counts[,<span class="fu">colData</span>(se)<span class="sc">$</span>sample<span class="sc">==</span>dupl[<span class="dv">1</span>]]</span></code></pre></div>
<div id="explore-the-data-of-two-technical-repeats"
class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> Explore the data of
two technical repeats</h3>
<p>We expect the counts to be very similar. Indeed, they are based on
the same cell culture, stimulated with the same stimulus and incubated
for the same time.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">qplot</span>(techreps[,<span class="dv">1</span>],techreps[,<span class="dv">2</span>]) <span class="sc">+</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope =</span> <span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Technical repeat 1&quot;</span>) <span class="sc">+</span></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Technical repeat 2&quot;</span>)</span></code></pre></div>
<p><img src="sequencing_countData_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<ul>
<li><p>The plot shows a systematic difference between the counts of both
technical repeats!</p></li>
<li><p>However, plotting the data on the original count scale does not
give a good overview of the data. The plot is dominated by a few very
abundant genes, i.e. genes with very high counts.</p></li>
</ul>
</div>
<div id="explore-the-data-of-two-technical-repeats-1"
class="section level3" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> Explore the data of
two technical repeats</h3>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="fu">qplot</span>(techreps[,<span class="dv">1</span>],techreps[,<span class="dv">2</span>], <span class="at">log=</span><span class="st">&quot;xy&quot;</span>) <span class="sc">+</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope =</span> <span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Technical repeat 1&quot;</span>) <span class="sc">+</span></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Technical repeat 2&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>## Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<p><img src="sequencing_countData_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<ul>
<li><p>The plot clearly shows that the counts for all genes are very
different in the two technical repeats.</p></li>
<li><p>However, we have to look at the plot via the diagonal.</p></li>
<li><p>In omics we therefore often use MA plots. With M (log ratio) and
A (log average).</p></li>
</ul>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>A <span class="ot">&lt;-</span> techreps <span class="sc">%&gt;%</span> log2 <span class="sc">%&gt;%</span> rowMeans</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>M <span class="ot">&lt;-</span> techreps <span class="sc">%&gt;%</span> log2 <span class="sc">%&gt;%</span> <span class="fu">apply</span>(.,<span class="dv">1</span>,diff)</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>w <span class="ot">&lt;-</span> techreps[,<span class="dv">1</span>]<span class="sc">==</span><span class="fu">min</span>(techreps[,<span class="dv">1</span>]) <span class="sc">|</span> techreps[,<span class="dv">2</span>]<span class="sc">==</span><span class="fu">min</span>(techreps[,<span class="dv">2</span>])</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">any</span>(w)) {</span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a>            A[w] <span class="ot">&lt;-</span> <span class="fu">runif</span>(<span class="fu">sum</span>(w), <span class="at">min =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">max =</span> .<span class="dv">1</span>)</span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a>            M[w] <span class="ot">&lt;-</span> <span class="fu">log2</span>(techreps[w,<span class="dv">2</span>] <span class="sc">+</span> <span class="dv">1</span>) <span class="sc">-</span> <span class="fu">log2</span>(techreps[w,<span class="dv">1</span>] <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a>}</span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a>MAplot <span class="ot">&lt;-</span> <span class="fu">qplot</span>(A, M, <span class="at">col=</span>w) <span class="sc">+</span></span>
<span id="cb18-10"><a href="#cb18-10" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span> </span>
<span id="cb18-11"><a href="#cb18-11" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;black&quot;</span>,<span class="st">&quot;orange&quot;</span>)) <span class="sc">+</span></span>
<span id="cb18-12"><a href="#cb18-12" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;A: log2 Average&quot;</span>) <span class="sc">+</span></span>
<span id="cb18-13"><a href="#cb18-13" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;M: log2 Fold Change&quot;</span>)</span>
<span id="cb18-14"><a href="#cb18-14" tabindex="-1"></a></span>
<span id="cb18-15"><a href="#cb18-15" tabindex="-1"></a>MAplot <span class="sc">+</span></span>
<span id="cb18-16"><a href="#cb18-16" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">intercept=</span><span class="dv">0</span>,<span class="at">slope=</span><span class="dv">0</span>)</span></code></pre></div>
<p><img src="sequencing_countData_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>The MA plot clearly shows that the counts of one technical repeat are
systematically higher than that of the other technical repeats.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">colSums</span>(techreps)</span></code></pre></div>
<pre><code>## [1]  5251911 19332369</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>logFCdepth <span class="ot">&lt;-</span> <span class="fu">colSums</span>(techreps) <span class="sc">%&gt;%</span> log2 <span class="sc">%&gt;%</span> diff</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>logFCdepth</span></code></pre></div>
<pre><code>## [1] 1.880104</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="dv">2</span><span class="sc">^</span>logFCdepth</span></code></pre></div>
<pre><code>## [1] 3.681016</code></pre>
<p>The technical repeats differ with a factor 3.7 in sequencing
depth!</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>MAplot <span class="sc">+</span> <span class="fu">geom_abline</span>(<span class="at">intercept =</span> logFCdepth, <span class="at">slope =</span> <span class="dv">0</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="sequencing_countData_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<ul>
<li>This log2 fold change in sequencing depth is also the baseline
around which the log2 fold changes between technical repeats of
individual genes are fluctuating!</li>
</ul>
<p>Hence, we will have to correct for differences in sequencing
depth.</p>
</div>
</div>
<div id="count-scaling-using-glm-offsets" class="section level2"
number="3.2">
<h2><span class="header-section-number">3.2</span> Count scaling using
GLM offsets</h2>
<ul>
<li><p>We have previously discussed count scaling transformations such
as CPM and TPM.</p></li>
<li><p>A more appropriate and natural way when working with GLMs is
through the use of offsets. The general use of an offset is to account
for the ‘effort’ performed in order to gather that observation of the
response variable.</p></li>
</ul>
<p><span class="math inline">\(\rightarrow\)</span> i.e. sample being
sequenced deeper contains more information as compared to a sample being
sequenced relatively shallow. We have more confidence in a count from a
deeply sequenced sample than that from a shallowly sequenced sample. We
can therefore use the sequencing depth <span
class="math inline">\(N_i=\sum_g Y_{gi}\)</span> as offset in the
model.</p>
<ul>
<li><p>Adding an offset to the model is different from adding a new
variable to the model. For each new variable we add, we will estimate
its average effect <span class="math inline">\(\beta\)</span> on the
response variable. When adding an offset, however, we implicitly set
<span class="math inline">\(\beta=1\)</span>.</p></li>
<li><p>Offsets are typically added on the scale of the linear predictor.
Suppose we have a gene g and sample i specific offset <span
class="math inline">\(O_{gi}\)</span>, then we can define our GLM
including the offset as</p>
<p><span class="math display">\[
\left\{
\begin{array}{ccc}
Y_{gi} &amp; \sim &amp; Poi(\mu_{gi}) \\
\log \mu_{gi} &amp; = &amp; \eta_{gi} \\
\eta_{gi} &amp; = &amp; \mathbf{X}^T_i \beta_g + log(O_{gi}) \\
\end{array}
\right.
\]</span></p></li>
</ul>
<p>When we would like to correct for the overall sequencing depth <span
class="math inline">\(O_{gi}=N_i\)</span>. However, better offsets
exist!</p>
<p>Note, that</p>
<p><span class="math display">\[
\mu_{ig}= \exp(\mathbf{X}^T_i \beta_g) \times O_{gi}
\]</span> or <span class="math display">\[
\log\left(\frac{\mu_{ig}}{O_{gi}}\right)= \mathbf{X}^T_i \beta_g
\]</span></p>
</div>
<div id="biological-repeats" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Biological
repeats</h2>
<p>We extract the four biological repeats from the study for the control
treatment at time 24h.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>bioreps <span class="ot">&lt;-</span> <span class="fu">colData</span>(se)<span class="sc">$</span>treatment<span class="sc">==</span><span class="st">&quot;Control&quot;</span> <span class="sc">&amp;</span> <span class="fu">colData</span>(se)<span class="sc">$</span>time<span class="sc">==</span><span class="st">&quot;24h&quot;</span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>biorepCounts <span class="ot">&lt;-</span> <span class="fu">assays</span>(se)<span class="sc">$</span>counts[,bioreps]</span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a><span class="fu">qplot</span>(<span class="fu">rowMeans</span>(biorepCounts),<span class="fu">rowVars</span>(biorepCounts),<span class="at">log=</span><span class="st">&quot;xy&quot;</span>) <span class="sc">+</span></span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>,<span class="at">slope =</span> <span class="dv">1</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">col=</span><span class="st">&quot;orange&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>## Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<pre><code>## Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>## Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<pre><code>## `geom_smooth()` using method = &#39;gam&#39; and formula &#39;y ~ s(x, bs = &quot;cs&quot;)&#39;</code></pre>
<pre><code>## Warning: Removed 37702 rows containing non-finite values (stat_smooth).</code></pre>
<p><img src="sequencing_countData_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<ul>
<li><p>Having data on thousands of genes provides the opportunity to
empirically assess the mean-variance relationship.</p></li>
<li><p>It is clear that the data is overdispersed with respect to the
Poisson distribution (red y=x line). There also seems to be a quadratic
trend of the variance as a function of the mean.</p></li>
</ul>
</div>
</div>
<div id="negative-binomial-distribution" class="section level1"
number="4">
<h1><span class="header-section-number">4</span> Negative binomial
distribution</h1>
<p>The observed quadratic mean variance trend has motivated the use of
the negative binomial distribution to model (bulk) RNA-seq gene
expression data.</p>
<p><span class="math display">\[
  \left\{
  \begin{array}{ccc}
  Y_{gi} &amp; \sim &amp; NB(\mu_{gi},\phi_g) \\
  \log \mu_{gi} &amp; = &amp; \eta_{gi} \\
  \eta_{gi} &amp; = &amp; \mathbf{X}^T_i \beta_g + log(O_{gi}) \\
  \end{array}
  \right.
  \]</span></p>
<p>with</p>
<p><span class="math display">\[
\text{var}[Y_{gi}] = \mu_{gi} + \phi_g \mu_{gi}^2
\]</span></p>
<table>
<colgroup>
<col width="20%" />
<col width="20%" />
<col width="20%" />
<col width="20%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th></th>
<th><span class="math inline">\(\quad\)</span></th>
<th>Seq. technology</th>
<th><span class="math inline">\(\quad\)</span></th>
<th>real expression</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>total variability</td>
<td>=</td>
<td>technical variability</td>
<td>+</td>
<td>biological variability</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\text{var}[Y_{gi}]\)</span></td>
<td>=</td>
<td><span class="math inline">\(\mu_{gi}\)</span></td>
<td>+</td>
<td><span class="math inline">\(\phi_g \mu_{gi}^2\)</span></td>
</tr>
<tr class="odd">
<td>total CV<span class="math inline">\(^2\)</span></td>
<td>=</td>
<td><span class="math inline">\(\frac{1}{\mu_{gi}}\)</span></td>
<td>+</td>
<td><span class="math inline">\(\phi_g\)</span></td>
</tr>
</tbody>
</table>
<div id="what-about-technical-repeats" class="section level2"
number="4.1">
<h2><span class="header-section-number">4.1</span> What about technical
repeats?</h2>
<ul>
<li>Technical repeats are Poisson distributed.</li>
<li>The sum of two Poisson distributed counts is again Poisson. So we
can summarize over technical repeats by taking the sum of the counts.
This enables us to collapse the technical repeats to the level of
biological repeats.</li>
<li>We will illustrate that in the tutorial session.</li>
</ul>
</div>
<div id="estimating-the-overdispersion" class="section level2"
number="4.2">
<h2><span class="header-section-number">4.2</span> Estimating the
overdispersion?</h2>
<ul>
<li>Gene wise: for every single gene, very unstable due to the lack of
data</li>
<li>Common dispersion for all genes</li>
<li>Trended dispersion</li>
<li>Tag-wise: EB shrinkage to a common (trended) dispersion: Borrow
strength across genes (McCarthy &amp; Smyth, 2012, <a
href="https://doi.org/10.1093/nar/gks042">DOI:
10.1093/nar/gks042</a>)</li>
</ul>
<p>In the tutorials we will analyse the entire study and we will focus
on assessing the main research questions: i.e. comparing the early and
late effects of the stimuli, and if the effect of stimuli is changes
over time.</p>
<p>Here, we illustrate the estimation using the edgeR tool on a subset
of control treatment for patients 1, 2 and 3. For this subset only
biological repeats are sequenced and a measurement on time 24 and 48h
has been taken for cells of each patient.<br />
So we can model the study using a simple block design with a time effect
and a block effect for patient.</p>
<ol style="list-style-type: decimal">
<li>Setup the data</li>
</ol>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>seSub <span class="ot">&lt;-</span> se[,<span class="fu">colData</span>(se)<span class="sc">$</span>treatment<span class="sc">==</span><span class="st">&quot;Control&quot;</span><span class="sc">&amp;</span><span class="fu">colData</span>(se)<span class="sc">$</span>patient<span class="sc">%in%</span><span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a><span class="fu">colData</span>(seSub)<span class="sc">$</span>patient <span class="ot">&lt;-</span> <span class="fu">colData</span>(seSub)<span class="sc">$</span>patient <span class="sc">%&gt;%</span> </span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>  as.double <span class="sc">%&gt;%</span> </span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a>  as.factor</span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts =</span> <span class="fu">assays</span>(seSub)<span class="sc">$</span>counts)</span>
<span id="cb33-6"><a href="#cb33-6" tabindex="-1"></a></span>
<span id="cb33-7"><a href="#cb33-7" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>time<span class="sc">+</span>patient,<span class="fu">colData</span>(seSub))</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Typically lowly expressed genes are filtered.</li>
</ol>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(y,design)</span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>y <span class="ot">&lt;-</span> y[keep,]</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Normalisation to correct for differences in library size.</li>
</ol>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(y)</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Estimate the dispersions using empirical Bayes (EB)</li>
</ol>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(y, design)</span></code></pre></div>
<ol start="5" style="list-style-type: decimal">
<li>Estimate gene-wise dispersions without shrinkage for comparison
purposes</li>
</ol>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>yNoEB <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(y, design, <span class="at">prior.df =</span> <span class="dv">0</span>)</span></code></pre></div>
<ol start="6" style="list-style-type: decimal">
<li>Visualisation and comparison of tag-wise (EB) and genewise
dispersion</li>
</ol>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a>o <span class="ot">&lt;-</span> <span class="fu">order</span>(y<span class="sc">$</span>AveLogCPM)</span>
<span id="cb38-2"><a href="#cb38-2" tabindex="-1"></a><span class="fu">data.frame</span>(</span>
<span id="cb38-3"><a href="#cb38-3" tabindex="-1"></a>  <span class="at">AveLogCPM=</span><span class="fu">rep</span>(yNoEB<span class="sc">$</span>AveLogCPM[o],<span class="dv">2</span>),</span>
<span id="cb38-4"><a href="#cb38-4" tabindex="-1"></a>  <span class="at">BCV=</span><span class="fu">sqrt</span>(</span>
<span id="cb38-5"><a href="#cb38-5" tabindex="-1"></a>    <span class="fu">c</span>(yNoEB<span class="sc">$</span>tagwise.dispersion[o],</span>
<span id="cb38-6"><a href="#cb38-6" tabindex="-1"></a>      y<span class="sc">$</span>tagwise.dispersion[o])</span>
<span id="cb38-7"><a href="#cb38-7" tabindex="-1"></a>    ), </span>
<span id="cb38-8"><a href="#cb38-8" tabindex="-1"></a>  <span class="at">method=</span><span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;genewise&quot;</span>,<span class="st">&quot;tagwise&quot;</span>), <span class="at">each=</span><span class="fu">nrow</span>(y))) <span class="sc">%&gt;%</span></span>
<span id="cb38-9"><a href="#cb38-9" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(AveLogCPM,BCV,<span class="at">color=</span>method)) <span class="sc">+</span></span>
<span id="cb38-10"><a href="#cb38-10" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> .<span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb38-11"><a href="#cb38-11" tabindex="-1"></a>  <span class="fu">geom_line</span>(</span>
<span id="cb38-12"><a href="#cb38-12" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">data.frame</span>(</span>
<span id="cb38-13"><a href="#cb38-13" tabindex="-1"></a>      <span class="at">AveLogCPM=</span>y<span class="sc">$</span>AveLogCPM[o],</span>
<span id="cb38-14"><a href="#cb38-14" tabindex="-1"></a>      <span class="at">BCV =</span> <span class="fu">sqrt</span>(y<span class="sc">$</span>trended.dispersion[o]),</span>
<span id="cb38-15"><a href="#cb38-15" tabindex="-1"></a>      <span class="at">method=</span><span class="st">&quot;trended&quot;</span>)) <span class="sc">+</span></span>
<span id="cb38-16"><a href="#cb38-16" tabindex="-1"></a>    <span class="fu">scale_colour_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">&#39;black&#39;</span>,<span class="st">&#39;orange&#39;</span>,<span class="st">&#39;blue&#39;</span>))<span class="sc">+</span></span>
<span id="cb38-17"><a href="#cb38-17" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Biological Coefficient of Variation&quot;</span>) <span class="sc">+</span></span>
<span id="cb38-18"><a href="#cb38-18" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Average log CPM&quot;</span>)</span></code></pre></div>
<p><img src="sequencing_countData_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<p>Note, that</p>
<ul>
<li>The gene-wise dispersion is very variable (black dots).</li>
<li>There is a dispersion - mean trend (blue line)</li>
<li>The tag-wise dispersions (orange dots) are the result of shrinking
the gene-wise dispersion (black dots) to the trend (blue line)</li>
</ul>
<p>By default edgeR provides the following plot</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="fu">plotBCV</span>(y)</span></code></pre></div>
<p><img src="sequencing_countData_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<p>In the methods paper of edgeR different dispersion estimators were
compared and the tag-wise dispersions seemed to provide the best
goodness of fit.</p>
<div class="figure">
<img src="images_sequencing/NARDispersion.png" alt="QQ-plots of goodness of fit statistics using common, trended or empirical Bayes. Image from McCarthy and Smyth, 2012, NAR, DOI: 10.1093/nar/gks042" width="2026" />
<p class="caption">
QQ-plots of goodness of fit statistics using common, trended or
empirical Bayes. Image from McCarthy and Smyth, 2012, NAR, DOI:
10.1093/nar/gks042
</p>
</div>
</div>
</div>
<div id="statistical-inference" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Statistical
Inference</h1>
<ul>
<li><p>Asymptotic statistical tests exist to test if (contrasts of the)
parameters of the GLM are different from zero.</p></li>
<li><p>Implemented in edgeR and DESeq2.</p></li>
<li><p>Again we have to correct for multiple testing !!! FDR correction
is done by default in the edgeR and DESeq2 output</p></li>
</ul>
<ol style="list-style-type: decimal">
<li>Fit the model</li>
</ol>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(y,design)</span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a><span class="fu">head</span>(fit<span class="sc">$</span>coefficients)</span></code></pre></div>
<pre><code>##                 (Intercept)     time48h   patient2   patient3
## ENSG00000000003   -9.349871  0.13638138 -0.5613559 -0.7834884
## ENSG00000000419  -10.391095 -0.09800617  0.1057271  0.1356434
## ENSG00000000457  -10.928625 -0.08069712  0.4491010 -0.1289007
## ENSG00000000460  -10.041597 -0.81035512  0.5044914 -0.5433122
## ENSG00000000938  -14.614695  0.39307169  1.4719170  0.1349905
## ENSG00000000971  -13.512986  0.46252084 -0.2107229  0.5042261</code></pre>
<p>Interpretation of model parameters?</p>
<ol start="2" style="list-style-type: decimal">
<li>Statistical test to assess the time effect</li>
</ol>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a>lrt <span class="ot">&lt;-</span> <span class="fu">glmLRT</span>(fit, <span class="at">coef =</span> <span class="st">&quot;time48h&quot;</span>)</span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a><span class="fu">topTags</span>(lrt)</span></code></pre></div>
<pre><code>## Coefficient:  time48h 
##                     logFC   logCPM       LR       PValue          FDR
## ENSG00000164089  4.284786 3.099739 213.7650 2.072422e-48 3.294944e-44
## ENSG00000148795  3.026969 3.686056 183.3434 9.025281e-42 7.174647e-38
## ENSG00000133110  1.293005 7.185439 164.0009 1.512056e-37 8.013393e-34
## ENSG00000211445  1.450374 8.966014 162.1696 3.798734e-37 1.509902e-33
## ENSG00000107796  1.398750 6.002617 146.0886 1.241600e-33 3.865574e-30
## ENSG00000169239  1.557744 7.377027 145.7683 1.458799e-33 3.865574e-30
## ENSG00000188404 -1.590059 9.851890 141.9293 1.007722e-32 2.288824e-29
## ENSG00000136235  1.375796 5.149565 139.4353 3.537613e-32 7.030564e-29
## ENSG00000163631  1.835300 4.898878 135.1416 3.074462e-31 5.431208e-28
## ENSG00000005189 -1.702284 5.615167 134.7941 3.662515e-31 5.823032e-28</code></pre>
</div>

<div id="rmd-source-code">---
title: 'Introduction to Sequencing: Working with RNA-seq count data'
author: "Koen Van den Berge & Lieven Clement"
date: "Last edited on `r format(Sys.time(), '%d %B, %Y')`"
output: 
  pdf_document:
    toc: true
    number_sections: true
    latex_engine: xelatex
  html_document:
    toc: true
    toc_float: true
    toc_depth: 4
---

```{r functions, include=FALSE}
# A function for captioning and referencing images
fig <- local({
    i <- 0
    ref <- list()
    list(
        cap=function(refName, text) {
            i <<- i + 1
            ref[[refName]] <<- i
            paste("Figure ", i, ": ", text, sep="")
        },
        ref=function(refName) {
            ref[[refName]]
        })
})
``` 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
suppressPackageStartupMessages({
  library(knitr)
  library(rmarkdown)
  library(ggplot2)
  library(tidyverse)
  library(SummarizedExperiment)
  library(edgeR)
  if(!"parathyroidSE" %in% installed.packages()[,1]) BiocManager::install("parathyroidSE")
})
```

In this lecture we will introduce the main principles of working with count data, and how to model these using generalized linear models (GLMs). We focus on introducing the concept of generalized linear models, and how  they can be applied to genomics data analysis.

# The Poisson distribution

 - The Poisson distribution is a typical count distribution that is generally popular and fairly easy to work with. It is defined by a single parameter: its mean $\mu$. For a Poisson distributed random variable $Y_i$ with observations $i \in \{1, \ldots, n\}$, its variance is equal to its mean. That is, if 
 
 $$Y_i \sim Poi(\mu)$$
 , then 
 
 $$E(Y_i) = Var(Y_i) = \mu.$$
 
 - This immediately shows an important feature of count data: the **mean-variance relationship**. Indeed, in count data, the variance will always be a function of the mean.
 
 - This is quite intuitive. Consider the following example. You have two bird cages, where in one bird cage there are $10$ birds, while in the other there are $100$ birds. You let a sample of people look to the number of birds in either one of the cages. It seems unlikely that a person in front of the 10-bird cage would come up with an estimate of $3$, while it seems likely that someone in front of the 100-bird cage would come up with an estimate of $80$. Even though the difference from the true value is the same, the exact value has an impact on the plausible deviation around it.
 
```{r}
set.seed(11)
y1 <- rpois(n=500, lambda=10)
y2 <- rpois(n=500, lambda=100)

par(mfrow = c(1,2))
qplot(y1, main="Poisson(10)", geom = "histogram", binwidth=.5,center=0)
qplot(y2, main="Poisson(100)", geom = "histogram", binwidth=.5,center=0)
```

## The Poisson distribution in RNA-seq

 - In RNA-seq, technical replicates represent different aliquots of the same sample being sequenced repeatedly. The underlying true expression of a gene can hence safely be assumed to be equal across these technical replicates.
 - [Marioni *et al.* (2008)](https://genome.cshlp.org/content/18/9/1509) have shown that, for most genes, the distribution of observed gene expression counts across technical replicates follow a Poisson distribution. A small proportion of genes ($\sim 0.5\%$) do not follow this Poisson model, however, and actually show evidence for *'extra-Poisson variation'*.
 
```{r, echo=FALSE, fig.cap=paste("Figure: Technical replication in RNA-seq. Figures from Marioni et al. (2008).")}
# All defaults
include_graphics("./images_sequencing/marioniFigs_cropped.png")
```

## Relative uncertainty for Poisson distributed random variables

Take a minute to consider the following question:

 - Suppose that we have a solid tumor sample from a cancer patient, as well as a sample of surrounding healthy tissue. For each sample, we have three technical replicates at our disposal. Let $Y_{grt}$ denote the observed gene expression values of gene $g$ in replicate $r \in \{1,2,3\}$ from tissue $t \in \{0,1\}$, where $t=0$ denotes healthy tissue and $t=1$ denotes tumoral tissue. 
 - We then know that the random variables $Y_{gr0}$ and $Y_{gr1}$ follow a Poisson distribution, and we would estimate its mean as $\bar{Y}_{g0} = \frac{1}{3} \sum_{r=1}^3 Y_{gr0}$ and $\bar{Y}_{g1} = \frac{1}{3} \sum_{r=1}^3 Y_{gr1}$, respectively. 
 - Similar, for another gene $k$, we observe $Y_{krt}$, and estimate $\bar{Y}_{k0}$ and $\bar{Y}_{k1}$ correspondingly.
 - Now suppose that $\beta_{k} = \bar{Y}_{k1} / \bar{Y}_{k0} = 5$, but also $\beta_g = \bar{Y}_{g1} / \bar{Y}_{g0} = 5$, i.e., the two genes have the same average expression ratio (also often called a fold-change) across samples. However, they are differently expressed as $\bar{Y}_{k1} = 100$, and $\bar{Y}_{g1} = 10$ (making $\bar{Y}_{k0} = 20$, and $\bar{Y}_{g0} = 2$).
 - For which of the two genes is the uncertainty on the expression ratio the highest? In other words, do we trust $\beta_k$ more or do we trust $\beta_g$ more?
 
 ---

Let's approximate the uncertainty in $beta_g$ and $\beta_k$ using simulation:
 
```{r}
N <- 1e3
beta_g <- beta_k <- vector(length=N)
for(ii in 1:N){
  ygr1 <- rpois(n=3, lambda=10)
  ygr0 <- rpois(n=3, lambda=2)
  ykr1 <- rpois(n=3, lambda=100)
  ykr0 <- rpois(n=3, lambda=20)
  beta_g[ii] <- mean(ygr1) / mean(ygr0)
  beta_k[ii] <- mean(ykr1) / mean(ykr0)
}

par(mfrow=c(1,2), mar=c(4,2,3,1))
hist(beta_g, breaks=seq(0,50,by=1), xlim=c(0,50))
hist(beta_k, breaks=seq(0,50,by=1), xlim=c(0,50))
```
 
 --- 
 
 We clearly see that the uncertainty on $\beta_k$ is much lower than on $\beta_g$. Even though the variance on the counts of gene $k$ is higher, since its mean is higher and it is distributed as a Poisson variable. How do we explain this?
 
 - We may explain this by considering the relative uncertainty on the mean. Relative uncertainty may be defined as the coefficient of variation $CV = \frac{\sigma}{\mu}$ (this is, the standard deviation divided by the mean). Indeed, the CV describes the relative deviation of the distribution relative to its mean, where a low CV indicates low dispersion with respect to the mean.
 - Calculating the CV shows that **the relative uncertainty for gene $k$ than for gene $g$, even though the variance on the raw counts is higher for gene $k$ than for gene $g$**. 
 - This lower relative uncertainty on the mean then propagates further to a lower uncertainty on the fold-change. This basic result will be essential for understanding the results of a differential expression analysis!
 
```{r}
sqrt(100)/100 #CV for gene k

sqrt(10)/10 #CV for gene g
```
 
# Modeling count data: Generalized linear models

Just like we have modeled protein abundances in the proteomics module of this course in order to assess differential protein abundance, we can model gene expression counts to identify genes with differences in average expression between groups of samples.

## Why we can('t) use linear models to model count data

 - If we are using a linear model to model a response $Y_i$, with $i \in \{1, \ldots, n\}$ in function of a single covariate $X_i$, the linear model can be defined as follows:

\[
\left\{
\begin{array}{ccc}
Y_i & = & \beta_0 + \beta_1 x_i + \epsilon_i \\
Y_i | x_i & \sim & N(\beta_0 + \beta_1 x_i, \sigma^2 ).
\end{array}
\right.
\]

 - Or, equivalently, we can write it in matrix form as
  $$
  \left\{
  \begin{array}{ccc}
  Y & = & \mathbf{X}^T \boldsymbol{\beta} + \epsilon \\
  Y | \mathbf{X} & \sim & N(\mathbf{X} \boldsymbol{\beta}, \sigma^2 \mathbf{I}),
  \end{array}
  \right.
  $$
where $\mathbf{X}$ now represents our $n \times p$ design matrix, with row $i$ corresponding to observation $i$.

 ---

 - The variance-covariance matrix of $\mathbf{Y}$ is assumed a diagonal matrix with $\sigma^2$ on the diagonal elements and zero everywhere else. This means that the data points are uncorrelated, and that every observation has the same variance $\sigma^2$, also referred to as homoscedasticity. 
 
 - The latter does not hold for count data, due to the mean-variance relationship. This makes linear models, in its basic form, unsuitable to model count data.
 
 - In addition, count data are non-negative, while there are no such constraints in the standard linear model to make sure that our estimates will be non-negative. 
 Indeed, $\hat{Y}_i = \hat\mu_i = \mathbf{X}^T_i \hat{\beta} \in ] -\infty, \infty[$.

## Generalized linear models

 - As the name suggests, generalized linear models (GLMs) extend linear models. In GLMs, we extend two things with respect to the linear model:
    - The **conditional distribution of the response variable $Y_i | X_i$** can be assumed to follow any distribution that belongs to the **exponential family** of distributions, which includes the Gaussian but also other commonly known distributions, such as the Binomial, Gamma and Poisson distribution.
    - The linear model assumed a linear relationship between $Y_i$ and $X_i$, since we assumed that $E(Y_i | X_i) = \mathbf{X}^T_i \beta$. In GLMs, we will allow a **link function** $g()$ that links the conditional mean to the covariates. Hence, in GLMs we have that $g(E(Y_i | X_i)) = \mathbf{X}^T_i \beta$. Note that each family has got a canonical link function, which is the identity link function $g(\mu) = \mu$ for Gaussian, the log link function $g(\mu) = \log \mu$ for Poisson, or the logit link function $g(\mu) = \log(\frac{\mu}{1-\mu})$ for Binomial.
    
### A Poisson GLM

 - We can define a Poisson GLM as follows
  $$
  \left\{
  \begin{array}{ccc}
  Y_i & \sim & Poi(\mu_i) \\
  \log \mu_i & = & \eta_i \\
  \eta_i & = & \mathbf{X}^T_i \beta \\
  \end{array}
  \right.
  $$
where $Y_i$ is the response variable, with mean $\mu_i$, $\eta_i$ is the linear predictor, $\mathbf{X}$ is the $n \times p$ model matrix and $\beta$ is the $p \times 1$ matrix of regression coefficients.
  - It is insightful to compare this model to a linear model where $Y_i$ is log-transformed. Indeed, in the linear model case, we would be modeling $E(\log Y_i )$, while in the GLM we are modeling $\log E(Y_i)$.
  - This shows that in the GLM setting we are modeling a transformed version of the expected value, and after retransforming we can interpret the fit in terms of the mean of our response variable. In the transformed linear model, however, we are working with the expected value of a transformed version of our response variable, and we will not be able to interpret the fit in terms of the mean (because $E( \log Y_i) \ne \log E(Y_i)$. In this specific case, we would have to resort to interpreting changes in terms of a geometric mean.
  - Also note that $\mathbf{X}^T_i \beta \in ]-\infty, \infty[$, while $Y_i$ must be non-negative $[0, \infty[$. The link function helps with this, since the exponential function transforms any real number to a non-negative number, i.e., $\exp(\mathbf{X}^T_i \beta) \in [0, \infty[$. 

# Sequencing Data

In this lecture we will use a subset of the real bulk RNA-seq dataset from [Haglund *et al.* (2012)](https://academic.oup.com/jcem/article/97/12/4631/2536573).

Lets try to work out the experimental design using the following paragraph from the Methods section of the paper.

```{r, echo=FALSE, fig.cap=paste("Figure: A paragraph from the Methods section.")}
# All defaults
include_graphics("./images_sequencing/expDesign_para.png")
```

## Technical repeats 

There are technical repeats in the data! 

```{r}
data("parathyroidGenesSE", package="parathyroidSE")
se <- parathyroidGenesSE
rm(parathyroidGenesSE)
colData(se) %>% knitr::kable(.)
```

```{r}
se %>% 
  colData %>%
  as.data.frame %>%
  pull(sample) %>% 
  nlevels

se %>% 
  colData %>% 
  nrow
```

Extract sample names of duplicates! 

```{r}
dupl <- which(table(colData(se)$sample) > 1) %>% names
techreps <- assays(se)$counts[,colData(se)$sample==dupl[1]]
```

### Explore the data of two technical repeats

We expect the counts to be very similar. Indeed, they are based on the same cell culture, stimulated with the same stimulus and incubated for the same time. 

```{r}
qplot(techreps[,1],techreps[,2]) +
  geom_abline(intercept = 0, slope = 1) + 
  xlab("Technical repeat 1") +
  ylab("Technical repeat 2")
```

- The plot shows a systematic difference between the counts of both technical repeats!

- However, plotting the data on the original count scale does not give a good overview of the data. The plot is dominated by a few very abundant genes, i.e. genes with very high counts. 

### Explore the data of two technical repeats


```{r}
qplot(techreps[,1],techreps[,2], log="xy") +
  geom_abline(intercept = 0, slope = 1) + 
  xlab("Technical repeat 1") +
  ylab("Technical repeat 2")
```

- The plot clearly shows that the counts for all genes are very different in the two technical repeats. 

- However, we have to look at the plot via the diagonal. 

- In omics we therefore often use MA plots. With M (log ratio) and A (log average). 

```{r}
A <- techreps %>% log2 %>% rowMeans
M <- techreps %>% log2 %>% apply(.,1,diff)
w <- techreps[,1]==min(techreps[,1]) | techreps[,2]==min(techreps[,2])
if (any(w)) {
            A[w] <- runif(sum(w), min = -1, max = .1)
            M[w] <- log2(techreps[w,2] + 1) - log2(techreps[w,1] + 1)
}

MAplot <- qplot(A, M, col=w) +
  theme(legend.position = "none") + 
  scale_color_manual(values = c("black","orange")) +
  xlab("A: log2 Average") +
  ylab("M: log2 Fold Change")

MAplot +
  geom_abline(intercept=0,slope=0)
```

The MA plot clearly shows that the counts of one technical repeat are systematically higher than that of the other technical repeats. 

```{r}
colSums(techreps)
logFCdepth <- colSums(techreps) %>% log2 %>% diff
logFCdepth
2^logFCdepth
```

The technical repeats differ with a factor `r round(2^logFCdepth,1)` in sequencing depth! 


```{r}
MAplot + geom_abline(intercept = logFCdepth, slope = 0, color="red")
```

- This log2 fold change in sequencing depth is also the baseline around which the log2 fold changes between technical repeats of individual genes are fluctuating! 

Hence, we will have to correct for differences in sequencing depth. 

## Count scaling using GLM offsets 

- We have previously discussed count scaling transformations such as CPM and TPM.

- A more appropriate and natural way when working with GLMs is through the use of offsets. The general use of an offset is to account for the ‘effort’ performed in order to gather that observation of the response variable.

$\rightarrow$ i.e. sample being sequenced deeper contains more information as compared to a sample being sequenced relatively shallow. We have more confidence in a count from a deeply sequenced sample than that from a shallowly sequenced sample. We can therefore use the sequencing depth $N_i=\sum_g Y_{gi}$ as offset in the model.


- Adding an offset to the model is different from adding a new variable to the model. For each new variable we add, we will estimate its average effect $\beta$ on the response variable. When adding an offset, however, we implicitly set $\beta=1$.

- Offsets are typically added on the scale of the linear predictor. Suppose we have a gene g and sample i specific offset $O_{gi}$, then we can define our GLM including the offset as

  $$
  \left\{
  \begin{array}{ccc}
  Y_{gi} & \sim & Poi(\mu_{gi}) \\
  \log \mu_{gi} & = & \eta_{gi} \\
  \eta_{gi} & = & \mathbf{X}^T_i \beta_g + log(O_{gi}) \\
  \end{array}
  \right.
  $$

When we would like to correct for the overall sequencing depth $O_{gi}=N_i$. However, better offsets exist!

Note, that 

$$
\mu_{ig}= \exp(\mathbf{X}^T_i \beta_g) \times O_{gi}
$$
or 
$$
\log\left(\frac{\mu_{ig}}{O_{gi}}\right)= \mathbf{X}^T_i \beta_g
$$

## Biological repeats

We extract the four biological repeats from the study for the control treatment at time 24h. 

```{r}
bioreps <- colData(se)$treatment=="Control" & colData(se)$time=="24h"
biorepCounts <- assays(se)$counts[,bioreps]
qplot(rowMeans(biorepCounts),rowVars(biorepCounts),log="xy") +
  geom_abline(intercept = 0,slope = 1,col="red") +
  geom_smooth(col="orange")
```

- Having data on thousands of genes provides the opportunity to empirically assess the mean-variance relationship.

- It is clear that the data is overdispersed with respect to the Poisson distribution (red y=x line). There also seems to be a quadratic trend of the variance as a function of the mean. 

# Negative binomial distribution

The observed quadratic mean variance trend has motivated the use of the negative binomial distribution to model (bulk) RNA-seq gene expression data.

$$
  \left\{
  \begin{array}{ccc}
  Y_{gi} & \sim & NB(\mu_{gi},\phi_g) \\
  \log \mu_{gi} & = & \eta_{gi} \\
  \eta_{gi} & = & \mathbf{X}^T_i \beta_g + log(O_{gi}) \\
  \end{array}
  \right.
  $$

with 

$$ 
\text{var}[Y_{gi}] = \mu_{gi} + \phi_g \mu_{gi}^2 
$$

|   | $\quad$ | Seq. technology  | $\quad$ | real expression  | 
|---|---|---|---|---|
| total variability | = | technical variability | + | biological variability   |
| $\text{var}[Y_{gi}]$ | = | $\mu_{gi}$  | + | $\phi_g \mu_{gi}^2$ | | total CV$^2$ | = | technical CV$^2$ | + | biological CV$^2$ |
| total CV$^2$ | = | $\frac{1}{\mu_{gi}}$ | + | $\phi_g$  |

##  What about technical repeats? 

- Technical repeats are Poisson distributed.
- The sum of two Poisson distributed counts is again Poisson. So we can summarize over technical repeats by taking the sum of the counts. This enables us to collapse the technical repeats to the level of biological repeats. 
- We will illustrate that in the tutorial session.

## Estimating the overdispersion?

- Gene wise: for every single gene, very unstable due to the lack of data 
- Common dispersion for all genes 
- Trended dispersion
- Tag-wise: EB shrinkage to a common (trended) dispersion: Borrow strength across genes (McCarthy & Smyth, 2012, [DOI: 10.1093/nar/gks042](https://doi.org/10.1093/nar/gks042))

In the tutorials we will analyse the entire study and we will focus on assessing the main research questions: i.e. comparing the early and late effects of the stimuli, and if the effect of stimuli is changes over time. 

Here, we illustrate the estimation using the edgeR tool on a subset of control treatment for patients 1, 2 and 3. For this subset only biological repeats are sequenced and a measurement on time 24 and 48h has been taken for cells of each patient.  
So we can model the study using a simple block design with a time effect and a block effect for patient.

1. Setup the data
 

```{r}
seSub <- se[,colData(se)$treatment=="Control"&colData(se)$patient%in%1:3]
colData(seSub)$patient <- colData(seSub)$patient %>% 
  as.double %>% 
  as.factor
y <- DGEList(counts = assays(seSub)$counts)

design <- model.matrix(~time+patient,colData(seSub))
```

2. Typically lowly expressed genes are filtered. 

```{r}
keep <- filterByExpr(y,design)
y <- y[keep,]
```

3. Normalisation to correct for differences in library size.

```{r}
y <- calcNormFactors(y)
```

4. Estimate the dispersions using empirical Bayes (EB)

```{r}
y <- estimateDisp(y, design)
```

5. Estimate gene-wise dispersions without shrinkage for comparison purposes 

```{r}
yNoEB <- estimateDisp(y, design, prior.df = 0)
```

6. Visualisation and comparison of tag-wise (EB) and genewise dispersion 

```{r}
o <- order(y$AveLogCPM)
data.frame(
  AveLogCPM=rep(yNoEB$AveLogCPM[o],2),
  BCV=sqrt(
    c(yNoEB$tagwise.dispersion[o],
      y$tagwise.dispersion[o])
    ), 
  method=rep(c("genewise","tagwise"), each=nrow(y))) %>%
  ggplot(aes(AveLogCPM,BCV,color=method)) +
  geom_point(size = .2) +
  geom_line(
    data = data.frame(
      AveLogCPM=y$AveLogCPM[o],
      BCV = sqrt(y$trended.dispersion[o]),
      method="trended")) +
    scale_colour_manual(values =c('black','orange','blue'))+
  ylab("Biological Coefficient of Variation") +
  xlab("Average log CPM")
```

Note, that 

- The gene-wise dispersion is very variable (black dots).
- There is a dispersion - mean trend (blue line)
- The tag-wise dispersions (orange dots) are the result of shrinking the gene-wise dispersion (black dots) to the trend (blue line) 

By default edgeR provides the following plot 

```{r}
plotBCV(y)
```

In the methods paper of edgeR different dispersion estimators were compared and the tag-wise dispersions seemed to provide the best goodness of fit. 

```{r refGenome, echo=FALSE, fig.cap="QQ-plots of goodness of fit statistics using common, trended or empirical Bayes. Image from McCarthy and Smyth, 2012, NAR, DOI: 10.1093/nar/gks042"}
knitr::include_graphics("./images_sequencing/NARDispersion.png")
```

# Statistical Inference 

- Asymptotic statistical tests exist to test if (contrasts of the) parameters of the GLM are different from zero.

- Implemented in edgeR and DESeq2.

- Again we have to correct for multiple testing !!! FDR correction is done by  default in the edgeR and DESeq2 output 

1. Fit the model

```{r}
fit <- glmFit(y,design)
head(fit$coefficients)
```

Interpretation of model parameters? 


2. Statistical test to assess the time effect 

```{r}
lrt <- glmLRT(fit, coef = "time48h")
topTags(lrt)
```

</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("sequencing_countData.Rmd");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
