<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Introduction to TargetDecoy</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="intro.html">1.1 Position</a>
    </li>
    <li>
      <a href="recapGeneralLinearModel.html">1.2 Recap linear model</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="docs/martens_proteomics_bioinformatics.pdf">1. Proteomics Bioinformatics</a>
    </li>
    <li>
      <a href="TargetDecoy.html">2.1 Identification - FDR - QC</a>
    </li>
    <li>
      <a href="pda_quantification_preprocessing.html">2.2.1. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">2.2.2. Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">2.3.1. Quantification - Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="pda_blocking_wrapup.html">2.3.2. Wrap-up DA - Blocking</a>
    </li>
    <li>
      <a href="technicalDetailsProteomics.html">2.3.3 Technical Details</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencing_intro.html">1.  Introduction to RNA-seq</a>
    </li>
    <li>
      <a href="sequencing_countData.html">2.  Working with count data and GLMs</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="multipleRegression_KPNA2.html">1 Intro KPNA2</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="https://www.compomics.com/bioinformatics-for-proteomics/">1. Bioinformatics for Proteomics</a>
    </li>
    <li>
      <a href="TargetDecoyTutorial.html">2.1. Target Decoy QC</a>
    </li>
    <li>
      <a href="pda_tutorialPreprocessing.html">2.2. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">2.3. Quantification - Design</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencingTutorial1_mapping.html">1. Read mapping</a>
    </li>
    <li>
      <a href="sequencingTutorial2_DE.html">2. Working with Count Data</a>
    </li>
    <li>
      <a href="scRNA-seq-Tutorial.html">3. Single Cell RNA-seq Data Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Introduction to TargetDecoy</h1>
<h4 class="author">Lieven Clement</h4>
<address class="author_afil">
Ghent University<br>
</div>


<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(TargetDecoy)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code></pre></div>
<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p><img src="figures/ProteomicsWorkflow.png" /></p>
<p><img src="figures/peptideSpectralMatching.png" /> Slide courtesy to
Lennart Martens</p>
<div id="why-did-lennart-mention-that-statistics-were-out-of-the-window"
class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Why did Lennart
mention that “statistics were out of the window”</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(TargetDecoy)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&gt; ── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2 ──</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#&gt; ✔ tibble  3.1.7     ✔ dplyr   1.0.9</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&gt; ✔ tidyr   1.2.0     ✔ stringr 1.4.1</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#&gt; ✔ readr   2.1.2     ✔ forcats 0.5.1</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&gt; ✔ purrr   0.3.4     </span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#&gt; ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::filter() masks stats::filter()</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::lag()    masks stats::lag()</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;ModSwissXT&quot;</span>)</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>hlp <span class="ot">&lt;-</span> TargetDecoy<span class="sc">:::</span><span class="fu">.getDF</span>(ModSwissXT) </span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="fu">names</span>(hlp) <span class="ot">&lt;-</span> <span class="fu">names</span>(hlp) <span class="sc">%&gt;%</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>    <span class="fu">str_replace</span>(<span class="at">pattern=</span><span class="st">&quot;:&quot;</span>,<span class="at">replacement=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>hlp <span class="ot">&lt;-</span> hlp <span class="sc">%&gt;%</span> </span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="fu">mutate</span>(<span class="at">omssa_evalue=</span><span class="fu">as.double</span>(omssa_evalue))</span></code></pre></div>
<ul>
<li><p>E-values: Probability that a random candidate peptide produces a
higher score than the observed PSM score.</p></li>
<li><p>Probability that a random candidate peptide produces a higher
score that the observed PSM score for a real search with OMSSA.</p></li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">library</span>(TargetDecoy)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;ModSwissXT&quot;</span>)</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>hlp <span class="ot">&lt;-</span> TargetDecoy<span class="sc">:::</span><span class="fu">.getDF</span>(ModSwissXT) </span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="fu">names</span>(hlp) <span class="ot">&lt;-</span> <span class="fu">names</span>(hlp) <span class="sc">%&gt;%</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>    <span class="fu">str_replace</span>(<span class="at">pattern=</span><span class="st">&quot;:&quot;</span>,<span class="at">replacement=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>hlp <span class="ot">&lt;-</span> hlp <span class="sc">%&gt;%</span> </span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="fu">mutate</span>(<span class="at">omssa_evalue=</span><span class="fu">as.double</span>(omssa_evalue))</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>hlp <span class="sc">%&gt;%</span> </span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(omssa_evalue)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(omssa_evalue)) <span class="sc">+</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;E-value (%)&quot;</span>) <span class="sc">+</span> </span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;frequency&quot;</span>)</span></code></pre></div>
<p><img src="TargetDecoy_files/figure-html/unnamed-chunk-2-1.png" width="432" /></p>
<ul>
<li>E-value we expect for an observed spectrum and a random candidate
peptide.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">evalue=</span><span class="fu">runif</span>(<span class="dv">20000</span>,<span class="dv">0</span>,<span class="dv">100</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(evalue)) <span class="sc">+</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;E-value (%)&quot;</span>) <span class="sc">+</span> </span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;frequency&quot;</span>)</span></code></pre></div>
<p><img src="TargetDecoy_files/figure-html/unnamed-chunk-3-1.png" width="432" /></p>
<ul>
<li>Probability that a random candidate peptide produces a higher score
than the observed PSM score for decoys in a real search with OMSSA.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>hlp <span class="sc">%&gt;%</span> </span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>    <span class="fu">filter</span>(isdecoy <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(omssa_evalue)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(omssa_evalue)) <span class="sc">+</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;E-value (%)&quot;</span>) <span class="sc">+</span> </span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;frequency&quot;</span>)</span></code></pre></div>
<p><img src="TargetDecoy_files/figure-html/unnamed-chunk-4-1.png" width="432" /></p>
<ul>
<li><p>A bad hit is the random hit with the best score so it is also
bound to have a low E-value.</p></li>
<li><p>If we look at E-values for all PSMs they are only useful as a
score.</p></li>
<li><p>We should know the distribution of the maximum score of random
candidate peptides when we want to do the statistics.</p></li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>hlp <span class="sc">%&gt;%</span> </span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(omssa_evalue)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="sc">-</span><span class="fu">log10</span>(omssa_evalue))) <span class="sc">+</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1000</span>,.<span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;-log10(E-value)&quot;</span>) <span class="sc">+</span> </span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;frequency&quot;</span>) <span class="sc">+</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>    <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">20</span>) </span></code></pre></div>
<p><img src="TargetDecoy_files/figure-html/unnamed-chunk-5-1.png" width="432" /></p>
</div>
</div>
<div id="concepts" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Concepts</h1>
<div id="basic-statistical-concepts" class="section level2"
number="2.1">
<h2><span class="header-section-number">2.1</span> Basic Statistical
Concepts</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">names</span>(hlp) <span class="ot">&lt;-</span> <span class="fu">names</span>(hlp) <span class="sc">%&gt;%</span> <span class="fu">str_replace</span>(<span class="at">pattern=</span><span class="st">&quot;-&quot;</span>,<span class="at">replacement =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>hlp <span class="ot">&lt;-</span> hlp <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">ms_gf_specevalue=</span><span class="fu">as.double</span>(ms_gf_specevalue))</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">library</span>(mgcv)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="co">#&gt; Loading required package: nlme</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="co">#&gt; Attaching package: &#39;nlme&#39;</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="co">#&gt; The following object is masked from &#39;package:dplyr&#39;:</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="co">#&gt;     collapse</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="co">#&gt; This is mgcv 1.8-40. For overview type &#39;help(&quot;mgcv-package&quot;)&#39;.</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>dec <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(hlp<span class="sc">$</span>ms_gf_specevalue[hlp<span class="sc">$</span>isdecoy]) <span class="sc">%&gt;%</span> <span class="fu">na.exclude</span>()</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>tar <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(hlp<span class="sc">$</span>ms_gf_specevalue[<span class="sc">!</span>hlp<span class="sc">$</span>isdecoy]) <span class="sc">%&gt;%</span> <span class="fu">na.exclude</span>()</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>breaks <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">30</span>,.<span class="dv">5</span>)</span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a><span class="co">#binWidth &lt;-2</span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a><span class="co">#breaks &lt;- seq(floor(min(c(dec,tar))/binWidth)*binWidth,ceiling(max(c(dec,tar))/binWidth)*binWidth,binWidth)</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a><span class="co">#code if we register the modes by substracting the mode from the target scores and  the decoy scores.</span></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="co">#breaks=seq(-(ceiling(abs(min(c(dec,tar))/binWidth))+.5)*binWidth,(ceiling(max(c(dec,tar))/binWidth)+.5)*binWidth,binWidth)</span></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a>histDec <span class="ot">&lt;-</span> <span class="fu">hist</span>(dec,<span class="at">breaks=</span>breaks,<span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a>histTar <span class="ot">&lt;-</span> <span class="fu">hist</span>(tar,<span class="at">breaks=</span>breaks,<span class="at">plot=</span><span class="cn">FALSE</span>)</span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a>histSam <span class="ot">&lt;-</span> <span class="fu">hist</span>(<span class="fu">c</span>(dec,tar),<span class="at">breaks=</span>breaks, <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a></span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a>grid<span class="ot">&lt;-</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">30</span>,.<span class="dv">1</span>)</span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a>countsTarG<span class="ot">&lt;-</span><span class="fu">data.frame</span>(<span class="at">y=</span>histTar<span class="sc">$</span>counts<span class="sc">-</span>histDec<span class="sc">$</span>counts,<span class="at">x=</span>histTar<span class="sc">$</span>mids)</span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a>countsTarG<span class="sc">$</span>y[countsTarG<span class="sc">$</span>y<span class="sc">&lt;</span><span class="dv">0</span>]<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a>fitTarG<span class="ot">&lt;-</span><span class="fu">gam</span>(y<span class="sc">~</span><span class="fu">s</span>(x),<span class="at">data=</span>countsTarG,<span class="at">family=</span>poisson)</span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a>fitTarGrid<span class="ot">&lt;-</span><span class="fu">exp</span>(<span class="fu">predict</span>(fitTarG,<span class="at">newdata=</span><span class="fu">data.frame</span>(<span class="at">x=</span>grid)))</span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a></span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a>countsDec<span class="ot">&lt;-</span><span class="fu">data.frame</span>(<span class="at">y=</span>histDec<span class="sc">$</span>counts,<span class="at">x=</span>histDec<span class="sc">$</span>mids)</span>
<span id="cb8-28"><a href="#cb8-28" tabindex="-1"></a>fitDec<span class="ot">&lt;-</span><span class="fu">gam</span>(y<span class="sc">~</span><span class="fu">s</span>(x),<span class="at">data=</span>countsDec,<span class="at">family=</span>poisson)</span>
<span id="cb8-29"><a href="#cb8-29" tabindex="-1"></a>fitSamBad<span class="ot">&lt;-</span><span class="fu">exp</span>(<span class="fu">predict</span>(fitDec,<span class="at">newdata=</span><span class="fu">data.frame</span>(<span class="at">x=</span>grid)))<span class="sc">*</span><span class="dv">2</span></span>
<span id="cb8-30"><a href="#cb8-30" tabindex="-1"></a></span>
<span id="cb8-31"><a href="#cb8-31" tabindex="-1"></a><span class="fu">plot</span>(histSam,<span class="at">xlab=</span><span class="st">&quot;MS-GF+ Score&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;# PSMs&quot;</span>,<span class="at">main=</span><span class="st">&quot;Pyrococcus Search&quot;</span>,<span class="at">border=</span><span class="st">&quot;white&quot;</span>,<span class="at">col=</span><span class="st">&quot;grey&quot;</span>,<span class="at">cex.axis=</span><span class="fl">1.5</span>,<span class="at">cex.main=</span><span class="fl">1.5</span>,<span class="at">cex.lab=</span><span class="fl">1.5</span>,<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1500</span>), <span class="at">axes =</span><span class="cn">FALSE</span>)</span>
<span id="cb8-32"><a href="#cb8-32" tabindex="-1"></a><span class="fu">axis</span>(<span class="at">side=</span><span class="dv">2</span>,<span class="at">at=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">750</span>,<span class="dv">1500</span>))</span>
<span id="cb8-33"><a href="#cb8-33" tabindex="-1"></a><span class="fu">axis</span>(<span class="at">side=</span><span class="dv">1</span>,<span class="at">at=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">10</span>,<span class="dv">20</span>,<span class="dv">30</span>))</span>
<span id="cb8-34"><a href="#cb8-34" tabindex="-1"></a><span class="fu">lines</span>(grid,fitSamBad<span class="sc">+</span>fitTarGrid,<span class="at">col=</span><span class="st">&quot;black&quot;</span>,<span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb8-35"><a href="#cb8-35" tabindex="-1"></a><span class="fu">lines</span>(grid,fitSamBad,<span class="at">col=</span><span class="st">&quot;#FF9900&quot;</span>,<span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb8-36"><a href="#cb8-36" tabindex="-1"></a><span class="fu">lines</span>(grid,fitTarGrid,<span class="at">col=</span><span class="st">&quot;#009900&quot;</span>,<span class="at">lwd=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="TargetDecoy_files/figure-html/unnamed-chunk-7-1.png" width="432" /></p>
<p>Let <span class="math inline">\(x\)</span> be the PSM score</p>
<p>The scores will follow a mixture distribution:</p>
<p><span class="math display">\[f(x) = \pi_b \mathbin{\color{orange}{
f_b(x)}}+(1-\pi_b)\mathbin{\color{green}{ f_g(x)}},\]</span> The local
fdr is also referred to as the posterior error probability (PEP), and is
the probability that a PSM with a score <span
class="math inline">\(x\)</span> is a bad hit.</p>
<p><span class="math display">\[
\begin{array}{lcl}
\text{lfdr}(x) &amp;=&amp;\text{PEP(x)}\\
&amp;=&amp; \text{P}[\text{bad hit} \vert X=x]\\
&amp;=&amp; \frac{\pi_b f_b(x)}{f(x)}
\end{array}
\]</span></p>
<p>We will return a list of PSMs by using the FDR:</p>
<p><span class="math display">\[FDR = E\left[\frac{FP}{TP +
FP}\right]\]</span></p>
<ul>
<li>FP: number of false positives, bad hits</li>
<li>TP: number of true positives, good hits</li>
</ul>
<p><span class="math display">\[
\begin{array}{lcl}
\text{FDR}(x\geq t) &amp;=&amp;
\text{P}[\text{bad hit }\vert X \geq t]\\\\
&amp;=&amp;\frac{\pi_b\int\limits_{x=t}^{+\infty}f_b(x)}{\int\limits_{x=t}^{+\infty}f(x)}\\\\
&amp;=&amp;\frac{\pi_b[1-F_b(t)]}{1-F(t)}
\end{array}
\]</span></p>
<ul>
<li>So the FDR is a set property, it measure the probability on a bad
hit in the set of PSMs with scores <span class="math inline">\(X\geq
t\)</span>.</li>
</ul>
<p>Our list, thus consists of all PSMs with a score <span
class="math inline">\(x\)</span> above a threshold t.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">plot</span>(histSam,<span class="at">xlab=</span><span class="st">&quot;MS-GF+ Score&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;# PSMs&quot;</span>,<span class="at">main=</span><span class="st">&quot;Pyrococcus Search&quot;</span>,<span class="at">border=</span><span class="st">&quot;white&quot;</span>,<span class="at">col=</span><span class="st">&quot;grey&quot;</span>,<span class="at">cex.axis=</span><span class="fl">1.5</span>,<span class="at">cex.main=</span><span class="fl">1.5</span>,<span class="at">cex.lab=</span><span class="fl">1.5</span>,<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1500</span>), <span class="at">axes =</span><span class="cn">FALSE</span>)</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="fu">axis</span>(<span class="at">side=</span><span class="dv">2</span>,<span class="at">at=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">750</span>,<span class="dv">1500</span>))</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="fu">axis</span>(<span class="at">side=</span><span class="dv">1</span>,<span class="at">at=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">10</span>,<span class="dv">20</span>,<span class="dv">30</span>))</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="fu">lines</span>(grid,fitSamBad<span class="sc">+</span>fitTarGrid,<span class="at">col=</span><span class="st">&quot;black&quot;</span>,<span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="fu">lines</span>(grid,fitSamBad,<span class="at">col=</span><span class="st">&quot;#FF9900&quot;</span>,<span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a><span class="fu">lines</span>(grid,fitTarGrid,<span class="at">col=</span><span class="st">&quot;#009900&quot;</span>,<span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a><span class="fu">text</span>(<span class="at">pos=</span><span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">1430</span>,<span class="at">label=</span><span class="fu">expression</span>(x <span class="sc">&gt;=</span> t),<span class="at">col=</span><span class="st">&quot;darkorchid4&quot;</span>,<span class="at">cex=</span><span class="dv">2</span>)</span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a><span class="fu">rect</span>(<span class="dv">5</span>,<span class="sc">-</span><span class="dv">10</span>,<span class="dv">30</span>,<span class="dv">1500</span>,<span class="at">lwd=</span><span class="dv">2</span>,<span class="at">border=</span><span class="st">&quot;darkorchid4&quot;</span>)</span></code></pre></div>
<p><img src="TargetDecoy_files/figure-html/unnamed-chunk-8-1.png" width="432" /></p>
<ul>
<li>So we know how many PSMs we return, i.e. TP + FP: <span
class="math display">\[\text{#PSMs with } x \geq t\]</span></li>
</ul>
<p>and we can also estimate the probability on a PSM above the threshold
empirically:</p>
<p><span class="math display">\[1-\hat{\text{F}}(t) = \frac{\text{#PSMs
with } x \geq t}{\text{#PSMs}}\]</span></p>
<ul>
<li>So to estimate the FDR we only have to estimate the expected number
of PSMs that are bad hits with a score <span
class="math inline">\(x\)</span> above the threshold <span
class="math inline">\(t\)</span>.</li>
</ul>
<p><span class="math display">\[\widehat{\text{FDR}}(t) =
\frac{E\left[\#\text{Bad PSM hits with } X \geq t\right]}{\text{#PSMs
with } x \geq t}\]</span></p>
</div>
<div id="competitive-target-decoy-approach" class="section level2"
number="2.2">
<h2><span class="header-section-number">2.2</span> Competitive target
decoy approach</h2>
<ul>
<li>Search against decoy database to generate representative bad
hits</li>
<li>Reverse database is popular</li>
<li>Concatenated search is most popular</li>
<li>Advantage, a number of bad hits already matches with decoys</li>
</ul>
<p><span class="math inline">\(\rightarrow\)</span> we know that these
are bad hits</p>
<p><span class="math inline">\(\rightarrow\)</span> we have to infer on
less target PSMs.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>hlp <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ms_gf_specevalue)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="sc">-</span><span class="fu">log10</span>(ms_gf_specevalue))) <span class="sc">+</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1000</span>,.<span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;-log10(E-value)&quot;</span>) <span class="sc">+</span> </span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;frequency&quot;</span>) <span class="sc">+</span></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>    <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">40</span>) <span class="sc">+</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>    <span class="fu">facet_grid</span>(isdecoy<span class="sc">~</span>.)</span></code></pre></div>
<p><img src="TargetDecoy_files/figure-html/unnamed-chunk-9-1.png" width="432" /></p>
<p>We estimate that by using the decoys: <span
class="math display">\[\text{# Decoys with x} \geq t\]</span> So our
estimated FDR becomes</p>
<p><span class="math display">\[\widehat{\text{FDR}}(x\geq t) =
\frac{\text{# Decoys} \geq t}{\text{# Targets} \geq t}\]</span> If we
rewrite the FDR we can see the TDA assumptions:</p>
<p><span class="math display">\[
\begin{array}{lcl}
\widehat{\text{FDR}}(x\geq t)&amp;=&amp;\frac{\text{# Decoys} \geq
t}{\text{# Targets} \geq t}\\\\
&amp;=&amp;\frac{\frac{\text{# Decoys}}{\text{# Targets}}\frac{\text{#
Decoys with }x \geq t}{\text{# Decoys}}}{\frac{\text{# Targets with } x
\geq t}{\text{# Targets}}}\\\\
&amp;=&amp;\frac{\frac{\hat{\text{E}}\left[\text{# Bad
Targets}\right]}{\text{# Targets}}\frac{\hat{\text{E}}\left[\text{# Bad
Targets with }X \geq t\right]}{\hat{\text{E}}\left[\text{# Bad
targets}\right]}}{\frac{\text{# Targets with } x \geq t}{\text{#
Targets}}} \\\\
&amp;=&amp;\frac{\hat{\text{P}}\left[\text{Bad  Target}
\right]\times\hat{\text{P}}\left[\text{Bad  Target}\vert X \geq t
\right]}{\hat{\text{P}}\left[\text{Target}\vert X \geq t\right]}\\\\
&amp;=&amp;\frac{\hat\pi_b [1-\hat{F}_b(t)]}{1-\hat{F}(t)}
\end{array}
\]</span></p>
<p>So the TDA has the following assumptions:</p>
<ol style="list-style-type: decimal">
<li><p>A bad hit is equally likely to match to a decoy as to a target
sequence. <span class="math inline">\(\rightarrow\)</span> we can thus
estimate the fraction of bad hits or the probability on a bad hit as
<span class="math display">\[\hat{\pi}_b = \frac{\# \text{Decoys}}{\#
\text{Targets}}\]</span></p></li>
<li><p>Bad target PSM scores and decoy PSM scores are equaly
distributed.</p></li>
</ol>
</div>
</div>
<div id="diagnostic-plots-for-the-tda" class="section level1"
number="3">
<h1><span class="header-section-number">3</span> Diagnostic plots for
the TDA</h1>
<p>We will evaluate the TDA assumptions using diagnostic plots that
compares the empirical distribution of decoy and target PSM scores.</p>
<ul>
<li>Histograms</li>
<li>P-P plots</li>
</ul>
<p>With P-P plots will plot for each observed PSM score <span
class="math inline">\(t\)</span> the empirical probability to observe a
decoy with score <span class="math inline">\(x \leq t\)</span> to the
empirical probability to observe a target with score <span
class="math inline">\(x \leq t\)</span>:</p>
<p><span class="math display">\[\hat{\text{P}}\left[\text{decoy with
score } x \leq t\right] = \frac{\# \text{decoys with score } x \leq
t}{\# \text{decoys}}\]</span></p>
<p><span class="math display">\[
\hat{\text{P}}\left[\text{target with score } x \leq t\right] = \frac{\#
\text{target with score } x \leq t}{\# \text{targets}}
\]</span></p>
<p>If the two distributions are the same the dots of the P-P plot should
follow the 1-1 line.</p>
<p>This will not be the case. We expect the distribution of the target
PSMs:</p>
<ul>
<li>to be similarly distributed as the decoys for low scores</li>
<li>and enriched with many high scores corresponding to target PSMs
which are matching to the proper peptide sequence in the data base.</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>hlp <span class="ot">&lt;-</span> hlp <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">score=</span><span class="sc">-</span><span class="fu">log10</span>(hlp<span class="sc">$</span>ms_gf_specevalue))</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>y1<span class="ot">&lt;-</span>hlp<span class="sc">$</span>score[hlp<span class="sc">$</span>isdecoy] <span class="sc">%&gt;%</span> na.exclude</span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>y2<span class="ot">&lt;-</span>hlp<span class="sc">$</span>score[<span class="sc">!</span>hlp<span class="sc">$</span>isdecoy] <span class="sc">%&gt;%</span> na.exclude</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>F1<span class="ot">&lt;-</span><span class="fu">ecdf</span>(y1)</span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>F2<span class="ot">&lt;-</span><span class="fu">ecdf</span>(y2)</span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>breaks<span class="ot">&lt;-</span><span class="fu">seq</span>(<span class="fu">floor</span>(<span class="fu">min</span>(<span class="fu">c</span>(y1,y2))),<span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">c</span>(y1,y2))),<span class="at">length.out=</span><span class="dv">50</span>)</span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a>pi0<span class="ot">&lt;-</span><span class="fu">length</span>(y1)<span class="sc">/</span><span class="fu">length</span>(y2)</span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="cf">for</span> (x <span class="cf">in</span> <span class="fu">quantile</span>(<span class="fu">c</span>(y1,y2),<span class="fu">c</span>(<span class="dv">0</span>,.<span class="dv">01</span>,.<span class="dv">02</span>,<span class="fu">seq</span>(<span class="fl">0.1</span>,<span class="dv">1</span>,.<span class="dv">1</span>))))</span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a>{</span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a><span class="fu">hist</span>(y2,<span class="at">breaks=</span>breaks,<span class="at">main=</span><span class="st">&quot;Pyrococcus MSGF+&quot;</span>,<span class="at">cex.axis=</span><span class="fl">1.5</span>,<span class="at">cex.lab=</span><span class="fl">1.5</span>,<span class="at">cex.main=</span><span class="fl">1.5</span>,<span class="at">col=</span><span class="st">&quot;grey&quot;</span>)</span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a>decHist<span class="ot">&lt;-</span><span class="fu">hist</span>(y1,<span class="at">breaks=</span>breaks,<span class="at">plot=</span><span class="cn">FALSE</span>)</span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a><span class="fu">points</span>(decHist<span class="sc">$</span>mids,decHist<span class="sc">$</span>counts,<span class="at">col=</span><span class="st">&quot;#FF9900&quot;</span>,<span class="at">type=</span><span class="st">&quot;h&quot;</span>,<span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v=</span>x,<span class="at">col=</span><span class="st">&quot;blue&quot;</span>,<span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">F1</span>(y2),<span class="fu">F2</span>(y2),<span class="at">xlab=</span><span class="st">&quot;ECDF Targets&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;ECDF Decoys&quot;</span>,<span class="at">cex=</span>.<span class="dv">4</span>,<span class="at">main=</span><span class="st">&quot;P-P plot&quot;</span>,<span class="at">cex.axis=</span><span class="fl">1.5</span>,<span class="at">cex.lab=</span><span class="fl">1.5</span>,<span class="at">cex.main=</span><span class="fl">1.5</span>,<span class="at">col=</span><span class="st">&quot;grey&quot;</span>,<span class="at">pch=</span><span class="dv">19</span>)</span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">a=</span><span class="dv">0</span>,<span class="at">b=</span>pi0)</span>
<span id="cb11-19"><a href="#cb11-19" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v=</span><span class="fu">F1</span>(x),<span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span>
<span id="cb11-20"><a href="#cb11-20" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fu">F2</span>(x),<span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span>
<span id="cb11-21"><a href="#cb11-21" tabindex="-1"></a><span class="fu">points</span>(<span class="fu">F1</span>(x),<span class="fu">F2</span>(x),<span class="at">col=</span><span class="st">&quot;blue&quot;</span>,<span class="at">cex=</span><span class="dv">2</span>,<span class="at">pch=</span><span class="dv">19</span>)</span>
<span id="cb11-22"><a href="#cb11-22" tabindex="-1"></a>}</span></code></pre></div>
<p><img src="TargetDecoy_files/figure-html/unnamed-chunk-10-1.png" width="432" /><img src="TargetDecoy_files/figure-html/unnamed-chunk-10-2.png" width="432" /><img src="TargetDecoy_files/figure-html/unnamed-chunk-10-3.png" width="432" /><img src="TargetDecoy_files/figure-html/unnamed-chunk-10-4.png" width="432" /><img src="TargetDecoy_files/figure-html/unnamed-chunk-10-5.png" width="432" /><img src="TargetDecoy_files/figure-html/unnamed-chunk-10-6.png" width="432" /><img src="TargetDecoy_files/figure-html/unnamed-chunk-10-7.png" width="432" /><img src="TargetDecoy_files/figure-html/unnamed-chunk-10-8.png" width="432" /><img src="TargetDecoy_files/figure-html/unnamed-chunk-10-9.png" width="432" /><img src="TargetDecoy_files/figure-html/unnamed-chunk-10-10.png" width="432" /><img src="TargetDecoy_files/figure-html/unnamed-chunk-10-11.png" width="432" /><img src="TargetDecoy_files/figure-html/unnamed-chunk-10-12.png" width="432" /><img src="TargetDecoy_files/figure-html/unnamed-chunk-10-13.png" width="432" /></p>
<p>Note, that</p>
<ul>
<li>the points corresponding to low score values follow a straight line
indicating that targets and decoys scores are similarly distributed</li>
<li>this line has an angle equal to fraction of bad hits, which is
estimated as <span class="math inline">\(\hat\pi_b=\frac{\#
\text{decoys}}{\#\text{targets}}\)</span> and is indicated by the black
line in the plot.</li>
<li>This line can be used to assess the assumption that bad hits are
equally likely matching to targets sequences as to decoy sequences.</li>
</ul>
</div>

<div id="rmd-source-code">---
title: "Introduction to TargetDecoy"
author: 
  - name: Lieven Clement
    affiliation:
    - Ghent University
output: 
    html_document:
      code_download: true
      theme: flatly
      toc: true
      toc_float: true
      highlight: tango
      number_sections: true
      code_folding: "hide"
linkcolor: blue
urlcolor: blue
citecolor: blue
---

```{r setup, include=FALSE}
library(knitr)
opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    crop = NULL,
    fig.width = 6,
    dpi = 72
)
```

```{r "libraries", message=FALSE, warning=FALSE}
library(TargetDecoy)
library(ggplot2)
```

# Introduction

![](figures/ProteomicsWorkflow.png)

![](figures/peptideSpectralMatching.png)
Slide courtesy to Lennart Martens 

## Why did Lennart mention that "statistics were out of the window"

```{r}
library(TargetDecoy)
library(tidyverse)
data("ModSwissXT")
hlp <- TargetDecoy:::.getDF(ModSwissXT) 
names(hlp) <- names(hlp) %>%
    str_replace(pattern=":",replacement="_")
hlp <- hlp %>% 
mutate(omssa_evalue=as.double(omssa_evalue))
```


- E-values: Probability that a random candidate peptide produces a higher score than the observed PSM score.


- Probability that a random candidate peptide produces a higher score that the observed PSM score for a real search with OMSSA.

```{r}
library(TargetDecoy)
library(tidyverse)
data("ModSwissXT")
hlp <- TargetDecoy:::.getDF(ModSwissXT) 
names(hlp) <- names(hlp) %>%
    str_replace(pattern=":",replacement="_")
hlp <- hlp %>% 
mutate(omssa_evalue=as.double(omssa_evalue))
hlp %>% 
    filter(!is.na(omssa_evalue)) %>% 
    ggplot(aes(omssa_evalue)) +
    geom_histogram(breaks=seq(0,100,5)) +
    xlab("E-value (%)") + 
    ylab("frequency")
```

- E-value we expect for an observed spectrum and a random candidate peptide.

```{r}
data.frame(evalue=runif(20000,0,100)) %>%
    ggplot(aes(evalue)) +
    geom_histogram(breaks=seq(0,100,5)) +
    xlab("E-value (%)") + 
    ylab("frequency")
```

- Probability that a random candidate peptide produces a higher score than the observed PSM score for decoys in a real search with OMSSA.

```{r}
hlp %>% 
    filter(isdecoy & !is.na(omssa_evalue)) %>% 
    ggplot(aes(omssa_evalue)) +
    geom_histogram(breaks=seq(0,100,5)) +
    xlab("E-value (%)") + 
    ylab("frequency")
```

- A bad hit is the random hit with the best score so it is also bound to have a low E-value.

- If we look at E-values for all PSMs they are only useful as a score.

- We should know the distribution of the maximum score of
random candidate peptides when we want to do the statistics.

```{r warning=FALSE}
hlp %>% 
    filter(!is.na(omssa_evalue)) %>% 
    ggplot(aes(-log10(omssa_evalue))) +
    geom_histogram(breaks=seq(0,1000,.5)) +
    xlab("-log10(E-value)") + 
    ylab("frequency") +
    xlim(0,20) 
```

# Concepts

## Basic Statistical Concepts

```{r}
names(hlp) <- names(hlp) %>% str_replace(pattern="-",replacement = "_")
hlp <- hlp %>% mutate(ms_gf_specevalue=as.double(ms_gf_specevalue))
```

```{r}
library(mgcv)
dec <- -log10(hlp$ms_gf_specevalue[hlp$isdecoy]) %>% na.exclude()
tar <- -log10(hlp$ms_gf_specevalue[!hlp$isdecoy]) %>% na.exclude()

breaks <- seq(0,30,.5)
#binWidth <-2
#breaks <- seq(floor(min(c(dec,tar))/binWidth)*binWidth,ceiling(max(c(dec,tar))/binWidth)*binWidth,binWidth)
#code if we register the modes by substracting the mode from the target scores and  the decoy scores.
#breaks=seq(-(ceiling(abs(min(c(dec,tar))/binWidth))+.5)*binWidth,(ceiling(max(c(dec,tar))/binWidth)+.5)*binWidth,binWidth)
histDec <- hist(dec,breaks=breaks,plot = FALSE)
histTar <- hist(tar,breaks=breaks,plot=FALSE)
histSam <- hist(c(dec,tar),breaks=breaks, plot = FALSE)

grid<-seq(0,30,.1)
countsTarG<-data.frame(y=histTar$counts-histDec$counts,x=histTar$mids)
countsTarG$y[countsTarG$y<0]<-0
fitTarG<-gam(y~s(x),data=countsTarG,family=poisson)
fitTarGrid<-exp(predict(fitTarG,newdata=data.frame(x=grid)))

countsDec<-data.frame(y=histDec$counts,x=histDec$mids)
fitDec<-gam(y~s(x),data=countsDec,family=poisson)
fitSamBad<-exp(predict(fitDec,newdata=data.frame(x=grid)))*2

plot(histSam,xlab="MS-GF+ Score",ylab="# PSMs",main="Pyrococcus Search",border="white",col="grey",cex.axis=1.5,cex.main=1.5,cex.lab=1.5,ylim=c(0,1500), axes =FALSE)
axis(side=2,at=c(0,750,1500))
axis(side=1,at=c(0,10,20,30))
lines(grid,fitSamBad+fitTarGrid,col="black",lwd=2)
lines(grid,fitSamBad,col="#FF9900",lwd=2)
lines(grid,fitTarGrid,col="#009900",lwd=2)
```

Let $x$ be the PSM score 

The scores will follow a mixture distribution:

$$f(x) = \pi_b \mathbin{\color{orange}{ f_b(x)}}+(1-\pi_b)\mathbin{\color{green}{ f_g(x)}},$$
The local fdr is also referred to as the posterior error probability (PEP), and is the probability that a PSM with a score $x$ is a bad hit. 

$$
\begin{array}{lcl}
\text{lfdr}(x) &=&\text{PEP(x)}\\
&=& \text{P}[\text{bad hit} \vert X=x]\\
&=& \frac{\pi_b f_b(x)}{f(x)}
\end{array}
$$


We will return a list of PSMs by using the FDR:

$$FDR = E\left[\frac{FP}{TP + FP}\right]$$

- FP: number of false positives, bad hits
- TP: number of true positives, good hits

$$
\begin{array}{lcl}
\text{FDR}(x\geq t) &=&
\text{P}[\text{bad hit }\vert X \geq t]\\\\
&=&\frac{\pi_b\int\limits_{x=t}^{+\infty}f_b(x)}{\int\limits_{x=t}^{+\infty}f(x)}\\\\
&=&\frac{\pi_b[1-F_b(t)]}{1-F(t)}
\end{array}
$$




- So the FDR is a set property, it measure the probability on a bad hit in the set of PSMs with scores $X\geq t$.

Our list, thus consists of all PSMs with a score $x$ above a threshold t. 

```{r warning=FALSE}
plot(histSam,xlab="MS-GF+ Score",ylab="# PSMs",main="Pyrococcus Search",border="white",col="grey",cex.axis=1.5,cex.main=1.5,cex.lab=1.5,ylim=c(0,1500), axes =FALSE)
axis(side=2,at=c(0,750,1500))
axis(side=1,at=c(0,10,20,30))
lines(grid,fitSamBad+fitTarGrid,col="black",lwd=2)
lines(grid,fitSamBad,col="#FF9900",lwd=2)
lines(grid,fitTarGrid,col="#009900",lwd=2)
text(pos=4,5,1430,label=expression(x >= t),col="darkorchid4",cex=2)
rect(5,-10,30,1500,lwd=2,border="darkorchid4")
```


- So we know how many PSMs we return, i.e. TP + FP: 
$$\text{#PSMs with } x \geq t$$ 

and we can also estimate the probability on a PSM above the threshold empirically:

$$1-\hat{\text{F}}(t) = \frac{\text{#PSMs with } x \geq t}{\text{#PSMs}}$$ 




- So to estimate the FDR we only have to estimate the expected number of PSMs that are bad hits with a score $x$ above the threshold $t$.

$$\widehat{\text{FDR}}(t) = \frac{E\left[\#\text{Bad PSM hits with } X \geq t\right]}{\text{#PSMs with } x \geq t}$$


## Competitive target decoy approach

- Search against decoy database to generate representative bad hits
- Reverse database is popular
- Concatenated search is most popular
- Advantage, a number of bad hits already matches with decoys

$\rightarrow$ we know that these are bad hits

$\rightarrow$ we have to infer on less target PSMs.

```{r warning=FALSE}
hlp %>% 
    filter(!is.na(ms_gf_specevalue)) %>% 
    ggplot(aes(-log10(ms_gf_specevalue))) +
    geom_histogram(breaks=seq(0,1000,.5)) +
    xlab("-log10(E-value)") + 
    ylab("frequency") +
    xlim(0,40) +
    facet_grid(isdecoy~.)
```

We estimate that by using the decoys: 
$$\text{# Decoys with x} \geq t$$
So our estimated FDR becomes 

$$\widehat{\text{FDR}}(x\geq t) = \frac{\text{# Decoys} \geq t}{\text{# Targets} \geq t}$$
If we rewrite the FDR we can see the TDA assumptions:

$$
\begin{array}{lcl}
\widehat{\text{FDR}}(x\geq t)&=&\frac{\text{# Decoys} \geq t}{\text{# Targets} \geq t}\\\\
&=&\frac{\frac{\text{# Decoys}}{\text{# Targets}}\frac{\text{# Decoys with }x \geq t}{\text{# Decoys}}}{\frac{\text{# Targets with } x \geq t}{\text{# Targets}}}\\\\
&=&\frac{\frac{\hat{\text{E}}\left[\text{# Bad Targets}\right]}{\text{# Targets}}\frac{\hat{\text{E}}\left[\text{# Bad Targets with }X \geq t\right]}{\hat{\text{E}}\left[\text{# Bad targets}\right]}}{\frac{\text{# Targets with } x \geq t}{\text{# Targets}}} \\\\
&=&\frac{\hat{\text{P}}\left[\text{Bad  Target} \right]\times\hat{\text{P}}\left[\text{Bad  Target}\vert X \geq t \right]}{\hat{\text{P}}\left[\text{Target}\vert X \geq t\right]}\\\\
&=&\frac{\hat\pi_b [1-\hat{F}_b(t)]}{1-\hat{F}(t)}
\end{array}
$$


So the TDA has the following assumptions: 

1. A bad hit is equally likely to match to a decoy as to a target sequence. $\rightarrow$ we can thus estimate the fraction of bad hits or the probability on a bad hit as $$\hat{\pi}_b = \frac{\# \text{Decoys}}{\# \text{Targets}}$$

2. Bad target PSM scores and decoy PSM scores are equaly distributed.

# Diagnostic plots for the TDA

We will evaluate the TDA assumptions using diagnostic plots that compares the empirical distribution of decoy and target PSM scores. 

- Histograms
- P-P plots


With P-P plots will plot for each observed PSM score $t$ the empirical probability to observe a decoy with score $x \leq t$ to the empirical probability to observe a target with score $x \leq t$: 

$$\hat{\text{P}}\left[\text{decoy with score } x \leq t\right] = \frac{\# \text{decoys with score } x \leq t}{\# \text{decoys}}$$

$$
\hat{\text{P}}\left[\text{target with score } x \leq t\right] = \frac{\# \text{target with score } x \leq t}{\# \text{targets}}
$$

If the two distributions are the same the dots of the P-P plot should follow the 1-1 line. 

This will not be the case. We expect the distribution of the target PSMs: 

- to be similarly distributed as the decoys for low scores
- and enriched with many high scores corresponding to target PSMs which are matching to the proper peptide sequence in the data base.


```{r}
hlp <- hlp %>% 
  mutate(score=-log10(hlp$ms_gf_specevalue))

y1<-hlp$score[hlp$isdecoy] %>% na.exclude
y2<-hlp$score[!hlp$isdecoy] %>% na.exclude
F1<-ecdf(y1)
F2<-ecdf(y2)
breaks<-seq(floor(min(c(y1,y2))),ceiling(max(c(y1,y2))),length.out=50)
pi0<-length(y1)/length(y2)
for (x in quantile(c(y1,y2),c(0,.01,.02,seq(0.1,1,.1))))
{
par(mfrow=c(1,2))
hist(y2,breaks=breaks,main="Pyrococcus MSGF+",cex.axis=1.5,cex.lab=1.5,cex.main=1.5,col="grey")
decHist<-hist(y1,breaks=breaks,plot=FALSE)
points(decHist$mids,decHist$counts,col="#FF9900",type="h",lwd=2)
abline(v=x,col="blue",lwd=2)
plot(F1(y2),F2(y2),xlab="ECDF Targets",ylab="ECDF Decoys",cex=.4,main="P-P plot",cex.axis=1.5,cex.lab=1.5,cex.main=1.5,col="grey",pch=19)
abline(a=0,b=pi0)
abline(v=F1(x),col="blue")
abline(h=F2(x),col="blue")
points(F1(x),F2(x),col="blue",cex=2,pch=19)
}
```

Note, that 

- the points corresponding to low score values follow a straight line indicating that targets and decoys scores are similarly distributed 
- this line has an angle equal to fraction of bad hits, which is estimated as $\hat\pi_b=\frac{\# \text{decoys}}{\#\text{targets}}$ and is indicated by the black line in the plot. 
- This line can be used to assess the assumption that bad hits are equally likely matching to targets sequences as to decoy sequences. </div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("TargetDecoy.Rmd");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
