<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>Technical details on linear regression for proteomics when starting from summarized protein expression values</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="intro.html">1.1 Position</a>
    </li>
    <li>
      <a href="recapGeneralLinearModel.html">1.2 Recap linear model</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="docs/martens_proteomics_bioinformatics.pdf">1. Proteomics Bioinformatics</a>
    </li>
    <li>
      <a href="TargetDecoy.html">2.1 Identification - FDR - QC</a>
    </li>
    <li>
      <a href="pda_quantification_preprocessing.html">2.2.1. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">2.2.2. Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">2.3.1. Quantification - Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="pda_blocking_wrapup.html">2.3.2. Wrap-up DA - Blocking</a>
    </li>
    <li>
      <a href="technicalDetailsProteomics.html">2.3.3 Technical Details</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencing_intro.html">1.  Introduction to RNA-seq</a>
    </li>
    <li>
      <a href="sequencing_countData.html">2.  Working with count data and GLMs</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="multipleRegression_KPNA2.html">1 Intro KPNA2</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="https://www.compomics.com/bioinformatics-for-proteomics/">1. Bioinformatics for Proteomics</a>
    </li>
    <li>
      <a href="TargetDecoyTutorial.html">2.1. Target Decoy QC</a>
    </li>
    <li>
      <a href="pda_tutorialPreprocessing.html">2.2. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">2.3. Quantification - Design</a>
    </li>
    <li class="dropdown-header">---- MODULE II RNA-seq ----</li>
    <li>
      <a href="sequencingTutorial1_mapping.html">1. Read mapping</a>
    </li>
    <li>
      <a href="sequencingTutorial2_DE.html">2. Working with Count Data</a>
    </li>
    <li>
      <a href="scRNA-seq-Tutorial.html">3. Single Cell RNA-seq Data Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Technical details on linear regression for
proteomics when starting from summarized protein expression values</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date">statOmics, Ghent University (<a
href="https://statomics.github.io"
class="uri">https://statomics.github.io</a>)</h4>

</div>


<div id="preamble" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Preamble</h1>
<p>Read stored results for heart data</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(QFeatures)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(msqrob2)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>  <span class="fu">url</span>(</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>    <span class="st">&quot;https://raw.githubusercontent.com/statOmics/SGA2020/gh-pages/assets/peHeart.rds&quot;</span>,</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>    <span class="st">&quot;rb&quot;</span>)</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>  )</span></code></pre></div>
</div>
<div id="linear-regression" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Linear regression</h1>
<ul>
<li>Consider a vector of predictors <span
class="math inline">\(\mathbf{x}_i=(x_1,\ldots,x_{p-1})\)</span>
and</li>
<li>a real-valued response <span class="math inline">\(Y_i\)</span></li>
<li>with <span class="math inline">\(i = 1, \ldots, n\)</span></li>
<li>then the linear regression model can be written as <span
class="math display">\[
Y_i=f(\mathbf{x}) +\epsilon=\beta_0+\sum\limits_{j=1}^{p-1} x_{ij}\beta
+ \epsilon_i
\]</span> with i.i.d. <span class="math inline">\(\epsilon_i\sim
N(0,\sigma^2)\)</span></li>
</ul>
<hr />
<ul>
<li><span class="math inline">\(n\)</span> observations <span
class="math inline">\((\mathbf{x}_1,y_1) \ldots
(\mathbf{x}_n,y_n)\)</span></li>
<li>Regression in matrix notation <span
class="math display">\[\mathbf{Y}=\mathbf{X\beta} +
\boldsymbol{\epsilon}\]</span> with <span
class="math inline">\(\mathbf{Y}=\left[\begin{array}{c}y_1\\
\vdots\\y_n\end{array}\right]\)</span>, <span
class="math inline">\(\mathbf{X}=\left[\begin{array}{cccc}
1&amp;x_{11}&amp;\ldots&amp;x_{1p-1}\\
\vdots&amp;\vdots&amp;&amp;\vdots\\
1&amp;x_{n1}&amp;\ldots&amp;x_{np-1}
\end{array}\right]\)</span>, <span
class="math inline">\(\boldsymbol{\beta}=\left[\begin{array}{c}\beta_0\\
\vdots\\ \beta_{p-1}\end{array}\right]\)</span> and <span
class="math inline">\(\boldsymbol{\epsilon}=\left[\begin{array}{c}
\epsilon_1 \\ \vdots \\ \epsilon_n\end{array}\right]\)</span></li>
</ul>
<hr />
<div id="least-squares-ls" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Least Squares
(LS)</h2>
<ul>
<li><p>Minimize the residual sum of squares <span
class="math display">\[\begin{eqnarray*}
RSS(\boldsymbol{\beta})&amp;=&amp;\sum\limits_{i=1}^n e^2_i\\
&amp;=&amp;\sum\limits_{i=1}^n \left(y_i-\beta_0-\sum\limits_{j=1}^p
x_{ij}\beta_j\right)^2
\end{eqnarray*}\]</span></p></li>
<li><p>or in matrix notation <span
class="math display">\[\begin{eqnarray*}
RSS(\boldsymbol{\beta})&amp;=&amp;(\mathbf{Y}-\mathbf{X\beta})^T(\mathbf{Y}-\mathbf{X\beta})\\
&amp;=&amp;\Vert \mathbf{Y}-\mathbf{X\beta}\Vert^2
\end{eqnarray*}\]</span> with the <span
class="math inline">\(L_2\)</span>-norm of a <span
class="math inline">\(p\)</span>-dim. vector <span
class="math inline">\(v\)</span> <span class="math inline">\(\Vert
\mathbf{v} \Vert=\sqrt{v_1^2+\ldots+v_p^2}\)</span></p></li>
<li><p><span class="math inline">\(\rightarrow\)</span> <span
class="math inline">\(\hat{\boldsymbol{\beta}}=\text{argmin}_\beta \Vert
\mathbf{Y}-\mathbf{X\beta}\Vert^2\)</span>}</p></li>
</ul>
<hr />
<div id="minimize-rss" class="section level3" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Minimize RSS</h3>
<p><span class="math display">\[
\begin{array}{ccc}
\frac{\partial RSS}{\partial
\boldsymbol{\beta}}&amp;=&amp;\mathbf{0}\\\\
\frac{(\mathbf{Y}-\mathbf{X\beta})^T(\mathbf{Y}-\mathbf{X\beta})}{\partial
\boldsymbol{\beta}}&amp;=&amp;\mathbf{0}\\\\
-2\mathbf{X}^T(\mathbf{Y}-\mathbf{X\beta})&amp;=&amp;\mathbf{0}\\\\
\mathbf{X}^T\mathbf{X\beta}&amp;=&amp;\mathbf{X}^T\mathbf{Y}\\\\
\hat{\boldsymbol{\beta}}&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{Y}
\end{array}
\]</span></p>
<hr />
<div id="heart-example" class="section level4" number="2.1.1.1">
<h4><span class="header-section-number">2.1.1.1</span> Heart
example</h4>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">assay</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])[<span class="dv">2</span>,]</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> location<span class="sc">*</span>tissue <span class="sc">+</span> patient, <span class="at">data =</span> <span class="fu">colData</span>(pe), <span class="at">x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">head</span>(fit<span class="sc">$</span>x,<span class="dv">4</span>)</span></code></pre></div>
<pre><code>##               (Intercept) locationR tissueV patient4 patient8 locationR:tissueV
## Intensity.LA3           1         0       0        0        0                 0
## Intensity.LA4           1         0       0        1        0                 0
## Intensity.LA8           1         0       0        0        1                 0
## Intensity.LV3           1         0       1        0        0                 0</code></pre>
<p>The model matrix can also be obtained without fitting the model:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>X <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> location <span class="sc">*</span> tissue <span class="sc">+</span> patient, <span class="fu">colData</span>(pe))</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="fu">head</span>(X,<span class="dv">4</span>)</span></code></pre></div>
<pre><code>##               (Intercept) locationR tissueV patient4 patient8 locationR:tissueV
## Intensity.LA3           1         0       0        0        0                 0
## Intensity.LA4           1         0       0        1        0                 0
## Intensity.LA8           1         0       0        0        1                 0
## Intensity.LV3           1         0       1        0        0                 0</code></pre>
<p>Least squares:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>betas <span class="ot">&lt;-</span> <span class="fu">solve</span>(<span class="fu">t</span>(X)<span class="sc">%*%</span>X) <span class="sc">%*%</span> <span class="fu">t</span>(X) <span class="sc">%*%</span> y</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="fu">cbind</span>(fit<span class="sc">$</span>coef, betas)</span></code></pre></div>
<pre><code>##                          [,1]        [,2]
## (Intercept)       26.57559360 26.57559360
## locationR          0.20636497  0.20636497
## tissueV            0.12756675  0.12756675
## patient4           0.30592435  0.30592435
## patient8          -0.33432636 -0.33432636
## locationR:tissueV -0.04130843 -0.04130843</code></pre>
<hr />
</div>
</div>
<div id="variance-estimator" class="section level3" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Variance
Estimator?</h3>
<p><span class="math display">\[
\begin{array}{ccl}
\hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}
&amp;=&amp;\text{var}\left[(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{Y}\right]\\\\
&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\text{var}\left[\mathbf{Y}\right]\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\\\\
&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T(\mathbf{I}\sigma^2)\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}
\\\\
&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{I}\quad\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\sigma^2\\\\
%\hat{\boldmath{\Sigma}}_{\hat{\boldsymbol{\beta}}}&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\var\left[\mathbf{Y}\right](\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}\\
&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\sigma^2\\\\
&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\sigma^2
\end{array}
\]</span></p>
<hr />
<div id="heart-example-1" class="section level4" number="2.1.2.1">
<h4><span class="header-section-number">2.1.2.1</span> Heart
example</h4>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">summary</span>(fit)<span class="sc">$</span>cov.unscaled <span class="sc">*</span> <span class="fu">sigma</span>(fit)<span class="sc">^</span><span class="dv">2</span></span></code></pre></div>
<pre><code>##                   (Intercept)     locationR       tissueV      patient4
## (Intercept)         0.3938774 -2.625849e-01 -2.625849e-01 -1.969387e-01
## locationR          -0.2625849  5.251699e-01  2.625849e-01  1.549906e-16
## tissueV            -0.2625849  2.625849e-01  5.251699e-01  1.549906e-16
## patient4           -0.1969387  1.549906e-16  1.549906e-16  3.938774e-01
## patient8           -0.1969387  1.249600e-16  1.249600e-16  1.969387e-01
## locationR:tissueV   0.2625849 -5.251699e-01 -5.251699e-01 -1.332518e-16
##                        patient8 locationR:tissueV
## (Intercept)       -1.969387e-01      2.625849e-01
## locationR          1.249600e-16     -5.251699e-01
## tissueV            1.249600e-16     -5.251699e-01
## patient4           1.969387e-01     -1.332518e-16
## patient8           3.938774e-01     -1.236848e-16
## locationR:tissueV -1.236848e-16      1.050340e+00</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">nrow</span>(X)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ncol</span>(X)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>mse <span class="ot">&lt;-</span> <span class="fu">sum</span>((y<span class="sc">-</span>X<span class="sc">%*%</span>betas)<span class="sc">^</span><span class="dv">2</span>)<span class="sc">/</span>(n<span class="sc">-</span>p)</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>SigmaBeta <span class="ot">&lt;-</span> <span class="fu">solve</span>(<span class="fu">t</span>(X)<span class="sc">%*%</span>X) <span class="sc">*</span> mse   </span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>SigmaBeta</span></code></pre></div>
<pre><code>##                   (Intercept)  locationR    tissueV   patient4   patient8
## (Intercept)         0.3938774 -0.2625849 -0.2625849 -0.1969387 -0.1969387
## locationR          -0.2625849  0.5251699  0.2625849  0.0000000  0.0000000
## tissueV            -0.2625849  0.2625849  0.5251699  0.0000000  0.0000000
## patient4           -0.1969387  0.0000000  0.0000000  0.3938774  0.1969387
## patient8           -0.1969387  0.0000000  0.0000000  0.1969387  0.3938774
## locationR:tissueV   0.2625849 -0.5251699 -0.5251699  0.0000000  0.0000000
##                   locationR:tissueV
## (Intercept)               0.2625849
## locationR                -0.5251699
## tissueV                  -0.5251699
## patient4                  0.0000000
## patient8                  0.0000000
## locationR:tissueV         1.0503398</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">range</span>(SigmaBeta <span class="sc">-</span> <span class="fu">summary</span>(fit)<span class="sc">$</span>cov.unscaled <span class="sc">*</span> <span class="fu">sigma</span>(fit)<span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## [1] -3.996803e-15  7.993606e-15</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">summary</span>(fit)<span class="sc">$</span>coef[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], <span class="at">betas =</span> betas, <span class="at">seBetas =</span> <span class="fu">diag</span>(SigmaBeta)<span class="sc">^</span>.<span class="dv">5</span>)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Estimate"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Std..Error"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["betas"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["seBetas"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"26.57559360","2":"0.6275965","3":"26.57559360","4":"0.6275965","_rn_":"(Intercept)"},{"1":"0.20636497","2":"0.7246861","3":"0.20636497","4":"0.7246861","_rn_":"locationR"},{"1":"0.12756675","2":"0.7246861","3":"0.12756675","4":"0.7246861","_rn_":"tissueV"},{"1":"0.30592435","2":"0.6275965","3":"0.30592435","4":"0.6275965","_rn_":"patient4"},{"1":"-0.33432636","2":"0.6275965","3":"-0.33432636","4":"0.6275965","_rn_":"patient8"},{"1":"-0.04130843","2":"1.0248609","3":"-0.04130843","4":"1.0248609","_rn_":"locationR:tissueV"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</div>
<div id="contrasts" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Contrasts</h2>
<p>When we assess a contrast we assess a linear combination of model
parameters:</p>
<p><span class="math display">\[ H_0: \mathbf{L^T\beta} = 0 \text{ vs }
H_1: \mathbf{L^T\beta} \neq 0 \]</span></p>
<p>Estimator of Contrast?</p>
<p><span
class="math display">\[\mathbf{L}^T\hat{\boldsymbol{\beta}}\]</span></p>
<p>Variance?</p>
<p><span class="math display">\[
\boldsymbol{\Sigma}_{\mathbf{L}\hat{\boldsymbol{\beta}}}=\mathbf{L}^T\boldsymbol{\Sigma}_{\hat{\boldsymbol{\beta}}}\mathbf{L}
\]</span></p>
<hr />
<div id="heart-example-2" class="section level3" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Heart example</h3>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>  <span class="fu">c</span>(</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>    <span class="st">&quot;tissueV = 0&quot;</span>,</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>    <span class="st">&quot;tissueV + locationR:tissueV = 0&quot;</span>,</span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>    <span class="st">&quot;tissueV + 0.5*locationR:tissueV = 0&quot;</span>,<span class="st">&quot;locationR:tissueV = 0&quot;</span>),</span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>  <span class="at">parameterNames =</span></span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a>    <span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>msqrobModels[[<span class="dv">2</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a>    getCoef <span class="sc">%&gt;%</span></span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a>    names</span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a>  )</span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a>L</span></code></pre></div>
<pre><code>##                   tissueV tissueV + locationR:tissueV
## (Intercept)             0                           0
## locationR               0                           0
## tissueV                 1                           1
## patient4                0                           0
## patient8                0                           0
## locationR:tissueV       0                           1
##                   tissueV + 0.5 * locationR:tissueV locationR:tissueV
## (Intercept)                                     0.0                 0
## locationR                                       0.0                 0
## tissueV                                         1.0                 0
## patient4                                        0.0                 0
## patient8                                        0.0                 0
## locationR:tissueV                               0.5                 1</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>contrasts <span class="ot">&lt;-</span> <span class="fu">t</span>(L) <span class="sc">%*%</span> betas</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>SigmaContrasts <span class="ot">&lt;-</span> <span class="fu">t</span>(L) <span class="sc">%*%</span> SigmaBeta <span class="sc">%*%</span> L</span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>seContrasts <span class="ot">&lt;-</span> SigmaContrasts <span class="sc">%&gt;%</span></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a>  diag <span class="sc">%&gt;%</span></span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a>  sqrt</span></code></pre></div>
<p>Comparison with lm and glht results</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">library</span>(multcomp)</span></code></pre></div>
<pre><code>## Loading required package: mvtnorm</code></pre>
<pre><code>## Loading required package: survival</code></pre>
<pre><code>## Loading required package: TH.data</code></pre>
<pre><code>## Loading required package: MASS</code></pre>
<pre><code>## 
## Attaching package: &#39;MASS&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     select</code></pre>
<pre><code>## 
## Attaching package: &#39;TH.data&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:MASS&#39;:
## 
##     geyser</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>fitGlht <span class="ot">&lt;-</span> <span class="fu">glht</span>(fit, <span class="at">linfct =</span> <span class="fu">t</span>(L))</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a><span class="fu">summary</span>(fitGlht, <span class="at">test =</span> <span class="fu">adjusted</span>(<span class="st">&quot;none&quot;</span>))</span></code></pre></div>
<pre><code>## 
##   Simultaneous Tests for General Linear Hypotheses
## 
## Fit: lm(formula = y ~ location * tissue + patient, data = colData(pe), 
##     x = TRUE)
## 
## Linear Hypotheses:
##                                        Estimate Std. Error t value Pr(&gt;|t|)
## tissueV == 0                            0.12757    0.72469   0.176    0.866
## tissueV + locationR:tissueV == 0        0.08626    0.72469   0.119    0.909
## tissueV + 0.5 * locationR:tissueV == 0  0.10691    0.51243   0.209    0.842
## locationR:tissueV == 0                 -0.04131    1.02486  -0.040    0.969
## (Adjusted p values reported -- none method)</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="fu">data.frame</span>(contrasts, seContrasts)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["contrasts"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["seContrasts"],"name":[2],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.12756675","2":"0.7246861","_rn_":"tissueV"},{"1":"0.08625832","2":"0.7246861","_rn_":"tissueV + locationR:tissueV"},{"1":"0.10691253","2":"0.5124304","_rn_":"tissueV + 0.5 * locationR:tissueV"},{"1":"-0.04130843","2":"1.0248609","_rn_":"locationR:tissueV"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<ul>
<li><p>Note, that the power for assessing <span
class="math inline">\(\log_2\)</span> FC between ventriculum and atrium
left and right is the same. Indeed, the standard errors are equal for
both effects.</p></li>
<li><p>Note, that the power for assessing <span
class="math inline">\(\log_2\)</span> FC between ventriculum and atrium
over both heart regions is higher than when assessing the effect left or
right.</p>
<ul>
<li>Indeed, the standard error is a factor <span
class="math inline">\(\sqrt{2}\)</span> smaller for the former
effect</li>
<li>We intuitively can explain this because we can use all samples
(double the number of samples) to assess the average effect.</li>
<li>Hence the variance is a factor two smaller, and the se with a factor
<span class="math inline">\(\sqrt{2}\)</span></li>
</ul></li>
<li><p>Note, that we have the lowest power to pick up an interaction
effect. Indeed, the se is a factor <span
class="math inline">\(\sqrt{2}\)</span> larger than for the ventriculum
- atrium effect left or right and a factor 2 larger than for the average
effect between ventriculum and atrium.</p></li>
</ul>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>seContrasts <span class="sc">/</span> seContrasts[<span class="dv">1</span>]</span></code></pre></div>
<pre><code>##                           tissueV       tissueV + locationR:tissueV 
##                         1.0000000                         1.0000000 
## tissueV + 0.5 * locationR:tissueV                 locationR:tissueV 
##                         0.7071068                         1.4142136</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a><span class="fu">sqrt</span>(<span class="dv">2</span>)</span></code></pre></div>
<pre><code>## [1] 1.414214</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="dv">1</span><span class="sc">/</span><span class="fu">sqrt</span>(<span class="dv">2</span>)</span></code></pre></div>
<pre><code>## [1] 0.7071068</code></pre>
</div>
<div id="t-tests" class="section level3" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> t-tests</h3>
<ul>
<li><p>When the assumptions of the linear model hold <span
class="math display">\[
\hat{\boldsymbol{\beta}} \sim
MVN\left[\boldsymbol{\beta},\left(\mathbf{X}^T\mathbf{X}\right)^{-1}\sigma^2\right]
\]</span></p></li>
<li><p>Hence, <span class="math display">\[
\mathbf{L}^T\hat{\boldsymbol{\beta}} \sim
MVN\left[\mathbf{L}^T\boldsymbol{\beta},\mathbf{L}^T\left[\left(\mathbf{X}^T\mathbf{X}\right)^{-1}\sigma^2\right]\mathbf{L}\right]
\]</span></p></li>
<li><p>We estimate <span class="math inline">\(\sigma^2\)</span> by MSE
<span
class="math display">\[\hat{\sigma}^2=\frac{\mathbf{e}^T\mathbf{e}}{n-p}
\rightarrow
\hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}=\left(\mathbf{X}^T\mathbf{X}\right)^{-1}\hat\sigma^2\]</span></p></li>
<li><p>When we test one contrast at the time (e.g. the <span
class="math inline">\(k^\text{th}\)</span> contrast) the statistic
reduces to</p></li>
</ul>
<p><span
class="math display">\[T=\frac{\mathbf{L}_k^T\hat{\boldsymbol{\beta}}}{\sqrt{\left(\mathbf{L}^T_k\hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}\mathbf{L}_k\right)}}
\underset{H_0}{\sim} t_{n-p}\]</span> follows a t distribution with n-p
degrees of freedom under <span class="math inline">\(H_0:
\mathbf{L}^T_k\hat{\boldsymbol{\beta}}=0\)</span></p>
<hr />
<div id="heart-example-3" class="section level4" number="2.2.2.1">
<h4><span class="header-section-number">2.2.2.1</span> Heart
example</h4>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>tContrasts <span class="ot">&lt;-</span> contrasts<span class="sc">/</span>seContrasts</span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a>pContrasts <span class="ot">&lt;-</span> <span class="fu">pt</span>(<span class="fu">abs</span>(tContrasts),</span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a>  <span class="at">df =</span> n <span class="sc">-</span> p,</span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a>  <span class="at">lower.tail =</span> <span class="cn">FALSE</span>) <span class="sc">*</span> <span class="dv">2</span></span></code></pre></div>
<p>Comparison with lm and glht results</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a><span class="fu">summary</span>(fitGlht, <span class="at">test =</span> <span class="fu">adjusted</span>(<span class="st">&quot;none&quot;</span>))</span></code></pre></div>
<pre><code>## 
##   Simultaneous Tests for General Linear Hypotheses
## 
## Fit: lm(formula = y ~ location * tissue + patient, data = colData(pe), 
##     x = TRUE)
## 
## Linear Hypotheses:
##                                        Estimate Std. Error t value Pr(&gt;|t|)
## tissueV == 0                            0.12757    0.72469   0.176    0.866
## tissueV + locationR:tissueV == 0        0.08626    0.72469   0.119    0.909
## tissueV + 0.5 * locationR:tissueV == 0  0.10691    0.51243   0.209    0.842
## locationR:tissueV == 0                 -0.04131    1.02486  -0.040    0.969
## (Adjusted p values reported -- none method)</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="fu">data.frame</span>(contrasts, seContrasts, tContrasts, pContrasts)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["contrasts"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["seContrasts"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["tContrasts"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["pContrasts"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.12756675","2":"0.7246861","3":"0.17603036","4":"0.8660611","_rn_":"tissueV"},{"1":"0.08625832","2":"0.7246861","3":"0.11902853","4":"0.9091380","_rn_":"tissueV + locationR:tissueV"},{"1":"0.10691253","2":"0.5124304","3":"0.20863814","4":"0.8416333","_rn_":"tissueV + 0.5 * locationR:tissueV"},{"1":"-0.04130843","2":"1.0248609","3":"-0.04030638","4":"0.9691566","_rn_":"locationR:tissueV"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<hr />
</div>
</div>
</div>
</div>
<div id="robust-regression" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Robust regression</h1>
<ul>
<li><p>With msqrob2 we perform robust regression to estimate the model
parameters of the regression model</p></li>
<li><p>No normality assumption needed</p></li>
<li><p>Robust fit minimises the maximal bias of the estimators</p></li>
<li><p>CI and statistical tests are based on asymptotic theory</p></li>
<li><p>If <span class="math inline">\(\epsilon\)</span> is normal, the
M-estimators have a high efficiency!</p></li>
<li><p>ordinary least squares (OLS): minimize loss function <span
class="math display">\[\sum\limits_{i=1}^n
(y_i-\mathbf{x}_i^T\boldsymbol{\beta})^2\]</span></p></li>
<li><p>M-estimation: minimize loss function <span
class="math display">\[\sum\limits_{i=1}^n  \rho\left(y_i-\mathbf{x}_i^T\boldsymbol{\beta}\right)\]</span>
with</p>
<ul>
<li><span class="math inline">\(\rho\)</span> is symmetric, i.e. <span
class="math inline">\(\rho(z)=\rho(-z)\)</span></li>
<li><span class="math inline">\(\rho\)</span> has a minimum at <span
class="math inline">\(\rho(0)=0\)</span>, is positive for all <span
class="math inline">\(z\neq 0\)</span></li>
<li><span class="math inline">\(\rho(z)\)</span> increases as <span
class="math inline">\(\vert z\vert\)</span> increases</li>
</ul></li>
</ul>
<hr />
<p>The estimator <span class="math inline">\(\hat{\mu}\)</span> is also
the solution to the equation <span class="math display">\[
   \sum_{i=1}^n \Psi(y_i - \mathbf{x}_i\boldsymbol{\beta}) =0,
\]</span> where <span class="math inline">\(\Psi\)</span> is the
derivative of <span class="math inline">\(\rho\)</span>. For <span
class="math inline">\(\hat{\beta}\)</span> possessing the robustness
property, <span class="math inline">\(\Psi\)</span> should be
bounded.</p>
<hr />
<p>Example: least squares</p>
<ul>
<li><p><span class="math inline">\(\rho(z) = z^2\)</span>, and thus
<span class="math inline">\(\Psi(z)=2z\)</span> (unbounded!). Not
robust!</p></li>
<li><p><span class="math inline">\(\hat{\boldsymbol{\beta}}\)</span> is
the solution of <span class="math display">\[
\sum_{i=1}^n 2 \mathbf{x}_i (y_i - \mathbf{x}_i^T\boldsymbol{\beta}) = 0
\text{ or } \hat{\boldsymbol{\beta}} =
(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}\mathbf{y}
\]</span> with <span class="math inline">\(\mathbf{X}=[\mathbf{x}_1
\ldots \mathbf{x}_G]^T\)</span></p></li>
</ul>
<hr />
<p>When a location and a scale parameter, say <span
class="math inline">\(\sigma\)</span>, have to be estimated
simultaneously, we write <span class="math display">\[
   (\hat{\boldsymbol{\beta}},\hat{\sigma}) =
\text{ArgMin}_{\boldsymbol{\beta},\sigma} \sum_{i=1}^n
\rho\left(\frac{y_i - \mathbf{x}_i^T\boldsymbol{\beta}}{\sigma}\right)
   \text{ and } \sum_{i=1}^n \Psi\left(\frac{y_i -
\mathbf{x}_i^T\boldsymbol{\beta}}{\sigma}\right) =0.
\]</span></p>
<p>Define <span class="math inline">\(u_i = \frac{y_i -
\mathbf{x}_i^T\boldsymbol{\beta}}{\sigma}\)</span>. The last estimation
equation is equivalent to <span class="math display">\[
   \sum_{i=1}^n w(u_i) u_i = 0 ,
\]</span> with weight function <span
class="math inline">\(w(u)=\Psi(u)/u\)</span>. This is the typical form
that appears when solving the <em>iteratively reweighted least squares
problem</em>, <span class="math display">\[
   (\hat{\boldsymbol{\beta}},\hat{\sigma}) = \text{ArgMin}_{\mu,\sigma}
\sum_{i=1}^n w(u_i^{(k-1)}) \left(u_i^{(k)}\right)^2 ,
\]</span> where <span class="math inline">\(k\)</span> represents the
iteration number.</p>
<hr />
<div id="some-examples-of-robust-functions" class="section level2"
number="3.1">
<h2><span class="header-section-number">3.1</span> Some Examples of
Robust Functions</h2>
<p><img
src="https://raw.githubusercontent.com/statOmics/SGA2020/gh-pages/assets/TableRobust.PNG" /></p>
<p>PhD thesis Bolstad 2004</p>
<hr />
</div>
<div id="the-rho-functions" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> The <span
class="math inline">\(\rho\)</span> functions</h2>
<p><img
src="https://raw.githubusercontent.com/statOmics/SGA2020/gh-pages/assets/RhoRobust.PNG" /></p>
<p>PhD thesis Bolstad 2004</p>
<hr />
<div id="common-psi-functions" class="section level3" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Common <span
class="math inline">\(\Psi\)</span>-Functions</h3>
<p><img
src="https://raw.githubusercontent.com/statOmics/SGA2020/gh-pages/assets/robustRegressionPsi.png" /></p>
<p>PhD thesis Bolstad 2004</p>
<hr />
</div>
<div id="corresponding-weight-functions" class="section level3"
number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> Corresponding
Weight Functions</h3>
<p><img
src="https://raw.githubusercontent.com/statOmics/SGA2020/gh-pages/assets/robustRegressionWeights.png" /></p>
<p>PhD thesis Bolstad 2004</p>
<hr />
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;MASS&quot;</span>)</span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a>rfit <span class="ot">&lt;-</span> <span class="fu">rlm</span>(y <span class="sc">~</span> location <span class="sc">*</span> tissue <span class="sc">+</span> patient, <span class="fu">colData</span>(pe), <span class="at">maxit=</span><span class="dv">1</span>)</span></code></pre></div>
<pre><code>## Warning in rlm.default(x, y, weights, method = method, wt.method = wt.method, :
## &#39;rlm&#39; failed to converge in 1 steps</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="fu">qplot</span>(fit<span class="sc">$</span>coefficient[<span class="sc">-</span><span class="dv">1</span>],</span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a>  rfit<span class="sc">$</span>coefficient[<span class="sc">-</span><span class="dv">1</span>],</span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a>  <span class="at">xlab=</span><span class="st">&quot;fit&quot;</span>,</span>
<span id="cb42-4"><a href="#cb42-4" tabindex="-1"></a>  <span class="at">ylab=</span><span class="st">&quot;robust fit&quot;</span>) <span class="sc">+</span></span>
<span id="cb42-5"><a href="#cb42-5" tabindex="-1"></a>  <span class="fu">geom_abline</span>() <span class="sc">+</span></span>
<span id="cb42-6"><a href="#cb42-6" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="fu">range</span>(<span class="fu">c</span>(fit<span class="sc">$</span>coefficient[<span class="sc">-</span><span class="dv">1</span>],rfit<span class="sc">$</span>coefficient[<span class="sc">-</span><span class="dv">1</span>]))) <span class="sc">+</span></span>
<span id="cb42-7"><a href="#cb42-7" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="fu">range</span>(<span class="fu">c</span>(fit<span class="sc">$</span>coefficient[<span class="sc">-</span><span class="dv">1</span>],rfit<span class="sc">$</span>coefficient[<span class="sc">-</span><span class="dv">1</span>])))</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<hr />
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>rfit<span class="sc">$</span>w</span></code></pre></div>
<pre><code>##  [1] 0.9516397 1.0000000 0.6628477 1.0000000 1.0000000 1.0000000 1.0000000
##  [8] 1.0000000 1.0000000 0.4805187 1.0000000 0.4344049</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="fu">plot</span>(</span>
<span id="cb45-2"><a href="#cb45-2" tabindex="-1"></a>  rfit<span class="sc">$</span>fitted,</span>
<span id="cb45-3"><a href="#cb45-3" tabindex="-1"></a>  rfit<span class="sc">$</span>res,</span>
<span id="cb45-4"><a href="#cb45-4" tabindex="-1"></a>  <span class="at">cex=</span>rfit<span class="sc">$</span>w,</span>
<span id="cb45-5"><a href="#cb45-5" tabindex="-1"></a>  <span class="at">pch=</span><span class="dv">19</span>,<span class="at">col=</span><span class="dv">2</span>,</span>
<span id="cb45-6"><a href="#cb45-6" tabindex="-1"></a>  <span class="at">cex.lab=</span><span class="fl">1.5</span>,</span>
<span id="cb45-7"><a href="#cb45-7" tabindex="-1"></a>  <span class="at">cex.axis=</span><span class="fl">1.5</span>,</span>
<span id="cb45-8"><a href="#cb45-8" tabindex="-1"></a>  <span class="at">ylab=</span><span class="st">&quot;residuals&quot;</span>,</span>
<span id="cb45-9"><a href="#cb45-9" tabindex="-1"></a>  <span class="at">xlab=</span><span class="st">&quot;fit&quot;</span>)</span>
<span id="cb45-10"><a href="#cb45-10" tabindex="-1"></a><span class="fu">points</span>(rfit<span class="sc">$</span>fitted, rfit<span class="sc">$</span>res , <span class="at">cex=</span> <span class="fl">1.5</span>)</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<hr />
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a><span class="fu">summary</span>(fit)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = y ~ location * tissue + patient, data = colData(pe), 
##     x = TRUE)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -1.29698 -0.32767  0.04041  0.24905  1.17251 
## 
## Coefficients:
##                   Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)       26.57559    0.62760  42.345 1.16e-08 ***
## locationR          0.20636    0.72469   0.285    0.785    
## tissueV            0.12757    0.72469   0.176    0.866    
## patient4           0.30592    0.62760   0.487    0.643    
## patient8          -0.33433    0.62760  -0.533    0.613    
## locationR:tissueV -0.04131    1.02486  -0.040    0.969    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.8876 on 6 degrees of freedom
## Multiple R-squared:  0.1687, Adjusted R-squared:  -0.524 
## F-statistic: 0.2436 on 5 and 6 DF,  p-value: 0.9286</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a><span class="fu">summary</span>(rfit)</span></code></pre></div>
<pre><code>## 
## Call: rlm(formula = y ~ location * tissue + patient, data = colData(pe), 
##     maxit = 1)
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -1.51703 -0.22326  0.05909  0.16009  1.26070 
## 
## Coefficients:
##                   Value   Std. Error t value
## (Intercept)       26.3378  0.5391    48.8585
## locationR          0.2883  0.6225     0.4631
## tissueV            0.2095  0.6225     0.3365
## patient4           0.4654  0.5391     0.8633
## patient8          -0.0261  0.5391    -0.0484
## locationR:tissueV -0.0555  0.8803    -0.0630
## 
## Residual standard error: 0.4189 on 6 degrees of freedom</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>msqrobModels[[<span class="dv">2</span>]] <span class="sc">%&gt;%</span> getCoef</span></code></pre></div>
<pre><code>##       (Intercept)         locationR           tissueV          patient4 
##       26.33779939        0.28826213        0.20946392        0.46536955 
##          patient8 locationR:tissueV 
##       -0.02608046       -0.05550111</code></pre>
<hr />
</div>
</div>
<div id="understanding-implementation-of-robust-regression"
class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Understanding
implementation of robust regression</h2>
<div
id="simulate-20-observations-from-a-linear-model-with-errors-that-follow-a-normal-distribution"
class="section level3" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Simulate 20
observations from a linear model with errors that follow a normal
distribution</h3>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" tabindex="-1"></a>set.seed <span class="ot">&lt;-</span> <span class="dv">112358</span></span>
<span id="cb52-2"><a href="#cb52-2" tabindex="-1"></a>nobs <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb52-3"><a href="#cb52-3" tabindex="-1"></a>sdy <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb52-4"><a href="#cb52-4" tabindex="-1"></a>xsim <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">length.out =</span> nobs)</span>
<span id="cb52-5"><a href="#cb52-5" tabindex="-1"></a>ysim <span class="ot">&lt;-</span> <span class="dv">10</span> <span class="sc">+</span> <span class="dv">5</span><span class="sc">*</span>xsim <span class="sc">+</span> <span class="fu">rnorm</span>(nobs, <span class="at">sd =</span> sdy)</span></code></pre></div>
</div>
<div id="add-outlier-at-high-leverage-point" class="section level3"
number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> add outlier at high
leverage point</h3>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" tabindex="-1"></a>ysim[nobs] <span class="ot">&lt;-</span> <span class="dv">7</span></span></code></pre></div>
</div>
<div id="fit-linear-model" class="section level3" number="3.3.3">
<h3><span class="header-section-number">3.3.3</span> fit linear
model</h3>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" tabindex="-1"></a>ols <span class="ot">&lt;-</span> <span class="fu">lm</span>(ysim <span class="sc">~</span> xsim)</span></code></pre></div>
</div>
<div id="fit-robust-linear-model" class="section level3" number="3.3.4">
<h3><span class="header-section-number">3.3.4</span> fit robust linear
model</h3>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb55-2"><a href="#cb55-2" tabindex="-1"></a>mEst <span class="ot">&lt;-</span> <span class="fu">rlm</span>(ysim <span class="sc">~</span> xsim)</span></code></pre></div>
<div id="plot-results" class="section level4" number="3.3.4.1">
<h4><span class="header-section-number">3.3.4.1</span> plot results</h4>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" tabindex="-1"></a><span class="fu">plot</span>(xsim, ysim)</span>
<span id="cb56-2"><a href="#cb56-2" tabindex="-1"></a><span class="fu">abline</span>(ols, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb56-3"><a href="#cb56-3" tabindex="-1"></a><span class="fu">abline</span>(mEst, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb56-4"><a href="#cb56-4" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topleft&quot;</span>,</span>
<span id="cb56-5"><a href="#cb56-5" tabindex="-1"></a>  <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;OLS&quot;</span>, <span class="st">&quot;M-estimation&quot;</span>),</span>
<span id="cb56-6"><a href="#cb56-6" tabindex="-1"></a>  <span class="at">lwd =</span> <span class="dv">2</span>,</span>
<span id="cb56-7"><a href="#cb56-7" tabindex="-1"></a>  <span class="at">col =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" tabindex="-1"></a><span class="fu">round</span>(mEst<span class="sc">$</span>w,<span class="dv">3</span>)</span></code></pre></div>
<pre><code>##  [1] 1.000 1.000 1.000 1.000 1.000 1.000 1.000 1.000 1.000 1.000 1.000 1.000
## [13] 1.000 1.000 1.000 1.000 1.000 1.000 1.000 0.255</code></pre>
<p>The plot clearly shows that the outlier has a high impact on the
slope estimate. This is because the outlier is at a high leverage point,
i.e. far from the average covariate pattern.</p>
</div>
</div>
<div id="implement-it-yourself" class="section level3" number="3.3.5">
<h3><span class="header-section-number">3.3.5</span> Implement it
yourself</h3>
<div id="start-from-ols-fit" class="section level4" number="3.3.5.1">
<h4><span class="header-section-number">3.3.5.1</span> start from ols
fit</h4>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" tabindex="-1"></a>lmMod <span class="ot">&lt;-</span> ols</span></code></pre></div>
</div>
<div id="use-robust-variance-estimator-to-calculate-the-z"
class="section level4" number="3.3.5.2">
<h4><span class="header-section-number">3.3.5.2</span> Use robust
variance estimator to calculate the z</h4>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> lmMod<span class="sc">$</span>res</span>
<span id="cb60-2"><a href="#cb60-2" tabindex="-1"></a>stdev <span class="ot">&lt;-</span> <span class="fu">mad</span>(res)</span>
<span id="cb60-3"><a href="#cb60-3" tabindex="-1"></a>stdev</span></code></pre></div>
<pre><code>## [1] 1.200969</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" tabindex="-1"></a><span class="fu">median</span>(<span class="fu">abs</span>(res<span class="sc">-</span><span class="fu">median</span>(res)))<span class="sc">*</span><span class="fl">1.4826</span></span></code></pre></div>
<pre><code>## [1] 1.200969</code></pre>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" tabindex="-1"></a>z <span class="ot">&lt;-</span> res<span class="sc">/</span>stdev</span></code></pre></div>
</div>
<div id="calculate-weights-use-psi.huber-function"
class="section level4" number="3.3.5.3">
<h4><span class="header-section-number">3.3.5.3</span> Calculate weights
use psi.huber function</h4>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" tabindex="-1"></a>w <span class="ot">&lt;-</span> <span class="fu">psi.huber</span>(z)</span>
<span id="cb65-2"><a href="#cb65-2" tabindex="-1"></a><span class="fu">plot</span>(xsim, ysim)</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" tabindex="-1"></a><span class="fu">plot</span>(xsim, lmMod<span class="sc">$</span>res, <span class="at">cex =</span> w, <span class="at">pch =</span> <span class="dv">19</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span>
<span id="cb66-2"><a href="#cb66-2" tabindex="-1"></a><span class="fu">points</span>(xsim,lmMod<span class="sc">$</span>res, <span class="at">cex =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-24-2.png" width="672" /></p>
</div>
<div id="perform-a-weighted-regression-use-lm-with-weightsw"
class="section level4" number="3.3.5.4">
<h4><span class="header-section-number">3.3.5.4</span> Perform a
weighted regression use lm with weights=w</h4>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" tabindex="-1"></a>lmMod <span class="ot">&lt;-</span> <span class="fu">lm</span>(ysim<span class="sc">~</span>xsim, <span class="at">weights =</span> w)</span></code></pre></div>
</div>
<div id="plot-results-1" class="section level4" number="3.3.5.5">
<h4><span class="header-section-number">3.3.5.5</span> Plot results</h4>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" tabindex="-1"></a><span class="fu">plot</span>(xsim, ysim)</span>
<span id="cb68-2"><a href="#cb68-2" tabindex="-1"></a><span class="fu">abline</span>(ols, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb68-3"><a href="#cb68-3" tabindex="-1"></a><span class="fu">abline</span>(mEst, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb68-4"><a href="#cb68-4" tabindex="-1"></a><span class="fu">abline</span>(lmMod, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb68-5"><a href="#cb68-5" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topleft&quot;</span>,</span>
<span id="cb68-6"><a href="#cb68-6" tabindex="-1"></a>  <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;OLS&quot;</span>,<span class="st">&quot;M-estimation&quot;</span>,<span class="st">&quot;Our Impl&quot;</span>),</span>
<span id="cb68-7"><a href="#cb68-7" tabindex="-1"></a>  <span class="at">lwd =</span> <span class="dv">2</span>,</span>
<span id="cb68-8"><a href="#cb68-8" tabindex="-1"></a>  <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>))</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
</div>
<div id="repeat-this-many-times" class="section level4"
number="3.3.5.6">
<h4><span class="header-section-number">3.3.5.6</span> Repeat this many
times</h4>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" tabindex="-1"></a>lmMod <span class="ot">&lt;-</span> ols</span>
<span id="cb69-2"><a href="#cb69-2" tabindex="-1"></a><span class="cf">for</span> (k <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)</span>
<span id="cb69-3"><a href="#cb69-3" tabindex="-1"></a>{</span>
<span id="cb69-4"><a href="#cb69-4" tabindex="-1"></a><span class="do">######repeat this part several times until convergence</span></span>
<span id="cb69-5"><a href="#cb69-5" tabindex="-1"></a><span class="co">#use robust variance estimator to calculate the z</span></span>
<span id="cb69-6"><a href="#cb69-6" tabindex="-1"></a>res <span class="ot">&lt;-</span> lmMod<span class="sc">$</span>res</span>
<span id="cb69-7"><a href="#cb69-7" tabindex="-1"></a>stdev <span class="ot">&lt;-</span> <span class="fu">mad</span>(res)</span>
<span id="cb69-8"><a href="#cb69-8" tabindex="-1"></a><span class="fu">median</span>(<span class="fu">abs</span>(res<span class="sc">-</span><span class="fu">median</span>(res)))<span class="sc">*</span><span class="fl">1.4826</span></span>
<span id="cb69-9"><a href="#cb69-9" tabindex="-1"></a></span>
<span id="cb69-10"><a href="#cb69-10" tabindex="-1"></a>z <span class="ot">&lt;-</span> res<span class="sc">/</span>stdev</span>
<span id="cb69-11"><a href="#cb69-11" tabindex="-1"></a></span>
<span id="cb69-12"><a href="#cb69-12" tabindex="-1"></a><span class="co">#calculate weights</span></span>
<span id="cb69-13"><a href="#cb69-13" tabindex="-1"></a><span class="co">#use psi.huber function</span></span>
<span id="cb69-14"><a href="#cb69-14" tabindex="-1"></a>w <span class="ot">&lt;-</span> <span class="fu">psi.huber</span>(z)</span>
<span id="cb69-15"><a href="#cb69-15" tabindex="-1"></a></span>
<span id="cb69-16"><a href="#cb69-16" tabindex="-1"></a><span class="co">#perform a weighted regression use lm with weights=w</span></span>
<span id="cb69-17"><a href="#cb69-17" tabindex="-1"></a>lmMod <span class="ot">&lt;-</span> <span class="fu">lm</span>(ysim <span class="sc">~</span> xsim, <span class="at">weights =</span> w)</span>
<span id="cb69-18"><a href="#cb69-18" tabindex="-1"></a></span>
<span id="cb69-19"><a href="#cb69-19" tabindex="-1"></a><span class="co">#plot results</span></span>
<span id="cb69-20"><a href="#cb69-20" tabindex="-1"></a><span class="fu">plot</span>(xsim,ysim)</span>
<span id="cb69-21"><a href="#cb69-21" tabindex="-1"></a><span class="fu">abline</span>(ols, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb69-22"><a href="#cb69-22" tabindex="-1"></a><span class="fu">abline</span>(mEst, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb69-23"><a href="#cb69-23" tabindex="-1"></a><span class="fu">abline</span>(lmMod, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb69-24"><a href="#cb69-24" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topleft&quot;</span>,</span>
<span id="cb69-25"><a href="#cb69-25" tabindex="-1"></a>  <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;OLS&quot;</span>,<span class="st">&quot;M-estimation&quot;</span>,<span class="st">&quot;Our Impl&quot;</span>),</span>
<span id="cb69-26"><a href="#cb69-26" tabindex="-1"></a>  <span class="at">lwd =</span> <span class="dv">2</span>,</span>
<span id="cb69-27"><a href="#cb69-27" tabindex="-1"></a>  <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb69-28"><a href="#cb69-28" tabindex="-1"></a>  )</span>
<span id="cb69-29"><a href="#cb69-29" tabindex="-1"></a><span class="do">####################################</span></span>
<span id="cb69-30"><a href="#cb69-30" tabindex="-1"></a>}</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-27-1.png" width="672" /><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-27-2.png" width="672" /><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-27-3.png" width="672" /></p>
</div>
</div>
</div>
</div>
<div id="empirical-bayesmoderated-t-test." class="section level1"
number="4">
<h1><span class="header-section-number">4</span> Empirical
Bayes/Moderated <span class="math inline">\(t\)</span>-test.</h1>
<p>A general class of moderated test statistics is given by</p>
<p><span class="math display">\[\tilde{T}_p = \frac{\mathbf{L}_k \hat{
\boldsymbol{\beta_p}}}{\mathbf{L}_k^T(\mathbf{X}^T\mathbf{WX})^{-1}\mathbf{L}_k^T
\tilde{s}_p^2}\]</span></p>
<p>where <span class="math inline">\(\tilde{s}_p\)</span> is a moderated
variance estimator.</p>
<p>Simple approach: set <span class="math inline">\(\tilde{s}_p=s_p +
s_0\)</span>: simply add a small positive constant to the denominator of
the t-statistic</p>
<p> theory provides formal framework for borrowing strength across genes
or proteins, e.g. popular bioconductor package <span
class="math display">\[\tilde{s}_g=\sqrt{\frac{d_ps_p^2+d_0s_0^2}{d_g+d_0}},\]</span>
and the moderated t-statistic is t-distributed with <span
class="math inline">\(d_0+d_g\)</span> degrees of freedom under the null
hypothesis <span class="math inline">\(H_0:
\mathbf{L}\boldsymbol{\beta}=0\)</span>.</p>
<ul>
<li>Note, that the degrees of freedom increase by borrowing strength
across proteins.</li>
</ul>
<hr />
<div id="intermezzo-bayesian-methods" class="section level2"
number="4.1">
<h2><span class="header-section-number">4.1</span> Intermezzo: Bayesian
Methods</h2>
<ul>
<li><p>Frequentists consider data as random and population parameters as
fixed but unknown</p></li>
<li><p>In Bayesian viewpoint a person has prior beliefs about the
population parameters and the uncertainty on this prior beliefs are
represented by a probability distribution placed on this parameter.</p>
<ul>
<li>This distribution reflects the person’s subjective prior opinion
about plausible values of the parameter.</li>
<li>And is referred to as the prior <span
class="math inline">\(g(\boldsymbol{\theta})\)</span>.</li>
</ul></li>
<li><p>Bayesian thinking will update the prior information on the
population parameters by confronting the model to data (<span
class="math inline">\(\mathbf{Y}\)</span>).</p></li>
<li><p>By using Bayes Theorem this results in a posterior distribution
on the model parameters.<br />
<span class="math display">\[
g(\boldsymbol{\theta}\vert\mathbf{Y})=\frac{f(Y\vert
\boldsymbol{\theta})g(\boldsymbol{\theta})}{\int f(Y\vert
\boldsymbol{\theta}) g(\boldsymbol{\theta}) d\boldsymbol{\theta}}
\text{     }\left(\text{ posterior}=\frac{\text{prior} \times \text{
likelihood}}{\text{Marginal distribution}}\right)
\]</span></p></li>
</ul>
<hr />
</div>
<div id="limma-approach" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Limma approach</h2>
<p>Developed for gene expression analysis with micro arrays. Let g be
the index for gene g. <span class="math display">\[
\begin{array}{cc}
&amp;\beta_{gk}\vert \sigma^2_g,\beta_{gk}\neq 0 \sim
N(0,v_{0k}\sigma_g^2)\\\\
\text{Prior}\\
&amp;\frac{1}{\sigma^2_g}\sim s^2_0\frac{\chi^2_{d_0}}{d_0}\\\\\\\\
&amp;\hat \beta_{gk} | \beta_{gk} , \sigma_g^2 \sim N( \beta_{gk} ,
v_{gk}\sigma_g^2)\\\\
\text{Data}\\
&amp;s_g^2\sim \sigma^2_g\frac{\chi^2_{d_g}}{d_g}\\\\
\end{array}
\]</span></p>
<hr />
</div>
<div id="limma-approach-1" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Limma approach</h2>
<p>Under this assumption, it can be shown that</p>
<ul>
<li><p>Posterior Mean for the variance parameter: <span
class="math display">\[\tilde{s}^2_p = \text{E}\left[\sigma^2_p\vert
s_p^2\right]=\frac{d_0 s_0^2+d_ps_p^2}{d_0+d_p}\]</span></p></li>
<li><p><span class="math display">\[\tilde{T}_p=\frac{\mathbf{L}_k \hat{
\boldsymbol{\beta_p}}}{\mathbf{L}_k^T(\mathbf{X}^T\mathbf{WX})^{-1}\mathbf{L}_k^T
\tilde{s}_p^2}\]</span></p></li>
</ul>
<p>is t-distributed under <span class="math inline">\(H_0:
\mathbf{L}_j\boldsymbol{\beta} = 0\)</span></p>
<p><span class="math display">\[\tilde{T}_p \vert H_0 \sim t(d_0 +
d_p)\]</span></p>
<hr />
</div>
<div id="empirical-bayes" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Empirical Bayes</h2>
<ul>
<li>A fully Bayesian
<ul>
<li>would define the prior distribution by carefully choosing the prior
parameters based on prior knowledge on the process</li>
<li>would confront the prior to the data and performs inference using
the posterior distribution of the model parameters.</li>
</ul></li>
<li>In an empirical Bayesian approach one estimates the prior parameters
based on the data.</li>
<li>In <strong>Limma</strong> moment estimators for <span
class="math inline">\(s_0\)</span> and <span
class="math inline">\(d_0\)</span> are derived using the information on
the gene (protein) wise variances of all genes (proteins).</li>
<li>In <strong>Limma</strong> one also does not work with the full
posterior distribution for the variances, but with the maximum
a-posterior estimate.</li>
</ul>
<hr />
</div>
<div id="illustration" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Illustration</h2>
<p>We borrow strength across proteins by</p>
<ol style="list-style-type: decimal">
<li>placing a scaled <span class="math inline">\(\chi^2\)</span> prior:
<span class="math inline">\(\chi^2(s_o,d_0)\)</span> on the precisions
(<span class="math inline">\(1/\sigma^2_p\)</span>)</li>
<li>estimating the prior parameters <span
class="math inline">\(s_0\)</span> and <span
class="math inline">\(df_0\)</span></li>
<li>replacing the estimated protein-wise variances (<span
class="math inline">\(s_p^2\)</span>) with the maximum a-posteriori
variance <span class="math display">\[\tilde{s}_p = \frac{d_p s^2_p +
d_0 s_0^2}{d_p+d_0}\]</span></li>
</ol>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" tabindex="-1"></a>sd <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb70-2"><a href="#cb70-2" tabindex="-1"></a>  <span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>msqrobModels,</span>
<span id="cb70-3"><a href="#cb70-3" tabindex="-1"></a>  getSigma) <span class="sc">%&gt;%</span></span>
<span id="cb70-4"><a href="#cb70-4" tabindex="-1"></a>  na.exclude</span>
<span id="cb70-5"><a href="#cb70-5" tabindex="-1"></a>sdPost <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb70-6"><a href="#cb70-6" tabindex="-1"></a>  <span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>msqrobModels,</span>
<span id="cb70-7"><a href="#cb70-7" tabindex="-1"></a>  getSigmaPosterior) <span class="sc">%&gt;%</span></span>
<span id="cb70-8"><a href="#cb70-8" tabindex="-1"></a>  na.exclude</span>
<span id="cb70-9"><a href="#cb70-9" tabindex="-1"></a></span>
<span id="cb70-10"><a href="#cb70-10" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">qplot</span>(sd,sdPost) <span class="sc">+</span></span>
<span id="cb70-11"><a href="#cb70-11" tabindex="-1"></a>  <span class="fu">geom_abline</span>()</span>
<span id="cb70-12"><a href="#cb70-12" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
<div id="how-do-we-get-to-the-posterior-standard-deviation"
class="section level3" number="4.5.1">
<h3><span class="header-section-number">4.5.1</span> How do we get to
the posterior standard deviation?</h3>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" tabindex="-1"></a>hlp <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">squeezeVar</span>(</span>
<span id="cb71-2"><a href="#cb71-2" tabindex="-1"></a>  <span class="at">var =</span> <span class="fu">sapply</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>msqrobModels, getVar),</span>
<span id="cb71-3"><a href="#cb71-3" tabindex="-1"></a>  <span class="at">df =</span> <span class="fu">sapply</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>msqrobModels, getDF)</span>
<span id="cb71-4"><a href="#cb71-4" tabindex="-1"></a>  )</span></code></pre></div>
<div id="degrees-of-freedom-of-prior" class="section level4"
number="4.5.1.1">
<h4><span class="header-section-number">4.5.1.1</span> Degrees of
freedom of prior</h4>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" tabindex="-1"></a>hlp<span class="sc">$</span>df.prior</span></code></pre></div>
<pre><code>## [1] 3.385413</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>msqrobModels[[<span class="dv">2</span>]]</span>
<span id="cb74-2"><a href="#cb74-2" tabindex="-1"></a><span class="fu">getDfPosterior</span>(model) <span class="sc">-</span> <span class="fu">getDF</span>(model)</span></code></pre></div>
<pre><code>## [1] 3.385413</code></pre>
</div>
<div id="posterior-variance" class="section level4" number="4.5.1.2">
<h4><span class="header-section-number">4.5.1.2</span> posterior
variance</h4>
<p><span class="math display">\[\tilde s_p=\sqrt{\frac{d_p\times s^2_p +
d_0 s_0^2}{d_p+d_0}} \]</span></p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" tabindex="-1"></a>hlp<span class="sc">$</span>var.prior</span></code></pre></div>
<pre><code>## [1] 0.2489859</code></pre>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" tabindex="-1"></a>varPost <span class="ot">&lt;-</span> (<span class="fu">getVar</span>(model) <span class="sc">*</span> <span class="fu">getDF</span>(model) <span class="sc">+</span> hlp<span class="sc">$</span>df.prior <span class="sc">*</span> hlp<span class="sc">$</span>var.prior)<span class="sc">/</span>(<span class="fu">getDF</span>(model)<span class="sc">+</span>hlp<span class="sc">$</span>df.prior)</span>
<span id="cb78-2"><a href="#cb78-2" tabindex="-1"></a><span class="fu">sqrt</span>(varPost)</span></code></pre></div>
<pre><code>## [1] 0.6607153</code></pre>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" tabindex="-1"></a><span class="fu">getSigmaPosterior</span>(model)</span></code></pre></div>
<pre><code>## [1] 0.6607153</code></pre>
<p>Hence, standard deviations are shrunken towards prior standard
deviation! Large standard deviations become smaller and smaller standard
deviations become larger!</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" tabindex="-1"></a>p1 <span class="sc">+</span></span>
<span id="cb82-2"><a href="#cb82-2" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> hlp<span class="sc">$</span>var.prior<span class="sc">^</span>.<span class="dv">5</span>)</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-32-1.png" width="672" /></p>
</div>
</div>
<div id="illustration-via-simulation" class="section level3"
number="4.5.2">
<h3><span class="header-section-number">4.5.2</span> Illustration via
Simulation</h3>
<p>Suppose that the standard deviations for all proteins are the same
and are equal to 1. We simulate proteins with the same mean as the
fitted mean in the experiment but with standard deviation of 1.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" tabindex="-1"></a>nCoefs <span class="ot">&lt;-</span> <span class="fu">getCoef</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>msqrobModels[[<span class="dv">2</span>]]) <span class="sc">%&gt;%</span> length</span>
<span id="cb83-2"><a href="#cb83-2" tabindex="-1"></a>coefs <span class="ot">&lt;-</span></span>
<span id="cb83-3"><a href="#cb83-3" tabindex="-1"></a><span class="fu">sapply</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>msqrobModels,</span>
<span id="cb83-4"><a href="#cb83-4" tabindex="-1"></a>    <span class="cf">function</span>(x) <span class="fu">getCoef</span>(x)[<span class="dv">1</span><span class="sc">:</span>nCoefs]</span>
<span id="cb83-5"><a href="#cb83-5" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb83-6"><a href="#cb83-6" tabindex="-1"></a>  t <span class="sc">%&gt;%</span></span>
<span id="cb83-7"><a href="#cb83-7" tabindex="-1"></a>  na.exclude</span>
<span id="cb83-8"><a href="#cb83-8" tabindex="-1"></a></span>
<span id="cb83-9"><a href="#cb83-9" tabindex="-1"></a></span>
<span id="cb83-10"><a href="#cb83-10" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">nrow</span>(coefs)</span>
<span id="cb83-11"><a href="#cb83-11" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(pe[[<span class="dv">1</span>]])</span>
<span id="cb83-12"><a href="#cb83-12" tabindex="-1"></a>f0_equalVar <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span>p,</span>
<span id="cb83-13"><a href="#cb83-13" tabindex="-1"></a>  <span class="at">FUN=</span><span class="cf">function</span>(i, n, betas, sd, design) {</span>
<span id="cb83-14"><a href="#cb83-14" tabindex="-1"></a>  <span class="fu">rnorm</span>(n, <span class="at">mean =</span> design <span class="sc">%*%</span> betas[i,], <span class="at">sd =</span> sd)},</span>
<span id="cb83-15"><a href="#cb83-15" tabindex="-1"></a>  <span class="at">n =</span> n,</span>
<span id="cb83-16"><a href="#cb83-16" tabindex="-1"></a>  <span class="at">betas =</span> coefs,</span>
<span id="cb83-17"><a href="#cb83-17" tabindex="-1"></a>  <span class="at">sd =</span> <span class="dv">1</span>,</span>
<span id="cb83-18"><a href="#cb83-18" tabindex="-1"></a>  <span class="at">design =</span> X</span>
<span id="cb83-19"><a href="#cb83-19" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb83-20"><a href="#cb83-20" tabindex="-1"></a>  t</span>
<span id="cb83-21"><a href="#cb83-21" tabindex="-1"></a><span class="fu">colnames</span>(f0_equalVar) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(pe[[<span class="dv">1</span>]])</span>
<span id="cb83-22"><a href="#cb83-22" tabindex="-1"></a>sims <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(f0_equalVar <span class="sc">%&gt;%</span> as.data.frame, <span class="at">ecol =</span> <span class="dv">1</span><span class="sc">:</span>n, <span class="at">name =</span> <span class="st">&quot;sim_equalVar&quot;</span>)</span>
<span id="cb83-23"><a href="#cb83-23" tabindex="-1"></a><span class="fu">colData</span>(sims) <span class="ot">&lt;-</span> <span class="fu">colData</span>(pe)</span>
<span id="cb83-24"><a href="#cb83-24" tabindex="-1"></a>sims <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(<span class="at">object =</span> sims, <span class="at">i =</span> <span class="st">&quot;sim_equalVar&quot;</span>, <span class="at">formula =</span> <span class="sc">~</span> location<span class="sc">*</span>tissue <span class="sc">+</span> patient)</span>
<span id="cb83-25"><a href="#cb83-25" tabindex="-1"></a></span>
<span id="cb83-26"><a href="#cb83-26" tabindex="-1"></a>sd0 <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb83-27"><a href="#cb83-27" tabindex="-1"></a>  <span class="fu">rowData</span>(sims[[<span class="st">&quot;sim_equalVar&quot;</span>]])<span class="sc">$</span>msqrobModels,</span>
<span id="cb83-28"><a href="#cb83-28" tabindex="-1"></a>  getSigma) <span class="sc">%&gt;%</span></span>
<span id="cb83-29"><a href="#cb83-29" tabindex="-1"></a>  na.exclude</span>
<span id="cb83-30"><a href="#cb83-30" tabindex="-1"></a>sdPost0 <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb83-31"><a href="#cb83-31" tabindex="-1"></a>  <span class="fu">rowData</span>(sims[[<span class="st">&quot;sim_equalVar&quot;</span>]])<span class="sc">$</span>msqrobModels,</span>
<span id="cb83-32"><a href="#cb83-32" tabindex="-1"></a>  getSigmaPosterior) <span class="sc">%&gt;%</span></span>
<span id="cb83-33"><a href="#cb83-33" tabindex="-1"></a>  na.exclude</span>
<span id="cb83-34"><a href="#cb83-34" tabindex="-1"></a></span>
<span id="cb83-35"><a href="#cb83-35" tabindex="-1"></a><span class="fu">qplot</span>(sd0,sdPost0) <span class="sc">+</span></span>
<span id="cb83-36"><a href="#cb83-36" tabindex="-1"></a>  <span class="fu">geom_abline</span>() <span class="sc">+</span></span>
<span id="cb83-37"><a href="#cb83-37" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="fu">range</span>(sd0))</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-33-1.png" width="672" /></p>
<ul>
<li>We observe a large variability in the individual protein level
standard deviation estimates.</li>
<li>We simulated proteins with standard deviation of 1, but the protein
estimates vary from 0.17, … , 1.97.</li>
<li>Large uncertainty on the estimation of variances in small
samples</li>
<li>The empirical Bayes method, however, recognises that all proteins
are simulated with the same variance.</li>
<li>Hence, it can borrow tremendous strength across proteins to
stabilize the variance estimation</li>
<li>Here, it shrinks all protein variance to the prior variance, which
is indeed very close to 1, the value we have adopted in the
simulation.</li>
</ul>
<p>Note, that the prior degrees of freedom are also set to infinity:</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" tabindex="-1"></a><span class="fu">getDF</span>(<span class="fu">rowData</span>(sims[[<span class="st">&quot;sim_equalVar&quot;</span>]])<span class="sc">$</span>msqrobModels[[<span class="dv">1</span>]])</span></code></pre></div>
<pre><code>## [1] 5.524613</code></pre>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" tabindex="-1"></a><span class="fu">getDfPosterior</span>(<span class="fu">rowData</span>(sims[[<span class="st">&quot;sim_equalVar&quot;</span>]])<span class="sc">$</span>msqrobModels[[<span class="dv">1</span>]])</span></code></pre></div>
<pre><code>## [1] Inf</code></pre>
<p>which imposes shrinkage to the prior standard deviation!</p>
<p>The empirical Bayes method can thus indeed recognize the common
variance that is shared across all simulated proteins!</p>
</div>
</div>
</div>
<div id="p-values" class="section level1" number="5">
<h1><span class="header-section-number">5</span> P-values</h1>
<div id="simulation-under-h_0." class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Simulation under
H_0.</h2>
<ul>
<li>Mean log2 protein intensity for atrium equals mean log2 protein
intensity for ventriculum in the left heart region.</li>
<li>sd equals the sd for the protein.</li>
</ul>
<ol style="list-style-type: decimal">
<li>Extract <span class="math inline">\(\hat \sigma\)</span> and <span
class="math inline">\(\beta\)</span>’s</li>
</ol>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" tabindex="-1"></a>sd <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb88-2"><a href="#cb88-2" tabindex="-1"></a>  <span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>msqrobModels,</span>
<span id="cb88-3"><a href="#cb88-3" tabindex="-1"></a>  getSigma) <span class="sc">%&gt;%</span></span>
<span id="cb88-4"><a href="#cb88-4" tabindex="-1"></a>  na.exclude</span>
<span id="cb88-5"><a href="#cb88-5" tabindex="-1"></a></span>
<span id="cb88-6"><a href="#cb88-6" tabindex="-1"></a>coefs <span class="ot">&lt;-</span></span>
<span id="cb88-7"><a href="#cb88-7" tabindex="-1"></a><span class="fu">sapply</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>msqrobModels,</span>
<span id="cb88-8"><a href="#cb88-8" tabindex="-1"></a>    <span class="cf">function</span>(x) <span class="fu">getCoef</span>(x)[<span class="dv">1</span><span class="sc">:</span>nCoefs]</span>
<span id="cb88-9"><a href="#cb88-9" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb88-10"><a href="#cb88-10" tabindex="-1"></a>  t <span class="sc">%&gt;%</span></span>
<span id="cb88-11"><a href="#cb88-11" tabindex="-1"></a>  na.exclude</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Set <span class="math inline">\(\beta_\text{tissue}\)</span> equal
to 0. No FC between atrium and ventriculum left.</li>
</ol>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" tabindex="-1"></a>coefs0 <span class="ot">&lt;-</span> coefs</span>
<span id="cb89-2"><a href="#cb89-2" tabindex="-1"></a>coefs0[,<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Simulate protein expressions for each protein from a Normal
distribution under <span class="math inline">\(H_0\)</span> for left
heart region (no FC between atrium and ventriculum left) and sd the sd
for the protein.</li>
</ol>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">104</span>)</span>
<span id="cb90-2"><a href="#cb90-2" tabindex="-1"></a>f0 <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span>p,</span>
<span id="cb90-3"><a href="#cb90-3" tabindex="-1"></a>  <span class="cf">function</span>(i, betas, sd, design)</span>
<span id="cb90-4"><a href="#cb90-4" tabindex="-1"></a>  <span class="fu">rnorm</span>(n, <span class="at">mean =</span> design <span class="sc">%*%</span> betas[i,], <span class="at">sd =</span> sd[i]),</span>
<span id="cb90-5"><a href="#cb90-5" tabindex="-1"></a>  <span class="at">betas =</span> coefs0,</span>
<span id="cb90-6"><a href="#cb90-6" tabindex="-1"></a>  <span class="at">sd =</span> sd,</span>
<span id="cb90-7"><a href="#cb90-7" tabindex="-1"></a>  <span class="at">design =</span> X</span>
<span id="cb90-8"><a href="#cb90-8" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb90-9"><a href="#cb90-9" tabindex="-1"></a>  t</span>
<span id="cb90-10"><a href="#cb90-10" tabindex="-1"></a><span class="fu">colnames</span>(f0) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(pe[[<span class="dv">1</span>]])</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Setup QFeatures object and perform MSqRob analysis</li>
</ol>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" tabindex="-1"></a>sims <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(f0 <span class="sc">%&gt;%</span> as.data.frame, <span class="at">ecol =</span> <span class="dv">1</span><span class="sc">:</span>n, <span class="at">name =</span> <span class="st">&quot;sim0&quot;</span>)</span>
<span id="cb91-2"><a href="#cb91-2" tabindex="-1"></a><span class="fu">colData</span>(sims) <span class="ot">&lt;-</span> <span class="fu">colData</span>(pe)</span>
<span id="cb91-3"><a href="#cb91-3" tabindex="-1"></a>sims <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(<span class="at">object =</span> sims, <span class="at">i =</span> <span class="st">&quot;sim0&quot;</span>, <span class="at">formula =</span> <span class="sc">~</span> location<span class="sc">*</span>tissue <span class="sc">+</span> patient)</span>
<span id="cb91-4"><a href="#cb91-4" tabindex="-1"></a>sims <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(<span class="at">object =</span> sims, <span class="at">i =</span> <span class="st">&quot;sim0&quot;</span>, <span class="at">contrast =</span> L)</span></code></pre></div>
<div id="evaluate-pvalues-under-h_0" class="section level3"
number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Evaluate pvalues
under H_0</h3>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" tabindex="-1"></a>volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">rowData</span>(sims[[<span class="st">&quot;sim0&quot;</span>]])<span class="sc">$</span>tissueV,</span>
<span id="cb92-2"><a href="#cb92-2" tabindex="-1"></a>                 <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> pval <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb92-3"><a href="#cb92-3" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb92-4"><a href="#cb92-4" tabindex="-1"></a> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme_minimal</span>()</span>
<span id="cb92-5"><a href="#cb92-5" tabindex="-1"></a>volcano</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-39-1.png" width="672" /></p>
<p>Number of false positives without multiple testing?</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" tabindex="-1"></a><span class="fu">rowData</span>(sims[[<span class="st">&quot;sim0&quot;</span>]])<span class="sc">$</span>tissueV <span class="sc">%&gt;%</span></span>
<span id="cb93-2"><a href="#cb93-2" tabindex="-1"></a>  <span class="fu">filter</span>(pval <span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb93-3"><a href="#cb93-3" tabindex="-1"></a>  nrow</span></code></pre></div>
<pre><code>## [1] 115</code></pre>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" tabindex="-1"></a><span class="fu">mean</span>(<span class="fu">rowData</span>(sims[[<span class="st">&quot;sim0&quot;</span>]])<span class="sc">$</span>tissueV<span class="sc">$</span>pval <span class="sc">&lt;</span> <span class="fl">0.05</span>)    </span></code></pre></div>
<pre><code>## [1] 0.05665025</code></pre>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">rowData</span>(sims[[<span class="st">&quot;sim0&quot;</span>]])<span class="sc">$</span>tissueV<span class="sc">$</span>pval,<span class="at">main =</span> <span class="st">&quot;simulation H0&quot;</span>)</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-40-1.png" width="672" /></p>
<ul>
<li>The p-values are uniform!</li>
<li>All p-values under the null are equally likely.</li>
<li>Statistical hypthesis testing leads to a uniform test strategy under
<span class="math inline">\(H_0\)</span></li>
<li>If use p-value cutoff at 0.05 we expect to return 5% of the non-DE
proteins as differentially expressed: many false positives can be
expected!</li>
</ul>
</div>
</div>
<div id="pvalue-distribution-in-real-experiment" class="section level2"
number="5.2">
<h2><span class="header-section-number">5.2</span> Pvalue distribution
in real experiment</h2>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>tissueV<span class="sc">$</span>pval, <span class="at">main =</span> <span class="st">&quot;realData&quot;</span>)</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-41-1.png" width="672" /></p>
<ul>
<li>A mixture of null proteins (non-DE): uniform, and, DE proteins:
enrichment of p-values at low p-values</li>
</ul>
</div>
</div>
<div id="correction-for-multiple-testing" class="section level1"
number="6">
<h1><span class="header-section-number">6</span> Correction for multiple
testing</h1>
<ul>
<li>We can adjust the p-values for multiple testing.</li>
</ul>
<div id="family-wise-error-rate-correction" class="section level2"
number="6.1">
<h2><span class="header-section-number">6.1</span> Family wise error
rate correction:</h2>
<ul>
<li><p>A list of returned proteins is considered to be in error as soon
as it contains at most one false positive protein.</p></li>
<li><p><span class="math inline">\(\text{FWER} = P(FP \leq
1)\)</span></p></li>
<li><p>FWER: probability of making at least one false positive decision
or probability to declare at least one protein differentially abundant
which is truly non differentially abundant</p></li>
</ul>
<div id="bonferroni-method" class="section level3" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> Bonferroni
method</h3>
<ul>
<li>Simple method</li>
<li><span class="math inline">\(m\)</span> tests are performed at the
level <span class="math inline">\(\alpha/m\)</span></li>
<li>FWER<span class="math inline">\(\leq\sum\limits_{p=1}^{m}P(reject
H_{0p}\vert H_{0p}\text{ is true})=m \alpha/m=\alpha\)</span></li>
<li>Provides strong control</li>
<li>Bonferroni is very conservative</li>
<li>Works for dependent tests</li>
<li>Adjusted p-value: <span class="math inline">\(\tilde{p}_p=\min(m\
p_p,1)\)</span></li>
</ul>
<hr />
</div>
<div id="bonferroni-in-practise" class="section level3" number="6.1.2">
<h3><span class="header-section-number">6.1.2</span> Bonferroni in
practise</h3>
<p>Via R functions</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" tabindex="-1"></a>padj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(</span>
<span id="cb99-2"><a href="#cb99-2" tabindex="-1"></a>  <span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>tissueV<span class="sc">$</span>pval,</span>
<span id="cb99-3"><a href="#cb99-3" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;bonferroni&quot;</span>)</span></code></pre></div>
<p>Own Implementation: adjust and make sure that p-value is smaller than
1.</p>
<div class="sourceCode" id="cb100"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>tissueV<span class="sc">$</span>pval))</span>
<span id="cb100-2"><a href="#cb100-2" tabindex="-1"></a>padjSelf <span class="ot">&lt;-</span>  <span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>tissueV<span class="sc">$</span>pval <span class="sc">*</span> m</span>
<span id="cb100-3"><a href="#cb100-3" tabindex="-1"></a>padjSelf[padjSelf <span class="sc">&gt;</span> <span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb100-4"><a href="#cb100-4" tabindex="-1"></a></span>
<span id="cb100-5"><a href="#cb100-5" tabindex="-1"></a><span class="fu">range</span>(padj <span class="sc">-</span> padjSelf, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## [1] 0 0</code></pre>
</div>
<div id="illustration-in-simulation-under-h_0-and-heart-case-study"
class="section level3" number="6.1.3">
<h3><span class="header-section-number">6.1.3</span> Illustration in
simulation under <span class="math inline">\(H_0\)</span> and heart case
study</h3>
<div class="sourceCode" id="cb102"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" tabindex="-1"></a>volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">rowData</span>(sims[[<span class="st">&quot;sim0&quot;</span>]])<span class="sc">$</span>tissueV,</span>
<span id="cb102-2"><a href="#cb102-2" tabindex="-1"></a>                 <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> <span class="fu">p.adjust</span>(pval,<span class="st">&quot;bonferroni&quot;</span>) <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb102-3"><a href="#cb102-3" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb102-4"><a href="#cb102-4" tabindex="-1"></a> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb102-5"><a href="#cb102-5" tabindex="-1"></a> <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb102-6"><a href="#cb102-6" tabindex="-1"></a> <span class="fu">ggtitle</span>(<span class="st">&quot;simulated heart data under H0&quot;</span>)</span>
<span id="cb102-7"><a href="#cb102-7" tabindex="-1"></a>volcano</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-44-1.png" width="672" /></p>
<ul>
<li>No false positives are returned for simulation under H_0. List is
correct according to FWER.</li>
</ul>
<div class="sourceCode" id="cb103"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" tabindex="-1"></a>volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>tissueV,</span>
<span id="cb103-2"><a href="#cb103-2" tabindex="-1"></a>                 <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> <span class="fu">p.adjust</span>(pval,<span class="st">&quot;bonferroni&quot;</span>) <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb103-3"><a href="#cb103-3" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb103-4"><a href="#cb103-4" tabindex="-1"></a> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb103-5"><a href="#cb103-5" tabindex="-1"></a> <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb103-6"><a href="#cb103-6" tabindex="-1"></a> <span class="fu">ggtitle</span>(<span class="st">&quot;real heart data&quot;</span>)</span>
<span id="cb103-7"><a href="#cb103-7" tabindex="-1"></a>volcano</span></code></pre></div>
<pre><code>## Warning: Removed 1263 rows containing missing values (geom_point).</code></pre>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-45-1.png" width="672" /></p>
<ul>
<li>Very few proteins are returned for real data. Very
conservative!</li>
</ul>
</div>
</div>
<div id="false-discovery-rate" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> False discovery
rate</h2>
<ul>
<li>Adjusted P-values with the Benjamini Hochberg correction correspond
to the estimated FDR of the set that is returned when the significance
level is set at this threshold. <span
class="math display">\[\begin{eqnarray}
FDR(p_0) &amp;=&amp; \text{E}\left[\frac{FP}{(FP + TP)}\right]\\
&amp;\approx&amp;\frac{p_0 \times m}{\#p_p \leq p_0}\\
\end{eqnarray}\]</span></li>
</ul>
<p>So adjusted p-value for protein j equals <span
class="math display">\[\tilde p_j = \frac{p_{0,j} \times m}{\#p_p \leq
p_{0,j}}\]</span></p>
<p>However, the FDR always has to be between 0 and 1 so:</p>
<p><span class="math display">\[\tilde p_j = \min\left[\frac{p_{0,j}
\times m}{\#p_p \leq p_{0,j}},1\right]\]</span></p>
<p>and the adjusted p-values should remain in the same order as the
original p-values.</p>
<p><span class="math display">\[\tilde p_j =  \min\limits_{\forall k:
p_k &gt; p_j} \min\left[\frac{p_{0,k} \times m}{\#p_p \leq
p_{0,k}},1\right]\]</span></p>
<ol style="list-style-type: decimal">
<li>Order pvalues</li>
</ol>
<div class="sourceCode" id="cb105"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" tabindex="-1"></a>pvals <span class="ot">&lt;-</span> <span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>tissueV<span class="sc">$</span>pval</span>
<span id="cb105-2"><a href="#cb105-2" tabindex="-1"></a>naInd <span class="ot">&lt;-</span> <span class="fu">is.na</span>(pvals)</span>
<span id="cb105-3"><a href="#cb105-3" tabindex="-1"></a>pHlp <span class="ot">&lt;-</span> pvals[<span class="sc">!</span>naInd]</span>
<span id="cb105-4"><a href="#cb105-4" tabindex="-1"></a>ord <span class="ot">&lt;-</span> pHlp <span class="sc">%&gt;%</span> order</span>
<span id="cb105-5"><a href="#cb105-5" tabindex="-1"></a>pHlp <span class="ot">&lt;-</span> pHlp[ord]</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Adjust ordered p-values</li>
</ol>
<div class="sourceCode" id="cb106"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" tabindex="-1"></a>pHlp <span class="ot">&lt;-</span> pHlp<span class="sc">*</span><span class="fu">length</span>(pHlp)<span class="sc">/</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pHlp))</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Ensure adjust p-values are smaller are equal than 1</li>
</ol>
<div class="sourceCode" id="cb107"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" tabindex="-1"></a>pHlp[pHlp<span class="sc">&gt;</span><span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="dv">1</span></span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Monotonicity constraint</li>
</ol>
<div class="sourceCode" id="cb108"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" tabindex="-1"></a>pmin <span class="ot">&lt;-</span> pHlp[<span class="fu">length</span>(pHlp)]</span>
<span id="cb108-2"><a href="#cb108-2" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> (<span class="fu">length</span>(pHlp)<span class="sc">-</span><span class="dv">1</span>)<span class="sc">:</span><span class="dv">1</span>)</span>
<span id="cb108-3"><a href="#cb108-3" tabindex="-1"></a>{</span>
<span id="cb108-4"><a href="#cb108-4" tabindex="-1"></a>  <span class="cf">if</span> (pHlp[j] <span class="sc">&lt;</span> pmin)</span>
<span id="cb108-5"><a href="#cb108-5" tabindex="-1"></a>    pmin <span class="ot">&lt;-</span> pHlp[j] <span class="cf">else</span></span>
<span id="cb108-6"><a href="#cb108-6" tabindex="-1"></a>    pHlp[j] <span class="ot">&lt;-</span> pmin   </span>
<span id="cb108-7"><a href="#cb108-7" tabindex="-1"></a>}</span></code></pre></div>
<ol start="5" style="list-style-type: decimal">
<li>Put p-values back in original order</li>
</ol>
<div class="sourceCode" id="cb109"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" tabindex="-1"></a>pHlp[ord] <span class="ot">&lt;-</span> pHlp</span>
<span id="cb109-2"><a href="#cb109-2" tabindex="-1"></a>pAdj <span class="ot">&lt;-</span> pvals</span>
<span id="cb109-3"><a href="#cb109-3" tabindex="-1"></a>pAdj[<span class="sc">!</span>naInd] <span class="ot">&lt;-</span> pHlp</span>
<span id="cb109-4"><a href="#cb109-4" tabindex="-1"></a></span>
<span id="cb109-5"><a href="#cb109-5" tabindex="-1"></a><span class="fu">head</span>(pAdj)</span></code></pre></div>
<pre><code>## [1]        NA 0.8351039        NA        NA 0.9062353        NA</code></pre>
<div class="sourceCode" id="cb111"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>tissueV)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["logFC"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["se"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["df"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["t"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["pval"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["adjPval"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"NA","2":"NA","3":"NA","4":"NA","5":"NA","6":"NA","_rn_":"A0M8Q6"},{"1":"0.20946392","2":"0.5601628","3":"7.914824","4":"0.3739340","5":"0.7182716","6":"0.8351039","_rn_":"A0PJW6"},{"1":"NA","2":"NA","3":"NA","4":"NA","5":"NA","6":"NA","_rn_":"A0PJZ3"},{"1":"NA","2":"NA","3":"NA","4":"NA","5":"NA","6":"NA","_rn_":"A0PK00"},{"1":"0.07123672","2":"0.3206733","3":"9.293515","4":"0.2221474","5":"0.8290044","6":"0.9062353","_rn_":"A1A4S6"},{"1":"NA","2":"NA","3":"NA","4":"NA","5":"NA","6":"NA","_rn_":"A1A5D9"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div class="sourceCode" id="cb112"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" tabindex="-1"></a><span class="fu">range</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>tissueV<span class="sc">$</span>adjPval <span class="sc">-</span> pAdj,<span class="at">na.rm=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## [1] -2.220446e-16  2.220446e-16</code></pre>
<div id="illustration-in-simulation-under-h_0-and-heart-case-study-1"
class="section level3" number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Illustration in
simulation under <span class="math inline">\(H_0\)</span> and heart case
study</h3>
<div class="sourceCode" id="cb114"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" tabindex="-1"></a>volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">rowData</span>(sims[[<span class="st">&quot;sim0&quot;</span>]])<span class="sc">$</span>tissueV,</span>
<span id="cb114-2"><a href="#cb114-2" tabindex="-1"></a>                 <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb114-3"><a href="#cb114-3" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb114-4"><a href="#cb114-4" tabindex="-1"></a> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb114-5"><a href="#cb114-5" tabindex="-1"></a> <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb114-6"><a href="#cb114-6" tabindex="-1"></a> <span class="fu">ggtitle</span>(<span class="st">&quot;simulated heart data under H0&quot;</span>)</span>
<span id="cb114-7"><a href="#cb114-7" tabindex="-1"></a>volcano</span></code></pre></div>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-51-1.png" width="672" /></p>
<ul>
<li>No false positives are returned for simulation under H_0. List is
correct according to FWER.</li>
<li>It can be shown that the FDR-method controls the FWER when <span
class="math inline">\(H_0\)</span> is true for all features.</li>
</ul>
<div class="sourceCode" id="cb115"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" tabindex="-1"></a>volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>tissueV,</span>
<span id="cb115-2"><a href="#cb115-2" tabindex="-1"></a>                 <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb115-3"><a href="#cb115-3" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb115-4"><a href="#cb115-4" tabindex="-1"></a> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb115-5"><a href="#cb115-5" tabindex="-1"></a> <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb115-6"><a href="#cb115-6" tabindex="-1"></a> <span class="fu">ggtitle</span>(<span class="st">&quot;real heart data&quot;</span>)</span>
<span id="cb115-7"><a href="#cb115-7" tabindex="-1"></a>volcano</span></code></pre></div>
<pre><code>## Warning: Removed 1263 rows containing missing values (geom_point).</code></pre>
<p><img src="technicalDetailsProteomics_files/figure-html/unnamed-chunk-52-1.png" width="672" /></p>
<p>The FDR method allows us to return much longer DA protein lists at
the expense of a few false positives. The FDR controls the fraction of
false positives in the list that you return on average on the
significance level that is adopted. So if you use <span
class="math inline">\(\alpha=0.05\)</span> we expect on average 5% of
false positives in the list that we return.</p>
</div>
</div>
</div>

<div id="rmd-source-code">---
title: "Technical details on linear regression for proteomics when starting from summarized protein expression values"
author: "Lieven Clement"
date: "statOmics, Ghent University (https://statomics.github.io)"
output:
    html_document:
      code_download: true
      theme: cosmo
      toc: true
      toc_float: true
      highlight: tango
      number_sections: true
---

# Preamble

Read stored results for heart data

```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(limma)
library(QFeatures)
library(msqrob2)
pe <- readRDS(
  url(
    "https://raw.githubusercontent.com/statOmics/SGA2020/gh-pages/assets/peHeart.rds",
    "rb")
  )
```

# Linear regression

- Consider a vector of predictors $\mathbf{x}_i=(x_1,\ldots,x_{p-1})$ and
- a real-valued response $Y_i$
- with $i = 1, \ldots, n$
- then the linear regression model can be written as
\[
Y_i=f(\mathbf{x}) +\epsilon=\beta_0+\sum\limits_{j=1}^{p-1} x_{ij}\beta + \epsilon_i
\]
with i.i.d. $\epsilon_i\sim N(0,\sigma^2)$

---

- $n$ observations $(\mathbf{x}_1,y_1) \ldots (\mathbf{x}_n,y_n)$
- Regression in matrix notation
\[\mathbf{Y}=\mathbf{X\beta} + \boldsymbol{\epsilon}\]
with $\mathbf{Y}=\left[\begin{array}{c}y_1\\ \vdots\\y_n\end{array}\right]$,
$\mathbf{X}=\left[\begin{array}{cccc} 1&x_{11}&\ldots&x_{1p-1}\\
\vdots&\vdots&&\vdots\\
1&x_{n1}&\ldots&x_{np-1}
\end{array}\right]$,
$\boldsymbol{\beta}=\left[\begin{array}{c}\beta_0\\ \vdots\\ \beta_{p-1}\end{array}\right]$ and
$\boldsymbol{\epsilon}=\left[\begin{array}{c} \epsilon_1 \\ \vdots \\ \epsilon_n\end{array}\right]$

---

## Least Squares (LS)

- Minimize the residual sum of squares
\begin{eqnarray*}
RSS(\boldsymbol{\beta})&=&\sum\limits_{i=1}^n e^2_i\\
&=&\sum\limits_{i=1}^n \left(y_i-\beta_0-\sum\limits_{j=1}^p x_{ij}\beta_j\right)^2
\end{eqnarray*}
- or in matrix notation
\begin{eqnarray*}
RSS(\boldsymbol{\beta})&=&(\mathbf{Y}-\mathbf{X\beta})^T(\mathbf{Y}-\mathbf{X\beta})\\
&=&\Vert \mathbf{Y}-\mathbf{X\beta}\Vert^2
\end{eqnarray*}
with the $L_2$-norm of a $p$-dim. vector $v$ $\Vert \mathbf{v} \Vert=\sqrt{v_1^2+\ldots+v_p^2}$

- $\rightarrow$ $\hat{\boldsymbol{\beta}}=\text{argmin}_\beta \Vert \mathbf{Y}-\mathbf{X\beta}\Vert^2$}


---

### Minimize RSS
\[
\begin{array}{ccc}
\frac{\partial RSS}{\partial \boldsymbol{\beta}}&=&\mathbf{0}\\\\
\frac{(\mathbf{Y}-\mathbf{X\beta})^T(\mathbf{Y}-\mathbf{X\beta})}{\partial \boldsymbol{\beta}}&=&\mathbf{0}\\\\
-2\mathbf{X}^T(\mathbf{Y}-\mathbf{X\beta})&=&\mathbf{0}\\\\
\mathbf{X}^T\mathbf{X\beta}&=&\mathbf{X}^T\mathbf{Y}\\\\
\hat{\boldsymbol{\beta}}&=&(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{Y}
\end{array}
\]

---

#### Heart example

```{r}
y <- assay(pe[["proteinRobust"]])[2,]
fit <- lm(y ~ location*tissue + patient, data = colData(pe), x = TRUE)
head(fit$x,4)
```

The model matrix can also be obtained without fitting the model:

```{r}
X <- model.matrix(~ location * tissue + patient, colData(pe))
head(X,4)
```

Least squares:
```{r}
betas <- solve(t(X)%*%X) %*% t(X) %*% y
cbind(fit$coef, betas)
```

---

### Variance Estimator?
\[
\begin{array}{ccl}
\hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}
&=&\text{var}\left[(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{Y}\right]\\\\
&=&(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\text{var}\left[\mathbf{Y}\right]\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\\\\
&=&(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T(\mathbf{I}\sigma^2)\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}
\\\\
&=&(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{I}\quad\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\sigma^2\\\\
%\hat{\boldmath{\Sigma}}_{\hat{\boldsymbol{\beta}}}&=&(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\var\left[\mathbf{Y}\right](\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}\\
&=&(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\sigma^2\\\\
&=&(\mathbf{X}^T\mathbf{X})^{-1}\sigma^2
\end{array}
\]

---

#### Heart example


```{r}
summary(fit)$cov.unscaled * sigma(fit)^2
```

```{r}
n <- nrow(X)
p <- ncol(X)
mse <- sum((y-X%*%betas)^2)/(n-p)
SigmaBeta <- solve(t(X)%*%X) * mse   
SigmaBeta
range(SigmaBeta - summary(fit)$cov.unscaled * sigma(fit)^2)
```

```{r}
data.frame(summary(fit)$coef[,1:2], betas = betas, seBetas = diag(SigmaBeta)^.5)
```

## Contrasts

When we assess a contrast we assess a linear combination of model parameters:

\[ H_0: \mathbf{L^T\beta} = 0 \text{ vs } H_1: \mathbf{L^T\beta} \neq 0 \]

Estimator of Contrast?

\[\mathbf{L}^T\hat{\boldsymbol{\beta}}\]


Variance?

\[
\boldsymbol{\Sigma}_{\mathbf{L}\hat{\boldsymbol{\beta}}}=\mathbf{L}^T\boldsymbol{\Sigma}_{\hat{\boldsymbol{\beta}}}\mathbf{L}
\]

---

### Heart example

```{r}
L <- makeContrast(
  c(
    "tissueV = 0",
    "tissueV + locationR:tissueV = 0",
    "tissueV + 0.5*locationR:tissueV = 0","locationR:tissueV = 0"),
  parameterNames =
    rowData(pe[["proteinRobust"]])$msqrobModels[[2]] %>%
    getCoef %>%
    names
  )
L
```

```{r}
contrasts <- t(L) %*% betas
SigmaContrasts <- t(L) %*% SigmaBeta %*% L
seContrasts <- SigmaContrasts %>%
  diag %>%
  sqrt
```

Comparison with lm and glht results

```{r}
library(multcomp)
fitGlht <- glht(fit, linfct = t(L))
summary(fitGlht, test = adjusted("none"))
data.frame(contrasts, seContrasts)
```

- Note, that the power for assessing $\log_2$ FC between ventriculum and atrium  left and right is the same. Indeed, the standard errors are equal for both effects.


- Note, that the power for assessing $\log_2$ FC between ventriculum and atrium over both heart regions is higher than when assessing the effect left or right.

  - Indeed, the standard error is a factor $\sqrt{2}$ smaller for the former effect
  - We intuitively can explain this because we can use all samples (double the number of samples) to assess the average effect.
  - Hence the variance is a factor two smaller, and the se with a factor $\sqrt{2}$

- Note, that we have the lowest power to pick up an interaction effect. Indeed, the se is a factor $\sqrt{2}$ larger than for the ventriculum - atrium effect left or right and a factor 2 larger than for the average effect between ventriculum and atrium.

```{r}
seContrasts / seContrasts[1]
sqrt(2)
1/sqrt(2)
```

### t-tests

 - When the assumptions of the linear model hold
\[
\hat{\boldsymbol{\beta}} \sim MVN\left[\boldsymbol{\beta},\left(\mathbf{X}^T\mathbf{X}\right)^{-1}\sigma^2\right]
\]
- Hence,
\[
\mathbf{L}^T\hat{\boldsymbol{\beta}} \sim MVN\left[\mathbf{L}^T\boldsymbol{\beta},\mathbf{L}^T\left[\left(\mathbf{X}^T\mathbf{X}\right)^{-1}\sigma^2\right]\mathbf{L}\right]
\]
- We estimate $\sigma^2$ by MSE
$$\hat{\sigma}^2=\frac{\mathbf{e}^T\mathbf{e}}{n-p} \rightarrow \hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}=\left(\mathbf{X}^T\mathbf{X}\right)^{-1}\hat\sigma^2$$

- When we test one contrast at the time (e.g. the $k^\text{th}$ contrast) the statistic reduces to

$$T=\frac{\mathbf{L}_k^T\hat{\boldsymbol{\beta}}}{\sqrt{\left(\mathbf{L}^T_k\hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}\mathbf{L}_k\right)}} \underset{H_0}{\sim} t_{n-p}$$
follows a t distribution with n-p degrees of freedom under $H_0: \mathbf{L}^T_k\hat{\boldsymbol{\beta}}=0$

---

#### Heart example

```{r}
tContrasts <- contrasts/seContrasts
pContrasts <- pt(abs(tContrasts),
  df = n - p,
  lower.tail = FALSE) * 2
```

Comparison with lm and glht results

```{r}
summary(fitGlht, test = adjusted("none"))
data.frame(contrasts, seContrasts, tContrasts, pContrasts)
```


---

# Robust regression

- With msqrob2 we perform robust regression to estimate the model parameters of the regression model

- No normality assumption needed
- Robust fit minimises the maximal bias of the estimators
- CI and statistical tests are based on asymptotic theory
- If $\epsilon$ is normal, the M-estimators have a high efficiency!
- ordinary least squares (OLS): minimize loss function \[\sum\limits_{i=1}^n (y_i-\mathbf{x}_i^T\boldsymbol{\beta})^2\]

- M-estimation: minimize loss function
\[\sum\limits_{i=1}^n  \rho\left(y_i-\mathbf{x}_i^T\boldsymbol{\beta}\right)\]
with

  - $\rho$ is symmetric, i.e. $\rho(z)=\rho(-z)$
  - $\rho$ has a minimum at $\rho(0)=0$, is positive for all $z\neq 0$
  - $\rho(z)$ increases as $\vert z\vert$ increases

---

 The estimator $\hat{\mu}$ is also the solution to the equation
 \[
   \sum_{i=1}^n \Psi(y_i - \mathbf{x}_i\boldsymbol{\beta}) =0,
 \]
 where $\Psi$ is the derivative of $\rho$. For $\hat{\beta}$ possessing the robustness property, $\Psi$ should be bounded.

---

 Example: least squares

 - $\rho(z) = z^2$, and thus $\Psi(z)=2z$ (unbounded!). Not robust!

- $\hat{\boldsymbol{\beta}}$ is the solution of
 \[
   \sum_{i=1}^n 2 \mathbf{x}_i (y_i - \mathbf{x}_i^T\boldsymbol{\beta}) = 0 \text{ or } \hat{\boldsymbol{\beta}} = (\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}\mathbf{y}
 \]
 with $\mathbf{X}=[\mathbf{x}_1 \ldots \mathbf{x}_G]^T$

---

 When a location and a scale parameter, say $\sigma$, have to be estimated simultaneously, we write
 \[
   (\hat{\boldsymbol{\beta}},\hat{\sigma}) = \text{ArgMin}_{\boldsymbol{\beta},\sigma} \sum_{i=1}^n \rho\left(\frac{y_i - \mathbf{x}_i^T\boldsymbol{\beta}}{\sigma}\right)
   \text{ and } \sum_{i=1}^n \Psi\left(\frac{y_i - \mathbf{x}_i^T\boldsymbol{\beta}}{\sigma}\right) =0.
 \]

 Define $u_i = \frac{y_i - \mathbf{x}_i^T\boldsymbol{\beta}}{\sigma}$. The last estimation equation is equivalent to
 \[
   \sum_{i=1}^n w(u_i) u_i = 0 ,
 \]
 with weight function $w(u)=\Psi(u)/u$. This is the typical form that appears when solving the
 *iteratively reweighted least squares problem*,
 \[
   (\hat{\boldsymbol{\beta}},\hat{\sigma}) = \text{ArgMin}_{\mu,\sigma} \sum_{i=1}^n w(u_i^{(k-1)}) \left(u_i^{(k)}\right)^2 ,
 \]
 where $k$ represents the iteration number.

---

## Some Examples of Robust Functions

![](https://raw.githubusercontent.com/statOmics/SGA2020/gh-pages/assets/TableRobust.PNG)

PhD thesis Bolstad 2004

---

## The $\rho$ functions

![](https://raw.githubusercontent.com/statOmics/SGA2020/gh-pages/assets/RhoRobust.PNG)

PhD thesis Bolstad 2004

---

### Common $\Psi$-Functions
![](https://raw.githubusercontent.com/statOmics/SGA2020/gh-pages/assets/robustRegressionPsi.png)

PhD thesis Bolstad 2004

---

### Corresponding Weight Functions
![](https://raw.githubusercontent.com/statOmics/SGA2020/gh-pages/assets/robustRegressionWeights.png)

PhD thesis Bolstad 2004

---

```{r}
library("MASS")
rfit <- rlm(y ~ location * tissue + patient, colData(pe), maxit=1)
qplot(fit$coefficient[-1],
  rfit$coefficient[-1],
  xlab="fit",
  ylab="robust fit") +
  geom_abline() +
  xlim(range(c(fit$coefficient[-1],rfit$coefficient[-1]))) +
  ylim(range(c(fit$coefficient[-1],rfit$coefficient[-1])))
```

---

```{r}
rfit$w
plot(
  rfit$fitted,
  rfit$res,
  cex=rfit$w,
  pch=19,col=2,
  cex.lab=1.5,
  cex.axis=1.5,
  ylab="residuals",
  xlab="fit")
points(rfit$fitted, rfit$res , cex= 1.5)
```

---

```{r}
summary(fit)
summary(rfit)
rowData(pe[["proteinRobust"]])$msqrobModels[[2]] %>% getCoef
```

---

## Understanding implementation of robust regression

### Simulate 20 observations from a linear model with errors that follow a normal distribution

```{r}
set.seed <- 112358
nobs <- 20
sdy <- 1
xsim <- seq(0, 1, length.out = nobs)
ysim <- 10 + 5*xsim + rnorm(nobs, sd = sdy)
```

### add outlier at high leverage point

```{r}
ysim[nobs] <- 7
```

### fit linear model

```{r}
ols <- lm(ysim ~ xsim)
```

### fit robust linear model

```{r}
library(MASS)
mEst <- rlm(ysim ~ xsim)
```

#### plot results

```{r}
plot(xsim, ysim)
abline(ols, lwd = 2)
abline(mEst, col = "red", lwd = 2)
legend("topleft",
  legend = c("OLS", "M-estimation"),
  lwd = 2,
  col = 1:2)
round(mEst$w,3)
```

The plot clearly shows that the outlier has a high impact on the slope estimate.
This is because the outlier is at a high leverage point, i.e. far from the average covariate pattern.

### Implement it yourself
#### start from ols fit

```{r}
lmMod <- ols
```

#### Use robust variance estimator to calculate the z

```{r}
res <- lmMod$res
stdev <- mad(res)
stdev
median(abs(res-median(res)))*1.4826
z <- res/stdev
```

#### Calculate weights use psi.huber function

```{r}
w <- psi.huber(z)
plot(xsim, ysim)
plot(xsim, lmMod$res, cex = w, pch = 19, col = "red")
points(xsim,lmMod$res, cex = 1.5)
```

#### Perform a weighted regression use lm with weights=w

```{r}
lmMod <- lm(ysim~xsim, weights = w)
```

#### Plot results

```{r}
plot(xsim, ysim)
abline(ols, lwd = 2)
abline(mEst, col = "red", lwd = 2)
abline(lmMod, col = "blue", lwd = 2)
legend("topleft",
  legend = c("OLS","M-estimation","Our Impl"),
  lwd = 2,
  col = c("black", "red", "blue"))
```

#### Repeat this many times
```{r}
lmMod <- ols
for (k in 1:3)
{
######repeat this part several times until convergence
#use robust variance estimator to calculate the z
res <- lmMod$res
stdev <- mad(res)
median(abs(res-median(res)))*1.4826

z <- res/stdev

#calculate weights
#use psi.huber function
w <- psi.huber(z)

#perform a weighted regression use lm with weights=w
lmMod <- lm(ysim ~ xsim, weights = w)

#plot results
plot(xsim,ysim)
abline(ols, lwd = 2)
abline(mEst, col = "red", lwd = 2)
abline(lmMod, col = "blue", lwd = 2)
legend("topleft",
  legend = c("OLS","M-estimation","Our Impl"),
  lwd = 2,
  col = c("black", "red", "blue")
  )
####################################
}
```

# Empirical Bayes/Moderated $t$-test.

 A general class of moderated test statistics is given by

 \[\tilde{T}_p = \frac{\mathbf{L}_k \hat{ \boldsymbol{\beta_p}}}{\mathbf{L}_k^T(\mathbf{X}^T\mathbf{WX})^{-1}\mathbf{L}_k^T \tilde{s}_p^2}\]

 where $\tilde{s}_p$ is a moderated variance estimator.

Simple approach: set $\tilde{s}_p=s_p + s_0$: simply add a small positive constant to the denominator of the t-statistic

\textbf{empirical Bayes} theory provides formal framework for borrowing strength across genes or proteins,
e.g. popular bioconductor package \textbf{limma}
\[\tilde{s}_g=\sqrt{\frac{d_ps_p^2+d_0s_0^2}{d_g+d_0}},\]
and the moderated t-statistic is t-distributed with $d_0+d_g$ degrees of freedom under the null hypothesis $H_0: \mathbf{L}\boldsymbol{\beta}=0$.

- Note, that the degrees of freedom increase by borrowing strength across proteins.

---

## Intermezzo: Bayesian Methods

- Frequentists consider data as random and population parameters as fixed but unknown
- In Bayesian viewpoint a person has prior beliefs about the population parameters and the uncertainty on this prior beliefs are represented by a probability distribution placed on this parameter.

  - This distribution reflects the person's subjective prior opinion about plausible values of the parameter.
  - And is referred to as the prior $g(\boldsymbol{\theta})$.

- Bayesian thinking will update the prior information on the population parameters by confronting the model to data ($\mathbf{Y}$).

- By using Bayes Theorem this results in a posterior distribution on the model parameters.  
\[
g(\boldsymbol{\theta}\vert\mathbf{Y})=\frac{f(Y\vert \boldsymbol{\theta})g(\boldsymbol{\theta})}{\int f(Y\vert \boldsymbol{\theta}) g(\boldsymbol{\theta}) d\boldsymbol{\theta}} \text{     }\left(\text{ posterior}=\frac{\text{prior} \times \text{ likelihood}}{\text{Marginal distribution}}\right)
\]

---

## Limma  approach

Developed for gene expression analysis with micro arrays.
Let g be the index for gene g.
\[
\begin{array}{cc}
&\beta_{gk}\vert \sigma^2_g,\beta_{gk}\neq 0 \sim N(0,v_{0k}\sigma_g^2)\\\\
\text{Prior}\\
&\frac{1}{\sigma^2_g}\sim s^2_0\frac{\chi^2_{d_0}}{d_0}\\\\\\\\
&\hat \beta_{gk} | \beta_{gk} , \sigma_g^2 \sim N( \beta_{gk} , v_{gk}\sigma_g^2)\\\\
\text{Data}\\
&s_g^2\sim \sigma^2_g\frac{\chi^2_{d_g}}{d_g}\\\\
\end{array}
\]

---

## Limma  approach

Under this assumption, it can be shown that

- Posterior Mean for the variance parameter: \[\tilde{s}^2_p = \text{E}\left[\sigma^2_p\vert s_p^2\right]=\frac{d_0 s_0^2+d_ps_p^2}{d_0+d_p}\]

- \[\tilde{T}_p=\frac{\mathbf{L}_k \hat{ \boldsymbol{\beta_p}}}{\mathbf{L}_k^T(\mathbf{X}^T\mathbf{WX})^{-1}\mathbf{L}_k^T \tilde{s}_p^2}\]

is t-distributed under $H_0: \mathbf{L}_j\boldsymbol{\beta} = 0$

\[\tilde{T}_p \vert H_0 \sim t(d_0 + d_p)\]

---

## Empirical Bayes
- A fully Bayesian
  - would define the prior distribution by carefully choosing the prior parameters based on prior knowledge on the process
  - would confront the prior to the data and performs inference using the posterior distribution of the model parameters.
- In an empirical Bayesian approach one estimates the prior parameters based on the data.
- In **Limma** moment estimators for $s_0$ and $d_0$ are derived using the information on the gene (protein) wise variances of all genes (proteins).
- In **Limma** one also does not work with the full posterior distribution for the variances, but with the maximum a-posterior estimate.

---

## Illustration

We borrow strength across proteins by

1. placing a scaled $\chi^2$ prior: $\chi^2(s_o,d_0)$ on the precisions ($1/\sigma^2_p$)
2. estimating the prior parameters $s_0$ and $df_0$
3. replacing the estimated protein-wise variances ($s_p^2$) with the maximum a-posteriori variance
\[\tilde{s}_p = \frac{d_p s^2_p + d_0 s_0^2}{d_p+d_0}\]


```{r}
sd <- sapply(
  rowData(pe[["proteinRobust"]])$msqrobModels,
  getSigma) %>%
  na.exclude
sdPost <- sapply(
  rowData(pe[["proteinRobust"]])$msqrobModels,
  getSigmaPosterior) %>%
  na.exclude

p1 <- qplot(sd,sdPost) +
  geom_abline()
p1
```

### How do we get to the posterior standard deviation?

```{r}
hlp <- limma::squeezeVar(
  var = sapply(rowData(pe[["proteinRobust"]])$msqrobModels, getVar),
  df = sapply(rowData(pe[["proteinRobust"]])$msqrobModels, getDF)
  )
```

#### Degrees of freedom of prior

```{r}
hlp$df.prior

model <- rowData(pe[["proteinRobust"]])$msqrobModels[[2]]
getDfPosterior(model) - getDF(model)
```

#### posterior variance

$$\tilde s_p=\sqrt{\frac{d_p\times s^2_p + d_0 s_0^2}{d_p+d_0}} $$

```{r}
hlp$var.prior

varPost <- (getVar(model) * getDF(model) + hlp$df.prior * hlp$var.prior)/(getDF(model)+hlp$df.prior)
sqrt(varPost)
getSigmaPosterior(model)
```

Hence, standard deviations are shrunken towards prior standard deviation!
Large standard deviations become smaller and smaller standard deviations become larger!

```{r}
p1 +
  geom_hline(yintercept = hlp$var.prior^.5)
```



### Illustration via Simulation

Suppose that the standard deviations for all proteins are the same and are equal to 1.
We simulate proteins with the same mean as the fitted mean in the experiment but with standard deviation of 1.

```{r warning=FALSE, message=FALSE}
nCoefs <- getCoef(rowData(pe[["proteinRobust"]])$msqrobModels[[2]]) %>% length
coefs <-
sapply(rowData(pe[["proteinRobust"]])$msqrobModels,
    function(x) getCoef(x)[1:nCoefs]
  ) %>%
  t %>%
  na.exclude


p <- nrow(coefs)
n <- ncol(pe[[1]])
f0_equalVar <- sapply(1:p,
  FUN=function(i, n, betas, sd, design) {
  rnorm(n, mean = design %*% betas[i,], sd = sd)},
  n = n,
  betas = coefs,
  sd = 1,
  design = X
  ) %>%
  t
colnames(f0_equalVar) <- colnames(pe[[1]])
sims <- readQFeatures(f0_equalVar %>% as.data.frame, ecol = 1:n, name = "sim_equalVar")
colData(sims) <- colData(pe)
sims <- msqrob(object = sims, i = "sim_equalVar", formula = ~ location*tissue + patient)

sd0 <- sapply(
  rowData(sims[["sim_equalVar"]])$msqrobModels,
  getSigma) %>%
  na.exclude
sdPost0 <- sapply(
  rowData(sims[["sim_equalVar"]])$msqrobModels,
  getSigmaPosterior) %>%
  na.exclude

qplot(sd0,sdPost0) +
  geom_abline() +
  ylim(range(sd0))
```

- We observe a large variability in the individual protein level standard deviation estimates.
- We simulated proteins with standard deviation of 1, but the protein estimates vary from `r paste(round(range(sd0),2),collapse=", ... , ")`.
- Large uncertainty on the estimation of variances in small samples
- The empirical Bayes method, however, recognises that all proteins are simulated with the same variance.
- Hence, it can borrow tremendous strength across proteins to stabilize the variance estimation
- Here, it shrinks all protein variance to the prior variance, which is indeed very close to 1, the value we have adopted in the simulation.

Note, that the prior degrees of freedom are also set to infinity:
```{r}
getDF(rowData(sims[["sim_equalVar"]])$msqrobModels[[1]])
getDfPosterior(rowData(sims[["sim_equalVar"]])$msqrobModels[[1]])
```

which imposes shrinkage to the prior standard deviation!

The empirical Bayes method can thus indeed recognize the common variance that is shared across all simulated proteins!

# P-values

## Simulation under H_0.

- Mean log2 protein intensity for atrium equals mean log2 protein intensity for ventriculum in the left heart region.
- sd equals the sd for the protein.


1. Extract $\hat \sigma$ and $\beta$'s  
```{r}
sd <- sapply(
  rowData(pe[["proteinRobust"]])$msqrobModels,
  getSigma) %>%
  na.exclude

coefs <-
sapply(rowData(pe[["proteinRobust"]])$msqrobModels,
    function(x) getCoef(x)[1:nCoefs]
  ) %>%
  t %>%
  na.exclude
```

2. Set $\beta_\text{tissue}$ equal to 0. No FC between atrium and ventriculum left.

```{r}
coefs0 <- coefs
coefs0[,3] <- 0
```

3. Simulate protein expressions for each protein from a Normal distribution under $H_0$ for left heart region (no FC between atrium and ventriculum left) and sd the sd for the protein.
```{r}
set.seed(104)
f0 <- sapply(1:p,
  function(i, betas, sd, design)
  rnorm(n, mean = design %*% betas[i,], sd = sd[i]),
  betas = coefs0,
  sd = sd,
  design = X
  ) %>%
  t
colnames(f0) <- colnames(pe[[1]])
```

4. Setup QFeatures object and perform MSqRob analysis

```{r warning = FALSE, message = FALSE}
sims <- readQFeatures(f0 %>% as.data.frame, ecol = 1:n, name = "sim0")
colData(sims) <- colData(pe)
sims <- msqrob(object = sims, i = "sim0", formula = ~ location*tissue + patient)
sims <- hypothesisTest(object = sims, i = "sim0", contrast = L)
```

### Evaluate  pvalues  under H_0

```{r}
volcano <- ggplot(rowData(sims[["sim0"]])$tissueV,
                 aes(x = logFC, y = -log10(pval), color = pval < 0.05)) +
 geom_point(cex = 2.5) +
 scale_color_manual(values = alpha(c("black", "red"), 0.5)) + theme_minimal()
volcano
```

Number of false positives without multiple testing?

```{r}
rowData(sims[["sim0"]])$tissueV %>%
  filter(pval <0.05) %>%
  nrow
mean(rowData(sims[["sim0"]])$tissueV$pval < 0.05)    
hist(rowData(sims[["sim0"]])$tissueV$pval,main = "simulation H0")
```


- The p-values are uniform!
- All p-values under the null are equally likely.
- Statistical hypthesis testing leads to a uniform test strategy under $H_0$
- If use p-value cutoff at 0.05 we expect to return 5% of the non-DE proteins  as differentially expressed: many false positives can be expected!


## Pvalue distribution in real experiment

```{r}
hist(rowData(pe[["proteinRobust"]])$tissueV$pval, main = "realData")
```

- A mixture of null proteins (non-DE): uniform, and, DE proteins: enrichment of p-values at low p-values


# Correction for multiple testing

- We can adjust the p-values for multiple testing.

## Family wise error rate correction:

- A list of returned proteins is considered to be in error as soon as it contains at most one false positive protein.

- $\text{FWER} = P(FP \leq 1)$

- FWER: probability of making at least one false positive decision or
probability to declare at least one protein differentially abundant which is truly non differentially abundant

### Bonferroni method

- Simple method
- $m$ tests are performed at the level $\alpha/m$
- FWER$\leq\sum\limits_{p=1}^{m}P(reject H_{0p}\vert H_{0p}\text{ is true})=m \alpha/m=\alpha$
- Provides strong control
- Bonferroni is very conservative
- Works for dependent tests
- Adjusted p-value: $\tilde{p}_p=\min(m\ p_p,1)$

---

### Bonferroni in practise

Via R functions

```{r}
padj <- p.adjust(
  rowData(pe[["proteinRobust"]])$tissueV$pval,
  method = "bonferroni")
```

Own Implementation: adjust and make sure that p-value is smaller than 1.

```{r}
m <- sum(!is.na(rowData(pe[["proteinRobust"]])$tissueV$pval))
padjSelf <-  rowData(pe[["proteinRobust"]])$tissueV$pval * m
padjSelf[padjSelf > 1] <- 1

range(padj - padjSelf, na.rm = TRUE)
```
### Illustration in simulation under $H_0$ and heart case study

```{r}
volcano <- ggplot(rowData(sims[["sim0"]])$tissueV,
                 aes(x = logFC, y = -log10(pval), color = p.adjust(pval,"bonferroni") < 0.05)) +
 geom_point(cex = 2.5) +
 scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
 theme_minimal() +
 ggtitle("simulated heart data under H0")
volcano
```

- No false positives are returned for simulation under H_0. List is correct according to FWER.

```{r}
volcano <- ggplot(rowData(pe[["proteinRobust"]])$tissueV,
                 aes(x = logFC, y = -log10(pval), color = p.adjust(pval,"bonferroni") < 0.05)) +
 geom_point(cex = 2.5) +
 scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
 theme_minimal() +
 ggtitle("real heart data")
volcano
```

- Very few proteins are returned for real data. Very conservative!

## False discovery rate

- Adjusted P-values with the Benjamini Hochberg correction correspond to the estimated FDR of the set that is returned when the significance level is set at this threshold.
\begin{eqnarray}
FDR(p_0) &=& \text{E}\left[\frac{FP}{(FP + TP)}\right]\\
&\approx&\frac{p_0 \times m}{\#p_p \leq p_0}\\
\end{eqnarray}

So adjusted p-value for protein j equals
\[\tilde p_j = \frac{p_{0,j} \times m}{\#p_p \leq p_{0,j}}\]

However, the FDR always has to be between 0 and 1 so:

\[\tilde p_j = \min\left[\frac{p_{0,j} \times m}{\#p_p \leq p_{0,j}},1\right]\]

and the adjusted p-values should remain in the same order as the original p-values.

\[\tilde p_j =  \min\limits_{\forall k: p_k > p_j} \min\left[\frac{p_{0,k} \times m}{\#p_p \leq p_{0,k}},1\right]\]

1. Order pvalues
```{r}
pvals <- rowData(pe[["proteinRobust"]])$tissueV$pval
naInd <- is.na(pvals)
pHlp <- pvals[!naInd]
ord <- pHlp %>% order
pHlp <- pHlp[ord]
```

2. Adjust ordered p-values
```{r}
pHlp <- pHlp*length(pHlp)/(1:length(pHlp))
```

3. Ensure adjust p-values are smaller are equal than 1
```{r}
pHlp[pHlp>1] <- 1
```

4. Monotonicity constraint
```{r}
pmin <- pHlp[length(pHlp)]
for (j in (length(pHlp)-1):1)
{
  if (pHlp[j] < pmin)
    pmin <- pHlp[j] else
    pHlp[j] <- pmin   
}
```

5. Put p-values back in original order

```{r}
pHlp[ord] <- pHlp
pAdj <- pvals
pAdj[!naInd] <- pHlp

head(pAdj)
head(rowData(pe[["proteinRobust"]])$tissueV)
range(rowData(pe[["proteinRobust"]])$tissueV$adjPval - pAdj,na.rm=TRUE)
```

### Illustration in simulation under $H_0$ and heart case study

```{r}
volcano <- ggplot(rowData(sims[["sim0"]])$tissueV,
                 aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)) +
 geom_point(cex = 2.5) +
 scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
 theme_minimal() +
 ggtitle("simulated heart data under H0")
volcano
```

- No false positives are returned for simulation under H_0. List is correct according to FWER.
- It can be shown that the FDR-method controls the FWER when $H_0$ is true for all features.   


```{r}
volcano <- ggplot(rowData(pe[["proteinRobust"]])$tissueV,
                 aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)) +
 geom_point(cex = 2.5) +
 scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
 theme_minimal() +
 ggtitle("real heart data")
volcano
```

The FDR method allows us to return much longer DA protein lists at the expense of a few false positives.
The FDR controls the fraction of false positives in the list that you return on average on the significance level that is adopted.
So if you use $\alpha=0.05$ we expect on average 5% of false positives in the list that we return.
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("technicalDetailsProteomics.Rmd");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
